<!DOCTYPE html>
<html>
<head>
<title>PepperProjectScript.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="scripts-of-the-pepper-project">Scripts of the Pepper Project</h1>
<h2 id="author-jiantao-zhao">Author: Jiantao Zhao</h2>
<h3 id="date-1252020">Date: 12/5/2020</h3>
<h1 id="table-of-contents">Table of Contents</h1>
<ul>
<li><a href="#scripts-of-the-pepper-project">Scripts of the Pepper Project</a>
<ul>
<li><a href="#author-jiantao-zhao">Author: Jiantao Zhao</a>
<ul>
<li><a href="#date-1252020">Date: 12/5/2020</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#table-of-contents">Table of Contents</a>
<ul>
<li><a href="#reference-genome-evaluation">Reference genome evaluation</a>
<ul>
<li><a href="#remove-contamination">Remove contamination</a></li>
<li><a href="#busco-evaluation">BUSCO evaluation</a></li>
<li><a href="#qualiy-evaluation-using-merqury">Qualiy evaluation using Merqury</a>
<ul>
<li><a href="#zhangshugang">Zhangshugang</a></li>
<li><a href="#calculate-the-genome-size-using-r">Calculate the genome size using R</a></li>
<li><a href="#grif_1614-merqury">Grif_1614 Merqury</a></li>
<li><a href="#pi632928-merqury">PI632928 Merqury</a></li>
</ul>
</li>
<li><a href="#genome-size-estimation-using-jellyfish">Genome size estimation using Jellyfish</a>
<ul>
<li><a href="#zhangshugang-1">Zhangshugang</a></li>
<li><a href="#grif_1614">Grif_1614</a></li>
<li><a href="#calculate-the-genome-size-using-r-1">Calculate the genome size using R</a></li>
<li><a href="#pi632928">PI632928</a></li>
</ul>
</li>
<li><a href="#ltr-assembly-index-lai">LTR Assembly Index (LAI)</a></li>
<li><a href="#ncbi-sumbission">NCBI sumbission</a>
<ul>
<li><a href="#remove-contamination-for-pi_632928">Remove contamination for PI_632928</a></li>
<li><a href="#remove-contamination-for-grif_1614">Remove contamination for Grif_1614</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#reference-genome-analysis">Reference Genome analysis</a>
<ul>
<li><a href="#circus-plot-of-s8zhangshugang">Circus plot of S8/Zhangshugang</a></li>
<li><a href="#genome-synteny-using-mummer">Genome synteny using mummer</a></li>
<li><a href="#genome-synteny-using-mcscanx">Genome synteny using MCScanX</a></li>
<li><a href="#map-the-chiltepin-reads-to-the-chiltepin-genome-to-check-the-read-count">Map the Chiltepin reads to the Chiltepin genome to check the read count</a></li>
<li><a href="#align-sequence-to-s8">Align sequence to S8</a></li>
</ul>
</li>
<li><a href="#vg-pangenome">vg pangenome</a>
<ul>
<li><a href="#trimmomatic">Trimmomatic</a></li>
<li><a href="#sv-estimation">SV estimation</a></li>
<li><a href="#vg-mapping">vg mapping</a></li>
<li><a href="#vg-mapping-on-cornell-server">vg mapping on cornell server</a></li>
<li><a href="#snp-calling-using-sentieon">SNP calling using sentieon</a></li>
<li><a href="#snp-calling-using-sentieon-for-tomato-as-outgroup">SNP calling using sentieon for tomato as outgroup</a></li>
<li><a href="#snp-calling-using-sentieon-on-cornell-server">SNP calling using sentieon on cornell server</a></li>
<li><a href="#join-the-vcf-file">Join the vcf file</a></li>
<li><a href="#extract-the-high-quality-snps-and-indels">Extract the high quality SNPs and Indels</a></li>
<li><a href="#snp-and-indel-snpeff">SNP and Indel snpeff</a>
<ul>
<li><a href="#snp-missing-rate-cutoff">SNP missing rate cutoff</a></li>
<li><a href="#snp-check">SNP check</a></li>
<li><a href="#snp-check-to-old-s8-references">SNP check to old S8 references</a></li>
<li><a href="#snp-check-to-vg-reference-for-bwa-mapping">SNP check to vg reference for bwa mapping</a></li>
<li><a href="#snp-check-to-vg-reference-for-vg-mapping">SNP check to vg reference for vg mapping</a></li>
</ul>
</li>
<li><a href="#snp-ld-prune">SNP LD prune</a></li>
<li><a href="#extract-4dtv-snps-and-iq-tree">Extract 4DTV SNPs and IQ-tree</a></li>
<li><a href="#newhyvbrid">Newhyvbrid</a></li>
<li><a href="#hyde">HyDe</a>
<ul>
<li><a href="#cha-and-bac-miss01">CHA and BAC Miss0.1</a></li>
<li><a href="#annuum-miss01">Annuum Miss0.1</a></li>
</ul>
</li>
<li><a href="#population-structure">Population structure</a></li>
<li><a href="#fst">Fst</a>
<ul>
<li><a href="#fst-permutation-for-babbap">Fst permutation for BAB.BAP</a></li>
<li><a href="#fst-permutation-for-pub-with-the-other-groups">Fst permutation for PUB with the other groups</a></li>
<li><a href="#top-1-fst">Top 1% Fst</a></li>
</ul>
</li>
<li><a href="#ld-estimation">LD estimation</a></li>
<li><a href="#ld-for-annuum-clade">LD for Annuum clade</a></li>
<li><a href="#pi">Pi</a></li>
<li><a href="#pi-with-only-window-of-100000">Pi with only window of 100000</a>
<ul>
<li><a href="#permutation-test-using-4dtv-snps">Permutation test using 4DTV SNPs</a></li>
<li><a href="#t-test-in-r">T-test in R</a></li>
</ul>
</li>
<li><a href="#effective-population-size">Effective population size</a>
<ul>
<li><a href="#ann">ANN</a></li>
<li><a href="#gla">GLA</a></li>
<li><a href="#bab">BAB</a></li>
<li><a href="#bap">BAP</a></li>
<li><a href="#cha">CHA</a></li>
<li><a href="#chn">CHN</a></li>
<li><a href="#fru">FRU</a></li>
<li><a href="#pub">PUB</a></li>
<li><a href="#final-combined-plot">Final combined plot</a></li>
</ul>
</li>
<li><a href="#gwas">GWAS</a>
<ul>
<li><a href="#glaann">GLA.ANN</a>
<ul>
<li><a href="#shape-variations">Shape variations</a></li>
</ul>
</li>
<li><a href="#babbap">BAB.BAP</a>
<ul>
<li><a href="#shape-variations-1">Shape variations</a></li>
</ul>
</li>
<li><a href="#seed-size">Seed size</a>
<ul>
<li><a href="#extract-the-phenotype">Extract the phenotype</a></li>
<li><a href="#glaann-1">GLA.ANN</a>
<ul>
<li><a href="#manhattan-plot">Manhattan plot</a></li>
</ul>
</li>
<li><a href="#fru-1">FRU</a>
<ul>
<li><a href="#manhattan-plot-1">Manhattan plot</a></li>
</ul>
</li>
<li><a href="#chn-1">CHN</a>
<ul>
<li><a href="#manhattan-plot-2">Manhattan plot</a></li>
</ul>
</li>
<li><a href="#annuum">Annuum</a>
<ul>
<li><a href="#manhattan-plot-3">Manhattan plot</a></li>
</ul>
</li>
<li><a href="#babbap-1">BAB.BAP</a>
<ul>
<li><a href="#manhattan-plot-4">Manhattan plot</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#chn-fruit-shape">CHN fruit shape</a>
<ul>
<li><a href="#manhattan-plot-5">Manhattan plot</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#sweed">SweeD</a>
<ul>
<li><a href="#installation">Installation</a></li>
<li><a href="#bap-with-10k-grid">BAP with 10k grid</a></li>
<li><a href="#bap-with-100k-size">BAP with 100k size</a></li>
<li><a href="#ann-with-10k-grid">ANN with 10k grid</a></li>
<li><a href="#ann-with-100k-grid">ANN with 100k grid</a></li>
<li><a href="#overlaped-regions">Overlaped regions</a></li>
<li><a href="#comparison-with-cao-et-al-2022">Comparison with Cao et al. 2022</a></li>
</ul>
</li>
<li><a href="#introgression">Introgression</a></li>
<li><a href="#introgression-with-ld-prune">Introgression with LD prune</a></li>
</ul>
</li>
<li><a href="#mapping-quality-evaluation">Mapping quality evaluation</a>
<ul>
<li><a href="#mapping-quality-of-the-1314-collection">Mapping quality of the 1314 collection</a></li>
<li><a href="#mapping-quality-of-the-500-core-collection">Mapping quality of the 500 core collection</a></li>
<li><a href="#genotype-check-of-c-baccatum">Genotype check of C. baccatum</a></li>
</ul>
</li>
<li><a href="#snp-calling-and-evaluation">SNP calling and evaluation</a>
<ul>
<li><a href="#snps-calling-and-quality-control-of-the-1300-pepper-collection">SNPs calling and quality control of the 1300 pepper collection</a></li>
<li><a href="#snp-calling-quality-check">SNP calling quality check</a></li>
<li><a href="#snp-count-for-each-group">SNP count for each group</a></li>
<li><a href="#snp-prunning">SNP prunning</a></li>
<li><a href="#snp-effects-and-distribution-of-different-subgroups">SNP effects and distribution of different subgroups.</a>
<ul>
<li><a href="#upset-plot-in-r">Upset plot in R</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#4dtv-snps">4DTV SNPs</a></li>
<li><a href="#iq-tree">IQ-tree</a>
<ul>
<li><a href="#iq-tree-for-the-1300-accessions">IQ-Tree for the ~1300 accessions</a></li>
<li><a href="#iq-tree-for-504-accessions">IQ-tree for 504 accessions</a></li>
<li><a href="#iq-tree-for-504-accessions-with-prunned-snps">IQ-tree for 504 accessions with prunned SNPs</a></li>
</ul>
</li>
<li><a href="#pca-analysis">PCA analysis</a></li>
<li><a href="#population-structure-analysis">Population structure analysis</a>
<ul>
<li><a href="#population-structure-use-pruned-4dtv-snps">Population structure use pruned 4DTV SNPs</a></li>
</ul>
</li>
<li><a href="#genome-wide-ld">Genome-wide LD</a></li>
<li><a href="#fst-differentiation">Fst differentiation</a>
<ul>
<li><a href="#fst-with-accessions-no-taxonomy-reassignment">Fst with accessions no taxonomy reassignment</a></li>
<li><a href="#fst-with-accessions-of-pure-background">Fst with accessions of pure background</a></li>
<li><a href="#fst-permutation-test">Fst permutation test</a>
<ul>
<li><a href="#bab_cha">BAB_CHA</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#nucleotide-diversity">Nucleotide diversity</a>
<ul>
<li><a href="#nucleotide-diversity-with-accessions-no-taxonomy-reassignment">Nucleotide diversity with accessions no taxonomy reassignment</a></li>
<li><a href="#nucleotide-diversity-with-accessions-of-pure-background">Nucleotide diversity with accessions of pure background</a></li>
</ul>
</li>
<li><a href="#treemix-analysis">Treemix analysis</a></li>
<li><a href="#abba-baba-tests">ABBA-BABA tests</a></li>
<li><a href="#introgression-1">Introgression</a>
<ul>
<li><a href="#introgression-chr074900001-10000000">Introgression Chr07:4,900,001-10,000,000</a></li>
<li><a href="#introgression-chr07258150001-259300000">Introgression Chr07:258,150,001-259,300,000</a></li>
<li><a href="#introgression-chr12250550001-255900000">Introgression Chr12:250550001-255900000</a></li>
<li><a href="#pubbacfruchngla">PUB.BAC.FRUCHN.GLA</a></li>
</ul>
</li>
<li><a href="#sweed-1">SweeD</a>
<ul>
<li><a href="#installation-1">Installation</a></li>
<li><a href="#bac-with-10k-grid">BAC with 10k grid</a></li>
<li><a href="#bac-with-pruned-snps">BAC with pruned SNPs</a></li>
<li><a href="#gla-with-10k-grid">GLA with 10k grid</a></li>
</ul>
</li>
<li><a href="#xp-clr-for-c-annuum">XP-CLR for C. annuum</a>
<ul>
<li><a href="#xp-clr-for-c-annuum-with-all-c-annuum-var-grabriusculum">XP-CLR for C. annuum with all C. annuum var. grabriusculum</a></li>
<li><a href="#xp-clr-for-c-bac-baccatum-versus-c-bac-pendulum">XP-CLR for C. bac. baccatum versus C. bac. pendulum</a></li>
<li><a href="#enrichment-analysis">Enrichment analysis</a></li>
</ul>
</li>
<li><a href="#effective-population-size-1">Effective population size</a>
<ul>
<li><a href="#smc-for-gla-with-different-geographical-background">SMC for GLA with different geographical background</a></li>
</ul>
</li>
<li><a href="#gwas-analysis-of-fruit-shape">GWAS analysis of fruit shape</a>
<ul>
<li><a href="#gwas-for-c-annuum">GWAS for <em>C. annuum</em></a></li>
<li><a href="#extract-the-genotypes-for-caovate">Extract the genotypes for CaOvate</a></li>
<li><a href="#manhattan-plot-for-glaann">Manhattan plot for GLA.ANN</a></li>
<li><a href="#manhattan-plot-for-babbap">Manhattan plot for BAB.BAP</a></li>
</ul>
</li>
<li><a href="#transcriptome-data">Transcriptome data</a></li>
<li><a href="#personal-scripts">Personal scripts</a>
<ul>
<li><a href="#4dtv_scanpl">4DTV_scan.pl</a></li>
<li><a href="#assign_geneticpl">assign_genetic.pl</a></li>
<li><a href="#pi_slide_winpl">PI_slide_win.pl</a></li>
<li><a href="#mergepl">merge.pl</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="reference-genome-evaluation">Reference genome evaluation</h2>
<h3 id="remove-contamination">Remove contamination</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/S8

<span class="hljs-comment"># Rename the chromosomal names of the reference genome</span>
cat pepper_S8.fasta | \
    sed <span class="hljs-string">'s/Superscaffold1_54_332676983/Chr01/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold2_27_177319215/Chr02/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold3_40_289790774/Chr03/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold4_26_248932513/Chr04/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold5_64_254874144/Chr05/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold6_82_253233553/Chr06/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold7_54_266382521/Chr07/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold8_30_174326481/Chr08/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold9_43_278428952/Chr09/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold10_31_210332287/Chr10/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold11_37_275185330/Chr11/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold12_60_260676116/Chr12/g'</span> \
    &gt; pepper_S8.fasta

<span class="hljs-comment"># Index the reference genome</span>
samtools faidx pepper_S8.fasta

<span class="hljs-comment"># Remove mitochondrion contaminations</span>
awk <span class="hljs-string">'{print $1}'</span> pepper_S8.fasta.fai | grep -v -f mitochondrion.id | tr -s <span class="hljs-string">"\r\n"</span> <span class="hljs-string">" "</span> &gt; clean.id

<span class="hljs-comment"># Rename chromosome names for genome gff file: pepper_s8.gff3</span>
zcat pepper_S8.gff3.gz | \
    sed <span class="hljs-string">'s/Superscaffold10/Chr10/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold11/Chr11/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold12/Chr12/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold1/Chr01/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold2/Chr02/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold3/Chr03/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold4/Chr04/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold5/Chr05/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold6/Chr06/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold7/Chr07/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold8/Chr08/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold9/Chr09/g'</span> \
    &gt; Ca.S8.gff

<span class="hljs-comment"># Remove contaminations</span>
samtools faidx Zhangshugang.clean.fasta

<span class="hljs-comment"># Create a bed file containing the chr, start and end+500 position and name as contamination.bed</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Contig00220 Contig00221 Contig00309 Contig00415 Contig00416 Contig00417 Contig00721 ; <span class="hljs-keyword">do</span> 
    sed -i <span class="hljs-string">"/<span class="hljs-variable">$i</span>/,+1d"</span> Neorickii.genome.fa
<span class="hljs-keyword">done</span>

bedtools maskfasta -mc M -<span class="hljs-keyword">fi</span> Zhangshugang.clean.fasta -bed contamination.bed -fo masked.fasta
sed <span class="hljs-string">'s/M//g'</span> masked.fasta &gt; Zhangshugang.updated.new.fasta
samtools faidx masked.fasta
samtools faidx Zhangshugang.updated.new.fasta

<span class="hljs-comment"># Update the gff files</span>
grep Chr01 Ca.S8.updated.gff  | awk <span class="hljs-string">'$4&lt;332426690 {print $0}'</span> &gt; Ca.S8.updated.Chr01.before.gff
grep Chr01 Ca.S8.updated.gff  | awk <span class="hljs-string">'$5&gt;332536023 {print $1,$2,$3,$4-61608,$5-61608,$6,$7,$8,$9}'</span> | \
    sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Ca.S8.updated.Chr01.after.gff
grep -v Chr01 Ca.S8.updated.gff | grep -v Chr09 | grep -v Chr10 | \
    grep -v Chr11 | grep -v Chr12 | grep -v Contig &gt; Ca.S8.updated.Chr02to08.gff
grep Chr09 Ca.S8.updated.gff  | awk <span class="hljs-string">'$5&lt;278410513 {print $0}'</span> &gt; Ca.S8.updated.Chr09.before.gff
grep Chr10 Ca.S8.updated.gff &gt; Ca.S8.updated.Chr10.gff
grep Chr11 Ca.S8.updated.gff &gt; Ca.S8.updated.Chr11.gff
grep Chr12 Ca.S8.updated.gff  | awk <span class="hljs-string">'$4&gt;939741 {print $1,$2,$3,$4-939741,$5-939741,$6,$7,$8,$9}'</span> | \
    sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Ca.S8.updated.Chr12.after.gff
grep Contig Ca.S8.updated.gff &gt; Ca.S8.updated.Contig.gff

cat Ca.S8.updated.Chr01.before.gff Ca.S8.updated.Chr01.after.gff Ca.S8.updated.Chr02to08.gff \
    Ca.S8.updated.Chr09.before.gff Ca.S8.updated.Chr10.gff Ca.S8.updated.Chr11.gff \
    Ca.S8.updated.Chr12.after.gff Ca.S8.updated.Contig.gff | \
    sed <span class="hljs-string">'s/gene-//g'</span> | sed <span class="hljs-string">'s/rna-//g'</span> \
    &gt; Ca.S8.updated.new.gff

<span class="hljs-comment"># Extract cds</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.S8.updated.Chr01.before Ca.S8.updated.Chr01.after Ca.S8.updated.Chr02to08 \
    Ca.S8.updated.Chr09.before Ca.S8.updated.Chr10 Ca.S8.updated.Chr11 Ca.S8.updated.Chr12.after ; <span class="hljs-keyword">do</span>
    gffread -x <span class="hljs-variable">$i</span>.cds.fasta -g Zhangshugang.updated.new.fasta <span class="hljs-variable">$i</span>.gff
<span class="hljs-keyword">done</span>

cat Ca.S8.updated.Chr01.before.cds.fasta Ca.S8.updated.Chr01.after.cds.fasta \
    Ca.S8.updated.Chr02to08.cds.fasta Ca.S8.updated.Chr09.before.cds.fasta \
    Ca.S8.updated.Chr10.cds.fasta Ca.S8.updated.Chr11.cds.fasta \
    Ca.S8.updated.Chr12.after.cds.fasta Ca.S8.updated.Contig.cds.fasta \
    &gt; Zhangshugang.updated.new.cds.fasta

<span class="hljs-comment"># Alternatively</span>
<span class="hljs-comment"># cds</span>
gffread -x Zhangshugang.updated.new.cds.fasta -g Zhangshugang.updated.new.fasta Ca.S8.updated.new.gff

<span class="hljs-comment"># protein</span>
gffread -y Zhangshugang.updated.new.protein.fasta -g Zhangshugang.updated.new.fasta Ca.S8.updated.new.gff

<span class="hljs-comment"># Upload to NCBI</span>

<span class="hljs-comment"># Update the vcf file</span>
<span class="hljs-comment"># For Chr01</span>
zcat Chr01.clean.vcf.gz | grep <span class="hljs-string">"#"</span> &gt; header
zcat Chr01.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | \
    awk <span class="hljs-string">'$2&lt;332426690 {print $0}'</span> &gt; Chr01.clean.vcf-1
zcat Chr01.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | \
    awk <span class="hljs-string">'$2&gt;332536023 {print $1,$2-61608,$0}'</span> | \
    sed <span class="hljs-string">'s/ /\t/g'</span> | cut -f1,2,5- \
    &gt; Chr01.clean.vcf-2
cat header Chr01.clean.vcf-1 Chr01.clean.vcf-2 | \
    gzip &gt; Chr01.clean.new.vcf.gz

<span class="hljs-comment"># Chr09</span>
zcat Chr09.clean.vcf.gz | grep <span class="hljs-string">"#"</span> &gt; header
zcat Chr09.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'$2&lt;278410513 {print $0}'</span> | \
    cat header - | gzip &gt; Chr09.clean.new.vcf.gz

<span class="hljs-comment"># Chr12</span>
zcat Chr12.clean.vcf.gz | grep <span class="hljs-string">"#"</span> &gt; header
zcat Chr12.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | \
    awk <span class="hljs-string">'$2&gt;939741 {print $1,$2-939741,$0}'</span> | \
    sed <span class="hljs-string">'s/ /\t/g'</span> | cut -f1,2,5- | cat header - | gzip &gt; Chr12.clean.new.vcf.gz

<span class="hljs-comment"># Cross-check of the genotype</span>
<span class="hljs-comment">## Chr01</span>
<span class="hljs-comment"># Extract the SNP position infos</span>
grep -v <span class="hljs-string">"#"</span> Chr01.clean.vcf | awk <span class="hljs-string">'{print $1,$2-1,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr01.bed
<span class="hljs-comment"># Extract the geno from references</span>
bedtools getfasta -<span class="hljs-keyword">fi</span> \
    /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \
    -bed Chr01.bed -name -tab \
    &gt; Chr01.bed.out

<span class="hljs-comment"># Extract the reference genotypes from vcf</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS %REF\n'</span> Chr01.clean.vcf.gz &gt; Chr01.vcf.bed

<span class="hljs-comment"># Check the consistency of genotypes</span>
paste Chr01.bed.out  Chr01.vcf.bed | awk <span class="hljs-string">'{if ($2==$5) print "TRUE"; else print "FALSE"}'</span> | grep FALSE | wc 
<span class="hljs-comment"># Number of inconsistency: 0</span>

<span class="hljs-comment">## Chr12 </span>
<span class="hljs-comment"># Extract the SNP position infos</span>
grep -v <span class="hljs-string">"#"</span> Chr12.clean.vcf | awk <span class="hljs-string">'{print $1,$2-1,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr12.bed
<span class="hljs-comment"># Extract the geno from references</span>
bedtools getfasta -<span class="hljs-keyword">fi</span> \
    /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \
    -bed Chr12.bed -name -tab \
    &gt; Chr12.bed.out

<span class="hljs-comment"># Extract the reference genotypes from vcf</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS %REF\n'</span> Chr12.clean.vcf.gz &gt; Chr12.vcf.bed

<span class="hljs-comment"># Check the consistency of genotypes</span>
paste Chr12.bed.out  Chr12.vcf.bed | awk <span class="hljs-string">'{if ($2==$5) print "TRUE"; else print "FALSE"}'</span> | grep FALSE | wc 
<span class="hljs-comment"># Number of inconsistency: 0</span>

</div></code></pre>
<h3 id="busco-evaluation">BUSCO evaluation</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/S8

<span class="hljs-comment"># Genome completeness</span>
busco -i /data/zhaojiantao/pepper/References/S8/Zhangshugang.updated.new.fasta \
    -l /data/zhaojiantao/database/embryophyta_odb10 \
    -c 20 -m genome -f --out Zhangshugang.embryophyta.genome

<span class="hljs-comment"># Gene completeness</span>
busco -i /data/zhaojiantao/pepper/References/S8/Zhangshugang.updated.new.protein.fasta \
    -l /data/zhaojiantao/database/embryophyta_odb10 \
    -c 20 -m proteins -f --out Zhangshugang.embryophyta.gene

</div></code></pre>
<h3 id="qualiy-evaluation-using-merqury">Qualiy evaluation using Merqury</h3>
<h4 id="zhangshugang">Zhangshugang</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># [Merqury](https://github.com/marbl/merqury)</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/S8/Merqury

<span class="hljs-comment"># Get the right k size</span>
best_k.sh 3020000000
<span class="hljs-comment"># Best K-mer size = 20</span>

<span class="hljs-comment"># Build meryl dbs</span>
meryl k=20 count output S8_R1.meryl S8_R1.fastq.gz
meryl k=20 count output S8_R2.meryl S8_R2.fastq.gz

<span class="hljs-comment"># Merge the files</span>
meryl union-sum output S8.genome.meryl S8_R*.meryl

<span class="hljs-comment"># Calculate QV score</span>
merqury.sh S8.genome.meryl ../pepper_S8.fasta S8.QV

</div></code></pre>
<h4 id="calculate-the-genome-size-using-r">Calculate the genome size using R</h4>
<pre class="hljs"><code><div>histo &lt;- read.table(<span class="hljs-string">"S8.reads.histo"</span>) 
dim(histo)
<span class="hljs-comment"># Estimate genome size</span>
sum(as.numeric(histo[<span class="hljs-number">2</span>:<span class="hljs-number">100896</span>,<span class="hljs-number">1</span>]*histo[<span class="hljs-number">2</span>:<span class="hljs-number">100896</span>,<span class="hljs-number">2</span>]))/<span class="hljs-number">45</span>

</div></code></pre>
<h4 id="grif1614-merqury">Grif_1614 Merqury</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># [Merqury](https://github.com/marbl/merqury)</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_raw_data/Grif_1614.HiFI/Merqury

<span class="hljs-comment"># Get the right k size</span>
best_k.sh 3927487386
<span class="hljs-comment"># Best K-mer size = 21</span>

<span class="hljs-comment"># Build meryl dbs</span>
meryl k=21 count output threads=80 Grif_1614.genome.meryl ../Grif1614.fastq.gz

<span class="hljs-comment"># Calculate QV score</span>
merqury.sh Grif_1614.genome.meryl /data/zhaojiantao/pepper/References/Grif_1614/Grif_1614.fa Grif_1614

<span class="hljs-comment"># Check the result summary</span>
Grif_1614.QV.qv
Grif_1614.QV.completeness.stats

</div></code></pre>
<h4 id="pi632928-merqury">PI632928 Merqury</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># [Merqury](https://github.com/marbl/merqury)</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_raw_data/PI632928_Illumina/Merqury

<span class="hljs-comment"># Build meryl dbs</span>
meryl k=21 count output B554.R1.meryl B554.R1.fastq.gz
meryl k=21 count output B554.R2.meryl B554.R2.fastq.gz

<span class="hljs-comment"># Merge the files</span>
meryl union-sum output B554.genome.meryl B554.R*.meryl

<span class="hljs-comment"># Calculate QV score</span>
merqury.sh B554.genome.meryl /data/zhaojiantao/pepper/References/PI_632928/PI_632928.fa PI_632928.QV

</div></code></pre>
<h3 id="genome-size-estimation-using-jellyfish">Genome size estimation using Jellyfish</h3>
<h4 id="zhangshugang">Zhangshugang</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># k-mer evaluation</span>
<span class="hljs-comment"># Obtain the k-mer frequencies of your raw data (fastq) using JellyFish</span>
jellyfish count -C -m 21 -s 1000000000 -t 30 &lt;(zcat S8_R1.fastq.gz) &lt;(zcat S8_R2.fastq.gz) -o S8.reads.jf

<span class="hljs-comment"># Export the kmer count histogram</span>
jellyfish histo -t 10 -h 1000000 S8.reads.jf &gt; S8.reads.histo

<span class="hljs-comment"># Run Genomescope</span>
genomescope.R S8.reads.histo 21 150 S8.Kmer

</div></code></pre>
<h4 id="grif1614">Grif_1614</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_raw_data/Grif_1614.HiFI/Jellyfish

<span class="hljs-comment"># k-mer evaluation</span>
<span class="hljs-comment"># Obtain the k-mer frequencies of your raw data (fastq) using JellyFish</span>
jellyfish bc -m 21 -s 100000000000 -t 80 -C -o Grif1614.bc ../Grif1614.fastq
<span class="hljs-comment"># Jellyfish count</span>
jellyfish count -m 21 -s 4000000000 -t 80 -C --bc Grif1614.bc ../Grif1614.fastq -o Grif_1614.reads.jf

<span class="hljs-comment"># Export the kmer count histogram</span>
jellyfish histo -t 100 -h 1000000 Grif_1614.reads.jf &gt; Grif_1614.reads.histo

<span class="hljs-comment"># Run Genomescope</span>
genomescope.R Grif_1614.reads.histo 21 14210 Grif_1614.GenomeScope

<span class="hljs-comment"># If you met an error showing: Failed to converge, which could possible due to some of the reads count x coverage exceeded the limited of maximum number. You can simply select a portion (eg. 3/4) of your data and run it again. </span>

</div></code></pre>
<h4 id="calculate-the-genome-size-using-r">Calculate the genome size using R</h4>
<pre class="hljs"><code><div>histo &lt;- read.table(<span class="hljs-string">"Grif_1614.reads.histo"</span>) 
dim(histo)
<span class="hljs-comment"># Estimate genome size</span>
sum(as.numeric(histo[<span class="hljs-number">2</span>:<span class="hljs-number">100896</span>,<span class="hljs-number">1</span>]*histo[<span class="hljs-number">2</span>:<span class="hljs-number">100896</span>,<span class="hljs-number">2</span>]))/<span class="hljs-number">20</span>

</div></code></pre>
<h4 id="pi632928">PI632928</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_raw_data/PI632928_Illumina/Jellyfish

jellyfish count -C -m 21 -s 3000000000 -t 80 PI632928.fastq -o PI632928.reads.jf

<span class="hljs-comment"># Export the kmer count histogram</span>
jellyfish histo -t 100 -h 1000000 PI632928.reads.jf &gt; PI632928.reads.histo

<span class="hljs-comment"># Run Genomescope</span>
genomescope.R PI632928.reads.histo 21 150 PI632928.Genomescope

</div></code></pre>
<h3 id="ltr-assembly-index-lai"><a href="https://github.com/oushujun/LTR_retriever">LTR Assembly Index (LAI)</a></h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
/data/zhaojiantao/pepper/References/S8

<span class="hljs-comment"># Install [RepeatMasker](http://www.repeatmasker.org/RepeatMasker/RepeatMasker-4.1.2-p1.tar.gz)</span>
wget http://www.repeatmasker.org/RepeatMasker/RepeatMasker-4.1.2-p1.tar.gz
tar -xvzf RepeatMasker-4.1.2-p1.tar.gz
<span class="hljs-comment"># Install and update RepeatMasker Libraries</span>
<span class="hljs-built_in">cd</span> RepeatMasker/Libraries
rm Dfam.h5

<span class="hljs-comment"># Install LTR_finder from https://github.com/xzhub/LTR_Finder</span>
wget https://github.com/xzhub/LTR_Finder/archive/refs/heads/master.zip
unzip master.zip
<span class="hljs-built_in">cd</span> LTR_Finder-master/<span class="hljs-built_in">source</span>

<span class="hljs-comment"># Index the genome</span>
gt suffixerator -db ../Zhangshugang.fasta -indexname ../Zhangshugang.fasta -tis -suf -lcp -des -ssp -sds -dna

gt ltrharvest \
    -index ../Zhangshugang.fasta \
    -similar 85 -vic 10 -seed 20 -seqids yes \
    -minlenltr 100 -maxlenltr 7000 -mintsd 4 -maxtsd 6 \
    -motif TGCA -motifmis 1 \
    &gt; Zhangshugang.harvest.scn

<span class="hljs-comment"># Use [LTR_FINDER_parallel](https://github.com/oushujun/LTR_FINDER_parallel) to accelerate the speed</span>
LTR_FINDER_parallel \
    -seq ../Zhangshugang.fasta \
    -harvest_out \
    -size 1000000 \
    -time 300 \
    -threads 50

cat Zhangshugang.harvest.scn Zhangshugang.fasta.finder.combine.scn &gt; Zhangshugang.rawLTR.scn

LTR_retriever \
    -genome ../Zhangshugang.fasta \
    -inharvest Zhangshugang.rawLTR.scn \
    -threads 60

mv Zhangshugang.fasta.mod.out.LAI Zhangshugang.fasta.mod.out.default.LAI

<span class="hljs-comment"># Add -unlock</span>
perl /data/zhaojiantao/tools/LTR_retriever-master/LAI \
    -genome Zhangshugang.fasta \
    -intact Zhangshugang.fasta.pass.list \
    -all Zhangshugang.fasta.out \
    -t 10 -q -unlock

</div></code></pre>
<h3 id="ncbi-sumbission">NCBI sumbission</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># For Grif_1614 genome assembly</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/NCBI.submit

<span class="hljs-comment"># Submit via ascp for Grif_1614</span>
ascp -i /data/zhaojiantao/pepper/References/NCBI.submit/aspera.openssh \
    -QT -l1000m -k1 -d Grif_1614 subasp@upload.ncbi.nlm.nih.gov:uploads/zf25_cornell.edu_cFD55kCx

<span class="hljs-comment"># Submit via ascp for PI_632928</span>
ascp -i /data/zhaojiantao/pepper/References/NCBI.submit/aspera.openssh \
    -QT -l1000m -k1 -d PI_632928 subasp@upload.ncbi.nlm.nih.gov:uploads/zf25_cornell.edu_cFD55kCx

</div></code></pre>
<h4 id="remove-contamination-for-pi632928">Remove contamination for PI_632928</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/PI_632928

<span class="hljs-comment"># Replace the contig ids</span>
seqkit replace -k chr.ids --pattern <span class="hljs-string">"^(\w+)"</span> --replacement <span class="hljs-string">"{kv}"</span> -w 0 PI_632928.fa &gt; ../../NCBI.submit/PI_632928/PI_632928.fa

<span class="hljs-comment"># Extract the contig id</span>
grep HiC_scaffold Genome.without.contig.merge/PI_632928.fa.fai | cut -f1 &gt; Chr00.ids
grep -v HiC_scaffold Genome.without.contig.merge/PI_632928.fa.fai | cut -f1 &gt; Chr.ids

<span class="hljs-comment"># Extract the Chr00 sequences</span>
seqkit grep -n -f Chr00.ids Genome.without.contig.merge/PI_632928.fa -w 0 -o Chr00.old.fa
seqkit grep -n -f Chr.ids Genome.without.contig.merge/PI_632928.fa -w 0 -o Chr.fa

<span class="hljs-comment"># Rename the seq ids</span>
seqkit replace -k Genome.without.contig.merge/contig.ids --pattern <span class="hljs-string">"^(\w+)"</span> --replacement <span class="hljs-string">"{kv}"</span> -w 0 Chr00.old.fa 

<span class="hljs-comment"># Remove contamination</span>
bedtools maskfasta -mc M -<span class="hljs-keyword">fi</span> PI_639682_genome_submit.fa -bed contamination.bed2 -fo PI_639682_genome_submit.fa-1
sed <span class="hljs-string">'s/M//g'</span> PI_639682_genome_submit.fa-1 &gt; PI_639682.withctg.updated.fa

<span class="hljs-comment"># If there are any contaminations, replace with N</span>
bedtools maskfasta -mc N -<span class="hljs-keyword">fi</span> PI_632928.fa -bed contamination.bed -fo PI_632928.fa2

<span class="hljs-comment"># Alternatively, replace it with others and remove it</span>
bedtools maskfasta -mc M -<span class="hljs-keyword">fi</span> PI_632928.old.fa -bed contamination.bed2 -fo PI_632928.old.fa2
sed <span class="hljs-string">'s/M//g'</span> PI_632928.old.fa2 &gt; PI_632928.updated.fa

</div></code></pre>
<h4 id="remove-contamination-for-grif1614">Remove contamination for Grif_1614</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/Grif_1614/Genome.without.contig.merge

<span class="hljs-comment"># Extract the contig id</span>
grep -v scaffold62 Grif_1614.fa.fai | cut -f1 &gt; Chr.ids
seqkit grep -n -f Chr.ids Grif_1614.fa -w 0 -o Chr.fa

<span class="hljs-comment"># Prepare the bed file for masking contaminations</span>
awk <span class="hljs-string">'{print $1,$2-1,$3}'</span> contamination.ctg | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; contamination.ctg2

<span class="hljs-comment"># Replace contaminations with N</span>
bedtools maskfasta -mc N -<span class="hljs-keyword">fi</span> Chr.fa -bed contamination.ctg2 -fo Grif_1614_genome_submit.fa

<span class="hljs-comment"># Re-submit to NCBI</span>
cp Grif_1614_genome_submit.fa /data/zhaojiantao/pepper/References/NCBI.submit/Grif_1614.update1

ascp -i /data/zhaojiantao/pepper/References/NCBI.submit/aspera.openssh \
    -QT -l1000m -k1 -d /data/zhaojiantao/pepper/References/NCBI.submit/Grif_1614.update1 subasp@upload.ncbi.nlm.nih.gov:uploads/zf25_cornell.edu_cFD55kCx

ascp -i /data/zhaojiantao/pepper/References/NCBI.submit/aspera.openssh \
    -QT -l1000m -k1 -d /data/zhaojiantao/pepper/References/NCBI.submit/Grif_1614.update3 subasp@upload.ncbi.nlm.nih.gov:uploads/zf25_cornell.edu_cFD55kCx


<span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/Grif_1614

<span class="hljs-comment"># Update the annotation</span>
<span class="hljs-comment"># Remove the following genes from Chr10 due to contaminations named as Contamination.genes</span>
Cpu10g10630
Cpu10g10640
Cpu10g10650
Cpu10g10750
Cpu10g10760
Cpu10g10770
Cpu10g10780
Cpu00g04710
Cpu00g04720
Cpu00g04730

<span class="hljs-comment"># Remove the genes from gff3</span>
grep -vf Contamination.genes Grif_1614.old.gff3 &gt; Grif_1614.gff3

<span class="hljs-comment"># Clean genome file</span>
bedtools maskfasta -mc N -<span class="hljs-keyword">fi</span> Grif_1614.old.fa -bed Genome.without.contig.merge/contamination.ctg2 -fo Grif_1614-1.fa
bedtools maskfasta -mc M -<span class="hljs-keyword">fi</span> Grif_1614-1.fa -bed Contamination.bed2 -fo Grif_1614-2.fa
sed <span class="hljs-string">'s/M//g'</span> Grif_1614-2.fa &gt; Grif_1614.fa

<span class="hljs-comment"># Extract the CDS and pep sequences</span>
gffread -x Grif_1614.CDS.fa -g Grif_1614.fa Grif_1614.gff3
gffread -y Grif_1614.pep.fa -g Grif_1614.fa Grif_1614.gff3
gffread -w Grif_1614.mRNA.fa -g Grif_1614.fa Grif_1614.gff3

</div></code></pre>
<h2 id="reference-genome-analysis">Reference Genome analysis</h2>
<h3 id="circus-plot-of-s8zhangshugang">Circus plot of S8/Zhangshugang</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/S8/Circos

<span class="hljs-comment"># Prepare genome bed file</span>
grep Super ../pepper_S8.fasta.ann | sed <span class="hljs-string">'s/_/ /g'</span> | awk <span class="hljs-string">'{print $2,$4}'</span> | \
    sed <span class="hljs-string">'s/Superscaffold/chr/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; pepper_S8.genome.bed

<span class="hljs-comment"># Split the genome into 1 Mb window</span>
bedtools makewindows -g pepper_S8.genome.bed -w 500000 | awk <span class="hljs-string">'{print $1,$2+1,$3}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8.win500kb.bed

<span class="hljs-comment"># Calculate the gene numbers within 1 Mb window</span>
awk <span class="hljs-string">'{if ($3=="gene") print $1,$4-1,$5}'</span> ../gff | sed <span class="hljs-string">'s/Chr0/chr/g'</span> | \
    sed <span class="hljs-string">'s/Chr/chr/g'</span> | grep -v chr0. | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    bedtools coverage -a S8.win500kb.bed -b - -counts | \
    sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; S8.gene.Win500kb.bed.txt

<span class="hljs-comment"># Calculate the transposon percentage within 1 Mb window</span>
zcat ../repeat/all.gff.gff3.gz | grep -v unanchor | grep Transposon | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print $1,$4-1,$5}'</span> | \
    sed <span class="hljs-string">'s/Superscaffold/chr/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    bedtools coverage -a S8.win500kb.bed -b - | \
    awk <span class="hljs-string">'{print $1,$2,$3,$7}'</span> | \
    sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; S8.TE.Win500kb.bed.txt

<span class="hljs-comment"># Run Circos</span>
circos -conf circos.conf -debug_group legend

<span class="hljs-comment"># Combine the plots together</span>
smc++ plot combined.pdf \
    ANN/smc.out/model.final.json \
    BAB/smc.out/model.final.json \
    BAP/smc.out/model.final.json \
    CHA/smc.out/model.final.json \
    CHN/smc.out/model.final.json \
    FRU/smc.out/model.final.json \
    GLA1/smc.out/model.final.json \
    GLA2/smc.out/model.final.json \
    PUB/smc.out/model.final.json
</div></code></pre>
<h3 id="genome-synteny-using-mummer">Genome synteny using mummer</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-comment"># Remove unanchorred sequences</span>
head -24 ../References/S8/pepper_S8.fasta | sed <span class="hljs-string">'s/Superscaffold1_54_332676983/Chr01/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold2_27_177319215/Chr02/g'</span> | sed <span class="hljs-string">'s/Superscaffold3_40_289790774/Chr03/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold4_26_248932513/Chr04/g'</span> | sed <span class="hljs-string">'s/Superscaffold5_64_254874144/Chr05/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold6_82_253233553/Chr06/g'</span> | sed <span class="hljs-string">'s/Superscaffold7_54_266382521/Chr07/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold8_30_174326481/Chr08/g'</span> | sed <span class="hljs-string">'s/Superscaffold9_43_278428952/Chr09/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold10_31_210332287/Chr10/g'</span> | sed <span class="hljs-string">'s/Superscaffold11_37_275185330/Chr11/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold12_60_260676116/Chr12/g'</span> &gt; S8.fa

head -24 ../References/CM334/v2.0/CM334.fasta | sed <span class="hljs-string">'s/chr/Chr/g'</span> &gt; CM334.fa
sed -n <span class="hljs-string">'1,44153421p'</span> ../References/Zunla-1/v2.0/Zunla-1.fasta &gt; Zunla-1.fa
sed -n <span class="hljs-string">'1,40892390p'</span> ../References/Chiltepin/v2.0/Chiltepin.fasta | cut -f1 &gt; Chiltepin.fa
head -24 ../References/PI159236/PI159236.fasta &gt; PI159236.fa
sed -n <span class="hljs-string">'1,44535272p'</span> ../References/CM334F1/CM334F1.fasta | sed <span class="hljs-string">'s/chr/Chr0/g'</span> | \
    sed <span class="hljs-string">'s/Chr010/Chr10/g'</span> | sed <span class="hljs-string">'s/Chr011/Chr11/g'</span> | sed <span class="hljs-string">'s/Chr012/Chr12/g'</span> &gt; CM334F1.fa

<span class="hljs-comment"># Use (Mummer)[https://github.com/mummer4/mummer]</span>
wget https://github.com/mummer4/mummer/releases/download/v4.0.0rc1/mummer-4.0.0rc1.tar.gz
tar -xvzf mummer-4.0.0rc1.tar.gz
<span class="hljs-built_in">cd</span> mummer-4.0.0rc1
./configure --prefix=/data/zhaojiantao/tools/mummer-4.0.0rc1/
make
make install

<span class="hljs-comment"># Genome-wide running takes a long time to complete, split the fasta by chromosomes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    nucmer --minmatch=1000 --mincluster=1000 --minalign=1000 -t 10 S8.Chr<span class="hljs-variable">$i</span>.fa Chiltepin.Chr<span class="hljs-variable">$i</span>.fa -p S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer
    delta-filter -r S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.delta -q &gt; S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.filter.delta
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the files</span>
head -2 S8.Chiltepin.Chr01.nucmer.filter.delta | sed <span class="hljs-string">'s/Chr01.//g'</span> | \
    cat - &lt;(cat S8.Chiltepin.Chr*.nucmer.filter.delta | grep -v zhaojiantao | grep -v NUCMER) \
    &gt; S8.Chiltepin.filter.delta

zcat S8.Chiltepin.Chr01.nucmer.filter.delta.gz | head -2 | sed <span class="hljs-string">'s/Chr01.//g'</span> | \
    cat - &lt;(zcat S8.Chiltepin.Chr*.nucmer.filter.delta.gz | grep -v zhaojiantao | grep -v NUCMER) \
    &gt; S8.Chiltepin.filter.delta

<span class="hljs-comment"># Generate the plot</span>
mummerplot --filter S8.Chiltepin.filter.delta --postscript --medium -p S8.Chiltepin.filter.delta

<span class="hljs-comment"># Convert ps to pdf</span>
ps2pdf S8.Chiltepin.filter.delta.ps S8.Chiltepin.filter.delta.pdf
convert -density 300 S8.Chiltepin.filter.delta.pdf S8.Chiltepin.filter.delta.png

<span class="hljs-comment"># Alternatively, generate the plot using (DotPlotly)[https://github.com/tpoorten/dotPlotly/tree/master/example]</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    delta-filter -r S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.delta &gt; S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.filter.delta
    show-coords -c S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.filter.delta &gt; S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.filter.delta.coords
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 2 12); <span class="hljs-keyword">do</span>  
    sed -n <span class="hljs-string">'6,$p'</span> S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.filter.delta.coords &gt; S8.Chiltepin.Chr<span class="hljs-variable">$i</span>.nucmer.filter.delta2.coords
<span class="hljs-keyword">done</span>

cat S8.Chiltepin.Chr01.nucmer.filter.delta.coords S8.Chiltepin.Chr*.nucmer.filter.delta2.coords | \
    sed <span class="hljs-string">'s/S8.Chr01/S8/g'</span> | sed <span class="hljs-string">'s/Chiltepin.Chr01/Chiltepin/g'</span> | \
    sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> &gt; S8.Chiltepin.coords

mummerCoordsDotPlotly.R -i S8.Chiltepin.coords -o S8.Chiltepin.nucmer.plot \
    -m 1000 -q 300000 -k 12 -s -t -p 5 -r 1,2,3,4,5,6,7,8,9,10,11,12

</div></code></pre>
<h3 id="genome-synteny-using-mcscanx">Genome synteny using <a href="https://github.com/wyp1125/MCScanX">MCScanX</a></h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/MCScanX

<span class="hljs-comment"># Use S8 as the reference</span>
<span class="hljs-comment"># Extract the gene id</span>
cat /data/zhaojiantao/pepper/References/S8/gene/pepper_S8.gff3 | grep -v unanchor | \
    sed <span class="hljs-string">'s/;/\t/g'</span> | grep mRNA | awk <span class="hljs-string">'{print "zs"$1,$9,$4,$5}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/Superscaffold//g'</span> &gt; S8.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 -d <span class="hljs-string">' '</span> S8.gene.gff) /data/zhaojiantao/pepper/References/S8/gene/pepper_S8.pep.fa &gt; S8.fa

<span class="hljs-comment"># Create index as the reference</span>
makeblastdb -<span class="hljs-keyword">in</span> S8.fa -dbtype prot -out index/S8 -parse_seqids

<span class="hljs-comment"># Create index as the reference</span>
makeblastdb -<span class="hljs-keyword">in</span> Chiltepin.fa -dbtype prot -out index/Chiltepin -parse_seqids

<span class="hljs-comment"># CA59</span>
cat /data/zhaojiantao/pepper/References/CA59/Ca_59.v1a.gff3 | grep -v scaffold | \
    sed <span class="hljs-string">'s/;/\t/g'</span> | grep mRNA | awk <span class="hljs-string">'{print "ca"$1,$9,$4,$5}'</span> | \
    sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/Ca_59Chr0//g'</span> | sed <span class="hljs-string">'s/Ca_59Chr//g'</span> &gt; CA59.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 -d <span class="hljs-string">' '</span> CA59.gene.gff) \
    /data/zhaojiantao/pepper/References/CA59/Ca_59.protein.fa &gt; CA59.fa

<span class="hljs-comment"># PBC81</span>
zcat /data/zhaojiantao/pepper/References/PBC81/PBC81.chromosome.gff3.gz | \
    grep gene | awk <span class="hljs-string">'{print "pb"$1,$9,$4,$5}'</span> | \
    grep Chr | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
    sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PBC81.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 PBC81.gene.gff) \
    /data/zhaojiantao/pepper/References/PBC81/PBC81.PEP.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; PBC81.fa

<span class="hljs-comment"># Chiltepin</span>
zcat /data/zhaojiantao/pepper/References/Chiltepin/v2.0/Chiltepin.genes.gff.gz | \
    grep mRNA | awk <span class="hljs-string">'{print "ch"$1,$9,$4,$5}'</span> | \
    grep Chr | grep -v Chr00 | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
    sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/;//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chiltepin.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 Chiltepin.gene.gff) \
    /data/zhaojiantao/pepper/References/Chiltepin/v2.0/Chiltepin.PEP.fasta | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; Chiltepin.fa

<span class="hljs-comment"># CM334</span>
cat /data/zhaojiantao/pepper/References/CM334/annotation/Pepper.v.1.55.chr.annotated.gff | \
    grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | \
    awk <span class="hljs-string">'{print "cm"$1,$9,$4,$5}'</span> | grep -v chr00 | \
    sed <span class="hljs-string">'s/Pepper.v.1.55.chr0//g'</span> | sed <span class="hljs-string">'s/Pepper.v.1.55.chr//g'</span> | \
    sed <span class="hljs-string">'s/ID=TC.//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; CM334.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 CM334.gene.gff) \
    /data/zhaojiantao/pepper/References/CM334/annotation/Pepper.v.1.55.PEP.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; CM334.fa

<span class="hljs-comment"># PI159236</span>
zcat /data/zhaojiantao/pepper/References/PI159236/PI159236.chromosome.gff3.gz | \
    grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | grep Chr | \
    awk <span class="hljs-string">'{print "pi"$1,$9,$4,$5}'</span> | sed <span class="hljs-string">'s/ID=TC.//g'</span> | \
    sed <span class="hljs-string">'s/piChr0/pi/g'</span> | sed <span class="hljs-string">'s/piChr/pi/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; PI159236.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 PI159236.gene.gff) \
    /data/zhaojiantao/pepper/References/PI159236/PI159236.PEP.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; PI159236.fa

<span class="hljs-comment"># Zunla-1</span>
cat /data/zhaojiantao/pepper/References/Zunla-1/v2.0/Zunla-1.genes.gff | \
    grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | grep -v Chr00 | \
    awk <span class="hljs-string">'{print "zu"$1,$9,$4,$5}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/zuChr0/zu/g'</span> | \
    sed <span class="hljs-string">'s/zuChr/zu/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Zunla-1.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 Zunla-1.gene.gff) \
    /data/zhaojiantao/pepper/References/Zunla-1/v2.0/Zunla-1.PEP.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; Zunla-1.fa

<span class="hljs-comment"># C. bac. pendulum PI_632928</span>
cat /data/zhaojiantao/pepper/References/PI_632928/PI_632928.gff3 | \
    grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | grep -v Chr00 | \
    awk <span class="hljs-string">'{print "cb"$1,$9,$4,$5}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/cbChr0/cb/g'</span> | \
    sed <span class="hljs-string">'s/cbChr/cb/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; PI_632928.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 PI_632928.gene.gff) \
    /data/zhaojiantao/pepper/References/PI_632928/PI_632928.pep.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; PI_632928.fa
<span class="hljs-comment"># Remove "." from the sequences</span>
perl fasta2rawseq.cgi PI_632928.fa
perl rawseq2fasta_protein.cgi new.fasta &gt; AA
mv AA PI_632928.fa

<span class="hljs-comment"># C.pubescens</span>
cat /data/zhaojiantao/pepper/References/Grif_1614/Grif_1614.gff3 | \
    grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | grep -v Chr00 | \
    awk <span class="hljs-string">'{print "cp"$1,$9,$4,$5}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/cpChr0/cp/g'</span> | \
    sed <span class="hljs-string">'s/cpChr/cp/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Grif_1614.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 Grif_1614.gene.gff) \
    /data/zhaojiantao/pepper/References/Grif_1614/Grif_1614.pep.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; Grif_1614.fa
perl fasta2rawseq.cgi Grif_1614.fa
perl rawseq2fasta_protein.cgi new.fasta &gt; AA
mv AA Grif_1614.fa

<span class="hljs-comment"># XLL</span>
cat /data/zhaojiantao/pepper/References/XLL/XLL.gff3 | \
    grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | grep -v Chr00 | \
    awk <span class="hljs-string">'{print "xl"$1,$9,$4,$5}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/xlChr0/xl/g'</span> | \
    sed <span class="hljs-string">'s/xlChr/xl/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; XLL.gene.gff
<span class="hljs-comment"># Extract the sequences</span>
seqkit grep -j 20 -f &lt;(cut -f2 XLL.gene.gff) \
    /data/zhaojiantao/pepper/References/XLL/XLL.pep.fa | \
    cut -f1 | sed <span class="hljs-string">'s/*//g'</span> &gt; XLL.fa
perl fasta2rawseq.cgi XLL.fa
perl rawseq2fasta_protein.cgi new.fasta &gt; AA
mv AA XLL.fa

<span class="hljs-comment"># Create index as the reference</span>
makeblastdb -<span class="hljs-keyword">in</span> PI_632928.fa -dbtype prot -out index/PI_632928 -parse_seqids
makeblastdb -<span class="hljs-keyword">in</span> Grif_1614.fa -dbtype prot -out index/Grif_1614 -parse_seqids

<span class="hljs-comment"># Chiltepin versus Zunla-1</span>

<span class="hljs-comment"># Align protein sequences to S8</span>
blastp -query PBC81.fa -db index/S8 -out S8_PBC81.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query XLL.fa -db index/S8 -out S8_XLL.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1

blastp -query S8.fa -db index/PI_632928 -out PI_632928_S8.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query S8.fa -db index/Grif_1614 -out Grif_1614_S8.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query Zhangshugang.fa -db index/PI_632928 -out PI_632928_Zhangshugang.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query Zhangshugang.fa -db index/Grif_1614 -out Grif_1614_Zhangshugang.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1

blastp -query CA59.fa -db index/S8 -out S8_CA59.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1

blastp -query Chiltepin.fa -db index/S8 -out S8_Chiltepin.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query CM334.fa -db index/S8 -out S8_CM334.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query PI159236.fa -db index/S8 -out S8_PI159236.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query Zunla-1.fa -db index/S8 -out S8_Zunla-1.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1

blastp -query Chiltepin.fa -db index/Zunla-1 -out Zunla-1_Chiltepin.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query Chiltepin.fa -db index/CM334 -out CM334_Chiltepin.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query PBC81.fa -db index/CM334 -out CM334_PBC81.blast -evalue 1e-10 -num_threads 30 -outfmt 6 -max_target_seqs 5
blastp -query Chiltepin.fa -db index/PI159236 -out PI159236_Chiltepin.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5

blastp -query Chiltepin.fa -db index/PBC81 -out PBC81_Chiltepin.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query CA59.fa -db index/PBC81 -out PBC81_CA59.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query CM334.fa -db index/PBC81 -out PBC81_CM334.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query Zunla-1.fa -db index/PBC81 -out PBC81_Zunla-1.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query PI159236.fa -db index/PBC81 -out PBC81_PI159236.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5

blastp -query CA59.fa -db index/Chiltepin -out Chiltepin_CA59.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query CM334.fa -db index/Chiltepin -out Chiltepin_CM334.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query Zunla-1.fa -db index/Chiltepin -out Chiltepin_Zunla-1.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query PI159236.fa -db index/Chiltepin -out Chiltepin_PI159236.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query PI_632928.fa -db index/Chiltepin -out Chiltepin_PI_632928.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5
blastp -query PI_632928.fa -db index/Grif_1614 -out Grif_1614_PI_632928.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 1
blastp -query Grif_1614.fa -db index/Chiltepin -out Chiltepin_Grif_1614.blast -evalue 1e-10 -num_threads 60 -outfmt 6 -max_target_seqs 5

<span class="hljs-comment"># Prepare the gene infos</span>
cat S8.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_PBC81.gff
cat S8.gene.gff XLL.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_XLL.gff

cat Grif_1614.gene.gff PI_632928.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Grif_1614_PI_632928.gff
cat Chiltepin.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_PBC81.gff
cat Chiltepin.gene.gff PI_632928.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_PI_632928.gff
cat Chiltepin.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_PBC81.gff

cat CA59.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PBC81_CA59.gff
cat CM334.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PBC81_CM334.gff
cat Zunla-1.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PBC81_Zunla-1.gff
cat PI159236.gene.gff PBC81.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PBC81_PI159236.gff

cat CA59.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_CA59.gff
cat CM334.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_CM334.gff
cat Zunla-1.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_Zunla-1.gff
cat PI159236.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_PI159236.gff
cat PI159236.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_PI159236.gff
cat PI_632928.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_PI_632928.gff
cat Grif_1614.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chiltepin_Grif_1614.gff

cat S8.gene.gff CA59.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_CA59.gff
cat S8.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_Chiltepin.gff
cat S8.gene.gff CM334.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_CM334.gff
cat S8.gene.gff PI159236.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_PI159236.gff
cat S8.gene.gff Zunla-1.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; S8_Zunla-1.gff
cat Zunla-1.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Zunla-1_Chiltepin.gff
cat CM334.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; CM334_Chiltepin.gff
cat PI159236.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PI159236_Chiltepin.gff
cat PBC81.gene.gff Chiltepin.gene.gff | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PBC81_Chiltepin.gff

<span class="hljs-comment"># MCScan</span>
MCScanX S8_PBC81 -a
MCScanX S8_XLL -a

MCScanX S8_PI_632928 -a
MCScanX S8_Grif_1614 -a
MCScanX PI_632928_S8 -a
MCScanX Grif_1614_S8 -a
MCScanX PI_632928_Zhangshugang -a
MCScanX Grif_1614_Zhangshugang -a
MCScanX Grif_1614_PI_632928 -a

MCScanX S8_CA59 -a
MCScanX S8_Chiltepin -a
MCScanX S8_CM334 -a
MCScanX S8_PI159236 -a
MCScanX S8_Zunla-1 -a
MCScanX Zunla-1_Chiltepin -a
MCScanX CM334_Chiltepin -a
MCScanX PI159236_Chiltepin -a

MCScanX PBC81_Chiltepin -a
MCScanX PBC81_CA59 -a
MCScanX PBC81_CM334 -a
MCScanX PBC81_Zunla-1 -a
MCScanX PBC81_PI159236 -a

MCScanX Chiltepin_CA59 -a
MCScanX Chiltepin_CM334 -a
MCScanX Chiltepin_Zunla-1 -a
MCScanX Chiltepin_PI159236 -a
MCScanX Chiltepin_PI_632928 -a
MCScanX Chiltepin_Grif_1614 -a

<span class="hljs-comment"># Generate the plot</span>
<span class="hljs-comment"># Important: modify the control file from the program folder instead of creating one using vim</span>
<span class="hljs-comment"># S8 versus PBC81</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_PBC81.gff \
    -s S8_PBC81.collinearity \
    -c S8_PBC81.dot.ctl \
    -o S8_PBC81.png

<span class="hljs-comment"># S8 versus XLL</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_XLL.gff \
    -s S8_XLL.collinearity \
    -c S8_XLL.dot.ctl \
    -o S8_XLL.png

<span class="hljs-comment"># Zhangshugang versus PI_632928</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PI_632928_Zhangshugang.gff \
    -s PI_632928_Zhangshugang.collinearity \
    -c PI_632928_Zhangshugang.dot.ctl \
    -o PI_632928_Zhangshugang.png

<span class="hljs-comment"># Zhangshugang vs Grif_1614</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Grif_1614_Zhangshugang.gff \
    -s Grif_1614_Zhangshugang.collinearity \
    -c Grif_1614_Zhangshugang.dot.ctl \
    -o Grif_1614_Zhangshugang.png
    
<span class="hljs-comment"># Grif_1614 versus PI_632928</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Grif_1614_PI_632928.gff \
    -s Grif_1614_PI_632928.collinearity \
    -c Grif_1614_PI_632928.dot.ctl \
    -o Grif_1614_PI_632928.png

<span class="hljs-comment"># PI_632928 vs S8</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PI_632928_S8.gff \
    -s PI_632928_S8.collinearity \
    -c PI_632928_S8.dot.ctl \
    -o PI_632928_S8.png

<span class="hljs-comment"># S8 versus Grif_1614</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_Grif_1614.gff \
    -s S8_Grif_1614.collinearity \
    -c S8_Grif_1614.dot.ctl \
    -o S8_Grif_1614.png

<span class="hljs-comment"># Grif_1614 vs S8</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Grif_1614_S8.gff \
    -s Grif_1614_S8.collinearity \
    -c Grif_1614_S8.dot.ctl \
    -o Grif_1614_S8.png

<span class="hljs-comment"># S8 versus CA59</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_CA59.gff \
    -s S8_CA59.collinearity \
    -c S8_CA59.dot.ctl \
    -o S8_CA59.png

<span class="hljs-comment"># S8 versus Chiltepin</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_Chiltepin.gff \
    -s S8_Chiltepin.collinearity \
    -c S8_Chiltepin.dot.ctl \
    -o S8_Chiltepin.png

<span class="hljs-comment"># S8 versus CM334</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_CM334.gff \
    -s S8_CM334.collinearity \
    -c S8_CM334.dot.ctl \
    -o S8_CM334.png

<span class="hljs-comment"># S8 versus PI159236</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_PI159236.gff \
    -s S8_PI159236.collinearity \
    -c S8_PI159236.dot.ctl \
    -o S8_PI159236.png

<span class="hljs-comment"># S8 versus Zunla-1</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g S8_Zunla-1.gff \
    -s S8_Zunla-1.collinearity \
    -c S8_Zunla-1.dot.ctl \
    -o S8_Zunla-1.png

<span class="hljs-comment"># PBC81 versus CA59</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PBC81_CA59.gff \
    -s PBC81_CA59.collinearity \
    -c PBC81_CA59.dot.ctl \
    -o PBC81_CA59.png

<span class="hljs-comment"># PBC81 versus CM334</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PBC81_CM334.gff \
    -s PBC81_CM334.collinearity \
    -c PBC81_CM334.dot.ctl \
    -o PBC81_CM334.png

<span class="hljs-comment"># PBC81 versus Zunla-1</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PBC81_Zunla-1.gff \
    -s PBC81_Zunla-1.collinearity \
    -c PBC81_Zunla-1.dot.ctl \
    -o PBC81_Zunla-1.png

<span class="hljs-comment"># PBC81 versus PI159236</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PBC81_PI159236.gff \
    -s PBC81_PI159236.collinearity \
    -c PBC81_PI159236.dot.ctl \
    -o PBC81_PI159236.png

<span class="hljs-comment"># PBC81 versus Chiltepin</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g PBC81_Chiltepin.gff \
    -s PBC81_Chiltepin.collinearity \
    -c PBC81_Chiltepin.dot.ctl \
    -o PBC81_Chiltepin.png

<span class="hljs-comment"># Chiltepin versus CA59</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Chiltepin_CA59.gff \
    -s Chiltepin_CA59.collinearity \
    -c Chiltepin_CA59.dot.ctl \
    -o Chiltepin_CA59.png

<span class="hljs-comment"># Chiltepin versus Zunla-1</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Chiltepin_Zunla-1.gff \
    -s Chiltepin_Zunla-1.collinearity \
    -c Chiltepin_Zunla-1.dot.ctl \
    -o Chiltepin_Zunla-1.png

<span class="hljs-comment"># Chiltepin versus PI159236</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Chiltepin_PI159236.gff \
    -s Chiltepin_PI159236.collinearity \
    -c Chiltepin_PI159236.dot.ctl \
    -o Chiltepin_PI159236.png

<span class="hljs-comment"># Chiltepin versus PI_632928</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Chiltepin_PI_632928.gff \
    -s Chiltepin_PI_632928.collinearity \
    -c Chiltepin_PI_632928.dot.ctl \
    -o Chiltepin_PI_632928.png

<span class="hljs-comment"># Chiltepin versus Grif_1614</span>
java /data/zhaojiantao/tools/MCScanX-master/downstream_analyses/dot_plotter.java \
    -g Chiltepin_Grif_1614.gff \
    -s Chiltepin_Grif_1614.collinearity \
    -c Chiltepin_Grif_1614.dot.ctl \
    -o Chiltepin_Grif_1614.png

<span class="hljs-comment"># Use MCsan python version</span>
<span class="hljs-comment"># Prepare the gene bed file</span>
<span class="hljs-comment"># S8</span>
sed <span class="hljs-string">'s/;/\t/g'</span> /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff | \
    awk <span class="hljs-string">'$3=="mRNA" {print $1,$4,$5,$9,0,$7}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Zhangshugang.bed
<span class="hljs-comment"># CA59</span>
sed <span class="hljs-string">'s/;/\t/g'</span> /data/zhaojiantao/pepper/References/CA59/Ca_59.v1a.gff3 | \
    awk <span class="hljs-string">'$3=="mRNA" {print $1,$4,$5,$9,0,$7}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    sed <span class="hljs-string">'s/Ca_59Chr/Chr/g'</span> &gt; CA59.bed

<span class="hljs-comment"># Prepare the cds file</span>
<span class="hljs-comment"># NCEBR</span>
sed <span class="hljs-string">'s/mRNA://g'</span> /data/zhaojiantao/pepper/References/S8/Zhangshugang.cds.fasta &gt; Zhangshugang.cds

<span class="hljs-comment"># Pimpi</span>
sed <span class="hljs-string">'s/mRNA://g'</span> /data/zhaojiantao/pepper/References/CA59/Ca_59.cds.fa &gt; CA59.cds 

<span class="hljs-comment"># Run MCScan</span>
python -m jcvi.compara.catalog ortholog Zhangshugang CA59 --no_strip_names

</div></code></pre>
<h3 id="map-the-chiltepin-reads-to-the-chiltepin-genome-to-check-the-read-count">Map the Chiltepin reads to the Chiltepin genome to check the read count</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Download the mate-paire raw Reads from [NCBI](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SAMN01906552&amp;o=acc_s%3Aa)</span>
<span class="hljs-comment"># SRR653457 has the largest insert size</span>
fasterq-dump -p SRR653457

<span class="hljs-comment"># Use NxTrim to remove mate-pair </span>
git <span class="hljs-built_in">clone</span> https://github.com/sequencing/NxTrim.git
<span class="hljs-built_in">cd</span> NxTrim
make

<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/Chiltepin/v2.0/SRA
<span class="hljs-comment"># Extract the region</span>
samtools faidx ../Chiltepin.Chr01.fasta
samtools faidx ../Chiltepin.Chr01.fasta Chr01:198941948-199029039 &gt; Chr01.region.fasta
bwa index Chr01.region.fasta

<span class="hljs-comment"># Trimm the adaptors</span>
nxtrim -1 SRR653457_1.fastq -2 SRR653457_2.fastq -O SRR653457
<span class="hljs-comment"># Reverse complementary sequences using [fastx_tookit](https://github.com/Debian/fastx-toolkit)</span>
fastx_reverse_complement -z -i &lt;(zcat SRR653457_R2.unknown.fastq.gz) -o SRR653457_R2.unknown.reverse.fastq.gz

<span class="hljs-comment"># Combine the mate-pare and read pairs had unknown orientation</span>
zcat SRR653457_R1.mp.fastq.gz SRR653457_R1.unknown.fastq.gz | gzip &gt; SRR653457_R1.mp.unknown.fastq.gz
zcat SRR653457_R2.mp.fastq.gz SRR653457_R2.unknown.reverse.fastq.gz | gzip &gt; SRR653457_R2.mp.unknown.fastq.gz
bwa aln -e 1 -t 80 ../Chiltepin.Chr01.fasta test_R1.unknown.fastq.gz &gt; test_R1.sai
bwa aln -e 1 -t 80 ../Chiltepin.Chr01.fasta test_R2.unknown.reverse.fastq.gz &gt; test_R2.sai


<span class="hljs-comment"># Map the read to the references</span>
bwa aln -e -1 -t 80 -n 1 ../Chiltepin.Chr01.fasta SRR653457.mp.uk.fastq.gz -f SRR653457.sai
bwa sampe -f SRR653457.sam ../Chiltepin.Chr01.fasta SRR653457.sai SRR653457.mp.uk.fastq.gz
samtools view -Shu SRR653457.sam | samtools sort -O BAM -@ 80 &gt; SRR653457.sort.bam

<span class="hljs-comment"># Extract the region</span>
samtools view -b SRR653457.sort.bam <span class="hljs-string">"Chr01:198942247-199028668"</span> &gt; Chr01.region.bam
samtools index Chr01.region.bam

head -100000 SRR653457_1.fastq &gt; test_1.fastq
head -100000 SRR653457_2.fastq &gt; test_2.fastq
nxtrim -1 test_1.fastq -2 test_2.fastq --separate -O <span class="hljs-built_in">test</span>
zcat SRR653457_R1.unknown.fastq.gz | head -1000 | gzip &gt; test_R1.unknown.fastq.gz
zcat SRR653457_R2.unknown.fastq.gz | head -1000 &gt; test_R2.unknown.fastq
fastx_reverse_complement -z -i test_R2.unknown.fastq -o test_R2.unknown.reverse.fastq.gz

bwa aln -e 1 -t 80 ../Chiltepin.Chr01.fasta test_R1.unknown.fastq.gz &gt; test_R1.sai
bwa aln -e 1 -t 80 ../Chiltepin.Chr01.fasta test_R2.unknown.reverse.fastq.gz &gt; test_R2.sai
bwa sampe ../Chiltepin.Chr01.fasta test_R1.sai test_R2.sai test_R1.unknown.fastq.gz test_R2.unknown.reverse.fastq.gz &gt; test.sam
samtools view -Shu test.sam | samtools sort -O BAM -@ 80 &gt; test.sort.bam

</div></code></pre>
<h3 id="align-sequence-to-s8">Align sequence to S8</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/S8

<span class="hljs-comment"># Install blast</span>
wget https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.11.0/ncbi-blast-2.11.0+-x64-linux.tar.gz
tar -xvzf ncbi-blast-2.11.0+-x64-linux.tar.gz
mv ncbi-blast-2.11.0+ blast-2.11.0+
<span class="hljs-built_in">export</span> PATH=<span class="hljs-string">"/data/zhaojiantao/tools/blast-2.11.0+/bin:<span class="hljs-variable">$PATH</span>"</span>

<span class="hljs-comment"># Build S8 database</span>
<span class="hljs-comment"># Nuclei database</span>
makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta \
    -dbtype nucl -parse_seqids \
    -out /data/zhaojiantao/pepper/References/S8/blastdb/index

makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \
    -dbtype nucl -parse_seqids \
    -out /data/zhaojiantao/pepper/References/S8/blastdb/Zhangshugang.index

makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/S8/Zhangshugang.cds.fa \
    -dbtype nucl -parse_seqids \
    -out /data/zhaojiantao/pepper/References/S8/blastdb/S8.cds.index

makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/CM334/assemblies/Pepper_1.55.fasta \
    -dbtype nucl -parse_seqids \
    -out /data/zhaojiantao/pepper/References/CM334/assemblies/blastdb/index
makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/Zunla-1/v1.0/Zunla-1.fasta \
    -dbtype nucl -parse_seqids \
    -out /data/zhaojiantao/pepper/References/Zunla-1/v1.0/blastdb/index

<span class="hljs-comment"># Protein database</span>
makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/CM334/annotation/Pepper.v.1.55.PEP.fa \
    -dbtype prot -parse_seqids -out /data/zhaojiantao/pepper/References/CM334/annotation/blastdb/index

makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/CM334/v2.0/CM334.PEP.fa \
    -dbtype prot -parse_seqids -out /data/zhaojiantao/pepper/References/CM334/v2.0/blastdb/index

makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/S8/gene/pepper_S8.pep.fa \
    -dbtype prot -parse_seqids -out /data/zhaojiantao/pepper/References/S8/gene/blastdb/index  
makeblastdb -<span class="hljs-keyword">in</span> /data/zhaojiantao/pepper/References/Zunla-1/v1.0/Zunla-1.pep.fa \
    -dbtype prot -parse_seqids -out /data/zhaojiantao/pepper/References/Zunla-1/v1.0/blastdb_protein/index

<span class="hljs-comment"># Blast the position on CM334</span>
blastn -query AT3-2.fa \
    -db /data/zhaojiantao/pepper/References/CM334/assemblies/blastdb/index \
    -evalue 1e-20 -num_threads 6 \
    -out AT3-2.CM334.blast

blastn -query P1.306622565.fa \
    -db /data/zhaojiantao/pepper/References/S8/blastdb/index \
    -evalue 1e-20 -num_threads 6 \
    -out P1.306622565.start.blast

blastn -query pATM.fa \
    -db /data/zhaojiantao/pepper/References/Zunla-1/v1.0/blastdb/index \
    -evalue 1e-20 -num_threads 6 \
    -out ZL1_1826.Zunla-1.blast

blastp -query XP_016580728.1.fa \
    -db /data/zhaojiantao/pepper/References/CM334/annotation/blastdb/index \
    -evalue 1e-20 -num_threads 6 \
    -out XP_016580728.CM334.pep.blast

blastp -query Ca9g33797.pro.fa \
    -db /data/zhaojiantao/pepper/References/CM334/v2.0/blastdb/index \
    -evalue 1e-20 -num_threads 6 \
    -out Ca9g33797.pro.blast

<span class="hljs-comment"># Find the gene ID</span>
grep Ca2g12681 /data/zhaojiantao/pepper/References/S8/gene/pepper_S8.cds.fa | sed <span class="hljs-string">'s/&gt;//g'</span> &gt; gene.id
grep CA11g14610 /data/zhaojiantao/pepper/References/CM334/annotation/Pepper.v.1.55.PEP.fa | sed <span class="hljs-string">'s/&gt;//g'</span> &gt; gene.id
seqtk subseq /data/zhaojiantao/pepper/References/CM334/annotation/Pepper.v.1.55.PEP.fa gene.id &gt; CA11g14610.CM334.fa
grep CA04g00860 /data/zhaojiantao/pepper/References/CM334/annotation/Pepper_1.55.gene_models.gff3

grep CA04g00860 /data/zhaojiantao/pepper/References/S8/gene/pepper_S8.gff3

seqtk subseq \
    /data/zhaojiantao/pepper/References/Zunla-1/v1.0/Zunla-1.cds.fasta \
    Zunla.pungency.genes.list \
    &gt; Zunla.pungency.genes.cds.fa

<span class="hljs-comment"># Split the fasta</span>
awk <span class="hljs-string">'/^&gt;/{s=++d".fasta"} {print &gt; s}'</span> Zunla.pungency.genes.cds.fa

<span class="hljs-comment"># Blast for each gene</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 154); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># to S8</span>
    blastn -query <span class="hljs-variable">$i</span>.fasta \
        -db /data/zhaojiantao/pepper/References/S8/blastdb/index \
        -evalue 1e-20 -num_threads 6 \
        -out <span class="hljs-variable">$i</span>.blast
    <span class="hljs-comment"># to cds</span>
    blastn -query <span class="hljs-variable">$i</span>.fasta \
        -db /data/zhaojiantao/pepper/References/S8/blastdb/S8.cds.index \
        -evalue 1e-20 -num_threads 60 -outfmt 6 \
        -out <span class="hljs-variable">$i</span>.blast3    
<span class="hljs-keyword">done</span>

<span class="hljs-comment">#----------------------------------------------------------------------------------------------------</span>
<span class="hljs-comment"># For the comparison of Cao et al. 2022</span>



<span class="hljs-comment"># Find genome synteny using [mcscan](https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version))</span>
<span class="hljs-comment"># Convert the GFF to BED file and rename them.</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/MCScanX

cut -f1 ../CM334/v2.0/Annuum.v.2.0.gff3 | sed <span class="hljs-string">'s/PGAv.1.6.scaffold/chr/g'</span> | \
    paste - ../CM334/v2.0/Annuum.v.2.0.gff3 | grep gene | \
    awk <span class="hljs-string">'{print $1,$5,$6,$10,$7,$8}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; CM334v2.bed
    
zcat ../S8/pepper_S8.gff3.gz | sed <span class="hljs-string">'s/;/\t/g'</span> | grep ID=gene | \
    awk <span class="hljs-string">'{print $1,$4,$5,$9,$6,$7}'</span> | sed <span class="hljs-string">'s/Superscaffold/chr/g'</span> | \
    sed <span class="hljs-string">'s/ID=gene-//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; S8.bed

<span class="hljs-comment"># Reformat fasta files</span>
python -m jcvi.formats.fasta format ../CM334/v2.0/Annuum.v.2.0.CDS.fa.gz CM334v2.cds
python -m jcvi.formats.fasta format ../S8/gene/pepper_S8.cds.fa S8.cds
sed <span class="hljs-string">'s/\.1//g'</span> S8.cds | sed <span class="hljs-string">'s/rna-//g'</span> &gt; S8.cds2
rm S8.cds
mv S8.cds2 S8.cds

<span class="hljs-comment"># Pairwise synteny search</span>
python -m jcvi.compara.catalog ortholog CM334v2 S8 --no_strip_names
python -m jcvi.compara.catalog ortholog CM334v2 S8 --cscore=.99 --no_strip_names
python -m jcvi.compara.synteny screen --minspan=30 --simple CM334v2.S8.anchors CM334v2.S8.anchors.new 

</div></code></pre>
<h2 id="vg-pangenome">vg pangenome</h2>
<h3 id="trimmomatic">Trimmomatic</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/00fastq

<span class="hljs-comment"># Trim adaptors and low quality reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat sample.list) ; <span class="hljs-keyword">do</span>
    trimmomatic PE <span class="hljs-variable">$i</span>.R1.fastq.gz <span class="hljs-variable">$i</span>.R2.fastq.gz \
        <span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>.R1U.fq \
        <span class="hljs-variable">$i</span>.R2P.fq <span class="hljs-variable">$i</span>.R2U.fq \
        -threads 60 \
        ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
        SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 ;
    <span class="hljs-comment"># Remove temporary files</span>
    rm <span class="hljs-variable">$i</span>.R1.fastq.gz <span class="hljs-variable">$i</span>.R2.fastq.gz <span class="hljs-variable">$i</span>.R1U.fq <span class="hljs-variable">$i</span>.R2U.fq
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># From the cornell server</span>
<span class="hljs-comment"># Trim adaptors and low quality reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> bailajiao ; <span class="hljs-keyword">do</span>
    trimmomatic PE <span class="hljs-variable">$i</span>.R1.fastq.gz <span class="hljs-variable">$i</span>.R2.fastq.gz \
        /workdir/zf25/01fastq.clean/<span class="hljs-variable">$i</span>.R1P.fq /workdir/zf25/01fastq.clean/<span class="hljs-variable">$i</span>.R1U.fq \
        /workdir/zf25/01fastq.clean/<span class="hljs-variable">$i</span>.R2P.fq /workdir/zf25/01fastq.clean/<span class="hljs-variable">$i</span>.R2U.fq \
        -threads 60 \
        ILLUMINACLIP:/home/zf25/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
        SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 ;
    <span class="hljs-comment"># Remove temporary files</span>
    rm /workdir/zf25/01fastq.clean/<span class="hljs-variable">$i</span>.R1U.fq /workdir/zf25/01fastq.clean/<span class="hljs-variable">$i</span>.R2U.fq
<span class="hljs-keyword">done</span>

nohup sh trim.sh 1&gt;trim.log1 2&gt;trim.log &amp;
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="sv-estimation">SV estimation</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/PanReference

<span class="hljs-comment"># PI_632928</span>
grep -v <span class="hljs-string">"#"</span> m6a.4gir.vcf | grep PASS | cut -f4,5,10 | grep -v <span class="hljs-string">"\."</span> | awk <span class="hljs-string">'{print length($1),length($2)}'</span> &gt; PI_632928.alt.length

<span class="hljs-comment"># Grif_1614</span>
grep -v <span class="hljs-string">"#"</span> m6a.4gir.vcf | grep PASS | cut -f4,5,11 | grep -v <span class="hljs-string">"\."</span> | awk <span class="hljs-string">'{print length($1),length($2)}'</span>  &gt; Grif_1614.alt.length

<span class="hljs-comment"># Calculate SV &lt;=10k</span>
awk <span class="hljs-string">'$1&lt;=10000 {sum += $7 } END { print sum }'</span> PI_632928.alt.length

</div></code></pre>
<h3 id="vg-mapping">vg mapping</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/01bam

<span class="hljs-comment"># vg mapping</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PI_355813 PI_355815 PI_355817 PI_355819 PI_355820 PI_357467 PI_358811 PI_358968; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Trim</span>
    trimmomatic PE ../00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz \
    ../00fastq/<span class="hljs-variable">$i</span>.R1P.fq ../00fastq/<span class="hljs-variable">$i</span>.R1U.fq \
    ../00fastq/<span class="hljs-variable">$i</span>.R2P.fq ../00fastq/<span class="hljs-variable">$i</span>.R2U.fq \
    -threads 60 \
    ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
    SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 2&gt;../00fastq/<span class="hljs-variable">$i</span>.trim.log
    <span class="hljs-comment"># Remove temporary files</span>
    rm ../00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R1U.fq ../00fastq/<span class="hljs-variable">$i</span>.R2U.fq
    <span class="hljs-comment"># Generate gam file</span>
    vg giraffe -t 60 \
        -Z m6a.giraffe.gbz \
        -m m6a.min -d m6a.dist \
        -f ../00fastq/<span class="hljs-variable">$i</span>.R1P.fq \
        -f ../00fastq/<span class="hljs-variable">$i</span>.R2P.fq \
        &gt; <span class="hljs-variable">$i</span>.gam
    <span class="hljs-comment"># Generate bam file</span>
    vg surject -t 60 \
        -N <span class="hljs-variable">$i</span> \
        -R <span class="hljs-variable">$i</span> \
        -x m6a.giraffe.gbz \
        -b -i <span class="hljs-variable">$i</span>.gam \
        &gt; <span class="hljs-variable">$i</span>.gam.bam
    <span class="hljs-comment"># Sort the bam</span>
    perl sort_bam_header_chrID.pl <span class="hljs-variable">$i</span> <span class="hljs-variable">$i</span>.gam.bam <span class="hljs-variable">${i}</span>_gam.bam
    <span class="hljs-comment"># Index the bam</span>
 	samtools index -@ 60 <span class="hljs-variable">${i}</span>_gam.bam
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="vg-mapping-on-cornell-server">vg mapping on cornell server</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /workdir/zf25

<span class="hljs-comment"># vg mapping</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat list) ; <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-string">"
date
trimmomatic PE /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz \
    /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R1P.fq /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R1U.fq \
    /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R2P.fq /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R2U.fq \
    -threads 110 \
    ILLUMINACLIP:/home/zf25/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
    SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 2&gt;/workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.trim.log

rm /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R1U.fq /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R2U.fq
rm /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz /workdir/zf25/00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz

vg giraffe -t 110 \
    -Z /workdir/zf25/db/m6a.giraffe.gbz \
    -m /workdir/zf25/db/m6a.min \
    -d /workdir/zf25/db/m6a.dist \
    -f 00fastq/<span class="hljs-variable">$i</span>.R1P.fq \
    -f 00fastq/<span class="hljs-variable">$i</span>.R2P.fq \
    &gt; <span class="hljs-variable">$i</span>.gam

vg surject -t 110 \
    -N <span class="hljs-variable">$i</span> \
    -R <span class="hljs-variable">$i</span> \
    -x /workdir/zf25/db/m6a.giraffe.gbz \
    -b -i <span class="hljs-variable">$i</span>.gam \
    &gt; <span class="hljs-variable">$i</span>.gam.bam

perl /home/zf25/sort_bam_header_chrID.pl <span class="hljs-variable">$i</span> <span class="hljs-variable">$i</span>.gam.bam <span class="hljs-variable">${i}</span>_gam.bam

samtools index -@ 110 <span class="hljs-variable">${i}</span>_gam.bam
date "</span> 
<span class="hljs-keyword">done</span> | sed <span class="hljs-string">'s/         / /g'</span> | sed <span class="hljs-string">'s/     / /g'</span> &gt; batch6.sh

</div></code></pre>
<h3 id="snp-calling-using-sentieon">SNP calling using sentieon</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/02gvcf

<span class="hljs-comment"># Run sentieon</span>
<span class="hljs-comment"># Call variants</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat list) ; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Collect read information to remove or mark duplicates</span>
    sentieon driver -t 100 -i ../01bam/<span class="hljs-variable">${i}</span>_gam.bam \
        --algo LocusCollector --fun score_info <span class="hljs-variable">$i</span>.score.gz
    <span class="hljs-comment"># Remove or mark duplicates</span>
    sentieon driver -t 100 -i ../01bam/<span class="hljs-variable">${i}</span>_gam.bam \
        --algo Dedup --score_info <span class="hljs-variable">$i</span>.score.gz \
        --metrics <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam
    <span class="hljs-comment"># Call gvcf</span>
    sentieon driver -t 100 \
        -r ../PanReference/Cann_Zhangshugang.chr.fa \
        -i <span class="hljs-variable">$i</span>.deduped.bam \
        --algo Haplotyper \
        --emit_mode gvcf \
        <span class="hljs-variable">$i</span>.gvcf.gz
    <span class="hljs-comment"># Remove unnecessary files</span>
    rm <span class="hljs-variable">$i</span>.score.gz* <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam*
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="snp-calling-using-sentieon-for-tomato-as-outgroup">SNP calling using sentieon for tomato as outgroup</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/02gvcf

<span class="hljs-comment"># Run sentieon</span>
<span class="hljs-comment"># Download tomato from NCBI as outgroup</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> ERR418102 ERR418103 ERR418107 ERR418106; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Download SRA</span>
    prefetch <span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Convert SRA to fastq</span>
    fasterq-dump --split-files  <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.sra
    <span class="hljs-comment"># Remove adaptor and low-quality sequences</span>
    trimmomatic PE <span class="hljs-variable">${i}</span>_1.fastq <span class="hljs-variable">${i}</span>_2.fastq \
            <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1U.fq \
            <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2U.fq \
            -threads 100 \
            ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
            SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40
    <span class="hljs-comment"># Remove temporary file</span>
    rm <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.sra <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1U.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2U.fq <span class="hljs-variable">${i}</span>_1.fastq <span class="hljs-variable">${i}</span>_2.fastq 
    <span class="hljs-comment"># vg mapping</span>
    vg giraffe -t 100 \
        -Z /workdir/zf25/db/m6a.giraffe.gbz \
        -m /workdir/zf25/db/m6a.min \
        -d /workdir/zf25/db/m6a.dist \
        -f 00fastq/<span class="hljs-variable">$i</span>.R1P.fq \
        -f 00fastq/<span class="hljs-variable">$i</span>.R2P.fq \
        &gt; <span class="hljs-variable">$i</span>.gam
    <span class="hljs-comment"># Generate bam</span>
    vg surject -t 100 \
        -N <span class="hljs-variable">$i</span> \
        -R <span class="hljs-variable">$i</span> \
        -x /workdir/zf25/db/m6a.giraffe.gbz \
        -b -i <span class="hljs-variable">$i</span>.gam \
        &gt; <span class="hljs-variable">$i</span>.gam.bam
    <span class="hljs-comment"># Sort the bam file</span>
    perl /home/zf25/sort_bam_header_chrID.pl <span class="hljs-variable">$i</span> <span class="hljs-variable">$i</span>.gam.bam <span class="hljs-variable">${i}</span>_gam.bam
    <span class="hljs-comment"># Index the bam</span>
    samtools index -@ 110 <span class="hljs-variable">${i}</span>_gam.bam        
    <span class="hljs-comment"># Call variants</span>
    <span class="hljs-comment"># Collect read information to remove or mark duplicates</span>
    sentieon driver -t 100 -i ../01bam/<span class="hljs-variable">${i}</span>_gam.bam \
        --algo LocusCollector --fun score_info <span class="hljs-variable">$i</span>.score.gz
    <span class="hljs-comment"># Remove or mark duplicates</span>
    sentieon driver -t 100 -i ../01bam/<span class="hljs-variable">${i}</span>_gam.bam \
        --algo Dedup --score_info <span class="hljs-variable">$i</span>.score.gz \
        --metrics <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam
    <span class="hljs-comment"># Call gvcf</span>
    sentieon driver -t 100 \
        -r ../PanReference/Cann_Zhangshugang.chr.fa \
        -i <span class="hljs-variable">$i</span>.deduped.bam \
        --algo Haplotyper \
        --emit_mode gvcf \
        <span class="hljs-variable">$i</span>.gvcf.gz
    <span class="hljs-comment"># Remove unnecessary files</span>
    rm <span class="hljs-variable">$i</span>.score.gz* <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam*
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="snp-calling-using-sentieon-on-cornell-server">SNP calling using sentieon on cornell server</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># work directory</span>
<span class="hljs-built_in">cd</span> /workdir/zf25/02gvcf

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> bailajiao ; <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-string">"
    # Collect read information to remove or mark duplicates
    /programs/sentieon-genomics-202112/bin/sentieon driver -t 110 -i ../01bam/<span class="hljs-variable">${i}</span>_gam.bam \
        --algo LocusCollector --fun score_info <span class="hljs-variable">$i</span>.score.gz
    # Remove or mark duplicates
    /programs/sentieon-genomics-202112/bin/sentieon driver -t 110 -i ../01bam/<span class="hljs-variable">${i}</span>_gam.bam \
        --algo Dedup --score_info <span class="hljs-variable">$i</span>.score.gz \
        --metrics <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam
    # Call gvcf
    /programs/sentieon-genomics-202112/bin/sentieon driver -t 110 \
        -r /home/zf25/PanReference/Cann_Zhangshugang.chr.fa \
        -i <span class="hljs-variable">$i</span>.deduped.bam \
        --algo Haplotyper \
        --emit_mode gvcf \
        <span class="hljs-variable">$i</span>.gvcf.gz
    # Remove unnecessary files
    rm <span class="hljs-variable">$i</span>.score.gz* <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam* "</span>
<span class="hljs-keyword">done</span> | sed <span class="hljs-string">'s/         / /g'</span> | sed <span class="hljs-string">'s/     / /g'</span> | sed <span class="hljs-string">'s/    //g'</span> &gt; batch1.sh

</div></code></pre>
<h3 id="join-the-vcf-file">Join the vcf file</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/02gvcf

<span class="hljs-comment"># Joint variant call</span>
sentieon driver -t 104 \
    -r /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa --algo GVCFtyper \
    ../03vcf/500.vcf.gz *gvcf.gz

</div></code></pre>
<h3 id="extract-the-high-quality-snps-and-indels">Extract the high quality SNPs and Indels</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/03vcf

<span class="hljs-comment"># Create the FASTA sequence directory file of the reference to be used in gatk</span>
java -jar /data/zhaojiantao/tools/picard/build/libs/picard.jar CreateSequenceDictionary \
    R=/data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
    O=/data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.dict

<span class="hljs-comment"># Split the file into chromosomes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    bcftools view 500.vcf.gz --regions CannZSG_Chr<span class="hljs-variable">$i</span> | bgzip -@ 100 &gt; Chr<span class="hljs-variable">$i</span>.vcf.gz;
    tabix -p vcf Chr<span class="hljs-variable">$i</span>.vcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># For contigs</span>
zcat 500.vcf.gz | grep <span class="hljs-string">"#"</span> | grep CannZSG_Contig | \
    sed <span class="hljs-string">'s/,/ /g'</span> | sed <span class="hljs-string">'s/##contig=&lt;ID=//g'</span> | sed <span class="hljs-string">'s/length=//g'</span> | awk <span class="hljs-string">'{print $1,1,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Contig.list
bcftools view 500.vcf.gz --regions-file Contig.list | bgzip -@ 20 &gt; Chr00.vcf.gz
tabix -p vcf Chr00.vcf.gz

<span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP

<span class="hljs-comment"># SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Select SNP</span>
    gatk SelectVariants \
        -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
        -V ../03vcf/Chr<span class="hljs-variable">$i</span>.vcf.gz \
        -select-type SNP \
        -O Chr<span class="hljs-variable">$i</span>.SNP.all.vcf.gz
    <span class="hljs-comment"># Hard filter for SNPs</span>
    gatk VariantFiltration \
        -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
        -V Chr<span class="hljs-variable">$i</span>.SNP.all.vcf.gz \
        --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || SOR &gt;3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0"</span> \
        --filter-name <span class="hljs-string">"my_snp_filter"</span> \
        -O Chr<span class="hljs-variable">$i</span>.filter.vcf.gz
    <span class="hljs-comment"># Extract the hard filtered</span>
    bcftools view Chr<span class="hljs-variable">$i</span>.filter.vcf.gz \
        --apply-filters PASS | \
        bgzip -@ 20 \
        &gt; Chr<span class="hljs-variable">$i</span>.filtered.vcf.gz
    <span class="hljs-comment"># Index</span>
    tabix -p vcf Chr<span class="hljs-variable">$i</span>.filtered.vcf.gz
    <span class="hljs-comment"># Remove loci with missing &lt;0.1 and maf &lt; 0.05</span>
    vcftools --gzvcf Chr<span class="hljs-variable">$i</span>.filtered.vcf.gz --maf 0.05 --max-missing 0.1 --recode --stdout | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.filtered2.vcf.gz
    <span class="hljs-comment"># Index</span>
    tabix -p vcf Chr<span class="hljs-variable">$i</span>.filtered2.vcf.gz
    <span class="hljs-comment"># Extract the high quality SNPs</span>
    bcftools view Chr<span class="hljs-variable">$i</span>.filtered2.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.biSNP.vcf.gz
    tabix -p vcf Chr<span class="hljs-variable">$i</span>.biSNP.vcf.gz
    <span class="hljs-comment"># Final SNP set with missing cut of 0.3</span>
    bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.3'</span> Chr<span class="hljs-variable">$i</span>.biSNP.vcf.gz | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Indels</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Select Indel</span>
    gatk SelectVariants \
        -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
        -V ../03vcf/Chr<span class="hljs-variable">$i</span>.vcf.gz \
        -select-type INDEL \
        -O Chr<span class="hljs-variable">$i</span>.INDEL.all.vcf.gz
    <span class="hljs-comment"># GATK hard filter</span>
    gatk VariantFiltration \
        -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
        -V Chr<span class="hljs-variable">$i</span>.INDEL.all.vcf.gz \
        --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0"</span> \
        --filter-name <span class="hljs-string">"my_snp_filter"</span> \
        -O Chr<span class="hljs-variable">$i</span>.INDEL.filter.vcf.gz
    <span class="hljs-comment"># Extract the Indels passing the hard filter and further filter</span>
    bcftools view Chr<span class="hljs-variable">$i</span>.INDEL.filtered.vcf.gz --apply-filters PASS \
        --types indels -m 2 -M 2 -i <span class="hljs-string">'F_MISSING&lt;0.3'</span> --min-af 0.05 | \
        bgzip -@ 20 \
        &gt; Chr<span class="hljs-variable">$i</span>.INDEL.filtered2.vcf.gz
    <span class="hljs-comment"># Rename the chromosome ids</span>
    bcftools annotate --rename-chrs chr.update.txt Chr<span class="hljs-variable">$i</span>.INDEL.filtered2.vcf.gz | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.INDEL.final.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment">## Missing&lt;0.3</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span> 
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.3'</span> ../Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz
bcftools query -f <span class="hljs-string">'x'</span> Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | wc -c &gt; Chr<span class="hljs-variable">$i</span>.SNP.count
bcftools query -f <span class="hljs-string">'%CHROM %POS\n'</span> Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | sed <span class="hljs-string">'s/CannZSG_//g'</span> &gt; Chr<span class="hljs-variable">$i</span>.SNP.info
grep -f Chr<span class="hljs-variable">$i</span>.SNP.info ../Chr<span class="hljs-variable">$i</span>.SNP.info.old | wc -l &gt; Chr<span class="hljs-variable">$i</span>.SNP.consis
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add SNP id to the vcf</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
bcftools annotate --<span class="hljs-built_in">set</span>-id <span class="hljs-string">'%CHROM\_%POS'</span> Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.SNP.new.vcf.gz
mv Chr<span class="hljs-variable">$i</span>.SNP.new.vcf.gz Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Count the number of SNPs and Indels</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
bcftools query -f <span class="hljs-string">'x'</span> Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | wc -c &gt; Chr<span class="hljs-variable">$i</span>.SNP.count
bcftools query -f <span class="hljs-string">'x'</span> Chr<span class="hljs-variable">$i</span>.INDEL.final.vcf.gz | wc -c &gt; Chr<span class="hljs-variable">$i</span>.INDEL.final.count
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Check the consistency of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-comment"># SNPs from pan</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS %REF %ALT\n'</span> Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | sed <span class="hljs-string">'s/CannZSG_//g'</span> | cut -f1,2 -d <span class="hljs-string">' '</span> &gt; Chr<span class="hljs-variable">$i</span>.SNP.info
<span class="hljs-comment"># SNP from old</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS %REF %ALT\n'</span> ../../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz  | cut -f1,2 -d <span class="hljs-string">' '</span> &gt; Chr<span class="hljs-variable">$i</span>.SNP.info.old
<span class="hljs-comment"># Rotal consistent</span>
grep -f Chr<span class="hljs-variable">$i</span>.SNP.info Chr<span class="hljs-variable">$i</span>.SNP.info.old | wc -l &gt; Chr<span class="hljs-variable">$i</span>.SNP.consis
grep -vf Chr<span class="hljs-variable">$i</span>.SNP.info Chr<span class="hljs-variable">$i</span>.SNP.info.old | wc -l &gt; Chr<span class="hljs-variable">$i</span>.SNP.inconsis
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Check the consistency of Indels</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
<span class="hljs-comment"># Indels from pan</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS %REF %ALT\n'</span> Chr<span class="hljs-variable">$i</span>.INDEL.vcf.gz | sed <span class="hljs-string">'s/CannZSG_//g'</span> | cut -f1,2 -d <span class="hljs-string">' '</span> &gt; Chr<span class="hljs-variable">$i</span>.Indels.info
<span class="hljs-comment"># Indels from old</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS %REF %ALT\n'</span> ../../Indels/Chr<span class="hljs-variable">$i</span>.fltr5.indel.vcf.gz  | cut -f1,2 -d <span class="hljs-string">' '</span> &gt; Chr<span class="hljs-variable">$i</span>.Indels.info.old
<span class="hljs-comment"># Total consistent</span>
grep -f Chr<span class="hljs-variable">$i</span>.Indels.info Chr<span class="hljs-variable">$i</span>.Indels.info.old | wc -l &gt; Chr<span class="hljs-variable">$i</span>.Indels.consis
grep -vf Chr<span class="hljs-variable">$i</span>.Indels.info Chr<span class="hljs-variable">$i</span>.Indels.info.old | wc -l &gt; Chr<span class="hljs-variable">$i</span>.Indels.inconsis
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Save the file to another folder</span>
mv *SNP* ../04SNP
mv *INDEL* ../05INDEL

</div></code></pre>
<h3 id="snp-and-indel-snpeff">SNP and Indel snpeff</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/tools/snpEff

<span class="hljs-comment"># Prepare and bulid the genome index</span>
cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.fa data/genomes/Zhangshugang.fa
cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.fa data/Zhangshugang/Zhangshugang.fa
cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.fa data/Zhangshugang/sequences.fa
cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff3 data/Zhangshugang/genes.gff

<span class="hljs-built_in">echo</span> <span class="hljs-string">"Zhangshugang.genome : Zhangshugang"</span> &gt;&gt; snpEff.config

<span class="hljs-comment"># Build the database</span>
java -jar snpEff.jar build -gff3 -v Zhangshugang

<span class="hljs-comment"># Working directory for SNPs</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3

<span class="hljs-comment"># Rename the chromosome ids</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    bcftools annotate --rename-chrs Old/chr.update.txt Old/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the SNP effects</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/SnpEff.Chr<span class="hljs-variable">$i</span>
    java -Xmx8g -jar /data/zhaojiantao/tools/snpEff/snpEff.jar \
        -c /data/zhaojiantao/tools/snpEff/snpEff.config \
        Zhangshugang \
        ../Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz | bgzip -@ 20 \
        &gt; Chr<span class="hljs-variable">$i</span>.ann.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Working directory for Indels</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/05INDEL

<span class="hljs-comment"># Rename the chromosome ids</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    bcftools annotate --rename-chrs Old/chr.update.txt Old/Chr<span class="hljs-variable">$i</span>.INDEL.vcf.gz | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.INDEL.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate indel effects</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/05INDEL/SnpEff.indel.Chr<span class="hljs-variable">$i</span>
    java -Xmx8g -jar /data/zhaojiantao/tools/snpEff/snpEff.jar \
        -c /data/zhaojiantao/tools/snpEff/snpEff.config \
        Zhangshugang \
        ../Chr<span class="hljs-variable">$i</span>.INDEL.final.vcf.gz | bgzip -@ 20 \
        &gt; Chr<span class="hljs-variable">$i</span>.indel.ann.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the uniq variants by taking the first effect</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    zcat SnpEff.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.ann.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f8 | sed <span class="hljs-string">'s/|/\t/g'</span> | cut -f2 | sort | uniq -c &gt; SnpEff.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.ann.count &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the uniq variants by taking the first effect</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/05INDEL
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    zcat SnpEff.indel.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.indel.ann.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f8 | sed <span class="hljs-string">'s/;/\t/g'</span> | \
    awk <span class="hljs-string">'{for(i=1;i&lt;=NF;i++){if ($i ~ /ANN/){print $i}}}'</span> | \
    sed <span class="hljs-string">'s/|/\t/g'</span> | cut -f2 | sort | uniq -c &gt; SnpEff.indel.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.indel.ann.count &amp;
<span class="hljs-keyword">done</span>

</div></code></pre>
<h4 id="snp-missing-rate-cutoff">SNP missing rate cutoff</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP

bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.1'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.1.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.2'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.2.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.3'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.3.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.4'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.4.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.5'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.5.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.6'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.6.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.7'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.7.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.8'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.8.SNP.vcf.gz &amp;
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.9'</span> Chr01.SNP.vcf.gz | bgzip &gt; Chr01.Miss_0.9.SNP.vcf.gz &amp;

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 9); <span class="hljs-keyword">do</span> 
tabix -p vcf Chr01.Miss_0.<span class="hljs-variable">$i</span>.SNP.vcf.gz &amp;
bcftools query -f <span class="hljs-string">'x'</span> Chr01.Miss_0.<span class="hljs-variable">$i</span>.SNP.vcf.gz  | wc -c &gt; Chr01.Miss_0.<span class="hljs-variable">$i</span>.SNP.count
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Compare with vcftools</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 9); <span class="hljs-keyword">do</span> 
vcftools --gzvcf Chr01.SNP.vcf.gz --max-missing 0.<span class="hljs-variable">$i</span> --maf 0.05 --out Chr01.Miss_0.<span class="hljs-variable">$i</span>.SNP.vcftools.count &amp;
<span class="hljs-keyword">done</span>


</div></code></pre>
<h4 id="snp-check">SNP check</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/SNP.check

<span class="hljs-comment"># Rename the sample id</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> chaotiaojiao Grif_1615 Grif_9355 PI_424732; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Rename</span>
    bcftools view <span class="hljs-variable">${i}</span>_bwa.gvcf.gz | \
        sed <span class="hljs-string">"s/<span class="hljs-variable">${i}</span>_1/<span class="hljs-variable">${i}</span>_bwa/g"</span> | bgzip -@ 20 \
        &gt; <span class="hljs-variable">${i}</span>_bwa.rename.gvcf.gz
    <span class="hljs-comment"># Index</span>
    tabix -p vcf <span class="hljs-variable">${i}</span>_bwa.rename.gvcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> chaotiaojiao Grif_1615 Grif_9355 PI_424732; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Rename</span>
    bcftools view <span class="hljs-variable">${i}</span>_gam.gvcf.gz | \
        sed <span class="hljs-string">"s/<span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_gam/g"</span> | bgzip -@ 20 \
        &gt; <span class="hljs-variable">${i}</span>_gam.rename.gvcf.gz
    <span class="hljs-comment"># Index</span>
    tabix -p vcf <span class="hljs-variable">${i}</span>_gam.rename.gvcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> chaotiaojiao Grif_1615 Grif_9355 PI_424732; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Rename</span>
    bcftools view <span class="hljs-variable">$i</span>.sppdup.g.vcf.gz | \
        sed <span class="hljs-string">"s/<span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.sppdup/g"</span> | bgzip -@ 20 \
        &gt; <span class="hljs-variable">$i</span>.sppdup.g.rename.vcf.gz
    <span class="hljs-comment"># Index</span>
    tabix -p vcf <span class="hljs-variable">$i</span>.sppdup.g.rename.vcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> chaotiaojiao Grif_1615 Grif_9355 PI_424732; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Rename</span>
    bcftools view <span class="hljs-variable">$i</span>.gvcf.gz | \
        sed <span class="hljs-string">"s/<span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.vg/g"</span> | bgzip -@ 20 \
        &gt; <span class="hljs-variable">$i</span>.rename.gvcf.gz
    <span class="hljs-comment"># Index</span>
    tabix -p vcf <span class="hljs-variable">$i</span>.rename.gvcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Join the gvcf files</span>
sentieon driver -t 100 \
    -r /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa --algo GVCFtyper \
    Check.vcf.gz *rename*gz
<span class="hljs-comment"># Hard filter</span>
gatk VariantFiltration \
    -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
    -V Check.vcf.gz \
    --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || SOR &gt;3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0"</span> \
    --filter-name <span class="hljs-string">"my_snp_filter"</span> \
    -O Check.filter.vcf.gz
<span class="hljs-comment"># Extract the hard filtered</span>
bcftools view Check.filter.vcf.gz \
    --apply-filters PASS | \
    bgzip -@ 20 \
    &gt; Check.filtered.vcf.gz
<span class="hljs-comment"># Index</span>
tabix -p vcf Check.filtered.vcf.gz
<span class="hljs-comment"># Extract biallelic    </span>
bcftools view Check.filtered.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Check.filtered.biSNP.vcf.gz
tabix -p vcf Check.filtered.biSNP.vcf.gz

<span class="hljs-comment"># Convert to plink biallelic</span>
plink --vcf Check.filtered.biSNP.vcf.gz \
    --recode 12 --transpose \
    --double-id \
    --allow-extra-chr \
    --out Check.filtered.biSNP

<span class="hljs-comment"># Count the number of missing</span>
awk <span class="hljs-string">'$5=="0" &amp;&amp; $6=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A01.missing &amp;
awk <span class="hljs-string">'$7=="0" &amp;&amp; $8=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A02.missing &amp;
awk <span class="hljs-string">'$9=="0" &amp;&amp; $10=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A03.missing &amp;
awk <span class="hljs-string">'$11=="0" &amp;&amp; $12=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A04.missing &amp;
awk <span class="hljs-string">'$13=="0" &amp;&amp; $14=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A05.missing &amp;
awk <span class="hljs-string">'$15=="0" &amp;&amp; $16=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A06.missing &amp;
awk <span class="hljs-string">'$17=="0" &amp;&amp; $18=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A07.missing &amp;
awk <span class="hljs-string">'$19=="0" &amp;&amp; $20=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A08.missing &amp;
awk <span class="hljs-string">'$21=="0" &amp;&amp; $22=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A09.missing &amp;
awk <span class="hljs-string">'$23=="0" &amp;&amp; $24=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A10.missing &amp;
awk <span class="hljs-string">'$25=="0" &amp;&amp; $26=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A11.missing &amp;
awk <span class="hljs-string">'$27=="0" &amp;&amp; $28=="0" {print}'</span> Check.filtered.biSNP.tped | wc -l &gt; A12.missing &amp;

<span class="hljs-comment"># Extract the results from previously mapping</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
bcftools view -S sample.list ../../../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz | bgzip -@ 20 &gt; Pepper4.Chr<span class="hljs-variable">$i</span>.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Convert to plink biallelic</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
plink --vcf Pepper4.Chr<span class="hljs-variable">$i</span>.vcf.gz \
    --recode 12 --transpose \
    --double-id \
    --out Pepper4.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Missing count</span>
cat Pepper4.Chr*.tped | awk <span class="hljs-string">'$5=="0" &amp;&amp; $6=="0" {print}'</span> | wc -l &gt; Pepper4.A01.missing &amp;
cat Pepper4.Chr*.tped | awk <span class="hljs-string">'$7=="0" &amp;&amp; $8=="0" {print}'</span> | wc -l &gt; Pepper4.A02.missing &amp;
cat Pepper4.Chr*.tped | awk <span class="hljs-string">'$9=="0" &amp;&amp; $10=="0" {print}'</span> | wc -l &gt; Pepper4.A03.missing &amp;
cat Pepper4.Chr*.tped | awk <span class="hljs-string">'$11=="0" &amp;&amp; $12=="0" {print}'</span> | wc -l &gt; Pepper4.A04.missing &amp;

</div></code></pre>
<h4 id="snp-check-to-old-s8-references">SNP check to old S8 references</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/SNP.check/notused

<span class="hljs-comment"># Join the gvcf files</span>
sentieon driver -t 100 \
    -r /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta --algo GVCFtyper \
    Check.vcf.gz *rename*gz
<span class="hljs-comment"># Hard filter</span>
gatk VariantFiltration \
    -R /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta \
    -V Check.vcf.gz \
    --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || SOR &gt;3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0"</span> \
    --filter-name <span class="hljs-string">"my_snp_filter"</span> \
    -O Check.filter.vcf.gz
<span class="hljs-comment"># Extract the hard filtered</span>
bcftools view Check.filter.vcf.gz \
    --apply-filters PASS | \
    bgzip -@ 20 \
    &gt; Check.filtered.vcf.gz
<span class="hljs-comment"># Index</span>
tabix -p vcf Check.filtered.vcf.gz
<span class="hljs-comment"># Extract biallelic    </span>
bcftools view Check.filtered.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Check.filtered.biSNP.vcf.gz
tabix -p vcf Check.filtered.biSNP.vcf.gz


</div></code></pre>
<h4 id="snp-check-to-vg-reference-for-bwa-mapping">SNP check to vg reference for bwa mapping</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/SNP.check

<span class="hljs-comment"># Join the gvcf files</span>
sentieon driver -t 100 \
    -r /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa --algo GVCFtyper \
    Check_bwa.vcf.gz *bwa.rename.gvcf.gz
<span class="hljs-comment"># Hard filter</span>
gatk VariantFiltration \
    -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
    -V Check_bwa.vcf.gz \
    --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || SOR &gt;3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0"</span> \
    --filter-name <span class="hljs-string">"my_snp_filter"</span> \
    -O Check_bwa.filter.vcf.gz
<span class="hljs-comment"># Extract the hard filtered</span>
bcftools view Check_bwa.filter.vcf.gz \
    --apply-filters PASS | \
    bgzip -@ 20 \
    &gt; Check_bwa.filtered.vcf.gz
<span class="hljs-comment"># Index</span>
tabix -p vcf Check_bwa.filtered.vcf.gz
<span class="hljs-comment"># Extract biallelic    </span>
bcftools view Check_bwa.filtered.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Check_bwa.filtered.biSNP.vcf.gz
tabix -p vcf Check_bwa.filtered.biSNP.vcf.gz

</div></code></pre>
<h4 id="snp-check-to-vg-reference-for-vg-mapping">SNP check to vg reference for vg mapping</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP/SNP.check

<span class="hljs-comment"># Join the gvcf files</span>
sentieon driver -t 100 \
    -r /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa --algo GVCFtyper \
    Check_gam.vcf.gz *gam.rename.gvcf.gz
<span class="hljs-comment"># Hard filter</span>
gatk VariantFiltration \
    -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
    -V Check_gam.vcf.gz \
    --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || SOR &gt;3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0"</span> \
    --filter-name <span class="hljs-string">"my_snp_filter"</span> \
    -O Check_gam.filter.vcf.gz
<span class="hljs-comment"># Extract the hard filtered</span>
bcftools view Check_gam.filter.vcf.gz \
    --apply-filters PASS | \
    bgzip -@ 20 \
    &gt; Check_gam.filtered.vcf.gz
<span class="hljs-comment"># Index</span>
tabix -p vcf Check_gam.filtered.vcf.gz
<span class="hljs-comment"># Extract biallelic    </span>
bcftools view Check_gam.filtered.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Check_gam.filtered.biSNP.vcf.gz
tabix -p vcf Check_gam.filtered.biSNP.vcf.gz

</div></code></pre>
<h3 id="snp-ld-prune">SNP LD prune</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/04SNP

<span class="hljs-comment"># Add the SNP positions in the vcf file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
zcat Chr00.SNP.vcf.gz | grep <span class="hljs-string">"#"</span> &gt; header
bgzip -d Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz -@ 20 -c | grep -v <span class="hljs-string">"#"</span> &gt; Chr<span class="hljs-variable">$i</span>.geno
cut -f1,2 Chr<span class="hljs-variable">$i</span>.geno | awk <span class="hljs-string">'{print $1,$2,$1"_"$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    paste - Chr<span class="hljs-variable">$i</span>.geno | cut -f1-3,7- | cat header - | bgzip -@ 20 \
    &gt; Chr<span class="hljs-variable">$i</span>.SNP.withid.vcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/08SNPprune

<span class="hljs-comment"># LD-prune</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-comment"># LD prune</span>
plink --vcf ../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
    --indep-pairwise 50 10 0.2 \
    --allow-extra-chr --double-id \
    --out Chr<span class="hljs-variable">$i</span>
<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --vcf ../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
    --extract Chr<span class="hljs-variable">$i</span>.prune.in \
    --recode vcf --double-id --allow-extra-chr \
    --out Chr<span class="hljs-variable">$i</span>.prune
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Use single sample id and bgzip the file</span>
grep <span class="hljs-string">"#"</span> Chr00.prune.vcf | grep -v <span class="hljs-string">"#CHROM"</span> | cat - &lt;(zcat ../04SNP/Miss_0.3/Chr00.SNP.withid.vcf.gz | grep <span class="hljs-string">"#CHROM"</span> ) &gt; header

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
grep -v <span class="hljs-string">"#"</span> Chr<span class="hljs-variable">$i</span>.prune.vcf | cat header - | bgzip -@ 20 &gt; Chr<span class="hljs-variable">$i</span>.prune.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove temporary files</span>
rm *<span class="hljs-built_in">log</span> *nosex *prune.in Chr*.prune.vcf header

</div></code></pre>
<h3 id="extract-4dtv-snps-and-iq-tree">Extract 4DTV SNPs and IQ-tree</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/064DTV

<span class="hljs-comment"># Call 4dtv SNPs</span>
4DTV_scan.pl -g ../PanReference/Cann_Zhangshugang.chr.fa \
    -f ../PanReference/Cann_Zhangshugang.gff3 \
    -o ffds.list

<span class="hljs-comment"># Select the ids</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
grep Chr<span class="hljs-variable">$i</span> ffds.list | cut -f1,2 &gt; Chr<span class="hljs-variable">$i</span>.ffds.list
<span class="hljs-keyword">done</span>
grep -v Chr ffds.list | cut -f1,2 &gt; Chr00.ffds.list

<span class="hljs-comment"># Extract the 4DTV SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
bcftools view ../04SNP/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
    --regions-file Chr<span class="hljs-variable">$i</span>.ffds.list | \
    bgzip -@ 20 \
    &gt; Chr<span class="hljs-variable">$i</span>.ffds.SNP.vcf.gz &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Merge the file</span>
bcftools concat Chr*.ffds.SNP.vcf.gz | bgzip -@ 20 &gt; 500.ffds.SNP.vcf.gz
tabix -p vcf 500.ffds.SNP.vcf.gz

<span class="hljs-comment"># Use tomato as outgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> ERR418102 ERR418107 ERR418103 ERR418106; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Download SRA</span>
    prefetch <span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Convert SRA to fastq</span>
    fasterq-dump --split-files  <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.sra
    <span class="hljs-comment"># Remove adaptor and low-quality sequences</span>
    trimmomatic PE <span class="hljs-variable">${i}</span>_1.fastq <span class="hljs-variable">${i}</span>_2.fastq \
            <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1U.fq \
            <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2U.fq \
            -threads 100 \
            ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
            SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40
    <span class="hljs-comment"># Remove temporary file</span>
    rm <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.sra <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1U.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2U.fq <span class="hljs-variable">${i}</span>_1.fastq <span class="hljs-variable">${i}</span>_2.fastq 
    <span class="hljs-comment"># vg mapping</span>
    vg giraffe -t 100 \
        -Z m6a.giraffe.gbz \
        -m m6a.min \
        -d m6a.dist \
        -f <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R1P.fq \
        -f <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.R2P.fq \
        &gt; <span class="hljs-variable">$i</span>.gam
    <span class="hljs-comment"># Generate bam</span>
    vg surject -t 100 \
        -N <span class="hljs-variable">$i</span> \
        -R <span class="hljs-variable">$i</span> \
        -x m6a.giraffe.gbz \
        -b -i <span class="hljs-variable">$i</span>.gam \
        &gt; <span class="hljs-variable">$i</span>.gam.bam
    <span class="hljs-comment"># Sort the bam file</span>
    perl sort_bam_header_chrID.pl <span class="hljs-variable">$i</span> <span class="hljs-variable">$i</span>.gam.bam <span class="hljs-variable">${i}</span>_gam.bam
    <span class="hljs-comment"># Index the bam</span>
    samtools index -@ 100 <span class="hljs-variable">${i}</span>_gam.bam        
    <span class="hljs-comment"># Call variants</span>
    <span class="hljs-comment"># Collect read information to remove or mark duplicates</span>
    sentieon driver -t 100 -i <span class="hljs-variable">${i}</span>_gam.bam \
        --algo LocusCollector --fun score_info <span class="hljs-variable">$i</span>.score.gz
    <span class="hljs-comment"># Remove or mark duplicates</span>
    sentieon driver -t 100 -i <span class="hljs-variable">${i}</span>_gam.bam \
        --algo Dedup --score_info <span class="hljs-variable">$i</span>.score.gz \
        --metrics <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam
    <span class="hljs-comment"># Call gvcf</span>
    sentieon driver -t 100 \
        -r ../PanReference/Cann_Zhangshugang.chr.fa \
        -i <span class="hljs-variable">$i</span>.deduped.bam \
        --algo Haplotyper \
        --emit_mode gvcf \
        <span class="hljs-variable">$i</span>.gvcf.gz
    <span class="hljs-comment"># Remove unnecessary files</span>
    rm <span class="hljs-variable">$i</span>.score.gz* <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam*
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Join the gvcf files</span>
sentieon driver -t 100 \
    -r /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa --algo GVCFtyper --emit_mode all \
    ERR418102.vcf.gz ERR418102.gvcf.gz

<span class="hljs-comment"># Hard filter</span>
gatk VariantFiltration \
    -R /data/zhaojiantao/pepper/vg/PanReference/Cann_Zhangshugang.chr.fa \
    -V Tomato.vcf.gz \
    --filter-expression <span class="hljs-string">"QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || SOR &gt;3.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0"</span> \
    --filter-name <span class="hljs-string">"my_snp_filter"</span> \
    -O Tomato.filter.vcf.gz
<span class="hljs-comment"># Extract the hard filtered</span>
bcftools view Tomato.filter.vcf.gz \
    --apply-filters PASS | \
    bgzip -@ 20 \
    &gt; Tomato.filtered.vcf.gz
<span class="hljs-comment"># Index</span>
tabix -p vcf Tomato.filtered.vcf.gz
<span class="hljs-comment"># Extract biallelic    </span>
bcftools view Tomato.filtered.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Tomato.filtered.biSNP.vcf.gz
bcftools view Tomato.filter.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; Tomato.filter.biSNP.vcf.gz
bcftools view 504.4DTV.vcf.gz --types snps -m 2 -M 2 | bgzip -@ 20 &gt; 504.4DTV.biSNP.vcf.gz

tabix -p vcf Tomato.filtered.biSNP.vcf.gz


<span class="hljs-comment"># Filter the 4DTV with missing &gt; 0.3 and maf &lt; 0.05</span>

<span class="hljs-comment">## Missing&lt;0.3</span>
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.3'</span> 500.ffds.SNP.vcf.gz | bgzip -@ 20 &gt; 500.ffds.filter.SNP.vcf.gz
tabix -p vcf 500.ffds.filter.SNP.vcf.gz
bcftools query -f <span class="hljs-string">'%CHROM %POS\n'</span> 500.ffds.filter.SNP.vcf.gz | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; 500.ffds.filter.SNP.info

<span class="hljs-comment"># Extract the common SNPs from tomato</span>
bcftools view Tomato.vcf.gz \
    --regions-file 500.ffds.filter.SNP.info | \
    bgzip -@ 20 \
    &gt; Tomato.ffds.filter.SNP.vcf.gz
tabix -p vcf Tomato.ffds.filtered.SNP.vcf.gz

bcftools view Tomato.ffds.filtered.SNP.vcf.gz \
    --regions-file comm.info | \
    bgzip -@ 20 \
    &gt; Tomato.ffds.filtered2.SNP.vcf.gz
tabix -p vcf 500.ffds.filtered2.SNP.vcf.gz

<span class="hljs-comment"># Extract the tomato info</span>
bcftools query -f <span class="hljs-string">'%CHROM %POS\n'</span> Tomato.ffds.filtered.SNP.vcf.gz | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Tomato.ffds.filtered.SNP.info
bcftools query -f <span class="hljs-string">'%CHROM %POS\n'</span> Tomato.filtered.biSNP.vcf.gz | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Tomato.filtered.biSNP.SNP.info
bcftools query -f <span class="hljs-string">'%CHROM %POS\n'</span> 500.ffds.filtered.SNP.vcf.gz | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; 500.ffds.filtered.SNP.info

Tomato.filtered.biSNP.vcf.gz

<span class="hljs-comment"># Find the overlapped SNPs</span>
comm -12 &lt;(sort 500.ffds.filter.SNP.info) &lt;(sort Tomato.ffds.filter.SNP.info) &gt; comm.SNP.info
comm -12 &lt;(sort 500.ffds.filtered.SNP.info) &lt;(sort Tomato.filtered.biSNP.SNP.info) &gt; comm.SNP.info

<span class="hljs-comment"># Extract the final results</span>
bcftools view Tomato.ffds.filter.SNP.vcf.gz \
    --regions-file comm.SNP.info | \
    bgzip -@ 20 \
    &gt; Tomato.ffds.filtered.SNP.vcf.gz

bcftools view Tomato.filtered.biSNP.vcf.gz \
    --regions-file comm.SNP.info | \
    bgzip -@ 20 \
    &gt; Tomato.ffds.filtered.biSNP.vcf.gz
tabix -p vcf Tomato.ffds.filtered.biSNP.vcf.gz

bcftools view 500.ffds.filter.SNP.vcf.gz \
    --regions-file comm.SNP.info | \
    bgzip -@ 20 \
    &gt; 500.ffds.filtered.SNP.vcf.gz

bcftools view 500.ffds.filtered.SNP.vcf.gz \
    --regions-file comm.SNP.info | \
    bgzip -@ 20 \
    &gt; 500.ffds.filtered.biSNP.vcf.gz
tabix -p vcf 500.ffds.filtered.biSNP.vcf.gz

<span class="hljs-comment"># Merge tomato and pepper vcfs</span>
bcftools merge Tomato.ffds.filtered.SNP.vcf.gz 500.ffds.filtered.SNP.vcf.gz -O z -o 504.4DTV.filtered.vcf.gz
bcftools merge Tomato.ffds.filtered.biSNP.vcf.gz 500.ffds.filtered.biSNP.vcf.gz -O z -o 504.4DTV.filtered.biSNP.vcf.gz


<span class="hljs-comment"># Number of SNPs with F_MISSING&lt;0.3</span>
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.2'</span> 504.4DTV.filtered.vcf.gz | bgzip -@ 20 &gt; 504.4DTV.filtered2.vcf.gz
bcftools view -i <span class="hljs-string">'F_MISSING&lt;0.1'</span> 504.4DTV.filtered.vcf.gz | bgzip -@ 20 &gt; 504.4DTV.filtered1.vcf.gz

<span class="hljs-comment"># LD-prune</span>
plink --vcf 504.4DTV.Miss0.1.vcf.gz \
    --indep-pairwise 50 10 0.4 \
    --allow-extra-chr --double-id \
    --out 504.4DTV.Miss0.1.LD0.4

<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --vcf 504.4DTV.Miss0.1.vcf.gz \
    --extract 504.4DTV.Miss0.1.LD0.4.prune.in \
    --recode vcf --double-id --allow-extra-chr \
    --out 504.4DTV.Miss0.1.LD0.4

zcat 504.4DTV.Miss0.1.vcf.gz | grep <span class="hljs-string">"#CHROM"</span> &gt; AA
grep -v <span class="hljs-string">"#"</span> 504.4DTV.Miss0.1.LD0.4.vcf &gt; BB
grep <span class="hljs-string">"#"</span> 504.4DTV.Miss0.1.LD0.4.vcf |grep -v <span class="hljs-string">"#CHROM"</span> | cat - AA BB | bgzip -@ 20 &gt; 504.4DTV.Miss0.1.LD0.4.gz

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i 504.4DTV.Miss0.1.LD0.4.gz
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i 504.4DTV.filtered.biSNP.vcf.gz

<span class="hljs-comment"># Run IQ-tree</span>
iqtree -s 504.4DTV.Miss0.1.LD0.4.gz.phy -nt 100 -m MFP
iqtree -s 504.4DTV.filtered.biSNP.min4.phy -nt 100 -m MFP

<span class="hljs-comment"># Save the tree for phylogenetic tree plot</span>
cp 504.4DTV.Miss0.1.LD0.4.gz.phy.treefile 504.4DTV.vg.phy.txt

<span class="hljs-comment"># LD-prune</span>
plink --vcf 504.4DTV.biSNP.vcf.gz \
    --indep-pairwise 50 10 0.4 \
    --allow-extra-chr --double-id \
    --out 504.4DTV.biSNP.Miss0.3.LD0.2

<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --vcf 504.4DTV.biSNP.vcf.gz \
    --extract 504.4DTV.biSNP.Miss0.3.LD0.2.prune.in \
    --recode vcf --double-id --allow-extra-chr \
    --out 504.4DTV.biSNP.Miss0.3.LD0.2

zcat 504.4DTV.biSNP.vcf.gz | grep <span class="hljs-string">"#CHROM"</span> &gt; AA
grep -v <span class="hljs-string">"#"</span> 504.4DTV.biSNP.Miss0.3.LD0.2.vcf &gt; BB
grep <span class="hljs-string">"#"</span> 504.4DTV.biSNP.Miss0.3.LD0.2.vcf |grep -v <span class="hljs-string">"#CHROM"</span> | cat - AA BB | bgzip -@ 20 &gt; 504.4DTV.biSNP.Miss0.3.LD0.2.vcf.gz

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i 504.4DTV.biSNP.Miss0.3.LD0.2.vcf.gz

<span class="hljs-comment"># Run IQ-tree</span>
iqtree -s 504.4DTV.biSNP.Miss0.3.LD0.2.phy -nt 100 -m MFP


<span class="hljs-comment"># LD-prune</span>
plink --vcf 504.4DTV.Miss0.1.vcf.gz \
    --indep-pairwise 50 10 0.5 \
    --allow-extra-chr --double-id \
    --out 504.4DTV.Miss0.1.LD0.5

<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --vcf 504.4DTV.Miss0.1.vcf.gz \
    --extract 504.4DTV.Miss0.1.LD0.5.prune.in \
    --recode vcf --double-id --allow-extra-chr \
    --out 504.4DTV.Miss0.1.LD0.5

zcat 504.4DTV.Miss0.1.vcf.gz | grep <span class="hljs-string">"#CHROM"</span> &gt; AA
grep -v <span class="hljs-string">"#"</span> 504.4DTV.Miss0.1.LD0.5.vcf&gt; BB
grep <span class="hljs-string">"#"</span> 504.4DTV.Miss0.1.LD0.5.vcf |grep -v <span class="hljs-string">"#CHROM"</span> | cat - AA BB | bgzip -@ 20 &gt; 504.4DTV.Miss0.1.LD0.5.vcf.gz

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i 504.4DTV.Miss0.1.LD0.5.vcf.gz

<span class="hljs-comment"># Run IQ-tree</span>
iqtree -s 504.4DTV.Miss0.1.LD0.5.min4.phy -nt 100 -m MFP
<span class="hljs-comment"># Save the tree for phylogenetic tree plot</span>
cp  504.4DTV.LD.min4.phy.treefile 504.4DTV.vg.phy.txt

iqtree -s 504.4DTV.Miss1.LD4.min4.phy -nt 70 -m MFP
iqtree -s 504.4DTV.LD0.min4.phy -nt AUTO -m MFP


<span class="hljs-comment"># LD-prune</span>
plink --vcf 501.4DTV.filtered.vcf.gz \
    --indep-pairwise 50 10 0.2 \
    --allow-extra-chr --double-id \
    --out 501.4DTV.filtered

<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --vcf 501.4DTV.filtered.vcf.gz \
    --extract 501.4DTV.filtered.prune.in \
    --recode vcf --double-id --allow-extra-chr \
    --out 501.4DTV.LD

zcat 501.4DTV.filtered.vcf.gz | grep <span class="hljs-string">"#CHROM"</span> &gt; AA
grep -v <span class="hljs-string">"#"</span> 501.4DTV.LD.vcf &gt; BB
grep <span class="hljs-string">"#"</span> 501.4DTV.LD.vcf | grep -v <span class="hljs-string">"#CHROM"</span> | cat - AA BB | bgzip -@ 20 &gt; 501.4DTV.LD.vcf.gz

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i 501.4DTV.LD.vcf.gz

<span class="hljs-comment"># Run IQ-tree</span>
iqtree -s 501.4DTV.LD.min4.phy -nt 60 -m MFP
<span class="hljs-comment"># Save the tree for phylogenetic tree plot</span>
cp  501.4DTV.LD.min4.phy.treefile 501.4DTV.vg.phy.txt

iqtree -s 501.4DTV.Miss1.LD4.min4.phy -nt 70 -m MFP
iqtree -s 501.4DTV.LD0.min4.phy -nt AUTO -m MFP


</div></code></pre>
<h3 id="newhyvbrid">Newhyvbrid</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/064DTV/newhybrid

<span class="hljs-comment"># Test for potential hybrids between C. chacoense and C. baccatum var. baccatum</span>
cat ../../Pop.list/CHA.list ../../Pop.list/BAB.list &gt; CHA.BAC.list
<span class="hljs-comment"># Add additional 17 BAP accessions</span>

<span class="hljs-comment"># Extract the non-missing SNPs</span>
bcftools view -i <span class="hljs-string">'F_MISSING=0'</span> ../504.4DTV.LD1.vcf.gz | bgzip -@ 20 &gt; 504.4DTV.LD1.Nomiss.vcf.gz
<span class="hljs-comment"># Extract the CHA.BAC list</span>
bcftools view -S CHA.BAC.list 504.4DTV.LD1.Nomiss.vcf.gz | bgzip -@ 20 &gt; CHA.BAC.vcf.gz

<span class="hljs-comment"># Convert to plink</span>
plink --vcf CHA.BAC.vcf.gz \
    --recode 12 --allow-extra-chr \
    --output-missing-genotype 9 \
    --out CHA.BAC

<span class="hljs-comment"># Combine every two columns</span>
cut -f7- -d <span class="hljs-string">' '</span> CHA.BAC.ped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; CHA.BAC.ped-1

<span class="hljs-comment"># Add the sample number</span>
seq 43 | paste - CHA.BAC.ped-1 &gt; CHA.BAC.ped-2

<span class="hljs-comment"># Add the SNP information</span>
sed <span class="hljs-string">'s/Ca.S8.//g'</span> CHA.BAC.map | awk <span class="hljs-string">'{print $2}'</span> | sed <span class="hljs-string">'1i\LocusNames'</span> &gt; CHA.BAC.snps

<span class="hljs-built_in">echo</span> <span class="hljs-string">"NumIndivs 43
NumLoci 1046
Digits 1
Format Lumped"</span> \
&gt; header

deal_table.pl -transpose CHA.BAC.snps | \
    cat header - CHA.BAC.ped-2 | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; CHA.BAC.NewHybrids.txt

<span class="hljs-comment"># Check the data format</span>
head -10 CHA.BAC.NewHybrids.txt | cut -f1-5

<span class="hljs-comment"># Test with regular SNPs</span>


<span class="hljs-comment"># Newhybrid has to be run on either window or mac</span>
E:\Project\Pepper\Pepper\newhybrids\newhybrids\bin\PC\newhybrids.exe -d E:\Project\Pepper\Pepper\newhybrids\newhybrids\test_data\<span class="hljs-built_in">test</span>\TestAFLP.txt -c E:\Project\Pepper\Pepper\newhybrids\newhybrids\test_data\<span class="hljs-built_in">test</span>\TwoGensGtypFreq.txt --burn-in 100 --num-sweeps 100 --no-gui

</div></code></pre>
<h3 id="hyde">HyDe</h3>
<h4 id="cha-and-bac-miss01">CHA and BAC Miss0.1</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/064DTV/HyDe/CHA.BAC

<span class="hljs-comment"># Extract the CHA + BAC accessions</span>
bcftools view -S CHA.BAC.list ../../504.4DTV.Miss0.1.vcf.gz | bgzip -@ 20 &gt; CHA.BAC.vcf.gz

<span class="hljs-comment"># Convert to phy</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i CHA.BAC.vcf.gz
sed <span class="hljs-string">'1d'</span> CHA.BAC.min4.phy &gt; CHA.BAC.min4.txt

<span class="hljs-comment"># Run a full hybridization detection analysis</span>
run_hyde_mp.py -i CHA.BAC.min4.txt -m map.txt -o C.pubescens -n 136 -t 4 -s 145492 -j 100

<span class="hljs-comment"># Test all individuals for the hybrid populations specified</span>
individual_hyde_mp.py -i CHA.BAC.min4.txt -m map.txt -tr hyde-out.txt -o C.pubescens -n 136 -t 4 -s 145492 -j 100

</div></code></pre>
<h4 id="annuum-miss01">Annuum Miss0.1</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/064DTV/HyDe/Annuum

<span class="hljs-comment"># Extract the Annuum accessions</span>
bcftools view -S Annuum.list ../../504.4DTV.Miss0.1.vcf.gz | bgzip -@ 20 &gt; Annuum.vcf.gz

<span class="hljs-comment"># Convert to phy</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i Annuum.vcf.gz
sed <span class="hljs-string">'1d'</span> Annuum.min4.phy &gt; Annuum.min4.txt

<span class="hljs-comment"># Run a full hybridization detection analysis</span>
run_hyde_mp.py -i Annuum.min4.txt -m map.txt -o CHA -n 327 -t 4 -s 145492 -j 100 -tr triples.txt

<span class="hljs-comment"># Test all individuals for the hybrid populations specified</span>
individual_hyde_mp.py -i Annuum.min4.txt -m map.txt -tr hyde-out.txt -o CHA -n 327 -t 4 -s 145492 -j 100
mv mv hyde-ind.txt hyde-ind.all.txt
<span class="hljs-comment"># Only for the significant</span>
individual_hyde_mp.py -i Annuum.min4.txt -m map.txt -tr hyde-out-filtered.txt -o CHA -n 327 -t 4 -s 145492 -j 100

<span class="hljs-comment"># Bootstrap</span>
bootstrap_hyde_mp.py -i Annuum.min4.txt -m map.txt -tr hyde-out-filtered.txt -o CHA -n 327 -t 4 -s 145492 -j 100

</div></code></pre>
<h3 id="population-structure">Population structure</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/07Structure

<span class="hljs-comment"># Requies python 2.7</span>
conda activate fastStructure
<span class="hljs-built_in">export</span> LD_LIBRARY_PATH=<span class="hljs-variable">$LD_LIBRARY_PATH</span>:/data/zhaojiantao/tools/miniconda3/pkgs/gsl-2.6-he838d99_2/lib

<span class="hljs-comment"># Make bed for the ped/map plink file</span>
plink --vcf ../064DTV/504.4DTV.LD1.vcf.gz --make-bed --double-id --allow-extra-chr --out 504.4DTV

<span class="hljs-comment"># fastStructure results at K = 7</span>
structure.py -K 7 --input=504.4DTV --output=K7 --full --seed=100

<span class="hljs-comment"># fastStructure</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 20); <span class="hljs-keyword">do</span> 
    python2 /data/zhaojiantao/tools/fastStructure/structure.py -K <span class="hljs-variable">$i</span> --input=504.4DTV --output=output --full --seed=100
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Choose the optimal K (K = 7)</span>
python2 /data/zhaojiantao/tools/fastStructure/chooseK.py --input=output

</div></code></pre>
<h3 id="fst">Fst</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/09Fst

<span class="hljs-comment"># Calculate Fst between PUB and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate Fst between CHA and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> CHA; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate Fst between BAB and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAB; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate Fst between BAP and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAP; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate Fst between FRU and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FRU; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> CHN GLA ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate Fst between CHN and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> CHN; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> GLA ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate Fst between CHN and the other subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> GLA; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> ANN; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../08SNPprune/Chr<span class="hljs-variable">$chr</span>.prune.vcf.gz \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out <span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *weir.fst | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.Chr/\t/g'</span> | cut -f1 | sort | uniq); <span class="hljs-keyword">do</span> 
    cat <span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | \
        awk <span class="hljs-string">'{print $7=$1"_"$2"_"$3,$1,$2,$3,$8=($2+$3)/2,$4,$5,$6}'</span> | \
        awk <span class="hljs-string">'$8 &gt; 0 {print $0}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS WEIGHTED_FST MEAN_FST'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.fst.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove chromosomal level files</span>
rm *.windowed.weir.fst *<span class="hljs-built_in">log</span> 

<span class="hljs-comment"># Calculate the mean Fst value</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *fst.txt.gz | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.fst.txt.gz//g'</span>); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; zcat <span class="hljs-variable">$i</span>.fst.txt.gz | grep -v MEAN | awk <span class="hljs-string">'{sum+=$8} END {print sum/NR}'</span>
<span class="hljs-keyword">done</span>

</div></code></pre>
<h4 id="fst-permutation-for-babbap">Fst permutation for BAB.BAP</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/09Fst/permutations

<span class="hljs-comment"># Prepare the list for permutation test</span>
cat ../../Pop.list/BAB.list ../../Pop.list/BAP.list &gt; BAB.BAP.list

<span class="hljs-comment"># Randomize the samples </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    sort --random-sort BAB.BAP.list | head -9 &gt; BAB.BAP.Random<span class="hljs-variable">$i</span>.BAB.list
    grep -vf BAB.BAP.Random<span class="hljs-variable">$i</span>.BAB.list BAB.BAP.list &gt; BAB.BAP.Random<span class="hljs-variable">$i</span>.BAP.list
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run the permutation test with 100 replicates</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../../08SNPprune/Chr<span class="hljs-variable">$j</span>.prune.vcf.gz \
        --weir-fst-pop BAB.BAP.Random<span class="hljs-variable">$i</span>.BAB.list \
        --weir-fst-pop BAB.BAP.Random<span class="hljs-variable">$i</span>.BAP.list \
        --fst-window-size 100000 \
        --fst-window-step 10000 \
        --out BAB.BAP.Random<span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the mean value</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    cat BAB.BAP.Random<span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | awk <span class="hljs-string">'$6&gt;=0 {print}'</span> | awk <span class="hljs-string">'{sum+=$6} END {print sum/NR}'</span> \
        &gt; BAB.BAP.Random<span class="hljs-variable">$i</span>.meanFst
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the permutation result</span>
cat *Random*.meanFst &gt; BAB.BAP.Random.meanFst

<span class="hljs-comment"># Remove tempory file</span>
rm *<span class="hljs-built_in">log</span> *weir.fst BAB.BAP.Random*.list
rm BAB.BAP.Random*.meanFst

<span class="hljs-comment"># Generate scripts for significant t test in R</span>
<span class="hljs-built_in">echo</span> <span class="hljs-string">"
data &lt;- read.table("</span>BAB.BAP.Random.meanFst<span class="hljs-string">", header=FALSE)
AA &lt;- t.test(x=data, mu=0.31, conf.level=0.95)
BAPrs &lt;- capture.output(print(AA))
writeLines(BAPrs, con = file("</span>t.test.txt<span class="hljs-string">")) "</span> | \
sed <span class="hljs-string">'s/BAB.BAP.Random.meanFst/"BAB.BAP.Random.meanFst"/g'</span> |\
sed <span class="hljs-string">'s/t.test.txt/"BAB.BAP.t.test.txt"/g'</span> \
&gt; BAB.BAP.t.test.R

<span class="hljs-comment"># Significant t test in R</span>
Rscript BAB.BAP.t.test.R

</div></code></pre>
<h4 id="fst-permutation-for-pub-with-the-other-groups">Fst permutation for PUB with the other groups</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/09Fst/permutations

<span class="hljs-comment"># Run Fst permutation</span>
<span class="hljs-keyword">for</span> group <span class="hljs-keyword">in</span> CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Prepare the list for permutation test</span>
    cat ../../Pop.list/PUB.list ../../Pop.list/<span class="hljs-variable">$group</span>.list &gt; PUB.<span class="hljs-variable">$group</span>.list
    <span class="hljs-comment"># Randomize the samples </span>
    sort --random-sort PUB.<span class="hljs-variable">$group</span>.list | head -38 &gt; PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.PUB.list
    grep -vf PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.PUB.list PUB.<span class="hljs-variable">$group</span>.list &gt; PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.<span class="hljs-variable">$group</span>.list
    
    <span class="hljs-comment"># Run the permutation test with 100 replicates</span>
    vcftools --gzvcf ../../08SNPprune/Chr<span class="hljs-variable">$j</span>.prune.vcf.gz \
        --weir-fst-pop PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.PUB.list \
        --weir-fst-pop PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.<span class="hljs-variable">$group</span>.list \
        --fst-window-size 100000 \
        --fst-window-step 10000 \
        --out PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>
    <span class="hljs-comment"># Calculate the mean value</span>
    cat PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | awk <span class="hljs-string">'$6&gt;=0 {print}'</span> | awk <span class="hljs-string">'{sum+=$6} END {print sum/NR}'</span> \
        &gt; PUB.<span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.meanFst
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the permutation result</span>
<span class="hljs-keyword">for</span> group <span class="hljs-keyword">in</span> CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
cat PUB.*Random*.meanFst &gt; PUB.<span class="hljs-variable">$group</span>.Random.meanFst
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove tempory file</span>
rm *<span class="hljs-built_in">log</span> *weir.fst PUB.<span class="hljs-variable">$group</span>.Random*.list
rm PUB.<span class="hljs-variable">$group</span>.Random*.meanFst

<span class="hljs-comment"># Generate scripts for significant t test in R</span>
<span class="hljs-built_in">echo</span> <span class="hljs-string">"
data &lt;- read.table("</span>PUB.<span class="hljs-variable">$group</span>.Random.meanFst<span class="hljs-string">", header=FALSE)
AA &lt;- t.test(x=data, mu=0.31, conf.level=0.95, alternative = "</span>two.sided<span class="hljs-string">", )
<span class="hljs-variable">$grouprs</span> &lt;- capture.output(print(AA))
writeLines(<span class="hljs-variable">$grouprs</span>, con = file("</span>t.test.txt<span class="hljs-string">")) "</span> | \
sed <span class="hljs-string">'s/PUB.$group.Random.meanFst/"PUB.$group.Random.meanFst"/g'</span> |\
sed <span class="hljs-string">'s/t.test.txt/"PUB.$group.t.test.txt"/g'</span> \
&gt; PUB.<span class="hljs-variable">$group</span>.t.test.R

<span class="hljs-comment"># Significant t test in R</span>
Rscript PUB.<span class="hljs-variable">$group</span>.t.test.R

</div></code></pre>
<h4 id="top-1-fst">Top 1% Fst</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/09Fst

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *fst.txt | sed <span class="hljs-string">'s/.txt//g'</span> | grep -v CHA | grep -v BAB | grep -v GLA); <span class="hljs-keyword">do</span> 
cut -f1,8 <span class="hljs-variable">$i</span>.txt | grep -v ID | sed <span class="hljs-string">'s/_/\t/g'</span> | awk <span class="hljs-string">'{print $1"_"$2,$1,$2,$3,$4}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; <span class="hljs-variable">$i</span>-2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Top 1% signal</span>
wc -l *fst-2.txt
sort -k5gr BAP.ANN.fst-2.txt | head -2854 &gt; BAP.ANN.fst.top1.txt
sort -k5gr BAP.CHN.fst-2.txt | head -2896 &gt; BAP.CHN.fst.top1.txt
sort -k5gr BAP.FRU.fst-2.txt | head -2896 &gt; BAP.FRU.fst.top1.txt
sort -k5gr CHN.ANN.fst-2.txt | head -2893 &gt; CHN.ANN.fst.top1.txt
sort -k5gr FRU.ANN.fst-2.txt | head -2894 &gt; FRU.ANN.fst.top1.txt
sort -k5gr FRU.CHN.fst-2.txt | head -2890 &gt; FRU.CHN.fst.top1.txt
sort -k5gr PUB.ANN.fst-2.txt | head -2805 &gt; PUB.ANN.fst.top1.txt
sort -k5gr PUB.BAP.fst-2.txt | head -2865 &gt; PUB.BAP.fst.top1.txt
sort -k5gr PUB.CHN.fst-2.txt | head -2883 &gt; PUB.CHN.fst.top1.txt
sort -k5gr PUB.FRU.fst-2.txt | head -2882 &gt; PUB.FRU.fst.top1.txt

<span class="hljs-comment"># Sort the file and add header in local excel</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *top1.txt | sed <span class="hljs-string">'s/.txt//g'</span>); <span class="hljs-keyword">do</span> perl merge.pl <span class="hljs-variable">$i</span>.txt &gt; <span class="hljs-variable">$i</span>.merged.txt ; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove temporary files</span>
rm rm *fst-2.txt *top1

<span class="hljs-comment"># Calculate the total differentiated size for the five domesticated species</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *top1.merged.txt | grep -v BAB | grep -v CHA | grep -v GLA); <span class="hljs-keyword">do</span>
    sed <span class="hljs-string">'1d'</span> <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{sum += $3-$2+1 } END { print sum }'</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Zhangshugang bed</span>
awk <span class="hljs-string">'$3=="gene" {print $1,$4,$5,$9}'</span> ../../References/S8/Zhangshugang.gff | \
    sed <span class="hljs-string">'s/;/ /g'</span> | cut -f1-4 -d <span class="hljs-string">' '</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Zhangshugang.bed

<span class="hljs-comment"># Find the overlapped genes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *top1.merged.txt | sed <span class="hljs-string">'s/.merged.txt//g'</span> | grep -v BAB | grep -v CHA | grep -v GLA); <span class="hljs-keyword">do</span>
    bedtools intersect -a Zhangshugang.bed \
        -b &lt;(sed <span class="hljs-string">'1d'</span> <span class="hljs-variable">$i</span>.merged.txt) | cut -f4 \
        &gt; <span class="hljs-variable">$i</span>.overlapped.genes
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="ld-estimation">LD estimation</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/10LD

<span class="hljs-comment"># Use all SNPs</span>
<span class="hljs-comment"># Use PopLDdecay</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
        PopLDdecay -InVCF ../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz -SubPop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            -MaxDist 2000 -MAF 0.05 -OutStat <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Plot the LD</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        /data/zhaojiantao/tools/PopLDdecay-master/bin/Plot_OnePop.pl -inFile <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.stat.gz -output <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add chromosome information to each stat files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    zcat <span class="hljs-variable">$i</span>.Chr01.bin.gz | awk <span class="hljs-string">'{print $1,$2,1}'</span> &gt; <span class="hljs-variable">$i</span>.Chr01.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr02.bin.gz | awk <span class="hljs-string">'{print $1,$2,2}'</span> &gt; <span class="hljs-variable">$i</span>.Chr02.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr03.bin.gz | awk <span class="hljs-string">'{print $1,$2,3}'</span> &gt; <span class="hljs-variable">$i</span>.Chr03.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr04.bin.gz | awk <span class="hljs-string">'{print $1,$2,4}'</span> &gt; <span class="hljs-variable">$i</span>.Chr04.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr05.bin.gz | awk <span class="hljs-string">'{print $1,$2,5}'</span> &gt; <span class="hljs-variable">$i</span>.Chr05.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr06.bin.gz | awk <span class="hljs-string">'{print $1,$2,6}'</span> &gt; <span class="hljs-variable">$i</span>.Chr06.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr07.bin.gz | awk <span class="hljs-string">'{print $1,$2,7}'</span> &gt; <span class="hljs-variable">$i</span>.Chr07.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr08.bin.gz | awk <span class="hljs-string">'{print $1,$2,8}'</span> &gt; <span class="hljs-variable">$i</span>.Chr08.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr09.bin.gz | awk <span class="hljs-string">'{print $1,$2,9}'</span> &gt; <span class="hljs-variable">$i</span>.Chr09.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr10.bin.gz | awk <span class="hljs-string">'{print $1,$2,10}'</span> &gt; <span class="hljs-variable">$i</span>.Chr10.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr11.bin.gz | awk <span class="hljs-string">'{print $1,$2,11}'</span> &gt; <span class="hljs-variable">$i</span>.Chr11.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr12.bin.gz | awk <span class="hljs-string">'{print $1,$2,12}'</span> &gt; <span class="hljs-variable">$i</span>.Chr12.bin.gz-2 ;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results for each subgroup</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
cat <span class="hljs-variable">$i</span>.Chr*.bin.gz-2 | grep -v <span class="hljs-string">"#"</span> &gt; <span class="hljs-variable">$i</span>.ld
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the average LD across 12 chromosomes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    paste <span class="hljs-variable">$i</span>.Chr*.bin.gz-2 | sed <span class="hljs-string">'1d'</span> | \
        awk <span class="hljs-string">'{print $1,($2+$5+$8+$11+$14+$17+$20+$23+$26+$29+$32+$35)/12}'</span> \
        &gt; <span class="hljs-variable">$i</span>.average.ld
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results</span>
paste *average.ld | awk <span class="hljs-string">'{print $1,$2,$4,$6,$8,$10,$12,$14,$16,$18}'</span> | \
    sed <span class="hljs-string">'1i\Dist A06 A03 A04 A02 A08 A07 A05 A01'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.average.ld.txt


<span class="hljs-comment"># Remove unnecessary files</span>
rm *png *stat.gz *pdf *bin.gz *bin.gz-2

<span class="hljs-comment"># Find the ld at LD =0.3</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ;
    cat <span class="hljs-variable">$i</span>.average.ld | awk <span class="hljs-string">'$2&gt;0.2999 &amp;&amp; $2&lt;0.3001 {print $0}'</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the minimum ld</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.pubescens; <span class="hljs-keyword">do</span> 
    cat <span class="hljs-variable">$i</span>.ld | awk <span class="hljs-string">'$3==1 {print $0}'</span> | sort -nk2 | head -1 | cut -f2 -d <span class="hljs-string">' '</span> ; 
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="ld-for-annuum-clade">LD for Annuum clade</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/10LD

<span class="hljs-comment"># Use PopLDdecay</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> FRU.high FRU.low FRU.mid CHN.high CHN.low CHN.mid ANN.high ANN.low ANN.mid ANN.30 ANN.60 CHN.30 CHN.60 FRU.30 FRU.60; <span class="hljs-keyword">do</span>
        PopLDdecay -InVCF ../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz -SubPop <span class="hljs-variable">$j</span>.list \
            -MaxDist 2000 -MAF 0.05 -OutStat <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Plot the LD</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FRU.high FRU.low FRU.mid CHN.high CHN.low CHN.mid ANN.high ANN.low ANN.mid ANN.30 ANN.60 CHN.30 CHN.60 FRU.30 FRU.60; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        /data/zhaojiantao/tools/PopLDdecay-master/bin/Plot_OnePop.pl -inFile <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.stat.gz -output <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add chromosome information to each stat files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FRU.high FRU.low FRU.mid CHN.high CHN.low CHN.mid ANN.high ANN.low ANN.mid ANN.30 ANN.60 CHN.30 CHN.60 FRU.30 FRU.60; <span class="hljs-keyword">do</span>
    zcat <span class="hljs-variable">$i</span>.Chr01.bin.gz | awk <span class="hljs-string">'{print $1,$2,1}'</span> &gt; <span class="hljs-variable">$i</span>.Chr01.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr02.bin.gz | awk <span class="hljs-string">'{print $1,$2,2}'</span> &gt; <span class="hljs-variable">$i</span>.Chr02.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr03.bin.gz | awk <span class="hljs-string">'{print $1,$2,3}'</span> &gt; <span class="hljs-variable">$i</span>.Chr03.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr04.bin.gz | awk <span class="hljs-string">'{print $1,$2,4}'</span> &gt; <span class="hljs-variable">$i</span>.Chr04.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr05.bin.gz | awk <span class="hljs-string">'{print $1,$2,5}'</span> &gt; <span class="hljs-variable">$i</span>.Chr05.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr06.bin.gz | awk <span class="hljs-string">'{print $1,$2,6}'</span> &gt; <span class="hljs-variable">$i</span>.Chr06.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr07.bin.gz | awk <span class="hljs-string">'{print $1,$2,7}'</span> &gt; <span class="hljs-variable">$i</span>.Chr07.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr08.bin.gz | awk <span class="hljs-string">'{print $1,$2,8}'</span> &gt; <span class="hljs-variable">$i</span>.Chr08.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr09.bin.gz | awk <span class="hljs-string">'{print $1,$2,9}'</span> &gt; <span class="hljs-variable">$i</span>.Chr09.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr10.bin.gz | awk <span class="hljs-string">'{print $1,$2,10}'</span> &gt; <span class="hljs-variable">$i</span>.Chr10.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr11.bin.gz | awk <span class="hljs-string">'{print $1,$2,11}'</span> &gt; <span class="hljs-variable">$i</span>.Chr11.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr12.bin.gz | awk <span class="hljs-string">'{print $1,$2,12}'</span> &gt; <span class="hljs-variable">$i</span>.Chr12.bin.gz-2 ;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results for each subgroup</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FRU.high FRU.low FRU.mid CHN.high CHN.low CHN.mid ANN.high ANN.low ANN.mid ANN.30 ANN.60 CHN.30 CHN.60 FRU.30 FRU.60; <span class="hljs-keyword">do</span>
cat <span class="hljs-variable">$i</span>.Chr*.bin.gz-2 | grep -v <span class="hljs-string">"#"</span> &gt; <span class="hljs-variable">$i</span>.ld
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the average LD across 12 chromosomes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FRU.high FRU.low FRU.mid CHN.high CHN.low CHN.mid ANN.high ANN.low ANN.mid ANN.30 ANN.60 CHN.30 CHN.60 FRU.30 FRU.60; <span class="hljs-keyword">do</span>
    paste <span class="hljs-variable">$i</span>.Chr*.bin.gz-2 | sed <span class="hljs-string">'1d'</span> | \
        awk <span class="hljs-string">'{print $1,($2+$5+$8+$11+$14+$17+$20+$23+$26+$29+$32+$35)/12,"Group"}'</span> | sed <span class="hljs-string">"s/Group/<span class="hljs-variable">$i</span>/g"</span> \
        &gt; <span class="hljs-variable">$i</span>.average.ld
<span class="hljs-keyword">done</span>

cat *average.ld | awk <span class="hljs-string">'{gsub(/\./, " ", $3); print}'</span> | cat - *average.ld-1 | sed <span class="hljs-string">'1i\Bin R2 Group1 Group2'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; ANN.ld.txt

</div></code></pre>
<h3 id="pi">Pi</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/11Pi

<span class="hljs-comment"># Estimate Pi for BAB accessions estimation</span>
<span class="hljs-comment"># Calculate per-site nucleotide diversity</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../04SNP/Miss_0.3/Chr<span class="hljs-variable">$j</span>.SNP.vcf.gz --keep ../Pop.list/<span class="hljs-variable">$i</span>.list --maf 0.05 --site-pi --out <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate windowed pi </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.sites.pi &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results of each chromosome</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    cat <span class="hljs-variable">$i</span>.Chr*.w100000.s10000.txt | \
        grep -v <span class="hljs-string">"#Chr"</span> | \
        awk <span class="hljs-string">'{print $1"_"$3"_"$4,$1,$3,$4,($3+$4)/2,$6,$7}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS PI'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.windowed.pi.txt &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the common ids</span>
awk <span class="hljs-string">'{print $1,$2,$3,$7,"A"}'</span> PUB.windowed.pi.txt &gt; PUB.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"B"}'</span> CHA.windowed.pi.txt &gt; CHA.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"C"}'</span> BAB.windowed.pi.txt &gt; BAB.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"D"}'</span> BAP.windowed.pi.txt &gt; BAP.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"E"}'</span> GLA.windowed.pi.txt &gt; GLA.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"F"}'</span> ANN.windowed.pi.txt &gt; ANN.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"G"}'</span> FRU.windowed.pi.txt &gt; FRU.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"H"}'</span> CHN.windowed.pi.txt &gt; CHN.windowed.pi.txt-2

cat *windowed.pi.txt-2 | grep -v ID | sed <span class="hljs-string">'1i\SNP Chromosome Position PI Taxonomy'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.pi.vg.txt

<span class="hljs-comment"># Calculate the mean value of Pi</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *windowed.pi.txt ; <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ;
    grep -v ID <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{ total += $7 } END { print total/NR }'</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the mean value of pi for all </span>
grep -v SNP Group.pi.vg.txt | awk <span class="hljs-string">'{ total += $4 } END { print total/NR }'</span> 

</div></code></pre>
<h3 id="pi-with-only-window-of-100000">Pi with only window of 100000</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/11Pi/window-pi

<span class="hljs-comment"># Calculate windowed nucleotide diversity</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$j</span>.SNP.vcf.gz --keep ../../Pop.list/<span class="hljs-variable">$i</span>.list --maf 0.05 --window-pi 100000 --out <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the common ids</span>
cat PUB.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"A"}'</span> &gt; PUB.windowed.pi-1
cat CHA.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"B"}'</span> &gt; CHA.windowed.pi-1
cat BAB.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"C"}'</span> &gt; BAB.windowed.pi-1
cat BAP.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"D"}'</span> &gt; BAP.windowed.pi-1
cat GLA.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"E"}'</span> &gt; GLA.windowed.pi-1
cat ANN.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"F"}'</span> &gt; ANN.windowed.pi-1
cat FRU.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"G"}'</span> &gt; FRU.windowed.pi-1
cat CHN.Chr*.windowed.pi | grep -v CHROM | awk <span class="hljs-string">'{print $5,"H"}'</span> &gt; CHN.windowed.pi-1

<span class="hljs-comment"># Combine the results</span>
cat *pi-1 | sed <span class="hljs-string">'1i\PI Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Combined.windowed.pi.txt

<span class="hljs-comment"># Pair-wise significant t-test in R</span>
R
group.pi &lt;- read.table(<span class="hljs-string">"Combined.windowed.pi.txt"</span>, header = TRUE)
pairwise_ttests &lt;- pairwise.t.test(group.pi<span class="hljs-variable">$PI</span>, group.pi<span class="hljs-variable">$Group</span>, method = <span class="hljs-string">"t.test"</span>, p.adjust.method = <span class="hljs-string">"bonferroni"</span>)
pairwise_ttests

</div></code></pre>
<h4 id="permutation-test-using-4dtv-snps">Permutation test using 4DTV SNPs</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/11Pi/permutations

<span class="hljs-comment"># Run the permutation test</span>
<span class="hljs-keyword">for</span> group <span class="hljs-keyword">in</span> PUB CHA BAB BAP FRU CHN GLA ANN; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Generate the random 10 accession list</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    sort --random-sort ../../Pop.list/<span class="hljs-variable">$group</span>.list | head -10 &gt; <span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.list
    <span class="hljs-keyword">done</span>
    <span class="hljs-comment"># Calculate per-site nucleotide diversity</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
            vcftools --gzvcf ../../064DTV/Chr<span class="hljs-variable">$j</span>.ffds.SNP.vcf.gz \
            --keep <span class="hljs-variable">$group</span>.Random<span class="hljs-variable">$i</span>.list --maf 0.05 --site-pi --out Random<span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
        <span class="hljs-keyword">done</span>
    <span class="hljs-keyword">done</span>
    <span class="hljs-comment"># Calculate the average site pi</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    cat Random<span class="hljs-variable">$i</span>.Chr*.sites.pi | grep -v PI | grep -v nan | awk <span class="hljs-string">'{sum += $3} END {print sum/NR}'</span> &gt; Random<span class="hljs-variable">$i</span>.average.pi
    <span class="hljs-keyword">done</span>
    <span class="hljs-comment"># Combine the summary</span>
    cat Random*.average.pi &gt; <span class="hljs-variable">$group</span>.averaged.pi
    <span class="hljs-comment"># Remove temporary files</span>
    rm *<span class="hljs-built_in">log</span> *sites.pi *list *average.pi
<span class="hljs-keyword">done</span>

</div></code></pre>
<h4 id="t-test-in-r">T-test in R</h4>
<pre class="hljs"><code><div>data &lt;- read.table(<span class="hljs-string">"PUB.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"CHA.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"BAB.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"BAP.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"FRU.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"CHN.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"GLA.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

data &lt;- read.table(<span class="hljs-string">"ANN.averaged.pi"</span>, header=<span class="hljs-literal">FALSE</span>)
mu &lt;- mean(data$V1)
t.test(x=data, mu=mu, conf.level=<span class="hljs-number">0.95</span>) 

</div></code></pre>
<h3 id="effective-population-size">Effective population size</h3>
<p>Software link: <a href="https://github.com/popgenmethods/smcpp#tips-for-using-smc">SMC++</a></p>
<h4 id="ann">ANN</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/ANN

<span class="hljs-comment"># Convert vcf to smc format</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    smc++ vcf2smc --cores 10 --missing-cutoff 10000 \
        /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
        ANN:PI_555642,PI_410407,chaotiaojiao,zhudachang,changanyingjiao,Grif_9122,PI_653659,PI_439378,PI_593500,Grif_9381,PI_645493,Grif_12454,PI_640503,PI_438538,PI_645508,PI_260611,PI_631152,PI_631129,qixingjiao,CXJ38,sanweijiao &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 30 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot ANN.smc20.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="gla">GLA</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/GLA

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat GLA.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            GLA:Grif_9142,PI_661082,PI_631136,PI_632930,PI_639662,PI_555616,PI_593527,PI_674459,PI_406948,PI_438567
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 60 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot GLA.smc.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="bab">BAB</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/BAB

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat BAB.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            BAB:PI_260595,PI_446909,PI_439528,PI_441656,PI_639129,PI_633751,PI_337524,PI_441699,PI_441685
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 60 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot BAB.smc.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="bap">BAP</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/BAP

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat BAP.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            BAP:PI_260538,PI_260567,PI_281310,PI_241679,Grif_9201,PI_585250,PI_632926,PI_257161,PI_439398,PI_159242
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 10 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot BAP.smc.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="cha">CHA</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/CHA

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat CHA.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            CHA:PI_260429,PI_439414,PI_260427,PI_260430,PI_260431,PI_439415,PI_639659,PI_659106,PI_639652,PI_555612
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 20 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot CHA.smc.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="chn">CHN</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/CHN

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat CHN.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            CHN:PI_260486,PI_441605,PI_257124,PI_439427,PI_355815,PI_666556,PI_438635,PI_238051,PI_281424,PI_159261
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 20 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot CHN.smc.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="fru">FRU</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/FRU

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat FRU.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            FRU:PI_645681,PI_441690,xiaomila,PI_257104,Grif_9226,PI_360732,PI_631144,PI_368080,PI_439512,PI_281421
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 20 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot FRU.smc.jpeg -g 1 -c smc.out/model.final.json

</div></code></pre>
<h4 id="pub">PUB</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++/PUB

<span class="hljs-comment"># Convert vcf to smc </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat PUB.list) ; <span class="hljs-keyword">do</span>
        smc++ vcf2smc --cores 10 --missing-cutoff 10000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            /data/zhaojiantao/pepper/vg/04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            PUB:Grif_1614,PI_585273,PI_593633,PI_585266,PI_585268,PI_585275,PI_593618,PI_593626,PI_593639,PI_593632
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 20 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot PUB.smc.jpeg -g 1 -c smc.out/model.final.json 

</div></code></pre>
<h4 id="final-combined-plot">Final combined plot</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/12SMC++

<span class="hljs-comment"># Combined final plot</span>
smc++ plot Final.smc.jpeg -g 1 */smc.out/model.final.json -x 100 14000 -y 600 20000

</div></code></pre>
<h3 id="gwas">GWAS</h3>
<h4 id="glaann">GLA.ANN</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/GLA.ANN

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep GLA.ANN.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> GLA.ANN.Chr<span class="hljs-variable">$i</span>.map &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.map2
    mv GLA.ANN.Chr<span class="hljs-variable">$i</span>.map2 GLA.ANN.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> --make-bed --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary GLA.ANN.Chr<span class="hljs-variable">$i</span>
    mv gec.sum GLA.ANN.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep GLA.ANN.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out GLA.ANN.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file GLA.ANN.ffds --pca --out GLA.ANN.ffds

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> GLA.ANN.ffds.eigenvec &gt; GLA.ANN.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file GLA.ANN.ffds --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.ffds --allow-extra-chr
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> GLA.ANN.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - GLA.ANN.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; GLA.ANN.ffds.tped2
mv GLA.ANN.ffds.tped2 GLA.ANN.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 GLA.ANN.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> FruitLength FruitWidth FruitRatio ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t GLA.ANN.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k GLA.ANN.ffds.BN.kinf -c GLA.ANN.ffds.PCA.txt -o GLA.ANN.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FruitLength FruitWidth FruitRatio ; <span class="hljs-keyword">do</span>
    cat GLA.ANN.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat GLA.ANN.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; GLA.ANN.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the LD for the peak SNP</span>
plink --file GLA.ANN.Chr02 \
    --r2 \
    --ld-snp 2_160951499 \
    --ld-window-kb 1000 \
    --ld-window 1050600 \
    --ld-window-r2 0.351 \
    --out 2_160951499

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAP.long BAP.bell ; <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../vcf.clean/Chr10.clean.vcf.gz --keep <span class="hljs-variable">$i</span>.list --site-pi --out <span class="hljs-variable">$i</span>.Chr10
    perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr10.sites.pi
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Step 15.2 Calculate windowed pi </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.sites.pi
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

</div></code></pre>
<h5 id="shape-variations">Shape variations</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/GLA.ANN/Shape

<span class="hljs-comment"># Extrac the samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
plink --file ../GLA.ANN.Chr<span class="hljs-variable">$i</span> --keep sample.list --make-bed --recode --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary GLA.ANN.Chr<span class="hljs-variable">$i</span>
    mv gec.sum GLA.ANN.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Merge the chromosomal-level map/ped files</span>
ls *map | sed <span class="hljs-string">'s/.map//g'</span> &gt; mergefiles.txt
plink --merge-list mergefiles.txt --make-bed --recode --out GLA.ANN

<span class="hljs-comment"># LD prune</span>
plink --file GLA.ANN \
    --indep-pairwise 50 10 0.2 \
    --allow-extra-chr --double-id \
    --out GLA.ANN
<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --file GLA.ANN \
    --extract GLA.ANN.prune.in \
    --recode --double-id \
    --out GLA.ANN.prune

<span class="hljs-comment"># Calculate PCA</span>
plink --file GLA.ANN.prune --pca --out GLA.ANN.prune

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> GLA.ANN.prune.eigenvec &gt; GLA.ANN.prune.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file GLA.ANN.prune --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.prune --allow-extra-chr

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 GLA.ANN.prune

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t GLA.ANN.Chr<span class="hljs-variable">$i</span> -p Shape.normal.txt -k GLA.ANN.prune.BN.kinf -c GLA.ANN.prune.PCA.txt -o GLA.ANN.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare Manhattan plot</span>
cat GLA.ANN.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
    paste - &lt;(cat GLA.ANN.Chr*.ps) | \
    awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; GLA.ANN.Shape.GWAS.txt

</div></code></pre>
<h4 id="babbap">BAB.BAP</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/BAB.BAP

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep BAB.BAP.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> BAB.BAP.Chr<span class="hljs-variable">$i</span>.map &gt; BAB.BAP.Chr<span class="hljs-variable">$i</span>.map2
    mv BAB.BAP.Chr<span class="hljs-variable">$i</span>.map2 BAB.BAP.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file BAB.BAP.Chr<span class="hljs-variable">$i</span> --make-bed --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary BAB.BAP.Chr<span class="hljs-variable">$i</span>
    mv gec.sum BAB.BAP.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep BAB.BAP.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out BAB.BAP.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file BAB.BAP.ffds --pca --out BAB.BAP.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> BAB.BAP.ffds.eigenvec &gt; BAB.BAP.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file BAB.BAP.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file BAB.BAP.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out BAB.BAP.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> BAB.BAP.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - BAB.BAP.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; BAB.BAP.ffds.tped2
mv BAB.BAP.ffds.tped2 BAB.BAP.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 BAB.BAP.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> FruitLength FruitWidth FruitRatio ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t BAB.BAP.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k BAB.BAP.ffds.BN.kinf -c BAB.BAP.ffds.PCA.txt -o BAB.BAP.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FruitLength FruitWidth FruitRatio ; <span class="hljs-keyword">do</span>
    cat BAB.BAP.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat BAB.BAP.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; BAB.BAP.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the LD for the peak SNP</span>
plink --file BAB.BAP.Chr02 \
    --r2 \
    --ld-snp Chr02_172705522 \
    --ld-window-kb 1000 \
    --ld-window 1050600 \
    --ld-window-r2 0.351 \
    --out Chr02_172705522

plink --file BAB.BAP.Chr12 \
    --r2 \
    --ld-snp Chr12_236447804 \
    --ld-window-kb 1000 \
    --ld-window 1050600 \
    --ld-window-r2 0.351 \
    --out Chr12_236447804

bedtools intersect -a ../../../References/S8/Zhangshugang.gff -b AA.bed | \
    grep gene | sed <span class="hljs-string">'s/;/\t/g'</span> | cut -f9 | sed <span class="hljs-string">'s/ID=//g'</span> &gt; Candidate.genes

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAP.long BAP.bell ; <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../../04SNP/Miss_0.3/Chr10.SNP.vcf.gz --keep <span class="hljs-variable">$i</span>.list --site-pi --out <span class="hljs-variable">$i</span>.Chr10 &amp;
    perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr10.sites.pi
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the pi plot for the candidate region</span>
grep -v <span class="hljs-string">"#"</span> BAP.bell.Chr10.sites.pi.w100000.s10000.txt | \
    awk <span class="hljs-string">'$3&gt;120500000 &amp;&amp; $4&lt;122600000 {print $4,$7,"bell"}'</span> &gt; BAP.bell.Chr10.candidate.pi
grep -v <span class="hljs-string">"#"</span> BAP.long.Chr10.sites.pi.w100000.s10000.txt | \
    awk <span class="hljs-string">'$3&gt;120500000 &amp;&amp; $4&lt;122600000 {print $4,$7,"long"}'</span> &gt; BAP.long.Chr10.candidate.pi
grep -v <span class="hljs-string">"CHROM"</span> ../../11Pi/BAB.windowed.pi.txt | grep Chr10 | \
    awk <span class="hljs-string">'$3&gt;120500000 &amp;&amp; $4&lt;122600000 {print $3,$7,"BAB"}'</span> &gt; BAB.Chr10.candidate.pi

</div></code></pre>
<h5 id="shape-variations">Shape variations</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/BAB.BAP/Shape

<span class="hljs-comment"># Extrac the samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
plink --file ../BAB.BAP.Chr<span class="hljs-variable">$i</span> --keep sample.list --make-bed --recode --out BAB.BAP.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary BAB.BAP.Chr<span class="hljs-variable">$i</span>
    mv gec.sum BAB.BAP.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Merge the chromosomal-level map/ped files</span>
ls *map | sed <span class="hljs-string">'s/.map//g'</span> &gt; mergefiles.txt
plink --merge-list mergefiles.txt --make-bed --recode --out BAB.BAP

<span class="hljs-comment"># Extrac the samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
plink --file ../BAB.BAP.Chr<span class="hljs-variable">$i</span> --keep sample.list --make-bed --recode --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary BAB.BAP.Chr<span class="hljs-variable">$i</span>
    mv gec.sum BAB.BAP.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Merge the chromosomal-level map/ped files</span>
ls *map | sed <span class="hljs-string">'s/.map//g'</span> &gt; mergefiles.txt
plink --merge-list mergefiles.txt --make-bed --recode --out BAB.BAP

<span class="hljs-comment"># LD prune</span>
plink --file BAB.BAP \
    --indep-pairwise 50 10 0.2 \
    --allow-extra-chr --double-id \
    --out BAB.BAP
<span class="hljs-comment"># Extract the prunned SNPs</span>
plink --file BAB.BAP \
    --extract BAB.BAP.prune.in \
    --recode --double-id \
    --out BAB.BAP.prune

<span class="hljs-comment"># Calculate PCA</span>
plink --file BAB.BAP.prune --pca --out BAB.BAP.prune

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> BAB.BAP.prune.eigenvec &gt; BAB.BAP.prune.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file BAB.BAP.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out BAB.BAP.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file BAB.BAP.prune --recode12 --output-missing-genotype 0 --transpose --out BAB.BAP.prune --allow-extra-chr

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 BAB.BAP.prune

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t BAB.BAP.Chr<span class="hljs-variable">$i</span> -p Shape.normal.txt -k BAB.BAP.prune.BN.kinf -c BAB.BAP.prune.PCA.txt -o BAB.BAP.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
cat BAB.BAP.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
    paste - &lt;(cat BAB.BAP.Chr*.ps) | \
    awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; BAB.BAP.Shape.GWAS.txt

</div></code></pre>
<h4 id="seed-size">Seed size</h4>
<h5 id="extract-the-phenotype">Extract the phenotype</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/Seed

<span class="hljs-comment"># Extract the samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAB.BAP GLA.ANN CHN FRU Annuum ; <span class="hljs-keyword">do</span>
grep -f ../../Pop.list/<span class="hljs-variable">$i</span>.list seed.txt | awk <span class="hljs-string">'{print $1,$1,$2}'</span> | sort -k1,1 | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; <span class="hljs-variable">$i</span>/L.txt
grep -f ../../Pop.list/<span class="hljs-variable">$i</span>.list seed.txt | awk <span class="hljs-string">'{print $1,$1,$3}'</span> | sort -k1,1 | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; <span class="hljs-variable">$i</span>/R.txt
grep -f ../../Pop.list/<span class="hljs-variable">$i</span>.list seed.txt | awk <span class="hljs-string">'{print $1,$1,$4}'</span> | sort -k1,1 | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; <span class="hljs-variable">$i</span>/S.txt
grep -f ../../Pop.list/<span class="hljs-variable">$i</span>.list seed.txt | awk <span class="hljs-string">'{print $1,$1,$5}'</span> | sort -k1,1 | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; <span class="hljs-variable">$i</span>/W.txt
<span class="hljs-keyword">done</span>

</div></code></pre>
<h5 id="glaann">GLA.ANN</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/Seed/GLA.ANN

<span class="hljs-comment"># Prepare the sample list</span>
awk <span class="hljs-string">'{print $1,$1}'</span> L.txt | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; GLA.ANN.list

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep GLA.ANN.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> GLA.ANN.Chr<span class="hljs-variable">$i</span>.map &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.map2
    mv GLA.ANN.Chr<span class="hljs-variable">$i</span>.map2 GLA.ANN.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> --make-bed --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary GLA.ANN.Chr<span class="hljs-variable">$i</span>
    mv gec.sum GLA.ANN.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep GLA.ANN.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out GLA.ANN.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file GLA.ANN.ffds --pca --out GLA.ANN.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> GLA.ANN.ffds.eigenvec &gt; GLA.ANN.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file GLA.ANN.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out GLA.ANN.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> GLA.ANN.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - GLA.ANN.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; GLA.ANN.ffds.tped2
mv GLA.ANN.ffds.tped2 GLA.ANN.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 GLA.ANN.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t GLA.ANN.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k GLA.ANN.ffds.BN.kinf -c GLA.ANN.ffds.PCA.txt -o GLA.ANN.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    cat GLA.ANN.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat GLA.ANN.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; GLA.ANN.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

plink --file GLA.ANN.Chr07 \
    --r2 \
    --ld-snp Chr07_13218329 \
    --ld-window-kb 1000 \
    --ld-window 1000000 \
    --ld-window-r2 0.5 \
    --out Chr07_13218329

</div></code></pre>
<h6 id="manhattan-plot">Manhattan plot</h6>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
GLA.ANN.L.GWAS &lt;- fread(<span class="hljs-string">"GLA.ANN.L.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.L.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.L.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

GLA.ANN.R.GWAS &lt;- fread(<span class="hljs-string">"GLA.ANN.R.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.R.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.R.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">30</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

GLA.ANN.S.GWAS &lt;- fread(<span class="hljs-string">"GLA.ANN.S.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.S.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.S.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

GLA.ANN.W.GWAS &lt;- fread(<span class="hljs-string">"GLA.ANN.W.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.W.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.W.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

</div></code></pre>
<h5 id="fru">FRU</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/Seed/FRU

<span class="hljs-comment"># Prepare the sample list</span>
awk <span class="hljs-string">'{print $1,$1}'</span> L.txt | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; FRU.list

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep FRU.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out FRU.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> FRU.Chr<span class="hljs-variable">$i</span>.map &gt; FRU.Chr<span class="hljs-variable">$i</span>.map2
    mv FRU.Chr<span class="hljs-variable">$i</span>.map2 FRU.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file FRU.Chr<span class="hljs-variable">$i</span> --make-bed --out FRU.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary FRU.Chr<span class="hljs-variable">$i</span>
    mv gec.sum FRU.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep FRU.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out FRU.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file FRU.ffds --pca --out FRU.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> FRU.ffds.eigenvec &gt; FRU.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file FRU.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out FRU.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file FRU.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out FRU.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> FRU.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - FRU.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; FRU.ffds.tped2
mv FRU.ffds.tped2 FRU.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 FRU.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t FRU.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k FRU.ffds.BN.kinf -c FRU.ffds.PCA.txt -o FRU.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    cat FRU.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat FRU.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; FRU.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

</div></code></pre>
<h6 id="manhattan-plot">Manhattan plot</h6>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
FRU.L.GWAS &lt;- fread(<span class="hljs-string">"FRU.L.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"FRU.L.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(FRU.L.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

FRU.R.GWAS &lt;- fread(<span class="hljs-string">"FRU.R.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"FRU.R.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(FRU.R.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

FRU.S.GWAS &lt;- fread(<span class="hljs-string">"FRU.S.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"FRU.S.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(FRU.S.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

FRU.W.GWAS &lt;- fread(<span class="hljs-string">"FRU.W.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"FRU.W.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(FRU.W.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.58721e+06</span>))
dev.off()

</div></code></pre>
<h5 id="chn">CHN</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/Seed/CHN

<span class="hljs-comment"># Prepare the sample list</span>
awk <span class="hljs-string">'{print $1,$1}'</span> L.txt | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; CHN.list

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep CHN.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out CHN.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> CHN.Chr<span class="hljs-variable">$i</span>.map &gt; CHN.Chr<span class="hljs-variable">$i</span>.map2
    mv CHN.Chr<span class="hljs-variable">$i</span>.map2 CHN.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file CHN.Chr<span class="hljs-variable">$i</span> --make-bed --out CHN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary CHN.Chr<span class="hljs-variable">$i</span>
    mv gec.sum CHN.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep CHN.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out CHN.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file CHN.ffds --pca --out CHN.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> CHN.ffds.eigenvec &gt; CHN.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file CHN.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out CHN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file CHN.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out CHN.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> CHN.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - CHN.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; CHN.ffds.tped2
mv CHN.ffds.tped2 CHN.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 CHN.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t CHN.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k CHN.ffds.BN.kinf -c CHN.ffds.PCA.txt -o CHN.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    cat CHN.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat CHN.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; CHN.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Candidate region</span>
plink --file CHN.Chr08 \
    --r2 \
    --ld-snp Chr08_107756202 \
    --ld-window-kb 1000 \
    --ld-window 1000000 \
    --ld-window-r2 0.5 \
    --out Chr08_107756202

</div></code></pre>
<h6 id="manhattan-plot">Manhattan plot</h6>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
CHN.L.GWAS &lt;- fread(<span class="hljs-string">"CHN.L.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"CHN.L.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(CHN.L.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.48239e+06</span>))
dev.off()

CHN.R.GWAS &lt;- fread(<span class="hljs-string">"CHN.R.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"CHN.R.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(CHN.R.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.48239e+06</span>))
dev.off()

CHN.S.GWAS &lt;- fread(<span class="hljs-string">"CHN.S.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"CHN.S.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(CHN.S.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.48239e+06</span>))
dev.off()

CHN.W.GWAS &lt;- fread(<span class="hljs-string">"CHN.W.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"CHN.W.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(CHN.W.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.48239e+06</span>))
dev.off()

</div></code></pre>
<h5 id="annuum">Annuum</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/Seed/Annuum

<span class="hljs-comment"># Prepare the sample list</span>
awk <span class="hljs-string">'{print $1,$1}'</span> L.txt | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Annuum.list

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep Annuum.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out Annuum.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> Annuum.Chr<span class="hljs-variable">$i</span>.map &gt; Annuum.Chr<span class="hljs-variable">$i</span>.map2
    mv Annuum.Chr<span class="hljs-variable">$i</span>.map2 Annuum.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file Annuum.Chr<span class="hljs-variable">$i</span> --make-bed --out Annuum.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary Annuum.Chr<span class="hljs-variable">$i</span>
    mv gec.sum Annuum.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep Annuum.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out Annuum.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file Annuum.ffds --pca --out Annuum.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> Annuum.ffds.eigenvec &gt; Annuum.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file Annuum.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out Annuum.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file Annuum.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out Annuum.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> Annuum.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - Annuum.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; Annuum.ffds.tped2
mv Annuum.ffds.tped2 Annuum.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 Annuum.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t Annuum.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k Annuum.ffds.BN.kinf -c Annuum.ffds.PCA.txt -o Annuum.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    cat Annuum.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat Annuum.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Annuum.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

</div></code></pre>
<h6 id="manhattan-plot">Manhattan plot</h6>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
Annuum.L.GWAS &lt;- fread(<span class="hljs-string">"Annuum.L.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"Annuum.L.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(Annuum.L.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

Annuum.R.GWAS &lt;- fread(<span class="hljs-string">"Annuum.R.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"Annuum.R.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(Annuum.R.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

Annuum.S.GWAS &lt;- fread(<span class="hljs-string">"Annuum.S.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"Annuum.S.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(Annuum.S.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

Annuum.W.GWAS &lt;- fread(<span class="hljs-string">"Annuum.W.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"Annuum.W.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(Annuum.W.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

</div></code></pre>
<h5 id="babbap">BAB.BAP</h5>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/Seed/BAB.BAP

<span class="hljs-comment"># Prepare the sample list</span>
awk <span class="hljs-string">'{print $1,$1}'</span> L.txt | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; BAB.BAP.list

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep BAB.BAP.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> BAB.BAP.Chr<span class="hljs-variable">$i</span>.map &gt; BAB.BAP.Chr<span class="hljs-variable">$i</span>.map2
    mv BAB.BAP.Chr<span class="hljs-variable">$i</span>.map2 BAB.BAP.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file BAB.BAP.Chr<span class="hljs-variable">$i</span> --make-bed --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary BAB.BAP.Chr<span class="hljs-variable">$i</span>
    mv gec.sum BAB.BAP.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes: 3.19274e+06</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep BAB.BAP.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out BAB.BAP.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file BAB.BAP.ffds --pca --out BAB.BAP.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> BAB.BAP.ffds.eigenvec &gt; BAB.BAP.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file BAB.BAP.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out BAB.BAP.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file BAB.BAP.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out BAB.BAP.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> BAB.BAP.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - BAB.BAP.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; BAB.BAP.ffds.tped2
mv BAB.BAP.ffds.tped2 BAB.BAP.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 BAB.BAP.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t BAB.BAP.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k BAB.BAP.ffds.BN.kinf -c BAB.BAP.ffds.PCA.txt -o BAB.BAP.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    cat BAB.BAP.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat BAB.BAP.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; BAB.BAP.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

</div></code></pre>
<h6 id="manhattan-plot">Manhattan plot</h6>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
BAB.BAP.L.GWAS &lt;- fread(<span class="hljs-string">"BAB.BAP.L.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.L.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.L.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

BAB.BAP.R.GWAS &lt;- fread(<span class="hljs-string">"BAB.BAP.R.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.R.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.R.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">20</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

BAB.BAP.S.GWAS &lt;- fread(<span class="hljs-string">"BAB.BAP.S.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.S.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.S.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

BAB.BAP.W.GWAS &lt;- fread(<span class="hljs-string">"BAB.BAP.W.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.W.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.W.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">3.19274e+06</span>))
dev.off()

</div></code></pre>
<h4 id="chn-fruit-shape">CHN fruit shape</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/13GWAS/CHN

<span class="hljs-comment"># Prepare the sample list</span>
awk <span class="hljs-string">'{print $1,$1}'</span> L.txt | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; CHN.list

<span class="hljs-comment"># Convert vcf to tped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        --keep CHN.list \
        --allow-extra-chr \
        --recode \
        --double-id \
        --geno 0.7 \
        --maf 0.05 \
        --out CHN.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Rename the chr id</span>
    sed <span class="hljs-string">'s/CannZSG_//g'</span> CHN.Chr<span class="hljs-variable">$i</span>.map &gt; CHN.Chr<span class="hljs-variable">$i</span>.map2
    mv CHN.Chr<span class="hljs-variable">$i</span>.map2 CHN.Chr<span class="hljs-variable">$i</span>.map
    <span class="hljs-comment"># Make bed</span>
    plink --file CHN.Chr<span class="hljs-variable">$i</span> --make-bed --out CHN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary CHN.Chr<span class="hljs-variable">$i</span>
    mv gec.sum CHN.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes: 3.19274e+06</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../064DTV/504.4DTV.Miss0.1.vcf.gz | grep -v Chr00) \
    --keep CHN.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.7 \
    --maf 0.05 \
    --out CHN.ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file CHN.ffds --pca --out CHN.ffds --allow-extra-chr

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> CHN.ffds.eigenvec &gt; CHN.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file CHN.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out CHN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file CHN.ffds --recode12 --output-missing-genotype 0 --transpose --allow-extra-chr --out CHN.ffds
<span class="hljs-comment"># Rename the chr id</span>
cut -f1 -d <span class="hljs-string">' '</span> CHN.ffds.tped | sed <span class="hljs-string">'s/CannZSG_Chr0//g'</span> | sed <span class="hljs-string">'s/CannZSG_Chr//g'</span> | \
    paste - CHN.ffds.tped | sed <span class="hljs-string">'s/\t/ /g'</span> | cut -f1,3- -d <span class="hljs-string">' '</span> &gt; CHN.ffds.tped2
mv CHN.ffds.tped2 CHN.ffds.tped

<span class="hljs-comment"># Estimate kinship</span>
emmax-kin -v -d 10 CHN.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t CHN.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k CHN.ffds.BN.kinf -c CHN.ffds.PCA.txt -o CHN.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> L R S W ; <span class="hljs-keyword">do</span>
    cat CHN.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
        paste - &lt;(cat CHN.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; CHN.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Candidate region</span>
plink --file CHN.Chr07 \
    --r2 \
    --ld-snp Chr07_49272382 \
    --ld-window-kb 1000 \
    --ld-window 1000000 \
    --ld-window-r2 0.5 \
    --out Chr07_49272382

</div></code></pre>
<h6 id="manhattan-plot">Manhattan plot</h6>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
CHN.L.GWAS &lt;- fread(<span class="hljs-string">"CHN.L.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"CHN.L.GWAS.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">16</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(CHN.L.GWAS, chr = <span class="hljs-string">"CHR"</span>, bp = <span class="hljs-string">"BP"</span>,
          snp = <span class="hljs-string">"SNP"</span>,col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>),
          p = <span class="hljs-string">"P"</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>), logp = <span class="hljs-literal">TRUE</span>,cex=<span class="hljs-number">0.3</span>, 
          cex.axis = <span class="hljs-number">0.5</span>,cex.main = <span class="hljs-number">1</span>, 
          suggestiveline = <span class="hljs-literal">FALSE</span>, genomewideline = -log10(<span class="hljs-number">1</span>/<span class="hljs-number">2.13784e+06</span>))
dev.off()


</div></code></pre>
<h3 id="sweed">SweeD</h3>
<h4 id="installation">Installation</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Installation</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/tools
git <span class="hljs-built_in">clone</span> https://github.com/alachins/sweed.git
<span class="hljs-built_in">cd</span> sweed
<span class="hljs-comment"># Install</span>
make -f Makefile.gcc
make -f Makefile.PTHREADS.gcc

<span class="hljs-comment"># Install DMTCP library</span>
DMTCPAWARELIB=/data/zhaojiantao/tools/sweed/dmtcp/lib/dmtcp

</div></code></pre>
<h4 id="bap-with-10k-grid">BAP with 10k grid</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/14SweeD/BAP10K

<span class="hljs-comment"># SweeD does not accept zipped vcf</span>
<span class="hljs-comment"># Extract the samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    bcftools view -S BAP.list \
        ../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        &gt; Chr<span class="hljs-variable">$i</span>.SNP.vcf &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep Chr ../../PanReference/Cann_Zhangshugang.chr.fa.fai | sed <span class="hljs-string">'s/CannZSG_//g'</span> | \
    awk <span class="hljs-string">'{print $1,int($2/10000+0.5)}'</span> \
    &gt; Chr.grids.10K

<span class="hljs-comment"># Run SweeD</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids.10K); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    <span class="hljs-built_in">echo</span> <span class="hljs-string">" SweeD-MPFR-P -name <span class="hljs-variable">$Chr</span> -input <span class="hljs-variable">$Chr</span>.SNP.vcf -grid <span class="hljs-variable">$grid</span> -maf 0.05 -missing 0.3 -threads 60 \
    -s 218 -eN 0.093 0.908 -eN 0.318 1.41 -eN 0.588 3.85 -eN 1.088 3.879 "</span>
<span class="hljs-keyword">done</span> &gt; sweed.sh

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "Chr01_"$1,"1",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "Chr02_"$1,"2",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "Chr03_"$1,"3",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "Chr04_"$1,"4",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "Chr05_"$1,"5",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "Chr06_"$1,"6",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "Chr07_"$1,"7",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "Chr08_"$1,"8",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "Chr09_"$1,"9",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "Chr10_"$1,"10",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "Chr11_"$1,"11",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "Chr12_"$1,"12",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | \
    sed <span class="hljs-string">'1i\SNP Chr Pos Start End SweeD'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.BAP.10k.txt

<span class="hljs-comment"># Reverse sort </span>
grep -v SNP SweeD_Report.BAP.10k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | head -3021 &gt; SweeD_Report.BAP.10k.top1.txt
grep -v SNP SweeD_Report.BAP.10k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | head -15106 &gt; SweeD_Report.BAP.10k.top5.txt
cut -f1 SweeD_Report.BAP.10k.top5.txt | sed <span class="hljs-string">'s/_/\t/g'</span> | paste - SweeD_Report.BAP.10k.top5.txt | \
    awk <span class="hljs-string">'{print $3,$1,int($2+0.5),int($2+1+0.5),$8}'</span> | sed <span class="hljs-string">'1i\SNP Chr Start End XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.BAP.10k.top5-1.txt

<span class="hljs-comment"># Sort the file in excel and add header</span>

<span class="hljs-comment"># cutoff</span>
grep -v SNP SweeD_Report.BAP.10k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | sed -n <span class="hljs-string">'3021p'</span>
grep -v SNP SweeD_Report.BAP.10k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | sed -n <span class="hljs-string">'15106p'</span>

<span class="hljs-comment"># Merge the sweeps with distance &lt;100000</span>
perl merge.pl SweeD_Report.BAP.10k.top1-1.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD_Report.BAP.10k.top1.merged.txt
perl merge.pl SweeD_Report.BAP.10k.top5-1.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD_Report.BAP.10k.top5.merged.txt
</div></code></pre>
<h4 id="bap-with-100k-size">BAP with 100k size</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/14SweeD/BAP100K

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep Chr ../../PanReference/Cann_Zhangshugang.chr.fa.fai | sed <span class="hljs-string">'s/CannZSG_//g'</span> | \
    awk <span class="hljs-string">'{print $1,int($2/100000+0.5)}'</span> \
    &gt; Chr.grids.100K

<span class="hljs-comment"># Generate command lines</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids.100K); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    <span class="hljs-built_in">echo</span> <span class="hljs-string">" SweeD-MPFR-P -name <span class="hljs-variable">$Chr</span> -input ../BAP10K/<span class="hljs-variable">$Chr</span>.SNP.vcf -grid <span class="hljs-variable">$grid</span> -threads 60 -maf 0.05 -missing 0.3 \
    -s 218 -eN 0.093 0.908 -eN 0.318 1.41 -eN 0.588 3.85 -eN 1.088 3.879 "</span>
<span class="hljs-keyword">done</span> &gt; sweed.100K.sh

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "Chr01_"$1,"1",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "Chr02_"$1,"2",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "Chr03_"$1,"3",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "Chr04_"$1,"4",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "Chr05_"$1,"5",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "Chr06_"$1,"6",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "Chr07_"$1,"7",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "Chr08_"$1,"8",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "Chr09_"$1,"9",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "Chr10_"$1,"10",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "Chr11_"$1,"11",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "Chr12_"$1,"12",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | \
    sed <span class="hljs-string">'1i\SNP Chr Pos Start End SweeD'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.BAP.100k.txt

<span class="hljs-comment"># Reverse sort </span>
grep -v SNP SweeD_Report.BAP.100k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | head -302 &gt; SweeD_Report.BAP.100k.top1.txt
<span class="hljs-comment"># Sort the file in excel and add header</span>

<span class="hljs-comment"># cutoff</span>
grep -v SNP SweeD_Report.BAP.100k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | sed -n <span class="hljs-string">'302p'</span>

<span class="hljs-comment"># Merge the sweeps with distance &lt;100000</span>
perl merge.pl SweeD_Report.BAP.100k.top1.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD_Report.BAP.100k.top1.merged.txt


</div></code></pre>
<h4 id="ann-with-10k-grid">ANN with 10k grid</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/14SweeD/ANN

<span class="hljs-comment"># Extract the samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    bcftools view -S ANN.list \
        ../../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        &gt; Chr<span class="hljs-variable">$i</span>.SNP.vcf
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep Chr ../../PanReference/Cann_Zhangshugang.chr.fa.fai | sed <span class="hljs-string">'s/CannZSG_//g'</span> | \
    awk <span class="hljs-string">'{print $1,int($2/10000+0.5)}'</span> \
    &gt; Chr.grids.10K

<span class="hljs-comment"># Run SweeD</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids.10K); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    <span class="hljs-built_in">echo</span> <span class="hljs-string">" SweeD-MPFR-P -name <span class="hljs-variable">$Chr</span> -input <span class="hljs-variable">$Chr</span>.SNP.vcf -grid <span class="hljs-variable">$grid</span> -threads 80 -maf 0.05 -missing 0.3 \
    -s 150 -eN 0.033 0.347 -eN 0.056 0.415 -eN 0.092 0.574 -eN 0.157 1.298 -eN 0.254 3.585 -eN 0.484 5.908 "</span>
<span class="hljs-keyword">done</span> &gt; sweed.sh

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "Chr01_"$1,"1",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "Chr02_"$1,"2",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "Chr03_"$1,"3",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "Chr04_"$1,"4",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "Chr05_"$1,"5",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "Chr06_"$1,"6",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "Chr07_"$1,"7",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "Chr08_"$1,"8",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "Chr09_"$1,"9",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "Chr10_"$1,"10",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "Chr11_"$1,"11",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "Chr12_"$1,"12",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | \
    sed <span class="hljs-string">'1i\SNP Chr Pos Start End SweeD'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.ANN.10k.txt

<span class="hljs-comment"># Reverse sort and top1% signals</span>
sort -k6gr SweeD_Report.ANN.10k.txt | head -3021 &gt; SweeD_Report.ANN.10k.top1.txt
sort -k6gr SweeD_Report.ANN.10k.txt | head -15106 &gt; SweeD_Report.ANN.10k.top5.txt
cut -f1 SweeD_Report.ANN.10k.top5.txt | sed <span class="hljs-string">'s/_/\t/g'</span> | paste - SweeD_Report.ANN.10k.top5.txt | \
    awk <span class="hljs-string">'{print $3,$1,int($2+0.5),int($2+1+0.5),$8}'</span> | sed <span class="hljs-string">'1i\SNP Chr Start End XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.ANN.10k.top5-1.txt

<span class="hljs-comment"># cutoff</span>
sort -k6gr SweeD_Report.ANN.10k.txt | sed -n <span class="hljs-string">'3021p'</span>
sort -k6gr SweeD_Report.ANN.10k.txt | sed -n <span class="hljs-string">'15106p'</span>

<span class="hljs-comment"># Merge the sweeps with distance &lt;100000</span>
perl merge.pl SweeD_Report.ANN.10k.top1-1.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD_Report.ANN.10k.top1.merged.txt
perl merge.pl SweeD_Report.ANN.10k.top5-1.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD_Report.ANN.10k.top5.merged.txt

</div></code></pre>
<h4 id="ann-with-100k-grid">ANN with 100k grid</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/14SweeD/ANN100K

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep Chr ../../PanReference/Cann_Zhangshugang.chr.fa.fai | sed <span class="hljs-string">'s/CannZSG_//g'</span> | \
    awk <span class="hljs-string">'{print $1,int($2/100000+0.5)}'</span> \
    &gt; Chr.grids.100K

<span class="hljs-comment"># Generate command lines</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids.100K); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    <span class="hljs-built_in">echo</span> <span class="hljs-string">" SweeD-MPFR-P -name <span class="hljs-variable">$Chr</span> -input ../ANN10K/<span class="hljs-variable">$Chr</span>.SNP.vcf -grid <span class="hljs-variable">$grid</span> -threads 80 -maf 0.05 -missing 0.3 \
    -s 150 -eN 0.033 0.347 -eN 0.056 0.415 -eN 0.092 0.574 -eN 0.157 1.298 -eN 0.254 3.585 -eN 0.484 5.908  "</span>
<span class="hljs-keyword">done</span> &gt; sweed.100K.sh

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "Chr01_"$1,"1",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "Chr02_"$1,"2",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "Chr03_"$1,"3",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "Chr04_"$1,"4",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "Chr05_"$1,"5",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "Chr06_"$1,"6",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "Chr07_"$1,"7",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "Chr08_"$1,"8",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "Chr09_"$1,"9",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "Chr10_"$1,"10",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "Chr11_"$1,"11",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "Chr12_"$1,"12",$1,$4,$5,$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | \
    sed <span class="hljs-string">'1i\SNP Chr Pos Start End SweeD'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.ANN.100k.txt

<span class="hljs-comment"># Reverse sort </span>
grep -v SNP SweeD_Report.ANN.100k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | head -302 &gt; SweeD_Report.ANN.100k.top1.txt
<span class="hljs-comment"># Sort the file in excel and add header</span>

<span class="hljs-comment"># cutoff</span>
grep -v SNP SweeD_Report.ANN.100k.txt | awk <span class="hljs-string">'$6&gt;0 {print}'</span> | sort -k6gr | sed -n <span class="hljs-string">'302p'</span>

<span class="hljs-comment"># Merge the sweeps with distance &lt;100000</span>
perl merge.pl SweeD_Report.ANN.10k.top1.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD_Report.ANN.10k.top1.merged.txt

</div></code></pre>
<h4 id="overlaped-regions">Overlaped regions</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/14SweeD/Overlapp

<span class="hljs-comment"># Top GLA.ANN pi ratio</span>
grep -v SNP Pi.GLA.ANN.txt  | head -28826 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.GLA.ANN.top10.bed
grep -v SNP Pi.GLA.ANN.txt  | head -57652 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.GLA.ANN.top20.bed
grep -v SNP Pi.GLA.ANN.txt  | head -86477 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.GLA.ANN.top30.bed
grep -v SNP Pi.GLA.ANN.txt  | head -115303 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.GLA.ANN.top40.bed
grep -v SNP Pi.GLA.ANN.txt  | head -144129 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.GLA.ANN.top50.bed

<span class="hljs-comment"># Top BAB.BAP pi ratio</span>
grep -v SNP Pi.BAB.BAP.txt  | head -28822 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.BAB.BAP.top10.bed
grep -v SNP Pi.BAB.BAP.txt  | head -57644 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.BAB.BAP.top20.bed
grep -v SNP Pi.BAB.BAP.txt  | head -86465 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.BAB.BAP.top30.bed
grep -v SNP Pi.BAB.BAP.txt  | head -115287 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.BAB.BAP.top40.bed
grep -v SNP Pi.BAB.BAP.txt  | head -144109 | cut -f1,7 | sed <span class="hljs-string">'s/_/\t/g'</span> &gt; Pi.BAB.BAP.top50.bed

<span class="hljs-comment"># ANN top 5% signal bed</span>
cut -f1 ../ANN10K/SweeD_Report.ANN.10k.top5.txt | sed <span class="hljs-string">'s/_/\t/g'</span> | \
    awk <span class="hljs-string">'{print $1,int($2)-5000,int($2+1)+5000,$1"_"$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | grep -v SNP \
    &gt; ANN.10k.top5.bed

<span class="hljs-comment"># BAP top 5% signal bed</span>
cut -f1 ../BAP10K/SweeD_Report.BAP.10k.top5.txt | sed <span class="hljs-string">'s/_/\t/g'</span> | \
    awk <span class="hljs-string">'{print $1,int($2)-5000,int($2+1)+5000,$1"_"$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | grep -v SNP \
    &gt; BAP.10k.top5.bed

paste BAP.10k.top5.bed ../BAP10K/SweeD_Report.BAP.10k.top5.txt | \
    awk <span class="hljs-string">'{print $4,$1,$2,$3,$10}'</span> | sed <span class="hljs-string">'1i\SNP Chr Start End XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; BAP.10k.top5.bed2
perl merge.pl BAP.10k.top5.bed2 | sed <span class="hljs-string">'1d'</span> &gt; BAP.10k.top5.bed2.merged

<span class="hljs-comment"># Extract the overlapped sweed scores for ANN</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span> 
bedtools intersect -a ANN.10k.top5.bed -b Pi.GLA.ANN.top<span class="hljs-variable">$i</span>.bed -wo  &gt; SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Extract the overlapped sweed scores for BAP</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span> 
bedtools intersect -a BAP.10k.top5.bed -b Pi.BAB.BAP.top<span class="hljs-variable">$i</span>.bed -wo  &gt; SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.txt
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span> 
cut -f4 SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.txt | sort | uniq | wc -l
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Merge the overlapped pi ratio windows</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
cut -f5-8 SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.txt | sort | uniq | awk <span class="hljs-string">'{print $1"_"$2,$1,$2,$3,$4}'</span> | \
    sed <span class="hljs-string">'1i\SNP Chr Start End XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq.txt
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
perl merge.pl SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the total size</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
awk <span class="hljs-string">'{ sum += $3-$2+1 } END { print sum }'</span> SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Merge the overlapped pi ratio windows</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
cut -f5-8 SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.txt | sort | uniq | awk <span class="hljs-string">'{print $1"_"$2,$1,$2,$3,$4}'</span> | \
    sed <span class="hljs-string">'1i\SNP Chr Start End XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq.txt
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
perl merge.pl SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq.txt | sed <span class="hljs-string">'1d'</span> &gt; SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the total size</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
awk <span class="hljs-string">'{ sum += $3-$2+1 } END { print sum }'</span> SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># BAP Genes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
bedtools intersect \
    -a /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff \
    -b SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt | \
    grep gene | sed <span class="hljs-string">'s/;/\t/g'</span> | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | cut -f1 -d <span class="hljs-string">'.'</span> | sort | uniq \
    &gt; SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.genes.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># ANN Genes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
bedtools intersect \
    -a /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff \
    -b SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt | \
    grep gene | sed <span class="hljs-string">'s/;/\t/g'</span> | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | cut -f1 -d <span class="hljs-string">'.'</span> | sort | uniq \
    &gt; SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq2.genes.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Overlapped genes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
comm -12 &lt;(sort SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq2.genes.txt) \
    &lt;(sort SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.genes.txt) \
    &gt; SweeD.ANN.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.genes.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Overlapped size</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
bedtools intersect \
    -a SweeD.ANN.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt \
    -b SweeD.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.txt \
    &gt; SweeD.ANN.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.size.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the total size</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 10 10 50); <span class="hljs-keyword">do</span>
awk <span class="hljs-string">'{ sum += $3-$2+1 } END { print sum }'</span> SweeD.ANN.BAP.overlapp.top<span class="hljs-variable">$i</span>.uniq2.size.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Overlapped region between Sweed and pi ratio</span>
bedtools intersect \
    -a ANN.10k.top5.bed \
    -b Pi.GLA.ANN.top10.bed -wo \
    &gt; SweeD.ANN.overlapp.top10.txt

<span class="hljs-comment"># Overlaped bed</span>
bedtools intersect \
    -a SweeD.ANN.overlapp.top30.uniq2.txt \
    -b SweeD.BAP.overlapp.top30.uniq2.txt \
    &gt; Overlapped.bed

<span class="hljs-comment"># Extract the overlapped SNPs</span>
<span class="hljs-comment"># ANN</span>
bedtools intersect \
    -a &lt;(cut -f1 ../ANN10K/SweeD_Report.ANN.10k.txt | sed <span class="hljs-string">'s/_/\t/g'</span> | \
        awk <span class="hljs-string">'{print $1,int($2),int($2+1),$1"_"$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | grep -v SNP) \
    -b Overlapped.bed | cut -f4 | deal_table.pl -transpose \
    &gt; SweeD.ANN.overlapp.SNPs.txt

<span class="hljs-comment"># BAP</span>
bedtools intersect \
    -a &lt;(cut -f1 ../BAP10K/SweeD_Report.BAP.10k.txt | sed <span class="hljs-string">'s/_/\t/g'</span> | \
        awk <span class="hljs-string">'{print $1,int($2),int($2+1),$1"_"$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | grep -v SNP) \
    -b Overlapped.bed | cut -f4 | deal_table.pl -transpose \
    &gt; SweeD.BAP.overlapp.SNPs.txt

<span class="hljs-comment"># GLA.ANN</span>
bedtools intersect \
    -a &lt;(awk <span class="hljs-string">'{print $2,$3,$4,$5,$1}'</span> ../GLA.ANN.ld8/SweeD_Report.GLA.ANN.10k.ld8.top1.txt | sed <span class="hljs-string">'s/ /\t/g'</span> | grep -v Pos) \
    -b Overlapp.bed | cut -f5 | deal_table.pl -transpose \
    &gt; GLA.ANN.overlapp.SNPs.txt

bedtools intersect \
    -a ../BAB.BAP.ld8/SweeD_Report.BAB.BAP.10k.ld8.top5%0.merged.txt \
    -b ../GLA.ANN.ld8/SweeD_Report.GLA.ANN.10k.ld8.top5%0.merged.txt \
    &gt; top5%0.merged.txt

bedtools intersect \
    -a ../BAB.BAP.ld8/SweeD_Report.BAB.BAP.10k.ld8.top1.merged.txt \
    -b ../GLA.ANN.ld8/SweeD_Report.GLA.ANN.10k.ld8.top1.merged.txt \
    &gt; top1.merged.txt

bedtools intersect \
    -a /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff \
    -b top5%0.merged.txt | \
    grep gene | sed <span class="hljs-string">'s/;/\t/g'</span> | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | cut -f1 -d <span class="hljs-string">'.'</span> | sort | uniq \
    &gt; top5%0.merged.genes.txt

bedtools intersect \
    -a /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff \
    -b top1.merged.txt | \
    grep gene | sed <span class="hljs-string">'s/;/\t/g'</span> | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | cut -f1 -d <span class="hljs-string">'.'</span> | sort | uniq \
    &gt; top1.merged.genes.txt

</div></code></pre>
<h4 id="comparison-with-cao-et-al-2022">Comparison with Cao et al. 2022</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/Zunla-1/v2.0/Cao

<span class="hljs-comment"># Prepare the gene bed</span>
cut -f1,4,5,9 ../Zunla-1.genes.gff | grep ID | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/;//g'</span> &gt; Zunla-1.genes.bed
<span class="hljs-comment"># Reformat the CDS to one row</span>
perl -pe <span class="hljs-string">'/^&gt;/ ? print "\n" : chomp'</span> ../Zunla-1.CDS.fa &gt; Zunla-1.CDS.fa

<span class="hljs-comment"># Extract the CDS</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> A B C D ; <span class="hljs-keyword">do</span>
grep -f Stage<span class="hljs-variable">$i</span>.genes -A 1 Zunla-1.CDS.fa | sed <span class="hljs-string">'/^--$/d'</span> &gt; Stage<span class="hljs-variable">$i</span>.cds.fa
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run blastn</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> A B C D ; <span class="hljs-keyword">do</span>
blastn -query Stage<span class="hljs-variable">$i</span>.cds.fa \
    -db /data/zhaojiantao/pepper/References/S8/blastdb/S8.cds.index \
    -evalue 1e-10 -num_threads 60 -outfmt 6 \
    -out Stage<span class="hljs-variable">$i</span>.cds.blastn6 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Extract the best matches</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> A B C D ; <span class="hljs-keyword">do</span>
awk <span class="hljs-string">'!seen[$1]++'</span> Stage<span class="hljs-variable">$i</span>.cds.blastn6 &gt; Stage<span class="hljs-variable">$i</span>.cds.best.blastn6
<span class="hljs-keyword">done</span>

<span class="hljs-comment">## Challenge: Many of the genes could not find alignment</span>

<span class="hljs-comment"># Use the liftoff approach</span>
liftoff -polish \
    -copies -sc 0.97 \
    -exclude_partial -s 0.97 -a 0.95 -p 80 \
    -dir intermediate.ncbi \
    -u unmapped.ncbi \
    -o Zunla.liftoff.gff3 \
    -g /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff3 \
    /data/zhaojiantao/pepper/References/Zunla-1/v2.0/Zunla-1.fasta \
    /data/zhaojiantao/pepper/References/S8/Zhangshugang.fa

<span class="hljs-comment"># Extrtact the remaining genes</span>
grep -v <span class="hljs-string">"#"</span> Zunla.liftoff.gff3_polished | awk <span class="hljs-string">'$3=="gene" {print $9}'</span> | \
    cut -f1 -d <span class="hljs-string">';'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | cut -f1 -d <span class="hljs-string">'_'</span> | sort | uniq | \
    grep -vf - /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff3 \
    &gt; Zhangshugang.remain.gff3

<span class="hljs-comment"># Liftoff for the remaining genes</span>
liftoff -polish \
    -copies -sc 0.97 \
    -exclude_partial -s 0.97 -a 0.95 -p 80 \
    -dir intermediate.ncbi \
    -u unmapped.ncbi \
    -o Zunla.remain.liftoff.gff3 \
    -g Zhangshugang.remain.gff3 \
    /data/zhaojiantao/pepper/References/Zunla-1/v2.0/Zunla-1.fasta \
    /data/zhaojiantao/pepper/References/S8/Zhangshugang.fa

<span class="hljs-comment"># Combine the results and filter the infos</span>
cat Zunla.liftoff.gff3_polished Zunla.remain.liftoff.gff3_polished | \
    grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'$3=="gene" {print}'</span> | sed <span class="hljs-string">'s/;/\t/g'</span> | \
    awk <span class="hljs-string">'{ for (i=1; i&lt;=NF; i++) if ($i ~ /coverage|sequence|valid_ORFs/) {cols[i] = $i} } {printf "%s %s %s %s %s ", $9, $1, $4, $5, $7; for (i=6; i&lt;=NF; i++) if (cols[i]) printf "%s ", $i; printf "\n" }'</span> | \
    sed <span class="hljs-string">'s/ID=gene://g'</span> | sed <span class="hljs-string">'s/coverage=//g'</span> | sed <span class="hljs-string">'s/sequence_ID=//g'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | \
    sed <span class="hljs-string">'s/valid_ORFs=0/Invalid/g'</span> | sed <span class="hljs-string">'s/valid_ORFs=1/Valid/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Zunla.liftoff.genes

<span class="hljs-comment"># Format Zhangshugang gene info</span>
grep mRNA ../Zunla-1.genes.gff | sed <span class="hljs-string">'s/;/\t/g'</span> | \
    awk <span class="hljs-string">'{print $9,$1,$4,$5,$7}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Zunla.genes

<span class="hljs-comment"># Find the overlapped genes</span>
bedtools intersect -wo -a &lt;(awk <span class="hljs-string">'{print $2,$3,$4,$1,$5}'</span> Zunla.genes | sed <span class="hljs-string">'s/ /\t/g'</span>) \
    -b &lt;(awk <span class="hljs-string">'{print $2,$3,$4,$1,$5,$6,$7,$8}'</span> Zunla.liftoff.genes | sed <span class="hljs-string">'s/ /\t/g'</span>) | \
        sed <span class="hljs-string">'1i\Chr Start End Zunla Strand Chr Start End Zunla.liftoff Strand Coverage Identity ValidORFs OverlapSize'</span> | \
        awk <span class="hljs-string">'{print $4,$1,$2,$3,$5,$9,$6,$7,$8,$10,$11,$12,$13,$14}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Zunla_Zhangshugang.genes-1

<span class="hljs-comment"># Remove those genes located on different chromosomes</span>
cut -f6,7 Zunla_M82.genes-1 | grep -v Zunla | sed <span class="hljs-string">'s/g/\t/g'</span> | cut -f1,3 | sed <span class="hljs-string">'s/Solyc/Chr/g'</span> | \
    paste - &lt;(cut -f6 Zunla_M82.genes-1 | grep -v Zunla) | \
    awk <span class="hljs-string">'{if ($1!=$2) print $3}'</span> &gt; Zunla_M82.genes-A
<span class="hljs-comment"># Those genes from Chr00</span>
cut -f6,7 Zunla_M82.genes-1 | grep -v Zunla | sed <span class="hljs-string">'s/g/\t/g'</span> | cut -f1,3 | sed <span class="hljs-string">'s/Solyc/Chr/g'</span> | \
    paste - &lt;(cut -f6 Zunla_M82.genes-1 | grep -v Zunla) | \
    awk <span class="hljs-string">'{if ($1=="Chr00" || $2=="Chr00" ) print $3}'</span> &gt; Zunla_M82.genes-B

<span class="hljs-comment"># Extract the final clean results        </span>
grep -vf Zunla_M82.genes-B Zunla_M82.genes-A | \
        grep -vf - Zunla_M82.genes-1 \
        &gt; Zunla_M82.genes-2

<span class="hljs-comment"># Solyc gene infos from Zunla v4.0</span>
grep -v <span class="hljs-string">"#"</span> ../../Zunla1706/ITAG4.0_gene_models.gff | sed <span class="hljs-string">'s/;/ /g'</span> | \
        awk <span class="hljs-string">' $3=="gene" {print $9,$1,$4,$5,$7}'</span> | sed <span class="hljs-string">'s/ID=gene://g'</span> | \
        sed <span class="hljs-string">'s/ch/Chr/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Zunla.genes

<span class="hljs-comment"># Extract the corresponding Zunla gene infos</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(grep -v M82 Zunla_M82.genes-2 | cut -f6 | cut -f1 -d <span class="hljs-string">"_"</span>); <span class="hljs-keyword">do</span> grep <span class="hljs-variable">$i</span> Zunla.genes ; <span class="hljs-keyword">done</span> | \
        sed <span class="hljs-string">'1i\Zunla Chr Start End Strand'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Zunla.genes-1

<span class="hljs-comment"># Combine the final output</span>
paste Zunla_M82.genes-2 Zunla.genes-1 &gt; Zunla_M82.genes.txt

<span class="hljs-comment"># For those with multiple copies, only keep the original one</span>
cut -f6 Zunla_M82.genes.txt | grep -v Zunla.liftoff | sort | uniq -c | awk <span class="hljs-string">'$1&gt;1 {print $2}'</span> &gt; Zunla_M82.genes.multi
grep -f Zunla_M82.genes.multi Zunla_M82.genes.txt

<span class="hljs-comment"># Extract those Zunla genes could liftoff to M82 but with no overlapps with M82 gene annotations</span>
grep -f &lt;(cut -f6 Zunla_M82.genes.txt | grep -v Zunla.liftoff) Zunla.liftoff.genes \
        &gt; Zunla.liftoff.uniq.genes 

<span class="hljs-comment"># Extract the corresponding Zunla gene infos</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cut -f1 Zunla.liftoff.uniq.genes | cut -f1 -d <span class="hljs-string">"_"</span>); <span class="hljs-keyword">do</span> grep <span class="hljs-variable">$i</span> Zunla.genes ; <span class="hljs-keyword">done</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Zunla.genes-2

<span class="hljs-comment"># Combine the Zunla liftoff to M82 with no overlapps with M82</span>
paste Zunla.liftoff.uniq.genes Zunla.genes-2 | awk <span class="hljs-string">'{print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13}'</span> | \
        sed <span class="hljs-string">'1i\Zunla.liftoff Chr Start End Strand Coverage Identity ORFs Zunla Chr Start End Strand'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Zunla.liftoff.uniq.genes.txt

<span class="hljs-comment"># Total Zunla v4 liftoff to M82</span>
cat Zunla.genes-1 Zunla.genes-2  | grep -v Zunla | sort | uniq



</div></code></pre>
<h3 id="introgression">Introgression</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/15ABBA-BABA

<span class="hljs-comment"># Prepare the Pop list file</span>
awk <span class="hljs-string">'{print $1,"Ca.annuum"}'</span> ../Pop.list/ANN.list &gt; ANN.list
awk <span class="hljs-string">'{print $1,"Ca.pubescens"}'</span> ../Pop.list/PUB.list &gt; PUB.list
cat ../Pop.list/BAB.list ../Pop.list/BAP.list | awk <span class="hljs-string">'{print $1,"Ca.baccatum"}'</span> &gt; BAC.list
cat ../Pop.list/FRU.list ../Pop.list/CHN.list | awk <span class="hljs-string">'{print $1,"Ca.FRU.CHN"}'</span> &gt; FRU.CHN.list
cat *list &gt; BAC2FRU.CHN.list
cut -f1 -d <span class="hljs-string">' '</span> BAC2FRU.CHN.list &gt; BAC2FRU.CHN.list.id

<span class="hljs-comment"># Extract the genotypes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    bcftools view -S BAC2FRU.CHN.list.id -i <span class="hljs-string">'F_MISSING&lt;0.3'</span> --min-af 0.05 \
        ../04SNP/Miss_0.3/Chr<span class="hljs-variable">$i</span>.SNP.vcf.gz \
        &gt; Chr<span class="hljs-variable">$i</span>.SNP.vcf &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Convert vcf to ABAB-ABBA geno file format</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    python /data/zhaojiantao/tools/ABBA-BABA/genomics_general-master/VCF_processing/parseVCF.py \
        -i Chr<span class="hljs-variable">$i</span>.SNP.vcf --minQual 30 --gtf flag=DP \
        &gt; Chr<span class="hljs-variable">$i</span>.clean.geno &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the files</span>
head -1 Chr01.clean.geno &gt; header
cat Chr*.clean.geno | grep -v <span class="hljs-string">"#CHROM"</span> &gt; clean.geno
cat header clean.geno | bgzip -@ 20 &gt; All.clean.geno.gz

<span class="hljs-comment"># Remove unnecessary files</span>
rm header clean.geno Chr*.clean.geno

<span class="hljs-comment"># ABBA-BABA</span>
ABBABABAwindows.py \
    -g All.clean.geno.gz \
    -o PUB.BAC.FRUCHN.ANN.out \
    -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 90 \
    -P1 Ca.annuum \
    -P2 Ca.FRU.CHN \
    -P3 Ca.baccatum \
    -O Ca.pubescens \
    --popsFile BAC2FRU.CHN.list \
    --writeFailedWindows

<span class="hljs-comment"># Select the top 5% highest fd values</span>
cut -f1 PUB.BAC.FRUCHN.ANN.boxplot.out | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
    paste - PUB.BAC.FRUCHN.ANN.boxplot.out | cut -f1,3- | sed <span class="hljs-string">'1i\Chr Start End Pos SNP fd'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; PUB.BAC.FRUCHN.ANN.boxplot.vg.out.txt

</div></code></pre>
<h3 id="introgression-with-ld-prune">Introgression with LD prune</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vg/15ABBA-BABA/LDprune

<span class="hljs-comment"># LD prune</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># LD estimation</span>
    plink --gzvcf ../Chr<span class="hljs-variable">$i</span>.SNP.vcf \
        --indep-pairwise 50 10 0.8 \
        --allow-extra-chr --double-id \
        --out Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Extract the prunned SNPs</span>
    bcftools view --include ID==@Chr<span class="hljs-variable">$i</span>.prune.in ../Chr<span class="hljs-variable">$i</span>.SNP.vcf &gt; Chr<span class="hljs-variable">$i</span>.prune.vcf
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Convert vcf to ABAB-ABBA geno file format</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    python /data/zhaojiantao/tools/ABBA-BABA/genomics_general-master/VCF_processing/parseVCF.py \
        -i Chr<span class="hljs-variable">$i</span>.prune.vcf --minQual 30 --gtf flag=DP \
        &gt; Chr<span class="hljs-variable">$i</span>.clean.geno
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the files</span>
head -1 Chr01.clean.geno &gt; header
cat Chr*.clean.geno | grep -v <span class="hljs-string">"#CHROM"</span> &gt; clean.geno
cat header clean.geno | bgzip -@ 20 &gt; All.clean.geno.gz

<span class="hljs-comment"># Remove unnecessary files</span>
rm header clean.geno Chr*.clean.geno

<span class="hljs-comment"># ABBA-BABA</span>
ABBABABAwindows.py \
    -g All.clean.geno.gz \
    -o PUB.BAC.FRUCHN.ANN.out \
    -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 90 \
    -P1 Ca.annuum \
    -P2 Ca.FRU.CHN \
    -P3 Ca.baccatum \
    -O Ca.pubescens \
    --popsFile BAC2FRU.CHN.list \
    --writeFailedWindows

<span class="hljs-comment"># Select the top 5% highest fd values</span>
cut -f1 PUB.BAC.FRUCHN.ANN.boxplot.out | sed <span class="hljs-string">'s/Chr0//g'</span> | sed <span class="hljs-string">'s/Chr//g'</span> | \
    paste - PUB.BAC.FRUCHN.ANN.boxplot.out | cut -f1,3- | sed <span class="hljs-string">'1i\Chr Start End Pos SNP fd'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; PUB.BAC.FRUCHN.ANN.boxplot.vg.out.ld.txt

<span class="hljs-comment"># Calculate the average value of fd values.</span>
cat PUB.BAC.FRUCHN.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;1{print $0}'</span> | grep -v nan  | awk <span class="hljs-string">'{s+=$10} END {print s/NR}'</span>
<span class="hljs-comment"># Calculate the total value of fd</span>
cat PUB.BAC.FRUCHN.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;1{print $0}'</span> | grep -v nan  | awk <span class="hljs-string">'{s+=$6*$10} END {print s}'</span> 
<span class="hljs-comment"># Calculate the total G</span>
cat PUB.BAC.FRUCHN.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;1{print $0}'</span> | grep -v nan  | awk <span class="hljs-string">'{s+=$6} END {print s}'</span> 


</div></code></pre>
<h2 id="mapping-quality-evaluation">Mapping quality evaluation</h2>
<h3 id="mapping-quality-of-the-1314-collection">Mapping quality of the 1314 collection</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_survey/01mapping

<span class="hljs-comment"># Trimm the raw sequences</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat sample.id); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Remove adaptors and low-quality sequences</span>
    trimmomatic PE ../00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz \
        <span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>.R1U.fq \
        <span class="hljs-variable">$i</span>.R2P.fq <span class="hljs-variable">$i</span>.R2U.fq \
        -threads 40 \
        ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
        SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 ;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_survey/02bam

<span class="hljs-comment"># Map the reads to S8</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat ../01mapping/sample.id); <span class="hljs-keyword">do</span>    
    sentieon bwa mem -M -R <span class="hljs-string">"@RG\tLB:Library1\tID:FeiLab_<span class="hljs-variable">${i}</span>\tSM:<span class="hljs-variable">$i</span>\tPL:ILLUMINA"</span> \
        -t 80 /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \
        ../01mapping/<span class="hljs-variable">$i</span>.R1P.fq ../01mapping/<span class="hljs-variable">$i</span>.R2P.fq | \
        sentieon util sort -o <span class="hljs-variable">$i</span>.bam -t 80 --sam2bam -i -
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Raw reads pairs</span>
grep <span class="hljs-string">"Input Read Pairs"</span> trimm2.log | cut -f4 -d <span class="hljs-string">' '</span>

<span class="hljs-comment"># Cleaned reads pairs</span>
grep <span class="hljs-string">"Input Read Pairs"</span> trimm2.log | cut -f7 -d <span class="hljs-string">' '</span>

<span class="hljs-comment"># Count the number of paired reads from fastq</span>
<span class="hljs-built_in">echo</span> $(cat yourfile.fastq|wc -l)/4|bc

<span class="hljs-comment"># Total mapped reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat ../01mapping/sample2.id); <span class="hljs-keyword">do</span>
    sentieon driver -t 20 \
        -r /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \
        -i <span class="hljs-variable">$i</span>.bam \
        --algo AlignmentStat \
        <span class="hljs-variable">$i</span>.<span class="hljs-built_in">stat</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Total cleaned reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *<span class="hljs-built_in">stat</span>); <span class="hljs-keyword">do</span> cut -f2 <span class="hljs-variable">$i</span> | sed -n <span class="hljs-string">'5p'</span> ; <span class="hljs-keyword">done</span> &gt; AA

<span class="hljs-comment"># Total aligned reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *<span class="hljs-built_in">stat</span>); <span class="hljs-keyword">do</span> cut -f6 <span class="hljs-variable">$i</span> | sed -n <span class="hljs-string">'5p'</span> ; <span class="hljs-keyword">done</span> &gt; BB

samtools flagstat bailajiao.bam &gt; bailajiao.flagstat

<span class="hljs-comment"># Total reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'1p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Total mapped Reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'7p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Unmapped singletons</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'11p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Total paired Reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'7p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Total reads + 0 paired in sequencing</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'6p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate genome coverage</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span>
    bedtools genomecov -ibam <span class="hljs-variable">$i</span>.bam -<span class="hljs-built_in">bg</span> &gt; <span class="hljs-variable">$i</span>.bed 
    awk <span class="hljs-string">'$4&gt;=1 {print $3-$2+1}'</span> <span class="hljs-variable">$i</span>.bed | awk <span class="hljs-string">'{sum+=$1} END {print 1,sum}'</span> &gt; <span class="hljs-variable">$i</span>.1.bases
    awk <span class="hljs-string">'$4&gt;=2 {print $3-$2+1}'</span> <span class="hljs-variable">$i</span>.bed | awk <span class="hljs-string">'{sum+=$1} END {print 2,sum}'</span> &gt; <span class="hljs-variable">$i</span>.2.bases
    awk <span class="hljs-string">'$4&gt;=5 {print $3-$2+1}'</span> <span class="hljs-variable">$i</span>.bed | awk <span class="hljs-string">'{sum+=$1} END {print 5,sum}'</span> &gt; <span class="hljs-variable">$i</span>.5.bases
    cat <span class="hljs-variable">$i</span>.1.bases <span class="hljs-variable">$i</span>.2.bases <span class="hljs-variable">$i</span>.5.bases &gt; <span class="hljs-variable">$i</span>.mapping.bases
    rm <span class="hljs-variable">$i</span>.1.bases <span class="hljs-variable">$i</span>.2.bases <span class="hljs-variable">$i</span>.5.bases <span class="hljs-variable">$i</span>.bed <span class="hljs-variable">$i</span>.bam <span class="hljs-variable">$i</span>.bam.bai
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Clean the files</span>
<span class="hljs-comment"># Remove those raw files with bam file generated</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *bai | sed <span class="hljs-string">'s/.bam.bai//g'</span>); <span class="hljs-keyword">do</span> rm ../00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz ; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove the cleaned paired fastq files with bam file generated</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *bai | sed <span class="hljs-string">'s/.bam.bai//g'</span>); <span class="hljs-keyword">do</span> rm ../01mapping/<span class="hljs-variable">$i</span>.R1P.fq ../01mapping/<span class="hljs-variable">$i</span>.R2P.fq ; <span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="mapping-quality-of-the-500-core-collection">Mapping quality of the 500 core collection</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Change working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_mapping_ratio

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat Sample.id); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Map the reads to S8</span>
    sentieon bwa mem -M -R <span class="hljs-string">"@RG\tLB:Library1\tID:FeiLab_<span class="hljs-variable">${i}</span>\tSM:<span class="hljs-variable">$i</span>\tPL:ILLUMINA"</span> \
        -t 100 /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta \
        /data/jiaochen/pepper_reseq/clean_reads/combined/<span class="hljs-variable">$i</span>.R1.comb.fastq.gz \
        /data/jiaochen/pepper_reseq/clean_reads/combined/<span class="hljs-variable">$i</span>.R2.comb.fastq.gz | \
        sentieon util sort -o <span class="hljs-variable">$i</span>.bam -t 100 --sam2bam -i -
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate mapped ratios</span>
samtools mpileup -f /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta PI_406847.bam -d 100 -o PI_406847.pileup
<span class="hljs-comment"># Base with mapped reads &gt; 1</span>
awk <span class="hljs-string">'$4&gt;1 {print $0}'</span> PI_406847.pileup | wc -l
<span class="hljs-comment"># Base with mapped reads &gt; 2</span>
awk <span class="hljs-string">'$4&gt;2 {print $0}'</span> PI_406847.pileup | wc -l
<span class="hljs-comment"># Total bases</span>
cat PI_406847.pileup | wc -l

<span class="hljs-comment"># Total mapped reads</span>
samtools flagstat PI_439367.bam &gt; PI_439367.flagstat

<span class="hljs-comment"># Total reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'1p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Total mapped Reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'5p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Unmapped singletons</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'11p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Total paired Reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'7p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Total reads + 0 paired in sequencing</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'6p'</span> <span class="hljs-variable">$i</span>.flagstat | awk <span class="hljs-string">'{print $1}'</span>; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Calculate genome coverage</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *flagstat | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.flagstat//g'</span>); <span class="hljs-keyword">do</span>
    bedtools genomecov -ibam <span class="hljs-variable">$i</span>.bam -<span class="hljs-built_in">bg</span> &gt; <span class="hljs-variable">$i</span>.bed 
    awk <span class="hljs-string">'$4&gt;=1 {print $3-$2}'</span> <span class="hljs-variable">$i</span>.bed | awk <span class="hljs-string">'{sum+=$1} END {print 1,sum}'</span> &gt; <span class="hljs-variable">$i</span>.1.bases
    awk <span class="hljs-string">'$4&gt;=2 {print $3-$2}'</span> <span class="hljs-variable">$i</span>.bed | awk <span class="hljs-string">'{sum+=$1} END {print 2,sum}'</span> &gt; <span class="hljs-variable">$i</span>.2.bases
    awk <span class="hljs-string">'$4&gt;=5 {print $3-$2}'</span> <span class="hljs-variable">$i</span>.bed | awk <span class="hljs-string">'{sum+=$1} END {print 5,sum}'</span> &gt; <span class="hljs-variable">$i</span>.5.bases
    cat <span class="hljs-variable">$i</span>.1.bases <span class="hljs-variable">$i</span>.2.bases <span class="hljs-variable">$i</span>.5.bases &gt; <span class="hljs-variable">$i</span>.mapping.bases
    rm <span class="hljs-variable">$i</span>.1.bases <span class="hljs-variable">$i</span>.2.bases <span class="hljs-variable">$i</span>.5.bases <span class="hljs-variable">$i</span>.bed <span class="hljs-variable">$i</span>.bam <span class="hljs-variable">$i</span>.bam.bai
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="genotype-check-of-c-baccatum">Genotype check of C. baccatum</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Ca.baccatum.check

<span class="hljs-comment"># Index the genome</span>
bwa index /data/zhaojiantao/pepper/References/PBC81/PBC81.fasta
samtools faidx /data/zhaojiantao/pepper/References/PBC81/PBC81.fasta
java -jar /data/zhaojiantao/tools/picard/build/libs/picard.jar CreateSequenceDictionary R=PBC81.fasta O=PBC81.dict

<span class="hljs-comment"># Create the FASTA sequence directory file of the reference to be used in gatk</span>
gatk CreateSequencedirectory -R /data/zhaojiantao/pepper/References/PBC81/PBC81.fasta

<span class="hljs-comment"># Call variants</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Grif_9354 Grif_9355 PI_257110 PI_260564 PI_260595 PI_281308 PI_281310 PI_281436 PI_337524 PI_370010; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Remove adaptors and low-quality sequences</span>
    trimmomatic PE ../00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz \
        <span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>.R1U.fq \
        <span class="hljs-variable">$i</span>.R2P.fq <span class="hljs-variable">$i</span>.R2U.fq \
        -threads 80 \
        ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
        SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 ;
    <span class="hljs-comment"># Map the reads to S8</span>
    sentieon bwa mem -M -R <span class="hljs-string">"@RG\tLB:Library1\tID:FeiLab_<span class="hljs-variable">${i}</span>\tSM:<span class="hljs-variable">$i</span>\tPL:ILLUMINA"</span> \
        -t 80 /data/zhaojiantao/pepper/References/PBC81/PBC81.fasta \
        <span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>.R2P.fq | \
        sentieon util sort -o <span class="hljs-variable">$i</span>.bam -t 80 --sam2bam -i -
    <span class="hljs-comment"># Collect read information to remove or mark duplicates</span>
    sentieon driver -t 80 -i <span class="hljs-variable">$i</span>.bam \
        --algo LocusCollector --fun score_info <span class="hljs-variable">$i</span>.score.gz
    <span class="hljs-comment"># Remove or mark duplicates</span>
    sentieon driver -t 80 -i <span class="hljs-variable">$i</span>.bam \
        --algo Dedup --score_info <span class="hljs-variable">$i</span>.score.gz \
        --metrics <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam
    <span class="hljs-comment"># Call gvcf</span>
    sentieon driver -t 80 -r /data/zhaojiantao/pepper/References/PBC81/PBC81.fasta \
        -i <span class="hljs-variable">$i</span>.deduped.bam \
        --algo Haplotyper \
        --emit_mode gvcf \
        <span class="hljs-variable">$i</span>.gvcf.gz
    <span class="hljs-comment"># Remove unnecessary files</span>
    rm <span class="hljs-variable">$i</span>.R1U.fq <span class="hljs-variable">$i</span>.R2U.fq *.score.gz* *dedup_metrix.txt *R1P.fq *R2P.fq
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Joint variant call</span>
sentieon driver -t 80 -r /data/zhaojiantao/pepper/References/PBC81/PBC81.fasta --algo GVCFtyper \
    Ca.baccatum10.vcf.gz *gvcf.gz 

<span class="hljs-comment"># Extract the SNPs could be mapped in both references</span>
<span class="hljs-comment"># BAC SNPs</span>
grep -v <span class="hljs-string">"NA"</span> rand500k.map_loc.REF_BACslct | cut -f5,6 &gt; BAC.snps
<span class="hljs-comment"># Extract the genotypes</span>
vcftools --gzvcf Ca.baccatum10.vcf.gz \
    --positions BAC.snps \
    --recode --recode-INFO-all \
    --out Ca.baccatum10

<span class="hljs-comment"># ANN SNPs</span>
grep -v <span class="hljs-string">"NA"</span> rand500k.map_loc.REF_BACslct | cut -f2,3 &gt; ANN.snps

<span class="hljs-comment"># Extract the genotypes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    vcftools --gzvcf /data/zhaojiantao/pepper/vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
        --keep Ca.baccatum10.list \
        --positions ANN.snps \
        --recode --recode-INFO-all \
        --out Ca.annuum10.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the vcfs for the 500 core collection</span>
grep <span class="hljs-string">"#"</span> Ca.annuum10.Chr00.recode.vcf &gt; header
cat Ca.annuum10.Chr*.recode.vcf | grep -v <span class="hljs-string">"#"</span> | cat header - &gt; Ca.annuum10.recode.vcf

<span class="hljs-comment"># Extract biallelic SNPs</span>
bcftools view --types snps -m 2 -M 2 Ca.baccatum10.comm.recode.vcf.gz -O v -o Ca.baccatum10.comm.biallelic.vcf.gz

zcat Ca.baccatum10.comm.biallelic.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f1,2 | head -50000 

<span class="hljs-comment"># Find the common SNPs</span>
zcat Ca.baccatum10.recode.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f1,2 | \
    grep -Fwf - rand500k.map_loc.REF_BACslct.filter \
    &gt; Ca.baccatum10.comm1.snps.txt

zcat Ca.baccatum10.biallelic.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f1,2 | \
    grep -Fwf - rand500k.map_loc.REF_BACslct.filter | \
    grep -Fwf &lt;(grep -v <span class="hljs-string">"#"</span> Ca.annuum10.recode.vcf | cut -f1,2) - \
    &gt; Ca.annuum10.comm1.snps.txt

<span class="hljs-comment"># Use the top 50k SNPs</span>
head -50000 Ca.baccatum10.comm1.snps.txt &gt; Ca.baccatum10.comm50k.snps.txt
grep -Fwf &lt;(zcat Ca.baccatum10.comm.biallelic.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f1,2 | \
    grep -v 43775082 | grep -v 37665850 | grep -v 40250533 | \
    grep -v 151351965 | grep -v 142012760 | grep -v 62417332 | grep -v 139406871) \
    Ca.baccatum10.comm1.snps.txt | head -50000 &gt; Ca.baccatum10.comm50k.snps.txt

<span class="hljs-comment"># Extract the common genotypes for C. baccatum</span>
vcftools --gzvcf Ca.baccatum10.comm.recode.vcf.gz \
    --positions &lt;(cut -f5,6 Ca.baccatum10.comm50k.snps.txt) \
    --recode --recode-INFO-all \
    --out Ca.baccatum10.comm50k

<span class="hljs-comment"># Extract the common genotypes for C. annuum</span>
vcftools --gzvcf Ca.annuum10.comm.recode.vcf.gz \
    --positions &lt;(cut -f2,3 Ca.baccatum10.comm50k.snps.txt) \
    --recode --recode-INFO-all \
    --out Ca.annuum10.comm50k

<span class="hljs-comment"># Prepare the file for genotype consistency calculation</span>
plink --vcf Ca.annuum10.comm50k.recode.vcf --recode --make-bed --allow-extra-chr --out Ca.annuum10.comm50k
plink --vcf Ca.baccatum10.comm50k.recode.vcf --recode --make-bed --allow-extra-chr --out Ca.baccatum10.comm50k

<span class="hljs-comment"># Replace the C. baccatum SNP position with correspoding position in C. annuum</span>
awk <span class="hljs-string">'{print "Chr0"$1,$2,$3,$4}'</span> Ca.baccatum10.comm50k.map | \
    sed <span class="hljs-string">'s/Chr010/Chr10/g'</span> | sed <span class="hljs-string">'s/Chr011/Chr11/g'</span> | \
    sed <span class="hljs-string">'s/Chr012/Chr12/g'</span> | sed <span class="hljs-string">'s/Chr0scaffold/scaffold/g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Ca.baccatum10.comm50k2.map

grep -Fwf &lt;(cut -f1,4 Ca.baccatum10.comm50k2.map) \
    Ca.baccatum10.comm50k.snps.txt | sort -k5,5 -k6n \
    &gt; Ca.baccatum10.comm50k3.map

<span class="hljs-comment"># Cross check the consistency of ids in excel</span>
paste Ca.baccatum10.comm50k2.map Ca.baccatum10.comm50k3.map | \
    awk <span class="hljs-string">'{if ($4=$10) print "TRUE",$0 ; else print "FALSE",$0}'</span>  | grep FALSE

awk <span class="hljs-string">'{print $2,".",0,$3}'</span> Ca.baccatum10.comm50k3.map | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Ca.baccatum10.comm.annuum.map

cp Ca.baccatum10.comm50k.ped Ca.baccatum10.comm.annuum.ped

<span class="hljs-comment"># Convert ped to tped</span>
plink --file Ca.annuum10.comm50k --recode transpose --out Ca.annuum10.comm50k --allow-extra-chr
plink --file Ca.baccatum10.comm.annuum --recode transpose --out Ca.baccatum10.comm.annuum --allow-extra-chr

<span class="hljs-comment"># 互补配对</span>
grep -Fwf &lt;(cut -f1 Ca.baccatum10.comm50k.snps.txt) &lt;(zcat rand500k.map_loc.REF_BACslct.filter.gz) | \
    awk <span class="hljs-string">'{if ($4!=$8) print $2"_"$3, "Negative"; else print $2"_"$3, "Positive"}'</span> | \
    grep Negative | cut -f1 -d <span class="hljs-string">' '</span> \
    &gt; negative.snps

grep -Fwf &lt;(cut -f1 Ca.baccatum10.comm50k.snps.txt) &lt;(zcat rand500k.map_loc.REF_BACslct.filter.gz) | \
    awk <span class="hljs-string">'{if ($4!=$8) print $2"_"$3, "Negative"; else print $2"_"$3, "Positive"}'</span> | \
    grep Positive | cut -f1 -d <span class="hljs-string">' '</span> \
    &gt; positive.snps

<span class="hljs-comment"># Negative genotype</span>
grep -Fwf negative.snps &lt;(awk <span class="hljs-string">'{print $1,"Chr0"$1"_"$4,$0}'</span> Ca.baccatum10.comm.annuum.tped | cut -f1,2,5- -d <span class="hljs-string">' '</span>) | cut -f5- -d <span class="hljs-string">' '</span> | \
    sed <span class="hljs-string">'s/A/1/g'</span> | sed <span class="hljs-string">'s/T/2/g'</span> | sed <span class="hljs-string">'s/C/3/g'</span> | sed <span class="hljs-string">'s/G/4/g'</span> | \
    sed <span class="hljs-string">'s/1/T/g'</span> | sed <span class="hljs-string">'s/2/A/g'</span> | sed <span class="hljs-string">'s/3/G/g'</span> | sed <span class="hljs-string">'s/4/C/g'</span> | \
    paste &lt;(grep -Fwf negative.snps &lt;(awk <span class="hljs-string">'{print $1,"Chr0"$1"_"$4,$0}'</span> Ca.baccatum10.comm.annuum.tped | cut -f1,2,5- -d <span class="hljs-string">' '</span>) | cut -f1-4 -d <span class="hljs-string">' '</span> ) - | sed <span class="hljs-string">'s/\t/ /g'</span> \
    &gt; Ca.baccatum10.comm.annuum.tped.negative

grep -Fvf negative.snps &lt;(awk <span class="hljs-string">'{print $1,"Chr0"$1"_"$4,$0}'</span> Ca.baccatum10.comm.annuum.tped | cut -f1,2,5- -d <span class="hljs-string">' '</span>) \
    &gt; Ca.baccatum10.comm.annuum.tped.positive

cat Ca.baccatum10.comm.annuum.tped.negative Ca.baccatum10.comm.annuum.tped.positive | \
    sort -k1n -k4n &gt; Ca.baccatum10.comm.annuum.tped2

<span class="hljs-comment"># Calculate the number of consistency</span>
cut -f5- -d <span class="hljs-string">' '</span> Ca.baccatum10.comm.annuum.tped2 | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Ca.baccatum10.comm.annuum.tped2-1
cut -f5- -d <span class="hljs-string">' '</span> Ca.annuum10.comm50k.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Ca.annuum10.comm50k.tped-1
cut -f5- -d <span class="hljs-string">' '</span> Ca.baccatum10.comm.annuum.tped2 &gt; Ca.baccatum10.comm.annuum.tped2-2
cut -f5- -d <span class="hljs-string">' '</span> Ca.annuum10.comm50k.tped &gt; Ca.annuum10.comm50k.tped-2


<span class="hljs-comment"># missing</span>
paste Ca.annuum10.comm50k.tped-2 Ca.baccatum10.comm.annuum.tped2-2 | \
    awk <span class="hljs-string">'{print $1,$2,$21,$22}'</span> | grep 0 | wc -l

<span class="hljs-comment"># Consistency of both allele for sample 1</span>
paste Ca.annuum10.comm50k.tped-2 Ca.baccatum10.comm.annuum.tped2-2 | \
    awk <span class="hljs-string">'{print $1,$2,$21,$22}'</span> | grep -v 0 |  sort | uniq -c \
    &gt; count.sample1

paste Ca.annuum10.comm50k.tped-2 Ca.baccatum10.comm.annuum.tped2-2 | awk <span class="hljs-string">'{print $1,$2,$21,$22}'</span> | grep -v 0 |  sort | uniq -c &gt; count.sample1

<span class="hljs-comment"># Consistency of both alleles</span>
<span class="hljs-comment"># Ref=Ref; Alt=Alt</span>
awk <span class="hljs-string">'{if ($2==$4 &amp;&amp; $3==$5) print $0,"TRUE"}'</span> count.sample1 | awk <span class="hljs-string">'{sum+=$1} END {print sum}'</span> 
<span class="hljs-comment"># Ref=Alt; Alt=Ref</span>
awk <span class="hljs-string">'{if ($2!=$3 &amp;&amp; $2==$5 &amp;&amp; $3==$4) print $0,"TRUE"}'</span> count.sample1 | awk <span class="hljs-string">'{sum+=$1} END {print sum}'</span> 

<span class="hljs-comment"># AA==TT</span>
grep <span class="hljs-string">"A A T T"</span> count.sample1 | awk <span class="hljs-string">'{sum+=$1} END {print sum}'</span> 

<span class="hljs-comment"># TT==AA</span>
grep <span class="hljs-string">"T T A A"</span> count.sample1 | awk <span class="hljs-string">'{sum+=$1} END {print sum}'</span> 

<span class="hljs-comment"># CC==GG</span>
grep <span class="hljs-string">"C C G G"</span> count.sample1 | awk <span class="hljs-string">'{sum+=$1} END {print sum}'</span> 

<span class="hljs-comment"># GG==CC</span>
grep <span class="hljs-string">"G G C C"</span> count.sample1 | awk <span class="hljs-string">'{sum+=$1} END {print sum}'</span> 


<span class="hljs-comment"># Consistency of one allele with Zhangshugang homo</span>
paste Ca.annuum10.comm50k.tped-2 Ca.baccatum10.comm.annuum.tped2-2 | \
    awk <span class="hljs-string">'{print $1,$2,$21,$22}'</span> | grep -v 0 | \
    awk <span class="hljs-string">'{if ($1==$3 &amp;&amp; $2==$4) print $0,"TRUE1"; else if ($1==$4 &amp;&amp; $2==$3) print $0,"TRUE2"; else if ($1==$3 || $1==$4 || $2==$3 || $2==$4) print $0,"HALF" ; else print $0,"FALSE"}'</span> | grep HALF | \
    awk <span class="hljs-string">'{if ($1==$2)print $0,"Zhangshugang.home.TRUE"; else print $0,"Zhangshugang.home.FALSE"}'</span> | \
    cut -f6 -d <span class="hljs-string">' '</span> | sort | uniq -c

</div></code></pre>
<h2 id="snp-calling-and-evaluation">SNP calling and evaluation</h2>
<h3 id="snps-calling-and-quality-control-of-the-1300-pepper-collection">SNPs calling and quality control of the 1300 pepper collection</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_survey/01mapping

<span class="hljs-comment"># Obtain all the sample ids</span>
ll *R1.fastq.gz | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.R1.fastq.gz//g'</span> &gt; sample.id

<span class="hljs-comment"># Index the genome</span>
bwa index /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta
samtools faidx /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta
java -jar /data/zhaojiantao/tools/picard/build/libs/picard.jar CreateSequenceDictionary R=pepper_S8.fasta O=pepper_S8.dict

<span class="hljs-comment"># Create the FASTA sequence directory file of the reference to be used in gatk</span>
gatk CreateSequencedirectory -R /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta

<span class="hljs-comment"># Call variants</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat sample.id); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Remove adaptors and low-quality sequences</span>
    trimmomatic PE ../00fastq/<span class="hljs-variable">$i</span>.R1.fastq.gz ../00fastq/<span class="hljs-variable">$i</span>.R2.fastq.gz \
        <span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>.R1U.fq \
        <span class="hljs-variable">$i</span>.R2P.fq <span class="hljs-variable">$i</span>.R2U.fq \
        -threads 80 \
        ILLUMINACLIP:/data/zhaojiantao/tools/miniconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
        SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40 ;
    <span class="hljs-comment"># Map the reads to S8</span>
    sentieon bwa mem -M -R <span class="hljs-string">"@RG\tLB:Library1\tID:FeiLab_<span class="hljs-variable">${i}</span>\tSM:<span class="hljs-variable">$i</span>\tPL:ILLUMINA"</span> \
        -t 80 /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta \
        <span class="hljs-variable">$i</span>.R1P.fq <span class="hljs-variable">$i</span>.R2P.fq | \
        sentieon util sort -o <span class="hljs-variable">$i</span>.bam -t 80 --sam2bam -i -
    <span class="hljs-comment"># Collect read information to remove or mark duplicates</span>
    sentieon driver -t 80 -i <span class="hljs-variable">$i</span>.bam \
        --algo LocusCollector --fun score_info <span class="hljs-variable">$i</span>.score.gz
    <span class="hljs-comment"># Remove or mark duplicates</span>
    sentieon driver -t 80 -i <span class="hljs-variable">$i</span>.bam \
        --algo Dedup --score_info <span class="hljs-variable">$i</span>.score.gz \
        --metrics <span class="hljs-variable">$i</span>.dedup_metrix.txt <span class="hljs-variable">$i</span>.deduped.bam
    <span class="hljs-comment"># Call gvcf</span>
    sentieon driver -t 80 -r /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta \
        -i <span class="hljs-variable">$i</span>.deduped.bam \
        --algo Haplotyper \
        --emit_mode gvcf \
        <span class="hljs-variable">$i</span>.gvcf.gz
    <span class="hljs-comment"># Remove unnecessary files</span>
    rm <span class="hljs-variable">$i</span>.R1U.fq <span class="hljs-variable">$i</span>.R2U.fq *bam* *.score.gz* *dedup_metrix.txt *R1P.fq *R2P.fq
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Joint variant call</span>
sentieon driver -t 80 -r /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta --algo GVCFtyper \
    1314.vcf.gz *gvcf.gz 

</div></code></pre>
<h3 id="snp-calling-quality-check">SNP calling quality check</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_survey

<span class="hljs-comment"># For 1314 collection</span>
vcftools --gzvcf All.filtered.4dtv.thin.vcf.gz \
    --keep ../Pop.list/500.from.1314.list \
    --recode --recode-INFO-all \
    --out All.4dtv.1314

<span class="hljs-comment"># For 500 collection</span>
vcftools --gzvcf ../4DTV/ffds.vcf.gz \
    --positions &lt;(zcat All.filtered.4dtv.thin.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f1,2) \
    --recode --recode-INFO-all \
    --out All.4dtv.500

<span class="hljs-comment"># Create chromosome map</span>
bcftools view -H All.4dtv.1314.recode.vcf | \
    cut -f 1 | uniq | awk <span class="hljs-string">'{print $0"\t"$0}'</span> &gt; \
    All.4dtv.1314.chrom-map.txt

bcftools view -H All.4dtv.500.recode.vcf | \
    cut -f 1 | uniq | awk <span class="hljs-string">'{print $0"\t"$0}'</span> &gt; \
    All.4dtv.500.chrom-map.txt

<span class="hljs-comment"># Make ped file using this chromosome map</span>
vcftools --vcf All.4dtv.1314.recode.vcf \
    --plink --chrom-map All.4dtv.1314.chrom-map.txt \
    --out All.4dtv.1314

vcftools --vcf All.4dtv.500.recode.vcf \
    --plink --chrom-map All.4dtv.500.chrom-map.txt \
    --out All.4dtv.500

<span class="hljs-comment"># The number of Nomissing values</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 4 503); <span class="hljs-keyword">do</span> 
    cut -f1-3,<span class="hljs-variable">$i</span> 1314.txt | awk <span class="hljs-string">'$4!=00 {print $0}'</span> | wc ; 
<span class="hljs-keyword">done</span> | awk <span class="hljs-string">'{print $1}'</span>

<span class="hljs-comment"># The number of overlapped Nomissing values</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 4 503); <span class="hljs-keyword">do</span> \
    comm -12 &lt;(cut -f1-3,<span class="hljs-variable">$i</span> 1314.txt | \awk <span class="hljs-string">'$4!=00 {print $0}'</span> | sort) \
    &lt;(cut -f1-3,<span class="hljs-variable">$i</span> 500.txt | awk <span class="hljs-string">'$4!=00 {print $0}'</span> | sort) | wc ; 
<span class="hljs-keyword">done</span> | awk <span class="hljs-string">'{print $1}'</span>

</div></code></pre>
<h3 id="snp-count-for-each-group">SNP count for each group</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SNP_Number_count

<span class="hljs-comment"># Extract the polymrphic SNPs for each group</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
            --keep ../Pop.list/Ca.<span class="hljs-variable">$j</span>.list \
            --recode --recode-INFO-all \
            --mac 1 \
            --out Ca.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># zip the file</span>

<span class="hljs-comment"># SNP count for each group</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum pubescens ; <span class="hljs-keyword">do</span>
    zcat Ca.<span class="hljs-variable">$i</span>.Chr*.recode.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print $1"_"$2}'</span> &gt; Ca.<span class="hljs-variable">$i</span>.snps
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Further filter with maf = 0.05</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf Ca.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>.recode.vcf.gz --maf 0.05 --recode --out Ca.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>.maf
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># SNP count for each group</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum pubescens ; <span class="hljs-keyword">do</span>
    cat Ca.<span class="hljs-variable">$i</span>.Chr*.maf.recode.vcf | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print $1"_"$2}'</span> &gt; Ca.<span class="hljs-variable">$i</span>.maf.snps
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum pubescens ; <span class="hljs-keyword">do</span>
    zcat Ca.<span class="hljs-variable">$i</span>.Chr*.maf.recode.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print $1"_"$2}'</span> &gt; Ca.<span class="hljs-variable">$i</span>.maf.snps
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="snp-prunning">SNP prunning</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SNP.prune

<span class="hljs-comment"># Convert vcf to ped/map</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
    --allow-extra-chr --double-id \
    --recode --make-bed \
    --out Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the map file format</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> Chr<span class="hljs-variable">$i</span>.map | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr<span class="hljs-variable">$i</span>.map-1
    mv Chr<span class="hljs-variable">$i</span>.map-1 Chr<span class="hljs-variable">$i</span>.map
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># SNP pruning</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file Chr<span class="hljs-variable">$i</span> \
    --indep-pairwise 500 10 0.1 \
    --allow-extra-chr --double-id \
    --out Chr<span class="hljs-variable">$i</span> &amp;
    <span class="hljs-comment"># Extract the prunned SNPs</span>
    plink --file Chr<span class="hljs-variable">$i</span> \
        --extract Chr<span class="hljs-variable">$i</span>.prune.in \
        --recode vcf --double-id --allow-extra-chr \
        --out Chr<span class="hljs-variable">$i</span>.prune &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Total number of SNPs:298,000</span>
change double ids to single id
sh single.id.sh
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    grep -v <span class="hljs-string">"#"</span> Chr<span class="hljs-variable">$i</span>.prune.vcf | cat single.id.info - &gt; Chr<span class="hljs-variable">$i</span>.prune.vcf-1
    mv Chr<span class="hljs-variable">$i</span>.prune.vcf-1 Chr<span class="hljs-variable">$i</span>.prune.vcf
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the vcf file</span>
cat &lt;(grep <span class="hljs-string">"#"</span> Chr01.prune.vcf) \
    &lt;(cat Chr*.prune.vcf | grep -v <span class="hljs-string">"#"</span>) \
    &gt; Combined.prune.vcf
 
<span class="hljs-comment"># Further filtering missing alleles</span>
plink --file Combined.prune \
    --geno 0 --maf 0.05 \
    --recode 12 \
    --output-missing-genotype 9 \
    --out Combined.prune.nomiss

</div></code></pre>
<h3 id="snp-effects-and-distribution-of-different-subgroups">SNP effects and distribution of different subgroups.</h3>
<pre class="hljs"><code><div><span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SNP

<span class="hljs-comment"># Generate the Plink map/ped format for each subgroup</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.pubescens; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../vcf.clean/Chr<span class="hljs-variable">$j</span>.clean.vcf.gz \
        --mac 2 --max-missing 0.3 \
        --keep ../Pop.list/<span class="hljs-variable">$i</span>.list \
        --chrom-map &lt;(<span class="hljs-built_in">echo</span> Chr<span class="hljs-variable">$j</span> Chr<span class="hljs-variable">$j</span> | sed <span class="hljs-string">'s/ /\t/g'</span>) \
        --plink --out <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>
    <span class="hljs-keyword">done</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.pubescens; <span class="hljs-keyword">do</span>
    cat <span class="hljs-variable">$i</span>.Chr*.map | awk <span class="hljs-string">'{print $2}'</span> &gt; <span class="hljs-variable">$i</span>.combined.map &amp;
<span class="hljs-keyword">done</span>

paste *combined.map | sed <span class="hljs-string">'1i\Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.pubescens'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; SNP.venn.txt

<span class="hljs-comment"># Evaluate the SNP effects using [SnpEff](http://pcingola.github.io/SnpEff/se_running/)</span>
<span class="hljs-comment"># Build the [databases](https://www.cnblogs.com/zhanmaomao/p/10964636.html)</span>

<span class="hljs-built_in">cd</span> /data/zhaojiantao/tools/snpEff
<span class="hljs-comment"># Prepare and bulid the genome index</span>
cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta data/Zhangshugang/sequences.fa
cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta data/genomes/Zhangshugang.fa

cp /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff data/Zhangshugang/genes.gff

<span class="hljs-built_in">echo</span> <span class="hljs-string">"Zhangshugang.genome : Zhangshugang"</span> &gt;&gt; snpEff.config

<span class="hljs-comment"># Build the database</span>
java -jar snpEff.jar build -gff3 -v Zhangshugang

<span class="hljs-comment"># Calculate the SNP effects</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vcf.clean/SnpEff.Chr<span class="hljs-variable">$i</span>
    java -Xmx8g -jar /data/zhaojiantao/tools/snpEff/snpEff.jar \
        -c /data/zhaojiantao/tools/snpEff/snpEff.config \
        Zhangshugang \
        ../Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
        &gt; Chr<span class="hljs-variable">$i</span>.ann.vcf &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate indel effects</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Indels/SnpEff.indel.Chr<span class="hljs-variable">$i</span>
    java -Xmx8g -jar /data/zhaojiantao/tools/snpEff/snpEff.jar \
        -c /data/zhaojiantao/tools/snpEff/snpEff.config \
        Zhangshugang \
        ../Chr<span class="hljs-variable">$i</span>.fltr5.indel.vcf.gz \
        &gt; Chr<span class="hljs-variable">$i</span>.indel.ann.vcf &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the uniq variants by taking the first effect</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/vcf.clean
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    zcat SnpEff.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.ann.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f8 | sed <span class="hljs-string">'s/|/\t/g'</span> | cut -f2 | sort | uniq -c &gt; SnpEff.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.ann.count
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the uniq variants by taking the first effect</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/all_indels_filter5
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    grep -v <span class="hljs-string">"#"</span> SnpEff.indel.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.indel.ann.vcf | cut -f8 | sed <span class="hljs-string">'s/;/\t/g'</span> | awk <span class="hljs-string">'{for(i=1;i&lt;=NF;i++){if ($i ~ /ANN/){print $i}}}'</span> | \
    sed <span class="hljs-string">'s/|/\t/g'</span> | cut -f2 | sort | uniq -c &gt; SnpEff.indel.Chr<span class="hljs-variable">$i</span>/Chr<span class="hljs-variable">$i</span>.indel.ann.count
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Upset distribution of cleaned SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *combined.map.gz | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.combined.map.gz//g'</span> ); <span class="hljs-keyword">do</span> 
    zcat <span class="hljs-variable">$i</span>.combined.map.gz &gt; <span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>

</div></code></pre>
<h4 id="upset-plot-in-r">Upset plot in R</h4>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(UpSetR)
<span class="hljs-keyword">library</span>(data.table)

Ca.annuum &lt;- fread(<span class="hljs-string">"Ca.annuum.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.bac.baccatum &lt;- fread(<span class="hljs-string">"Ca.bac.baccatum.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.bac.pendulum &lt;- fread(<span class="hljs-string">"Ca.bac.pendulum.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.chacoense &lt;- fread(<span class="hljs-string">"Ca.chacoense.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.chinense &lt;- fread(<span class="hljs-string">"Ca.chinense.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.frutescens &lt;- fread(<span class="hljs-string">"Ca.frutescens.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.glabriusculum &lt;- fread(<span class="hljs-string">"Ca.glabriusculum.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)
Ca.pubescens &lt;- fread(<span class="hljs-string">"Ca.pubescens.snp"</span>, header=<span class="hljs-literal">FALSE</span>, sep = <span class="hljs-string">"\n"</span>)

Ca.annuum &lt;- as.vector(Ca.annuum$V1)
Ca.bac.baccatum &lt;- as.vector(Ca.bac.baccatum$V1)
Ca.bac.pendulum &lt;- as.vector(Ca.bac.pendulum$V1)
Ca.chacoense &lt;- as.vector(Ca.chacoense$V1)
Ca.chinense &lt;- as.vector(Ca.chinense$V1)
Ca.frutescens &lt;- as.vector(Ca.frutescens$V1)
Ca.glabriusculum &lt;- as.vector(Ca.glabriusculum$V1)
Ca.pubescens &lt;- as.vector(Ca.pubescens$V1)

read_sets = list(C.annuum = Ca.annuum, C.baccatum.var.baccatum = Ca.bac.baccatum,
    C.baccatum.var.pendulum = Ca.bac.pendulum, C.chacoense = Ca.chacoense,
    C.chinense = Ca.chinense, C.frutescens = Ca.frutescens,
    C.annuum.var.glabriusculum = Ca.glabriusculum, C.pubescens = Ca.pubescens)

jpeg(<span class="hljs-string">"SNP_upset.jpeg"</span>, height = <span class="hljs-number">15</span>, width = <span class="hljs-number">25</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
pdf(<span class="hljs-string">"SNP_upset.pdf"</span>, height = <span class="hljs-number">6</span>, width = <span class="hljs-number">10</span>)
upset(fromList(read_sets), order.by = <span class="hljs-string">"freq"</span>,number.angles = <span class="hljs-number">90</span>,
    mainbar.y.label = <span class="hljs-string">"SNP Intersections"</span>, sets.x.label = <span class="hljs-string">"Total cleaned SNPs"</span>,
    sets = c(<span class="hljs-string">"C.annuum"</span>, <span class="hljs-string">"C.baccatum.var.baccatum"</span>, 
    <span class="hljs-string">"C.baccatum.var.pendulum"</span>,<span class="hljs-string">"C.chacoense"</span>,
    <span class="hljs-string">"C.chinense"</span>,<span class="hljs-string">"C.frutescens"</span>,
    <span class="hljs-string">"C.annuum.var.glabriusculum"</span>,<span class="hljs-string">"C.pubescens"</span>))
dev.off()

</div></code></pre>
<h2 id="4dtv-snps">4DTV SNPs</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/4DTV

<span class="hljs-comment"># Call 4dtv SNPs</span>
4DTV_scan.pl -g ../pepper_S8_annotation/fasta \
    -f ../pepper_S8_annotation/gff \
    -o ffds.list \
    &gt; ffds.log

<span class="hljs-comment"># Select the ids</span>
cat ../4DTV/ffds.list | cut -f1,2 &gt; ../4DTV/ffds.list.clean.txt

</div></code></pre>
<h2 id="iq-tree">IQ-tree</h2>
<h3 id="iq-tree-for-the-1300-accessions">IQ-Tree for the ~1300 accessions</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/pepper_survey

<span class="hljs-comment"># Select the Header</span>
grep <span class="hljs-string">"#"</span> All.snp.filter.vcf &gt; aa

<span class="hljs-comment"># Select those PASSED SNPs </span>
awk <span class="hljs-string">'$7=="PASS"'</span> All.snp.filter.vcf &gt; bb

<span class="hljs-comment"># Combine with the header and rename the chromosomal ids consistent with reference</span>
cat aa bb | \
    sed <span class="hljs-string">'s/Superscaffold1_54_332676983/Chr01/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold2_27_177319215/Chr02/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold3_40_289790774/Chr03/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold4_26_248932513/Chr04/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold5_64_254874144/Chr05/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold6_82_253233553/Chr06/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold7_54_266382521/Chr07/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold8_30_174326481/Chr08/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold9_43_278428952/Chr09/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold10_31_210332287/Chr10/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold11_37_275185330/Chr11/g'</span> | \
    sed <span class="hljs-string">'s/Superscaffold12_60_260676116/Chr12/g'</span> | \
    sed <span class="hljs-string">'s/unanchor1_1_17280/Contig01/g'</span> | sed <span class="hljs-string">'s/unanchor2_1_14241/Contig02/g'</span> | \
    sed <span class="hljs-string">'s/unanchor3_1_14184/Contig03/g'</span> | sed <span class="hljs-string">'s/unanchor4_1_14135/Contig04/g'</span> | \
    sed <span class="hljs-string">'s/unanchor5_1_13935/Contig05/g'</span> | sed <span class="hljs-string">'s/unanchor6_1_13927/Contig06/g'</span> | \
    sed <span class="hljs-string">'s/unanchor7_1_13733/Contig07/g'</span> | sed <span class="hljs-string">'s/unanchor8_1_13524/Contig08/g'</span> | \
    sed <span class="hljs-string">'s/unanchor9_1_13450/Contig09/g'</span> | sed <span class="hljs-string">'s/unanchor10_1_13288/Contig10/g'</span> | \
    sed <span class="hljs-string">'s/unanchor11_1_13186/Contig11/g'</span> | sed <span class="hljs-string">'s/unanchor12_1_12934/Contig12/g'</span> | \
    sed <span class="hljs-string">'s/unanchor13_1_12859/Contig13/g'</span> | sed <span class="hljs-string">'s/unanchor14_1_12026/Contig14/g'</span> | \
    sed <span class="hljs-string">'s/unanchor15_1_12004/Contig15/g'</span> | sed <span class="hljs-string">'s/unanchor16_1_11909/Contig16/g'</span> | \
    sed <span class="hljs-string">'s/unanchor17_1_11861/Contig17/g'</span> | sed <span class="hljs-string">'s/unanchor18_1_11798/Contig18/g'</span> | \
    sed <span class="hljs-string">'s/unanchor19_1_11724/Contig19/g'</span> | sed <span class="hljs-string">'s/unanchor20_1_11540/Contig20/g'</span> | \
    sed <span class="hljs-string">'s/unanchor21_1_10681/Contig21/g'</span> | sed <span class="hljs-string">'s/unanchor22_1_10492/Contig22/g'</span> | \
    sed <span class="hljs-string">'s/unanchor23_1_10491/Contig23/g'</span> | sed <span class="hljs-string">'s/unanchor24_1_10144/Contig24/g'</span> | \
    sed <span class="hljs-string">'s/unanchor25_1_10010/Contig25/g'</span> | sed <span class="hljs-string">'s/unanchor26_1_9833/Contig26/g'</span> | \
    sed <span class="hljs-string">'s/unanchor27_1_9659/Contig27/g'</span> | sed <span class="hljs-string">'s/unanchor28_1_9285/Contig28/g'</span> | \
    sed <span class="hljs-string">'s/unanchor29_1_9105/Contig29/g'</span> | sed <span class="hljs-string">'s/unanchor30_1_9048/Contig30/g'</span> | \
    sed <span class="hljs-string">'s/unanchor31_1_9046/Contig31/g'</span> | sed <span class="hljs-string">'s/unanchor32_1_9019/Contig32/g'</span> | \
    sed <span class="hljs-string">'s/unanchor33_1_8665/Contig33/g'</span> | sed <span class="hljs-string">'s/unanchor34_1_8639/Contig34/g'</span> | \
    sed <span class="hljs-string">'s/unanchor35_1_8180/Contig35/g'</span> | sed <span class="hljs-string">'s/unanchor36_1_8163/Contig36/g'</span> | \
    sed <span class="hljs-string">'s/unanchor37_1_8109/Contig37/g'</span> | sed <span class="hljs-string">'s/unanchor38_1_8032/Contig38/g'</span> | \
    sed <span class="hljs-string">'s/unanchor39_1_7734/Contig39/g'</span> | sed <span class="hljs-string">'s/unanchor40_1_7545/Contig40/g'</span> | \
    sed <span class="hljs-string">'s/unanchor41_1_7514/Contig41/g'</span> | sed <span class="hljs-string">'s/unanchor42_1_7491/Contig42/g'</span> | \
    sed <span class="hljs-string">'s/unanchor43_1_7293/Contig43/g'</span> | sed <span class="hljs-string">'s/unanchor44_1_7178/Contig44/g'</span> | \
    sed <span class="hljs-string">'s/unanchor45_1_6978/Contig45/g'</span> | sed <span class="hljs-string">'s/unanchor46_1_6489/Contig46/g'</span> | \
    sed <span class="hljs-string">'s/unanchor47_1_6375/Contig47/g'</span> | sed <span class="hljs-string">'s/unanchor48_1_6084/Contig48/g'</span> | \
    sed <span class="hljs-string">'s/unanchor49_1_5994/Contig49/g'</span> | sed <span class="hljs-string">'s/unanchor50_1_5934/Contig50/g'</span> | \
    sed <span class="hljs-string">'s/unanchor51_1_5922/Contig51/g'</span> | sed <span class="hljs-string">'s/unanchor52_1_5576/Contig52/g'</span> | \
    sed <span class="hljs-string">'s/unanchor53_1_5065/Contig53/g'</span> | sed <span class="hljs-string">'s/unanchor54_1_4952/Contig54/g'</span> | \
    sed <span class="hljs-string">'s/unanchor55_1_4829/Contig55/g'</span> | sed <span class="hljs-string">'s/unanchor56_1_4778/Contig56/g'</span> | \
    sed <span class="hljs-string">'s/unanchor57_1_4529/Contig57/g'</span> | sed <span class="hljs-string">'s/unanchor58_1_4262/Contig58/g'</span> | \
    sed <span class="hljs-string">'s/unanchor59_1_3903/Contig59/g'</span> | sed <span class="hljs-string">'s/unanchor60_1_3887/Contig60/g'</span> | \
    sed <span class="hljs-string">'s/unanchor61_1_3635/Contig61/g'</span> | sed <span class="hljs-string">'s/unanchor62_1_3597/Contig62/g'</span> | \
    sed <span class="hljs-string">'s/unanchor63_1_3447/Contig63/g'</span> | sed <span class="hljs-string">'s/unanchor64_1_3080/Contig64/g'</span> | \
    sed <span class="hljs-string">'s/unanchor65_1_2873/Contig65/g'</span> | sed <span class="hljs-string">'s/unanchor66_1_2852/Contig66/g'</span> | \
    sed <span class="hljs-string">'s/unanchor67_1_2512/Contig67/g'</span> | sed <span class="hljs-string">'s/unanchor68_1_2453/Contig68/g'</span> | \
    sed <span class="hljs-string">'s/unanchor69_1_2175/Contig69/g'</span> | sed <span class="hljs-string">'s/unanchor70_1_2025/Contig70/g'</span> | \
    sed <span class="hljs-string">'s/unanchor71_1_1781/Contig71/g'</span> | sed <span class="hljs-string">'s/unanchor72_1_1707/Contig72/g'</span> | \
    sed <span class="hljs-string">'s/unanchor73_1_1626/Contig73/g'</span> | sed <span class="hljs-string">'s/unanchor74_1_1578/Contig74/g'</span> | \
    sed <span class="hljs-string">'s/unanchor75_1_1279/Contig75/g'</span> | sed <span class="hljs-string">'s/unanchor76_1_1221/Contig76/g'</span> | \
    sed <span class="hljs-string">'s/unanchor77_1_936/Contig77/g'</span> | sed <span class="hljs-string">'s/unanchor78_1_594/Contig78/g'</span> | \
    sed <span class="hljs-string">'s/unanchor79_1_50000/Contig79/g'</span> | sed <span class="hljs-string">'s/unanchor80_1_36440/Contig80/g'</span> | \
    sed <span class="hljs-string">'s/unanchor81_1_188396/Contig81/g'</span> | sed <span class="hljs-string">'s/unanchor82_1_233858/Contig82/g'</span> | \
    sed <span class="hljs-string">'s/unanchor83_1_262094/Contig83/g'</span> | sed <span class="hljs-string">'s/unanchor84_1_55865/Contig84/g'</span> | \
    sed <span class="hljs-string">'s/unanchor85_1_116314/Contig85/g'</span> | sed <span class="hljs-string">'s/unanchor86_1_125000/Contig86/g'</span> | \
    sed <span class="hljs-string">'s/unanchor87_1_547308/Contig87/g'</span> | sed <span class="hljs-string">'s/unanchor88_1_126556/Contig88/g'</span> | \
    sed <span class="hljs-string">'s/unanchor89_1_444622/Contig89/g'</span> | gzip \
    &gt; All.snp.filtered.vcf.gz

<span class="hljs-comment"># 4DTV snps ids</span>
zcat ../IQ-Tree504/Ca.S8.ffds.vcf.gz | \
    grep -v <span class="hljs-string">"#"</span> | cut -f1,2 | grep -v Ca.S8.Chr00 | \
    sed <span class="hljs-string">'s/Ca.S8.Chr01/Superscaffold1_54_332676983/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr02/Superscaffold2_27_177319215/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr03/Superscaffold3_40_289790774/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr04/Superscaffold4_26_248932513/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr05/Superscaffold5_64_254874144/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr06/Superscaffold6_82_253233553/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr07/Superscaffold7_54_266382521/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr08/Superscaffold8_30_174326481/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr09/Superscaffold9_43_278428952/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr10/Superscaffold10_31_210332287/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr11/Superscaffold11_37_275185330/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr12/Superscaffold12_60_260676116/g'</span> \
    &gt; 504.4DTV.bisnps

<span class="hljs-comment"># Extract the SNPs</span>
vcftools --gzvcf 1314.vcf.gz \
    --positions 504.4DTV.bisnps \
    --recode --stdout &gt; 1314.4DTV-1.vcf

<span class="hljs-comment"># Select the SNPs</span>
gatk SelectVariants \
    -R /data/zhaojiantao/pepper/References/S8/pepper_S8.fasta \
    -V 1314.4DTV-1.vcf \
    -select-type SNP \
    -O 1314.4DTV.biSNPs.vcf

<span class="hljs-comment"># Filter the SNPs</span>
<span class="hljs-built_in">echo</span> Chr<span class="hljs-variable">$i</span> Chr<span class="hljs-variable">$i</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr<span class="hljs-variable">$i</span>.clean-map.txt ; 
<span class="hljs-comment"># Make ped file using this chromosome map</span>
vcftools --vcf 1314.4DTV.biSNPs.vcf \
    --max-missing 0.5 --maf 0.05 \
    --recode --stdout | bgzip -c &gt; 1314.4DTV.biSNPs.filter.vcf.gz

<span class="hljs-comment"># Extract the tomato accessions as outgroup</span>
bcftools view -s ERR418106,ERR418107,ERR418103,ERR418102 \
    /data/zhaojiantao/pepper/IQ-Tree504/Ca.S8.ffds.addTomato.vcf | \
    sed <span class="hljs-string">'s/Ca.S8.Chr01/Superscaffold1_54_332676983/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr02/Superscaffold2_27_177319215/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr03/Superscaffold3_40_289790774/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr04/Superscaffold4_26_248932513/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr05/Superscaffold5_64_254874144/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr06/Superscaffold6_82_253233553/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr07/Superscaffold7_54_266382521/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr08/Superscaffold8_30_174326481/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr09/Superscaffold9_43_278428952/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr10/Superscaffold10_31_210332287/g'</span> | \
    sed <span class="hljs-string">'s/Ca.S8.Chr11/Superscaffold11_37_275185330/g'</span> | sed <span class="hljs-string">'s/Ca.S8.Chr12/Superscaffold12_60_260676116/g'</span> \
    &gt; Tomato.vcf

<span class="hljs-comment"># Find the common SNPs of tomato</span>
zcat 1314.4DTV.biSNPs.filter.vcf | grep -v <span class="hljs-string">"#"</span> | cut -f1,2 &gt; 1314.4DTV.biSNPs.filter.snps
vcftools --vcf Tomato.vcf \
    --positions 1314.4DTV.biSNPs.filter.snps \
    --recode --stdout | bgzip -c &gt; Tomato.4DTV.vcf.gz

<span class="hljs-comment"># Extract biSNPs</span>
tabix  Tomato.4DTV.vcf.gz
bcftools view --types snps -m 2 -M 2 \
    Tomato.4DTV.vcf.gz -O z \
    -o Tomato.4DTV.biSNPs.vcf.gz

zcat Tomato.4DTV.biSNPs.vcf.gz | grep -v <span class="hljs-string">"#"</span> | cut -f1,2 &gt; Tomato.4DTV.biSNPs
vcftools --gzvcf 1314.4DTV.biSNPs.filter.vcf.gz \
    --positions Tomato.4DTV.biSNPs \
    --recode --stdout | bgzip -c &gt; 1314.4DTV.biSNPs.filter2.vcf.gz

<span class="hljs-comment"># Combine the vcf files</span>
bcftools index 1314.4DTV.biSNPs.filter2.vcf.gz
bcftools index Tomato.4DTV.biSNPs.vcf.gz

<span class="hljs-comment"># Merge samples</span>
bcftools merge -Oz \
    Tomato.4DTV.biSNPs.vcf.gz 1314.4DTV.biSNPs.filter2.vcf.gz \
    &gt; 1314.4DTV.biSNPs.filter.withtomato.vcf.gz

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i 1314.4DTV.biSNPs.filter.withtomato.vcf.gz
mv 1314.4DTV.biSNPs.filter.withtomato.min4.phy 1314.withtomato.phy

<span class="hljs-comment"># Run IQ-tree</span>
iqtree -s 1314.withtomato.phy -nt AUTO -m MFP

</div></code></pre>
<h3 id="iq-tree-for-504-accessions">IQ-tree for 504 accessions</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/IQ-Tree504

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i Ca.S8.ffds.addTomato.vcf.gz
mv ffds.addTomato.min4.phy ffds.addTomato.phy

<span class="hljs-comment"># IQ-tree</span>
iqtree -s ffds.addTomato.phy -nt AUTO -m MFP -o ERR418106

<span class="hljs-comment"># SNP pruning for 4DTV SNPs</span>
<span class="hljs-comment">## Add the SNP infos in the vcf files</span>
grep -v <span class="hljs-string">"#"</span> Ca.S8.ffds.addTomato.vcf | \
    awk <span class="hljs-string">'{print $1,$2,$1"_"$2,$0}'</span> | sed <span class="hljs-string">'s/Ca.S8.//g'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    cut -f1-3,7- &gt; Ca.S8.ffds.addTomato.vcf-1
grep <span class="hljs-string">"#"</span> Ca.S8.ffds.addTomato.vcf | sed <span class="hljs-string">'s/Ca.S8.//g'</span> | \
    cat - Ca.S8.ffds.addTomato.vcf-1 \
    &gt; Ca.S8.ffds.addTomato.vcf-2
mv Ca.S8.ffds.addTomato.vcf-2 Ca.S8.ffds.addTomato.vcf

</div></code></pre>
<h3 id="iq-tree-for-504-accessions-with-prunned-snps">IQ-tree for 504 accessions with prunned SNPs</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SNP.prune

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i Combined.prune.vcf
<span class="hljs-comment"># Shorten the sample ids</span>
cat Combined.prune.min4.phy | \
    sed <span class="hljs-string">'s/ERR418102_ERR418102/ERR418102/g'</span> | sed <span class="hljs-string">'s/ERR418103_ERR418103/ERR418103/g'</span> | \
    sed <span class="hljs-string">'s/ERR418106_ERR418106/ERR418106/g'</span> | sed <span class="hljs-string">'s/ERR418107_ERR418107/ERR418107/g'</span> | \
    sed <span class="hljs-string">'s/CPPSIH_1_05_CPPSIH_1_05/CPPSIH_1_05/g'</span> | sed <span class="hljs-string">'s/CXJ14_CXJ14/CXJ14/g'</span> | \
    sed <span class="hljs-string">'s/CXJ34_CXJ34/CXJ34/g'</span> | sed <span class="hljs-string">'s/CXJ38_CXJ38/CXJ38/g'</span> | \
    sed <span class="hljs-string">'s/CXJ52_CXJ52/CXJ52/g'</span> | sed <span class="hljs-string">'s/CXJ69_CXJ69/CXJ69/g'</span> | \
    sed <span class="hljs-string">'s/CXJ92_CXJ92/CXJ92/g'</span> | sed <span class="hljs-string">'s/CXJ96_CXJ96/CXJ96/g'</span> | \
    sed <span class="hljs-string">'s/Grif_12454_Grif_12454/Grif_12454/g'</span> | sed <span class="hljs-string">'s/Grif_1613_Grif_1613/Grif_1613/g'</span> | \
    sed <span class="hljs-string">'s/Grif_1614_Grif_1614/Grif_1614/g'</span> | sed <span class="hljs-string">'s/Grif_1615_Grif_1615/Grif_1615/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9118_Grif_9118/Grif_9118/g'</span> | sed <span class="hljs-string">'s/Grif_9122_Grif_9122/Grif_9122/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9138_Grif_9138/Grif_9138/g'</span> | sed <span class="hljs-string">'s/Grif_9140_Grif_9140/Grif_9140/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9142_Grif_9142/Grif_9142/g'</span> | sed <span class="hljs-string">'s/Grif_9152_Grif_9152/Grif_9152/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9184_Grif_9184/Grif_9184/g'</span> | sed <span class="hljs-string">'s/Grif_9188_Grif_9188/Grif_9188/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9194_Grif_9194/Grif_9194/g'</span> | sed <span class="hljs-string">'s/Grif_9197_Grif_9197/Grif_9197/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9199_Grif_9199/Grif_9199/g'</span> | sed <span class="hljs-string">'s/Grif_9201_Grif_9201/Grif_9201/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9226_Grif_9226/Grif_9226/g'</span> | sed <span class="hljs-string">'s/Grif_9229_Grif_9229/Grif_9229/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9230_Grif_9230/Grif_9230/g'</span> | sed <span class="hljs-string">'s/Grif_9236_Grif_9236/Grif_9236/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9241_Grif_9241/Grif_9241/g'</span> | sed <span class="hljs-string">'s/Grif_9249_Grif_9249/Grif_9249/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9256_Grif_9256/Grif_9256/g'</span> | sed <span class="hljs-string">'s/Grif_9257_Grif_9257/Grif_9257/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9267_Grif_9267/Grif_9267/g'</span> | sed <span class="hljs-string">'s/Grif_9289_Grif_9289/Grif_9289/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9301_Grif_9301/Grif_9301/g'</span> | sed <span class="hljs-string">'s/Grif_9312_Grif_9312/Grif_9312/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9313_Grif_9313/Grif_9313/g'</span> | sed <span class="hljs-string">'s/Grif_9315_Grif_9315/Grif_9315/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9352_Grif_9352/Grif_9352/g'</span> | sed <span class="hljs-string">'s/Grif_9354_Grif_9354/Grif_9354/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9355_Grif_9355/Grif_9355/g'</span> | sed <span class="hljs-string">'s/Grif_9357_Grif_9357/Grif_9357/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9359_Grif_9359/Grif_9359/g'</span> | sed <span class="hljs-string">'s/Grif_9365_Grif_9365/Grif_9365/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9376_Grif_9376/Grif_9376/g'</span> | sed <span class="hljs-string">'s/Grif_9381_Grif_9381/Grif_9381/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9388_Grif_9388/Grif_9388/g'</span> | sed <span class="hljs-string">'s/Grif_9391_Grif_9391/Grif_9391/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9393_Grif_9393/Grif_9393/g'</span> | sed <span class="hljs-string">'s/Grif_9400_Grif_9400/Grif_9400/g'</span> | \
    sed <span class="hljs-string">'s/Grif_9433_Grif_9433/Grif_9433/g'</span> | sed <span class="hljs-string">'s/OLJ20_OLJ20/OLJ20/g'</span> | \
    sed <span class="hljs-string">'s/PI_148629_PI_148629/PI_148629/g'</span> | sed <span class="hljs-string">'s/PI_159235_PI_159235/PI_159235/g'</span> | \
    sed <span class="hljs-string">'s/PI_159242_PI_159242/PI_159242/g'</span> | sed <span class="hljs-string">'s/PI_159246_PI_159246/PI_159246/g'</span> | \
    sed <span class="hljs-string">'s/PI_159249_PI_159249/PI_159249/g'</span> | sed <span class="hljs-string">'s/PI_159255_PI_159255/PI_159255/g'</span> | \
    sed <span class="hljs-string">'s/PI_159260_PI_159260/PI_159260/g'</span> | sed <span class="hljs-string">'s/PI_159261_PI_159261/PI_159261/g'</span> | \
    sed <span class="hljs-string">'s/PI_188481_PI_188481/PI_188481/g'</span> | sed <span class="hljs-string">'s/PI_193470_PI_193470/PI_193470/g'</span> | \
    sed <span class="hljs-string">'s/PI_194879_PI_194879/PI_194879/g'</span> | sed <span class="hljs-string">'s/PI_194880_PI_194880/PI_194880/g'</span> | \
    sed <span class="hljs-string">'s/PI_194910_PI_194910/PI_194910/g'</span> | sed <span class="hljs-string">'s/PI_195296_PI_195296/PI_195296/g'</span> | \
    sed <span class="hljs-string">'s/PI_195770_PI_195770/PI_195770/g'</span> | sed <span class="hljs-string">'s/PI_199506_PI_199506/PI_199506/g'</span> | \
    sed <span class="hljs-string">'s/PI_208738_PI_208738/PI_208738/g'</span> | sed <span class="hljs-string">'s/PI_209590_PI_209590/PI_209590/g'</span> | \
    sed <span class="hljs-string">'s/PI_215699_PI_215699/PI_215699/g'</span> | sed <span class="hljs-string">'s/PI_215731_PI_215731/PI_215731/g'</span> | \
    sed <span class="hljs-string">'s/PI_215741_PI_215741/PI_215741/g'</span> | sed <span class="hljs-string">'s/PI_224405_PI_224405/PI_224405/g'</span> | \
    sed <span class="hljs-string">'s/PI_224416_PI_224416/PI_224416/g'</span> | sed <span class="hljs-string">'s/PI_224426_PI_224426/PI_224426/g'</span> | \
    sed <span class="hljs-string">'s/PI_224427_PI_224427/PI_224427/g'</span> | sed <span class="hljs-string">'s/PI_224428_PI_224428/PI_224428/g'</span> | \
    sed <span class="hljs-string">'s/PI_238049_PI_238049/PI_238049/g'</span> | sed <span class="hljs-string">'s/PI_238051_PI_238051/PI_238051/g'</span> | \
    sed <span class="hljs-string">'s/PI_238053_PI_238053/PI_238053/g'</span> | sed <span class="hljs-string">'s/PI_238054_PI_238054/PI_238054/g'</span> | \
    sed <span class="hljs-string">'s/PI_238055_PI_238055/PI_238055/g'</span> | sed <span class="hljs-string">'s/PI_238056_PI_238056/PI_238056/g'</span> | \
    sed <span class="hljs-string">'s/PI_238060_PI_238060/PI_238060/g'</span> | sed <span class="hljs-string">'s/PI_241650_PI_241650/PI_241650/g'</span> | \
    sed <span class="hljs-string">'s/PI_241671_PI_241671/PI_241671/g'</span> | sed <span class="hljs-string">'s/PI_241673_PI_241673/PI_241673/g'</span> | \
    sed <span class="hljs-string">'s/PI_241675_PI_241675/PI_241675/g'</span> | sed <span class="hljs-string">'s/PI_241676_PI_241676/PI_241676/g'</span> | \
    sed <span class="hljs-string">'s/PI_241679_PI_241679/PI_241679/g'</span> | sed <span class="hljs-string">'s/PI_257046_PI_257046/PI_257046/g'</span> | \
    sed <span class="hljs-string">'s/PI_257050_PI_257050/PI_257050/g'</span> | sed <span class="hljs-string">'s/PI_257062_PI_257062/PI_257062/g'</span> | \
    sed <span class="hljs-string">'s/PI_257074_PI_257074/PI_257074/g'</span> | sed <span class="hljs-string">'s/PI_257080_PI_257080/PI_257080/g'</span> | \
    sed <span class="hljs-string">'s/PI_257083_PI_257083/PI_257083/g'</span> | sed <span class="hljs-string">'s/PI_257104_PI_257104/PI_257104/g'</span> | \
    sed <span class="hljs-string">'s/PI_257109_PI_257109/PI_257109/g'</span> | sed <span class="hljs-string">'s/PI_257110_PI_257110/PI_257110/g'</span> | \
    sed <span class="hljs-string">'s/PI_257116_PI_257116/PI_257116/g'</span> | sed <span class="hljs-string">'s/PI_257117_PI_257117/PI_257117/g'</span> | \
    sed <span class="hljs-string">'s/PI_257121_PI_257121/PI_257121/g'</span> | sed <span class="hljs-string">'s/PI_257124_PI_257124/PI_257124/g'</span> | \
    sed <span class="hljs-string">'s/PI_257141_PI_257141/PI_257141/g'</span> | sed <span class="hljs-string">'s/PI_257148_PI_257148/PI_257148/g'</span> | \
    sed <span class="hljs-string">'s/PI_257157_PI_257157/PI_257157/g'</span> | sed <span class="hljs-string">'s/PI_257161_PI_257161/PI_257161/g'</span> | \
    sed <span class="hljs-string">'s/PI_257173_PI_257173/PI_257173/g'</span> | sed <span class="hljs-string">'s/PI_257174_PI_257174/PI_257174/g'</span> | \
    sed <span class="hljs-string">'s/PI_257181_PI_257181/PI_257181/g'</span> | sed <span class="hljs-string">'s/PI_257182_PI_257182/PI_257182/g'</span> | \
    sed <span class="hljs-string">'s/PI_260427_PI_260427/PI_260427/g'</span> | sed <span class="hljs-string">'s/PI_260429_PI_260429/PI_260429/g'</span> | \
    sed <span class="hljs-string">'s/PI_260430_PI_260430/PI_260430/g'</span> | sed <span class="hljs-string">'s/PI_260431_PI_260431/PI_260431/g'</span> | \
    sed <span class="hljs-string">'s/PI_260435_PI_260435/PI_260435/g'</span> | sed <span class="hljs-string">'s/PI_260436_PI_260436/PI_260436/g'</span> | \
    sed <span class="hljs-string">'s/PI_260437_PI_260437/PI_260437/g'</span> | sed <span class="hljs-string">'s/PI_260478_PI_260478/PI_260478/g'</span> | \
    sed <span class="hljs-string">'s/PI_260479_PI_260479/PI_260479/g'</span> | sed <span class="hljs-string">'s/PI_260485_PI_260485/PI_260485/g'</span> | \
    sed <span class="hljs-string">'s/PI_260486_PI_260486/PI_260486/g'</span> | sed <span class="hljs-string">'s/PI_260488_PI_260488/PI_260488/g'</span> | \
    sed <span class="hljs-string">'s/PI_260492_PI_260492/PI_260492/g'</span> | sed <span class="hljs-string">'s/PI_260498_PI_260498/PI_260498/g'</span> | \
    sed <span class="hljs-string">'s/PI_260501_PI_260501/PI_260501/g'</span> | sed <span class="hljs-string">'s/PI_260505_PI_260505/PI_260505/g'</span> | \
    sed <span class="hljs-string">'s/PI_260508_PI_260508/PI_260508/g'</span> | sed <span class="hljs-string">'s/PI_260509_PI_260509/PI_260509/g'</span> | \
    sed <span class="hljs-string">'s/PI_260515_PI_260515/PI_260515/g'</span> | sed <span class="hljs-string">'s/PI_260516_PI_260516/PI_260516/g'</span> | \
    sed <span class="hljs-string">'s/PI_260535_PI_260535/PI_260535/g'</span> | sed <span class="hljs-string">'s/PI_260538_PI_260538/PI_260538/g'</span> | \
    sed <span class="hljs-string">'s/PI_260545_PI_260545/PI_260545/g'</span> | sed <span class="hljs-string">'s/PI_260549_PI_260549/PI_260549/g'</span> | \
    sed <span class="hljs-string">'s/PI_260550_PI_260550/PI_260550/g'</span> | sed <span class="hljs-string">'s/PI_260551_PI_260551/PI_260551/g'</span> | \
    sed <span class="hljs-string">'s/PI_260564_PI_260564/PI_260564/g'</span> | sed <span class="hljs-string">'s/PI_260567_PI_260567/PI_260567/g'</span> | \
    sed <span class="hljs-string">'s/PI_260575_PI_260575/PI_260575/g'</span> | sed <span class="hljs-string">'s/PI_260582_PI_260582/PI_260582/g'</span> | \
    sed <span class="hljs-string">'s/PI_260590_PI_260590/PI_260590/g'</span> | sed <span class="hljs-string">'s/PI_260595_PI_260595/PI_260595/g'</span> | \
    sed <span class="hljs-string">'s/PI_260611_PI_260611/PI_260611/g'</span> | sed <span class="hljs-string">'s/PI_266041_PI_266041/PI_266041/g'</span> | \
    sed <span class="hljs-string">'s/PI_267729_PI_267729/PI_267729/g'</span> | sed <span class="hljs-string">'s/PI_267730_PI_267730/PI_267730/g'</span> | \
    sed <span class="hljs-string">'s/PI_273415_PI_273415/PI_273415/g'</span> | sed <span class="hljs-string">'s/PI_273419_PI_273419/PI_273419/g'</span> | \
    sed <span class="hljs-string">'s/PI_281308_PI_281308/PI_281308/g'</span> | sed <span class="hljs-string">'s/PI_281309_PI_281309/PI_281309/g'</span> | \
    sed <span class="hljs-string">'s/PI_281310_PI_281310/PI_281310/g'</span> | sed <span class="hljs-string">'s/PI_281314_PI_281314/PI_281314/g'</span> | \
    sed <span class="hljs-string">'s/PI_281341_PI_281341/PI_281341/g'</span> | sed <span class="hljs-string">'s/PI_281396_PI_281396/PI_281396/g'</span> | \
    sed <span class="hljs-string">'s/PI_281398_PI_281398/PI_281398/g'</span> | sed <span class="hljs-string">'s/PI_281407_PI_281407/PI_281407/g'</span> | \
    sed <span class="hljs-string">'s/PI_281421_PI_281421/PI_281421/g'</span> | sed <span class="hljs-string">'s/PI_281422_PI_281422/PI_281422/g'</span> | \
    sed <span class="hljs-string">'s/PI_281423_PI_281423/PI_281423/g'</span> | sed <span class="hljs-string">'s/PI_281424_PI_281424/PI_281424/g'</span> | \
    sed <span class="hljs-string">'s/PI_281428_PI_281428/PI_281428/g'</span> | sed <span class="hljs-string">'s/PI_281430_PI_281430/PI_281430/g'</span> | \
    sed <span class="hljs-string">'s/PI_281436_PI_281436/PI_281436/g'</span> | sed <span class="hljs-string">'s/PI_281440_PI_281440/PI_281440/g'</span> | \
    sed <span class="hljs-string">'s/PI_281442_PI_281442/PI_281442/g'</span> | sed <span class="hljs-string">'s/PI_281444_PI_281444/PI_281444/g'</span> | \
    sed <span class="hljs-string">'s/PI_290980_PI_290980/PI_290980/g'</span> | sed <span class="hljs-string">'s/PI_290982_PI_290982/PI_290982/g'</span> | \
    sed <span class="hljs-string">'s/PI_290983_PI_290983/PI_290983/g'</span> | sed <span class="hljs-string">'s/PI_291999_PI_291999/PI_291999/g'</span> | \
    sed <span class="hljs-string">'s/PI_293349_PI_293349/PI_293349/g'</span> | sed <span class="hljs-string">'s/PI_310435_PI_310435/PI_310435/g'</span> | \
    sed <span class="hljs-string">'s/PI_311126_PI_311126/PI_311126/g'</span> | sed <span class="hljs-string">'s/PI_315007_PI_315007/PI_315007/g'</span> | \
    sed <span class="hljs-string">'s/PI_315020_PI_315020/PI_315020/g'</span> | sed <span class="hljs-string">'s/PI_315021_PI_315021/PI_315021/g'</span> | \
    sed <span class="hljs-string">'s/PI_315023_PI_315023/PI_315023/g'</span> | sed <span class="hljs-string">'s/PI_321387_PI_321387/PI_321387/g'</span> | \
    sed <span class="hljs-string">'s/PI_337524_PI_337524/PI_337524/g'</span> | sed <span class="hljs-string">'s/PI_338992_PI_338992/PI_338992/g'</span> | \
    sed <span class="hljs-string">'s/PI_355813_PI_355813/PI_355813/g'</span> | sed <span class="hljs-string">'s/PI_355815_PI_355815/PI_355815/g'</span> | \
    sed <span class="hljs-string">'s/PI_355817_PI_355817/PI_355817/g'</span> | sed <span class="hljs-string">'s/PI_355819_PI_355819/PI_355819/g'</span> | \
    sed <span class="hljs-string">'s/PI_355820_PI_355820/PI_355820/g'</span> | sed <span class="hljs-string">'s/PI_357467_PI_357467/PI_357467/g'</span> | \
    sed <span class="hljs-string">'s/PI_358811_PI_358811/PI_358811/g'</span> | sed <span class="hljs-string">'s/PI_358968_PI_358968/PI_358968/g'</span> | \
    sed <span class="hljs-string">'s/PI_360725_PI_360725/PI_360725/g'</span> | sed <span class="hljs-string">'s/PI_360732_PI_360732/PI_360732/g'</span> | \
    sed <span class="hljs-string">'s/PI_368073_PI_368073/PI_368073/g'</span> | sed <span class="hljs-string">'s/PI_368080_PI_368080/PI_368080/g'</span> | \
    sed <span class="hljs-string">'s/PI_368081_PI_368081/PI_368081/g'</span> | sed <span class="hljs-string">'s/PI_368085_PI_368085/PI_368085/g'</span> | \
    sed <span class="hljs-string">'s/PI_370008_PI_370008/PI_370008/g'</span> | sed <span class="hljs-string">'s/PI_370009_PI_370009/PI_370009/g'</span> | \
    sed <span class="hljs-string">'s/PI_370010_PI_370010/PI_370010/g'</span> | sed <span class="hljs-string">'s/PI_379193_PI_379193/PI_379193/g'</span> | \
    sed <span class="hljs-string">'s/PI_387833_PI_387833/PI_387833/g'</span> | sed <span class="hljs-string">'s/PI_406847_PI_406847/PI_406847/g'</span> | \
    sed <span class="hljs-string">'s/PI_406948_PI_406948/PI_406948/g'</span> | sed <span class="hljs-string">'s/PI_406987_PI_406987/PI_406987/g'</span> | \
    sed <span class="hljs-string">'s/PI_407450_PI_407450/PI_407450/g'</span> | sed <span class="hljs-string">'s/PI_410407_PI_410407/PI_410407/g'</span> | \
    sed <span class="hljs-string">'s/PI_414729_PI_414729/PI_414729/g'</span> | sed <span class="hljs-string">'s/PI_415094_PI_415094/PI_415094/g'</span> | \
    sed <span class="hljs-string">'s/PI_420377_PI_420377/PI_420377/g'</span> | sed <span class="hljs-string">'s/PI_420378_PI_420378/PI_420378/g'</span> | \
    sed <span class="hljs-string">'s/PI_420379_PI_420379/PI_420379/g'</span> | sed <span class="hljs-string">'s/PI_424732_PI_424732/PI_424732/g'</span> | \
    sed <span class="hljs-string">'s/PI_435917_PI_435917/PI_435917/g'</span> | sed <span class="hljs-string">'s/PI_438535_PI_438535/PI_438535/g'</span> | \
    sed <span class="hljs-string">'s/PI_438538_PI_438538/PI_438538/g'</span> | sed <span class="hljs-string">'s/PI_438567_PI_438567/PI_438567/g'</span> | \
    sed <span class="hljs-string">'s/PI_438619_PI_438619/PI_438619/g'</span> | sed <span class="hljs-string">'s/PI_438624_PI_438624/PI_438624/g'</span> | \
    sed <span class="hljs-string">'s/PI_438630_PI_438630/PI_438630/g'</span> | sed <span class="hljs-string">'s/PI_438635_PI_438635/PI_438635/g'</span> | \
    sed <span class="hljs-string">'s/PI_438637_PI_438637/PI_438637/g'</span> | sed <span class="hljs-string">'s/PI_438638_PI_438638/PI_438638/g'</span> | \
    sed <span class="hljs-string">'s/PI_438641_PI_438641/PI_438641/g'</span> | sed <span class="hljs-string">'s/PI_438642_PI_438642/PI_438642/g'</span> | \
    sed <span class="hljs-string">'s/PI_438643_PI_438643/PI_438643/g'</span> | sed <span class="hljs-string">'s/PI_438667_PI_438667/PI_438667/g'</span> | \
    sed <span class="hljs-string">'s/PI_439245_PI_439245/PI_439245/g'</span> | sed <span class="hljs-string">'s/PI_439256_PI_439256/PI_439256/g'</span> | \
    sed <span class="hljs-string">'s/PI_439311_PI_439311/PI_439311/g'</span> | sed <span class="hljs-string">'s/PI_439359_PI_439359/PI_439359/g'</span> | \
    sed <span class="hljs-string">'s/PI_439360_PI_439360/PI_439360/g'</span> | sed <span class="hljs-string">'s/PI_439363_PI_439363/PI_439363/g'</span> | \
    sed <span class="hljs-string">'s/PI_439366_PI_439366/PI_439366/g'</span> | sed <span class="hljs-string">'s/PI_439367_PI_439367/PI_439367/g'</span> | \
    sed <span class="hljs-string">'s/PI_439369_PI_439369/PI_439369/g'</span> | sed <span class="hljs-string">'s/PI_439371_PI_439371/PI_439371/g'</span> | \
    sed <span class="hljs-string">'s/PI_439378_PI_439378/PI_439378/g'</span> | sed <span class="hljs-string">'s/PI_439380_PI_439380/PI_439380/g'</span> | \
    sed <span class="hljs-string">'s/PI_439384_PI_439384/PI_439384/g'</span> | sed <span class="hljs-string">'s/PI_439386_PI_439386/PI_439386/g'</span> | \
    sed <span class="hljs-string">'s/PI_439390_PI_439390/PI_439390/g'</span> | sed <span class="hljs-string">'s/PI_439395_PI_439395/PI_439395/g'</span> | \
    sed <span class="hljs-string">'s/PI_439398_PI_439398/PI_439398/g'</span> | sed <span class="hljs-string">'s/PI_439401_PI_439401/PI_439401/g'</span> | \
    sed <span class="hljs-string">'s/PI_439402_PI_439402/PI_439402/g'</span> | sed <span class="hljs-string">'s/PI_439403_PI_439403/PI_439403/g'</span> | \
    sed <span class="hljs-string">'s/PI_439405_PI_439405/PI_439405/g'</span> | sed <span class="hljs-string">'s/PI_439407_PI_439407/PI_439407/g'</span> | \
    sed <span class="hljs-string">'s/PI_439410_PI_439410/PI_439410/g'</span> | sed <span class="hljs-string">'s/PI_439411_PI_439411/PI_439411/g'</span> | \
    sed <span class="hljs-string">'s/PI_439414_PI_439414/PI_439414/g'</span> | sed <span class="hljs-string">'s/PI_439415_PI_439415/PI_439415/g'</span> | \
    sed <span class="hljs-string">'s/PI_439421_PI_439421/PI_439421/g'</span> | sed <span class="hljs-string">'s/PI_439426_PI_439426/PI_439426/g'</span> | \
    sed <span class="hljs-string">'s/PI_439427_PI_439427/PI_439427/g'</span> | sed <span class="hljs-string">'s/PI_439431_PI_439431/PI_439431/g'</span> | \
    sed <span class="hljs-string">'s/PI_439439_PI_439439/PI_439439/g'</span> | sed <span class="hljs-string">'s/PI_439450_PI_439450/PI_439450/g'</span> | \
    sed <span class="hljs-string">'s/PI_439463_PI_439463/PI_439463/g'</span> | sed <span class="hljs-string">'s/PI_439466_PI_439466/PI_439466/g'</span> | \
    sed <span class="hljs-string">'s/PI_439472_PI_439472/PI_439472/g'</span> | sed <span class="hljs-string">'s/PI_439475_PI_439475/PI_439475/g'</span> | \
    sed <span class="hljs-string">'s/PI_439477_PI_439477/PI_439477/g'</span> | sed <span class="hljs-string">'s/PI_439481_PI_439481/PI_439481/g'</span> | \
    sed <span class="hljs-string">'s/PI_439488_PI_439488/PI_439488/g'</span> | sed <span class="hljs-string">'s/PI_439492_PI_439492/PI_439492/g'</span> | \
    sed <span class="hljs-string">'s/PI_439498_PI_439498/PI_439498/g'</span> | sed <span class="hljs-string">'s/PI_439502_PI_439502/PI_439502/g'</span> | \
    sed <span class="hljs-string">'s/PI_439504_PI_439504/PI_439504/g'</span> | sed <span class="hljs-string">'s/PI_439509_PI_439509/PI_439509/g'</span> | \
    sed <span class="hljs-string">'s/PI_439511_PI_439511/PI_439511/g'</span> | sed <span class="hljs-string">'s/PI_439512_PI_439512/PI_439512/g'</span> | \
    sed <span class="hljs-string">'s/PI_439521_PI_439521/PI_439521/g'</span> | sed <span class="hljs-string">'s/PI_439526_PI_439526/PI_439526/g'</span> | \
    sed <span class="hljs-string">'s/PI_439528_PI_439528/PI_439528/g'</span> | sed <span class="hljs-string">'s/PI_441516_PI_441516/PI_441516/g'</span> | \
    sed <span class="hljs-string">'s/PI_441523_PI_441523/PI_441523/g'</span> | sed <span class="hljs-string">'s/PI_441531_PI_441531/PI_441531/g'</span> | \
    sed <span class="hljs-string">'s/PI_441532_PI_441532/PI_441532/g'</span> | sed <span class="hljs-string">'s/PI_441536_PI_441536/PI_441536/g'</span> | \
    sed <span class="hljs-string">'s/PI_441537_PI_441537/PI_441537/g'</span> | sed <span class="hljs-string">'s/PI_441541_PI_441541/PI_441541/g'</span> | \
    sed <span class="hljs-string">'s/PI_441547_PI_441547/PI_441547/g'</span> | sed <span class="hljs-string">'s/PI_441548_PI_441548/PI_441548/g'</span> | \
    sed <span class="hljs-string">'s/PI_441558_PI_441558/PI_441558/g'</span> | sed <span class="hljs-string">'s/PI_441578_PI_441578/PI_441578/g'</span> | \
    sed <span class="hljs-string">'s/PI_441590_PI_441590/PI_441590/g'</span> | sed <span class="hljs-string">'s/PI_441592_PI_441592/PI_441592/g'</span> | \
    sed <span class="hljs-string">'s/PI_441594_PI_441594/PI_441594/g'</span> | sed <span class="hljs-string">'s/PI_441597_PI_441597/PI_441597/g'</span> | \
    sed <span class="hljs-string">'s/PI_441602_PI_441602/PI_441602/g'</span> | sed <span class="hljs-string">'s/PI_441605_PI_441605/PI_441605/g'</span> | \
    sed <span class="hljs-string">'s/PI_441608_PI_441608/PI_441608/g'</span> | sed <span class="hljs-string">'s/PI_441615_PI_441615/PI_441615/g'</span> | \
    sed <span class="hljs-string">'s/PI_441625_PI_441625/PI_441625/g'</span> | sed <span class="hljs-string">'s/PI_441631_PI_441631/PI_441631/g'</span> | \
    sed <span class="hljs-string">'s/PI_441656_PI_441656/PI_441656/g'</span> | sed <span class="hljs-string">'s/PI_441657_PI_441657/PI_441657/g'</span> | \
    sed <span class="hljs-string">'s/PI_441674_PI_441674/PI_441674/g'</span> | sed <span class="hljs-string">'s/PI_441685_PI_441685/PI_441685/g'</span> | \
    sed <span class="hljs-string">'s/PI_441690_PI_441690/PI_441690/g'</span> | sed <span class="hljs-string">'s/PI_441699_PI_441699/PI_441699/g'</span> | \
    sed <span class="hljs-string">'s/PI_441710_PI_441710/PI_441710/g'</span> | sed <span class="hljs-string">'s/PI_446900_PI_446900/PI_446900/g'</span> | \
    sed <span class="hljs-string">'s/PI_446909_PI_446909/PI_446909/g'</span> | sed <span class="hljs-string">'s/PI_487450_PI_487450/PI_487450/g'</span> | \
    sed <span class="hljs-string">'s/PI_487451_PI_487451/PI_487451/g'</span> | sed <span class="hljs-string">'s/PI_487452_PI_487452/PI_487452/g'</span> | \
    sed <span class="hljs-string">'s/PI_487457_PI_487457/PI_487457/g'</span> | sed <span class="hljs-string">'s/PI_497972_PI_497972/PI_497972/g'</span> | \
    sed <span class="hljs-string">'s/PI_497974_PI_497974/PI_497974/g'</span> | sed <span class="hljs-string">'s/PI_497984_PI_497984/PI_497984/g'</span> | \
    sed <span class="hljs-string">'s/PI_497985_PI_497985/PI_497985/g'</span> | sed <span class="hljs-string">'s/PI_511885_PI_511885/PI_511885/g'</span> | \
    sed <span class="hljs-string">'s/PI_532990_PI_532990/PI_532990/g'</span> | sed <span class="hljs-string">'s/PI_543181_PI_543181/PI_543181/g'</span> | \
    sed <span class="hljs-string">'s/PI_543182_PI_543182/PI_543182/g'</span> | sed <span class="hljs-string">'s/PI_543187_PI_543187/PI_543187/g'</span> | \
    sed <span class="hljs-string">'s/PI_543195_PI_543195/PI_543195/g'</span> | sed <span class="hljs-string">'s/PI_543197_PI_543197/PI_543197/g'</span> | \
    sed <span class="hljs-string">'s/PI_543198_PI_543198/PI_543198/g'</span> | sed <span class="hljs-string">'s/PI_543199_PI_543199/PI_543199/g'</span> | \
    sed <span class="hljs-string">'s/PI_543204_PI_543204/PI_543204/g'</span> | sed <span class="hljs-string">'s/PI_543207_PI_543207/PI_543207/g'</span> | \
    sed <span class="hljs-string">'s/PI_555611_PI_555611/PI_555611/g'</span> | sed <span class="hljs-string">'s/PI_555612_PI_555612/PI_555612/g'</span> | \
    sed <span class="hljs-string">'s/PI_555616_PI_555616/PI_555616/g'</span> | sed <span class="hljs-string">'s/PI_555620_PI_555620/PI_555620/g'</span> | \
    sed <span class="hljs-string">'s/PI_555638_PI_555638/PI_555638/g'</span> | sed <span class="hljs-string">'s/PI_555642_PI_555642/PI_555642/g'</span> | \
    sed <span class="hljs-string">'s/PI_555645_PI_555645/PI_555645/g'</span> | sed <span class="hljs-string">'s/PI_560944_PI_560944/PI_560944/g'</span> | \
    sed <span class="hljs-string">'s/PI_573337_PI_573337/PI_573337/g'</span> | sed <span class="hljs-string">'s/PI_585247_PI_585247/PI_585247/g'</span> | \
    sed <span class="hljs-string">'s/PI_585250_PI_585250/PI_585250/g'</span> | sed <span class="hljs-string">'s/PI_585254_PI_585254/PI_585254/g'</span> | \
    sed <span class="hljs-string">'s/PI_585264_PI_585264/PI_585264/g'</span> | sed <span class="hljs-string">'s/PI_585265_PI_585265/PI_585265/g'</span> | \
    sed <span class="hljs-string">'s/PI_585266_PI_585266/PI_585266/g'</span> | sed <span class="hljs-string">'s/PI_585267_PI_585267/PI_585267/g'</span> | \
    sed <span class="hljs-string">'s/PI_585268_PI_585268/PI_585268/g'</span> | sed <span class="hljs-string">'s/PI_585269_PI_585269/PI_585269/g'</span> | \
    sed <span class="hljs-string">'s/PI_585272_PI_585272/PI_585272/g'</span> | sed <span class="hljs-string">'s/PI_585273_PI_585273/PI_585273/g'</span> | \
    sed <span class="hljs-string">'s/PI_585274_PI_585274/PI_585274/g'</span> | sed <span class="hljs-string">'s/PI_585275_PI_585275/PI_585275/g'</span> | \
    sed <span class="hljs-string">'s/PI_585276_PI_585276/PI_585276/g'</span> | sed <span class="hljs-string">'s/PI_592528_PI_592528/PI_592528/g'</span> | \
    sed <span class="hljs-string">'s/PI_593490_PI_593490/PI_593490/g'</span> | sed <span class="hljs-string">'s/PI_593500_PI_593500/PI_593500/g'</span> | \
    sed <span class="hljs-string">'s/PI_593507_PI_593507/PI_593507/g'</span> | sed <span class="hljs-string">'s/PI_593517_PI_593517/PI_593517/g'</span> | \
    sed <span class="hljs-string">'s/PI_593519_PI_593519/PI_593519/g'</span> | sed <span class="hljs-string">'s/PI_593526_PI_593526/PI_593526/g'</span> | \
    sed <span class="hljs-string">'s/PI_593527_PI_593527/PI_593527/g'</span> | sed <span class="hljs-string">'s/PI_593545_PI_593545/PI_593545/g'</span> | \
    sed <span class="hljs-string">'s/PI_593546_PI_593546/PI_593546/g'</span> | sed <span class="hljs-string">'s/PI_593547_PI_593547/PI_593547/g'</span> | \
    sed <span class="hljs-string">'s/PI_593555_PI_593555/PI_593555/g'</span> | sed <span class="hljs-string">'s/PI_593557_PI_593557/PI_593557/g'</span> | \
    sed <span class="hljs-string">'s/PI_593574_PI_593574/PI_593574/g'</span> | sed <span class="hljs-string">'s/PI_593576_PI_593576/PI_593576/g'</span> | \
    sed <span class="hljs-string">'s/PI_593577_PI_593577/PI_593577/g'</span> | sed <span class="hljs-string">'s/PI_593606_PI_593606/PI_593606/g'</span> | \
    sed <span class="hljs-string">'s/PI_593607_PI_593607/PI_593607/g'</span> | sed <span class="hljs-string">'s/PI_593608_PI_593608/PI_593608/g'</span> | \
    sed <span class="hljs-string">'s/PI_593611_PI_593611/PI_593611/g'</span> | sed <span class="hljs-string">'s/PI_593613_PI_593613/PI_593613/g'</span> | \
    sed <span class="hljs-string">'s/PI_593616_PI_593616/PI_593616/g'</span> | sed <span class="hljs-string">'s/PI_593617_PI_593617/PI_593617/g'</span> | \
    sed <span class="hljs-string">'s/PI_593618_PI_593618/PI_593618/g'</span> | sed <span class="hljs-string">'s/PI_593619_PI_593619/PI_593619/g'</span> | \
    sed <span class="hljs-string">'s/PI_593620_PI_593620/PI_593620/g'</span> | sed <span class="hljs-string">'s/PI_593621_PI_593621/PI_593621/g'</span> | \
    sed <span class="hljs-string">'s/PI_593623_PI_593623/PI_593623/g'</span> | sed <span class="hljs-string">'s/PI_593624_PI_593624/PI_593624/g'</span> | \
    sed <span class="hljs-string">'s/PI_593625_PI_593625/PI_593625/g'</span> | sed <span class="hljs-string">'s/PI_593626_PI_593626/PI_593626/g'</span> | \
    sed <span class="hljs-string">'s/PI_593627_PI_593627/PI_593627/g'</span> | sed <span class="hljs-string">'s/PI_593628_PI_593628/PI_593628/g'</span> | \
    sed <span class="hljs-string">'s/PI_593630_PI_593630/PI_593630/g'</span> | sed <span class="hljs-string">'s/PI_593632_PI_593632/PI_593632/g'</span> | \
    sed <span class="hljs-string">'s/PI_593633_PI_593633/PI_593633/g'</span> | sed <span class="hljs-string">'s/PI_593634_PI_593634/PI_593634/g'</span> | \
    sed <span class="hljs-string">'s/PI_593636_PI_593636/PI_593636/g'</span> | sed <span class="hljs-string">'s/PI_593637_PI_593637/PI_593637/g'</span> | \
    sed <span class="hljs-string">'s/PI_593638_PI_593638/PI_593638/g'</span> | sed <span class="hljs-string">'s/PI_593639_PI_593639/PI_593639/g'</span> | \
    sed <span class="hljs-string">'s/PI_593640_PI_593640/PI_593640/g'</span> | sed <span class="hljs-string">'s/PI_593641_PI_593641/PI_593641/g'</span> | \
    sed <span class="hljs-string">'s/PI_593642_PI_593642/PI_593642/g'</span> | sed <span class="hljs-string">'s/PI_593921_PI_593921/PI_593921/g'</span> | \
    sed <span class="hljs-string">'s/PI_593922_PI_593922/PI_593922/g'</span> | sed <span class="hljs-string">'s/PI_593924_PI_593924/PI_593924/g'</span> | \
    sed <span class="hljs-string">'s/PI_593926_PI_593926/PI_593926/g'</span> | sed <span class="hljs-string">'s/PI_594137_PI_594137/PI_594137/g'</span> | \
    sed <span class="hljs-string">'s/PI_594138_PI_594138/PI_594138/g'</span> | sed <span class="hljs-string">'s/PI_594139_PI_594139/PI_594139/g'</span> | \
    sed <span class="hljs-string">'s/PI_595905_PI_595905/PI_595905/g'</span> | sed <span class="hljs-string">'s/PI_595907_PI_595907/PI_595907/g'</span> | \
    sed <span class="hljs-string">'s/PI_596059_PI_596059/PI_596059/g'</span> | sed <span class="hljs-string">'s/PI_631129_PI_631129/PI_631129/g'</span> | \
    sed <span class="hljs-string">'s/PI_631135_PI_631135/PI_631135/g'</span> | sed <span class="hljs-string">'s/PI_631136_PI_631136/PI_631136/g'</span> | \
    sed <span class="hljs-string">'s/PI_631137_PI_631137/PI_631137/g'</span> | sed <span class="hljs-string">'s/PI_631144_PI_631144/PI_631144/g'</span> | \
    sed <span class="hljs-string">'s/PI_631146_PI_631146/PI_631146/g'</span> | sed <span class="hljs-string">'s/PI_631151_PI_631151/PI_631151/g'</span> | \
    sed <span class="hljs-string">'s/PI_631152_PI_631152/PI_631152/g'</span> | sed <span class="hljs-string">'s/PI_632918_PI_632918/PI_632918/g'</span> | \
    sed <span class="hljs-string">'s/PI_632923_PI_632923/PI_632923/g'</span> | sed <span class="hljs-string">'s/PI_632926_PI_632926/PI_632926/g'</span> | \
    sed <span class="hljs-string">'s/PI_632928_PI_632928/PI_632928/g'</span> | sed <span class="hljs-string">'s/PI_632930_PI_632930/PI_632930/g'</span> | \
    sed <span class="hljs-string">'s/PI_633751_PI_633751/PI_633751/g'</span> | sed <span class="hljs-string">'s/PI_633755_PI_633755/PI_633755/g'</span> | \
    sed <span class="hljs-string">'s/PI_633757_PI_633757/PI_633757/g'</span> | sed <span class="hljs-string">'s/PI_633758_PI_633758/PI_633758/g'</span> | \
    sed <span class="hljs-string">'s/PI_635818_PI_635818/PI_635818/g'</span> | sed <span class="hljs-string">'s/PI_639126_PI_639126/PI_639126/g'</span> | \
    sed <span class="hljs-string">'s/PI_639129_PI_639129/PI_639129/g'</span> | sed <span class="hljs-string">'s/PI_639132_PI_639132/PI_639132/g'</span> | \
    sed <span class="hljs-string">'s/PI_639140_PI_639140/PI_639140/g'</span> | sed <span class="hljs-string">'s/PI_639639_PI_639639/PI_639639/g'</span> | \
    sed <span class="hljs-string">'s/PI_639640_PI_639640/PI_639640/g'</span> | sed <span class="hljs-string">'s/PI_639646_PI_639646/PI_639646/g'</span> | \
    sed <span class="hljs-string">'s/PI_639648_PI_639648/PI_639648/g'</span> | sed <span class="hljs-string">'s/PI_639651_PI_639651/PI_639651/g'</span> | \
    sed <span class="hljs-string">'s/PI_639652_PI_639652/PI_639652/g'</span> | sed <span class="hljs-string">'s/PI_639653_PI_639653/PI_639653/g'</span> | \
    sed <span class="hljs-string">'s/PI_639657_PI_639657/PI_639657/g'</span> | sed <span class="hljs-string">'s/PI_639659_PI_639659/PI_639659/g'</span> | \
    sed <span class="hljs-string">'s/PI_639661_PI_639661/PI_639661/g'</span> | sed <span class="hljs-string">'s/PI_639662_PI_639662/PI_639662/g'</span> | \
    sed <span class="hljs-string">'s/PI_639682_PI_639682/PI_639682/g'</span> | sed <span class="hljs-string">'s/PI_640503_PI_640503/PI_640503/g'</span> | \
    sed <span class="hljs-string">'s/PI_640770_PI_640770/PI_640770/g'</span> | sed <span class="hljs-string">'s/PI_640882_PI_640882/PI_640882/g'</span> | \
    sed <span class="hljs-string">'s/PI_640884_PI_640884/PI_640884/g'</span> | sed <span class="hljs-string">'s/PI_640889_PI_640889/PI_640889/g'</span> | \
    sed <span class="hljs-string">'s/PI_640897_PI_640897/PI_640897/g'</span> | sed <span class="hljs-string">'s/PI_640900_PI_640900/PI_640900/g'</span> | \
    sed <span class="hljs-string">'s/PI_640901_PI_640901/PI_640901/g'</span> | sed <span class="hljs-string">'s/PI_640902_PI_640902/PI_640902/g'</span> | \
    sed <span class="hljs-string">'s/PI_640909_PI_640909/PI_640909/g'</span> | sed <span class="hljs-string">'s/PI_642956_PI_642956/PI_642956/g'</span> | \
    sed <span class="hljs-string">'s/PI_643124_PI_643124/PI_643124/g'</span> | sed <span class="hljs-string">'s/PI_645493_PI_645493/PI_645493/g'</span> | \
    sed <span class="hljs-string">'s/PI_645508_PI_645508/PI_645508/g'</span> | sed <span class="hljs-string">'s/PI_645534_PI_645534/PI_645534/g'</span> | \
    sed <span class="hljs-string">'s/PI_645555_PI_645555/PI_645555/g'</span> | sed <span class="hljs-string">'s/PI_645556_PI_645556/PI_645556/g'</span> | \
    sed <span class="hljs-string">'s/PI_645557_PI_645557/PI_645557/g'</span> | sed <span class="hljs-string">'s/PI_645560_PI_645560/PI_645560/g'</span> | \
    sed <span class="hljs-string">'s/PI_645681_PI_645681/PI_645681/g'</span> | sed <span class="hljs-string">'s/PI_653659_PI_653659/PI_653659/g'</span> | \
    sed <span class="hljs-string">'s/PI_653671_PI_653671/PI_653671/g'</span> | sed <span class="hljs-string">'s/PI_653677_PI_653677/PI_653677/g'</span> | \
    sed <span class="hljs-string">'s/PI_653679_PI_653679/PI_653679/g'</span> | sed <span class="hljs-string">'s/PI_653681_PI_653681/PI_653681/g'</span> | \
    sed <span class="hljs-string">'s/PI_653744_PI_653744/PI_653744/g'</span> | sed <span class="hljs-string">'s/PI_653747_PI_653747/PI_653747/g'</span> | \
    sed <span class="hljs-string">'s/PI_655064_PI_655064/PI_655064/g'</span> | sed <span class="hljs-string">'s/PI_659105_PI_659105/PI_659105/g'</span> | \
    sed <span class="hljs-string">'s/PI_659106_PI_659106/PI_659106/g'</span> | sed <span class="hljs-string">'s/PI_660971_PI_660971/PI_660971/g'</span> | \
    sed <span class="hljs-string">'s/PI_660972_PI_660972/PI_660972/g'</span> | sed <span class="hljs-string">'s/PI_661082_PI_661082/PI_661082/g'</span> | \
    sed <span class="hljs-string">'s/PI_661088_PI_661088/PI_661088/g'</span> | sed <span class="hljs-string">'s/PI_665008_PI_665008/PI_665008/g'</span> | \
    sed <span class="hljs-string">'s/PI_666511_PI_666511/PI_666511/g'</span> | sed <span class="hljs-string">'s/PI_666550_PI_666550/PI_666550/g'</span> | \
    sed <span class="hljs-string">'s/PI_666556_PI_666556/PI_666556/g'</span> | sed <span class="hljs-string">'s/PI_666563_PI_666563/PI_666563/g'</span> | \
    sed <span class="hljs-string">'s/PI_666566_PI_666566/PI_666566/g'</span> | sed <span class="hljs-string">'s/PI_666567_PI_666567/PI_666567/g'</span> | \
    sed <span class="hljs-string">'s/PI_666580_PI_666580/PI_666580/g'</span> | sed <span class="hljs-string">'s/PI_666586_PI_666586/PI_666586/g'</span> | \
    sed <span class="hljs-string">'s/PI_666589_PI_666589/PI_666589/g'</span> | sed <span class="hljs-string">'s/PI_666591_PI_666591/PI_666591/g'</span> | \
    sed <span class="hljs-string">'s/PI_666594_PI_666594/PI_666594/g'</span> | sed <span class="hljs-string">'s/PI_666595_PI_666595/PI_666595/g'</span> | \
    sed <span class="hljs-string">'s/PI_666596_PI_666596/PI_666596/g'</span> | sed <span class="hljs-string">'s/PI_674456_PI_674456/PI_674456/g'</span> | \
    sed <span class="hljs-string">'s/PI_674459_PI_674459/PI_674459/g'</span> | sed <span class="hljs-string">'s/PI_675162_PI_675162/PI_675162/g'</span> | \
    sed <span class="hljs-string">'s/PI_678386_PI_678386/PI_678386/g'</span> | sed <span class="hljs-string">'s/PI_688675_PI_688675/PI_688675/g'</span> | \
    sed <span class="hljs-string">'s/YMH38_YMH38/YMH38/g'</span> | sed <span class="hljs-string">'s/Z111_Z111/Z111/g'</span> | \
    sed <span class="hljs-string">'s/bailajiao_bailajiao/bailajiao/g'</span> | sed <span class="hljs-string">'s/changanyingjiao_changanyingjiao/changanyingjiao/g'</span> | \
    sed <span class="hljs-string">'s/chaotiaojiao_chaotiaojiao/chaotiaojiao/g'</span> | sed <span class="hljs-string">'s/qixingjiao_qixingjiao/qixingjiao/g'</span> | \
    sed <span class="hljs-string">'s/sanweijiao_sanweijiao/sanweijiao/g'</span> | sed <span class="hljs-string">'s/sanyingjiao_sanyingjiao/sanyingjiao/g'</span> | \
    sed <span class="hljs-string">'s/xiaoguojiao_xiaoguojiao/xiaoguojiao/g'</span> | sed <span class="hljs-string">'s/xiaomila_xiaomila/xiaomila/g'</span> | \
    sed <span class="hljs-string">'s/zhudachang_zhudachang/zhudachang/g'</span> | sed <span class="hljs-string">'s/zoukejiao_zoukejiao/zoukejiao/g'</span> \
    &gt; Combined.prune.min4.phy2
mv Combined.prune.min4.phy2 Combined.prune.min4.phy

<span class="hljs-comment"># IQ-tree</span>
iqtree -s Combined.prune.min4.phy -nt 80 -m MFP

</div></code></pre>
<h2 id="pca-analysis">PCA analysis</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/PCA

<span class="hljs-comment"># Create chromosome map</span>
bcftools view -H ../4DTV/ffds.vcf.gz | cut -f 1 | uniq | awk <span class="hljs-string">'{print $0"\t"$0}'</span> &gt; ffds.chrom-map.txt

<span class="hljs-comment"># Make ped file using this chromosome map</span>
vcftools --gzvcf ../4DTV/ffds.vcf.gz --plink --chrom-map ffds.chrom-map.txt --out ffds

<span class="hljs-comment"># Make bed for the ped/map plink file</span>
plink --file ffds --make-bed --allow-extra-chr --out ffds

<span class="hljs-comment"># Calculate PCA</span>
plink --file ffds --allow-extra-chr --pca --out ffds

</div></code></pre>
<h2 id="population-structure-analysis">Population structure analysis</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/fastStructure

<span class="hljs-comment"># Install fastStructure</span>
<span class="hljs-comment"># Be careful: fastStructure is based on python2, rathor than python3</span>
<span class="hljs-comment"># Copy fastStructure file</span>
git <span class="hljs-built_in">clone</span> https://github.com/rajanil/fastStructure
<span class="hljs-built_in">cd</span> fastStructure
git fetch
git merge origin/master
<span class="hljs-comment"># Build library extensions, with force extention -f, which is crucial to run</span>
<span class="hljs-built_in">cd</span> ~/proj/fastStructure/vars
python setup.py build_ext -f --inplace
<span class="hljs-built_in">cd</span> ~/proj/fastStructure
python setup.py build_ext -f --inplace

<span class="hljs-comment"># Run fastStructure</span>
<span class="hljs-comment"># Create chromosome map</span>
bcftools view -H ../IQ-Tree504/ffds.thin.vcf.gz | cut -f 1 | uniq | awk <span class="hljs-string">'{print $0"\t"$0}'</span> &gt; ffds.thin.chrom-map.txt

<span class="hljs-comment"># Make ped file using this chromosome map</span>
vcftools --gzvcf ../IQ-Tree504/ffds.thin.vcf.gz --plink --chrom-map ffds.thin.chrom-map.txt --out ffds.thin

<span class="hljs-comment"># Make bed for the ped/map plink file</span>
plink --file ffds.thin --make-bed --allow-extra-chr --out ffds.thin

<span class="hljs-comment"># Remove unnecessary files</span>
rm ffds.thin.chrom-map.txt ffds.thin.log ffds.thin.nosex 

<span class="hljs-comment"># fastStructure</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 20); <span class="hljs-keyword">do</span> 
    python ~/tools/fastStructure/structure.py -K <span class="hljs-variable">$i</span> --input=ffds.thin --output=fastStructure/output --full --seed=100
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Choose the optimal K (K = 7)</span>
python ~/tools/fastStructure/chooseK.py --input=fastStructure/output

<span class="hljs-comment"># Combine the fastStructure results at K = 7</span>
<span class="hljs-comment"># Run the fastStructure for 30 times using all ffds SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 30); <span class="hljs-keyword">do</span> 
    python ~/tools/fastStructure/structure.py -K 7 --input=ffds --output=fastStructure/K7/Run<span class="hljs-variable">$i</span> --full --seed=100
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files Run*.7.meanQ.ind </span>
rm id empty Run*.7.meanP Run*.7.meanQ.ind Run*.7.varP Run*.7.varQ Run*.7.log K7.miscfile K7.perm_datafile 

</div></code></pre>
<h3 id="population-structure-use-pruned-4dtv-snps">Population structure use pruned 4DTV SNPs</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SNP.prune

<span class="hljs-comment"># fastStructure use pruned 4DTV</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 1 10); <span class="hljs-keyword">do</span> 
    python2 /data/zhaojiantao/tools/fastStructure/structure.py -K <span class="hljs-variable">$i</span> \
        --input=Combined.prune \
        --output=output \
        --full --seed=100
<span class="hljs-keyword">done</span>

</div></code></pre>
<h2 id="genome-wide-ld">Genome-wide LD</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Check the subgroup of Ca.glabriusculum</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/LD/Phylogeny
<span class="hljs-comment"># Extract the Ca.glabriusculum and one Ca.pubescens to test</span>
vcftools --gzvcf ../../IQ-Tree504/ffds.thin.vcf.gz \
    --keep ../../Pop.list/Ca.glabriusculum.check.list \
    --recode --recode-INFO-all \
    --out Ca.glabriusculum.check

<span class="hljs-comment"># Convert vcf to phylip</span>
python /data/zhaojiantao/tools/PythonScripts/vcf2phylip.py -i Ca.glabriusculum.check.recode.vcf

<span class="hljs-comment"># IQ-tree</span>
iqtree -s Ca.glabriusculum.check.recode.min4.phy -nt AUTO -m MFP

<span class="hljs-comment"># Quality control of genotypes: maf &gt;= 0.05</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 0 12); <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../vcf.raw/Chr<span class="hljs-variable">$i</span>.vcf.gz --maf 0.05 --recode --stdout | bgzip -c &gt; ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Use PopLDdecay</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
        PopLDdecay -InVCF ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz -SubPop ../Pop.list/<span class="hljs-variable">$j</span>.list \
            -MaxDist 2000 -MAF 0.05 -OutStat <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span> ;
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># LD for the candidate region of chromosome 6</span>
<span class="hljs-comment"># Extract the region</span>
plink --vcf ../vcf.clean/Chr06.clean.vcf.gz  \
    --allow-extra-chr --chr Chr06 --geno 0.2 --maf 0.05 \
    --from-bp 243400000 --to-bp 244400000 \
    --recode vcf --threads 40 --double-id \
    --out Chr06.region

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chinense Ca.frutescens Ca.pubescens ; <span class="hljs-keyword">do</span>
    PopLDdecay -InVCF Chr06.region.vcf -SubPop &lt;(awk <span class="hljs-string">'{print $1"_"$1}'</span> ../Pop.list/<span class="hljs-variable">$i</span>.list) \
        -MaxDist 1000 -MAF 0.05 -OutStat Chr06.region.<span class="hljs-variable">$i</span> ;
    /data/zhaojiantao/tools/PopLDdecay-master/bin/Plot_OnePop.pl -inFile Chr06.region.<span class="hljs-variable">$i</span>.stat.gz -output Chr06.region.<span class="hljs-variable">$i</span>  
<span class="hljs-keyword">done</span>

less Chr06.region.Ca.bac.baccatum.bin.gz | awk <span class="hljs-string">'$1&lt;400000 {print $1,$2,"A01"}'</span> &gt; Chr06.BAB.region.ld
less Chr06.region.Ca.bac.pendulum.bin.gz | awk <span class="hljs-string">'$1&lt;400000 {print $1,$2,"A02"}'</span> &gt; Chr06.BAP.region.ld
less Chr06.region.Ca.frutescens.bin.gz | awk <span class="hljs-string">'$1&lt;400000 {print $1,$2,"A03"}'</span> &gt; Chr06.FRU.region.ld
less Chr06.region.Ca.chinense.bin.gz | awk <span class="hljs-string">'$1&lt;400000 {print $1,$2,"A04"}'</span> &gt; Chr06.CHN.region.ld
less Chr06.region.Ca.annuum.bin.gz | awk <span class="hljs-string">'$1&lt;400000 {print $1,$2,"A05"}'</span> &gt; Chr06.ANN.region.ld
cat *ld 

<span class="hljs-comment"># Plot the LD</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        /data/zhaojiantao/tools/PopLDdecay-master/bin/Plot_OnePop.pl -inFile <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.stat.gz -output <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add chromosome information to each stat files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    zcat <span class="hljs-variable">$i</span>.Chr01.bin.gz | awk <span class="hljs-string">'{print $1,$2,1}'</span> &gt; <span class="hljs-variable">$i</span>.Chr01.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr02.bin.gz | awk <span class="hljs-string">'{print $1,$2,2}'</span> &gt; <span class="hljs-variable">$i</span>.Chr02.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr03.bin.gz | awk <span class="hljs-string">'{print $1,$2,3}'</span> &gt; <span class="hljs-variable">$i</span>.Chr03.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr04.bin.gz | awk <span class="hljs-string">'{print $1,$2,4}'</span> &gt; <span class="hljs-variable">$i</span>.Chr04.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr05.bin.gz | awk <span class="hljs-string">'{print $1,$2,5}'</span> &gt; <span class="hljs-variable">$i</span>.Chr05.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr06.bin.gz | awk <span class="hljs-string">'{print $1,$2,6}'</span> &gt; <span class="hljs-variable">$i</span>.Chr06.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr07.bin.gz | awk <span class="hljs-string">'{print $1,$2,7}'</span> &gt; <span class="hljs-variable">$i</span>.Chr07.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr08.bin.gz | awk <span class="hljs-string">'{print $1,$2,8}'</span> &gt; <span class="hljs-variable">$i</span>.Chr08.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr09.bin.gz | awk <span class="hljs-string">'{print $1,$2,9}'</span> &gt; <span class="hljs-variable">$i</span>.Chr09.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr10.bin.gz | awk <span class="hljs-string">'{print $1,$2,10}'</span> &gt; <span class="hljs-variable">$i</span>.Chr10.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr11.bin.gz | awk <span class="hljs-string">'{print $1,$2,11}'</span> &gt; <span class="hljs-variable">$i</span>.Chr11.bin.gz-2 ;
    zcat <span class="hljs-variable">$i</span>.Chr12.bin.gz | awk <span class="hljs-string">'{print $1,$2,12}'</span> &gt; <span class="hljs-variable">$i</span>.Chr12.bin.gz-2 ;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results for each subgroup</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.pubescens ; <span class="hljs-keyword">do</span>
cat <span class="hljs-variable">$i</span>.Chr*.bin.gz-2 | grep -v <span class="hljs-string">"#"</span> &gt; <span class="hljs-variable">$i</span>.ld
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine all subgroups together</span>
paste Ca.annuum.ld Ca.bac.baccatum.ld Ca.bac.pendulum.ld Ca.chacoense.ld \
    Ca.chinense.ld Ca.frutescens.ld Ca.glabriusculum.ld Ca.glabriusculum_II.ld Ca.pubescens.ld | \
    awk <span class="hljs-string">'{print $1,$3,$2,$5,$8,$11,$14,$17,$20,$23}'</span> | \
    sed <span class="hljs-string">'1i\Dist Chr Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.ld.txt

<span class="hljs-comment"># Calculate the average LD across 12 chromosomes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens \
    Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    paste <span class="hljs-variable">$i</span>.Chr*.bin.gz-2 | sed <span class="hljs-string">'1d'</span> | \
        awk <span class="hljs-string">'{print $1,($2+$5+$8+$11+$14+$17+$20+$23+$26+$29+$32+$35)/12}'</span> \
        &gt; <span class="hljs-variable">$i</span>.average.ld
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results</span>
paste *average.ld | awk <span class="hljs-string">'{print $1,$2,$4,$6,$8,$10,$12,$14,$16,$18}'</span> | \
    sed <span class="hljs-string">'1i\Dist Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.average.ld.txt


<span class="hljs-comment"># Remove unnecessary files</span>
rm *png *stat.gz *pdf *bin.gz *bin.gz-2 *average.ld

<span class="hljs-comment"># Find the ld at LD =0.3</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens \
    Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens; <span class="hljs-keyword">do</span> 
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ;
    cat <span class="hljs-variable">$i</span>.average.ld | awk <span class="hljs-string">'$2&gt;0.2999 &amp;&amp; $2&lt;0.3001 {print $0}'</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the minimum ld</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.pubescens; <span class="hljs-keyword">do</span> 
    cat <span class="hljs-variable">$i</span>.ld | awk <span class="hljs-string">'$3==1 {print $0}'</span> | sort -nk2 | head -1 | cut -f2 -d <span class="hljs-string">' '</span> ; 
<span class="hljs-keyword">done</span>

</div></code></pre>
<h2 id="fst-differentiation">Fst differentiation</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Fst/

<span class="hljs-comment"># Calculate Fst between subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum glabriusculum_II pubescens; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum glabriusculum_II pubescens; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../vcf.clean/Chr<span class="hljs-variable">$chr</span>.clean.vcf.gz \
            --weir-fst-pop ../Pop.list/Ca.<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop ../Pop.list/Ca.<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out Ca.<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> 
        <span class="hljs-keyword">done</span> &amp;
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../vcf.clean/Chr<span class="hljs-variable">$chr</span>.clean.vcf.gz \
        --weir-fst-pop ../Pop.list/Ca.bac.pendulum.list \
        --weir-fst-pop ../Pop.list/Ca.glabriusculum_I.list \
        --fst-window-size 100000 \
        --fst-window-step 10000 \
        --out Ca.bac.pendulum.glabriusculum_I.Chr<span class="hljs-variable">$chr</span> &amp;
<span class="hljs-keyword">done</span>


<span class="hljs-comment"># Combine the results</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *weir.fst | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.Chr/\t/g'</span> | cut -f1 | sort | uniq); <span class="hljs-keyword">do</span> 
    cat <span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | \
        awk <span class="hljs-string">'{print $7=$1"_"$2"_"$3,$1,$2,$3,$8=($2+$3)/2,$4,$5,$6}'</span> | \
        awk <span class="hljs-string">'$8 &gt; 0 {print $0}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS WEIGHTED_FST MEAN_FST'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.fst.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove chromosomal level files</span>
rm *.windowed.weir.fst *<span class="hljs-built_in">log</span> 

<span class="hljs-comment"># Calculate the mean Fst value</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *fst.txt.gz | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.fst.txt//g'</span>); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; zcat <span class="hljs-variable">$i</span>.fst.txt.gz | grep -v MEAN | awk <span class="hljs-string">'{sum+=$8} END {print sum/NR}'</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *fst.txt-2 common.id

<span class="hljs-comment"># Check the total line numbers to determine the top 1% lines</span>
zcat Ca.frutescens.annuum.fst.txt.gz | wc -l
zcat Ca.frutescens.annuum.fst.txt.gz | sort -rgk8 | head -2970 | cut -f1-4,8 | sed <span class="hljs-string">'1i\SNP\tChr\tStart\tEnd\tFst'</span> \
    &gt; Ca.frutescens.annuum.top1.fst
<span class="hljs-comment"># Numberic sort for each chromosome</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    grep Chr<span class="hljs-variable">$i</span> Ca.frutescens.annuum.top1.fst | sort -nk3 &gt; Ca.frutescens.annuum.top1.fst.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Combine the sort</span>
cat Ca.frutescens.annuum.top1.fst.Chr* &gt; Ca.frutescens.annuum.top1.fst.sorted
<span class="hljs-comment"># Merge the top 1% windows</span>
perl merge.windows.pl Ca.frutescens.annuum.top1.fst.sorted | sed <span class="hljs-string">'1d'</span> &gt; Ca.frutescens.annuum.top1.fst.merged.txt
<span class="hljs-comment"># Remove unnecessary files</span>
rm Ca.frutescens.annuum.top1.fst Ca.frutescens.annuum.top1.fst.Chr*

<span class="hljs-comment"># Find the number of genes </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *merged.txt | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.txt//g'</span>) ; <span class="hljs-keyword">do</span>
    bedtools intersect \
        -a &lt;(sed <span class="hljs-string">'s/Chr0//g'</span> ../References/S8/Ca.S8.gff | sed <span class="hljs-string">'s/Chr//g'</span>) \
        -b &lt;(cut -f1-3 <span class="hljs-variable">$i</span>.txt) | \
        grep mRNA | sed <span class="hljs-string">'s/;/\t/g'</span> | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | grep rna \
        &gt; <span class="hljs-variable">$i</span>.genes.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Uniq gene count</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *merged.genes.txt | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.txt//g'</span>) ; <span class="hljs-keyword">do</span>
    sed <span class="hljs-string">'s/\./\t/g'</span> <span class="hljs-variable">$i</span>.txt | cut -f1 | sort | uniq &gt; <span class="hljs-variable">$i</span>.uniq
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the total size of differentiated regions</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *top1.fst.merged.txt ; <span class="hljs-keyword">do</span>
    awk <span class="hljs-string">'{s+=$3-$2+1}END{print s}'</span> <span class="hljs-variable">$i</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the file for significant test</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *fst.txt.gz | sed <span class="hljs-string">'s/.fst.txt.gz//g'</span> | sed <span class="hljs-string">'s/Ca.//g'</span>); <span class="hljs-keyword">do</span> \
    zcat Ca.<span class="hljs-variable">$i</span>.fst.txt.gz | grep -v MEAN | awk <span class="hljs-string">'{print $8,"'</span><span class="hljs-variable">$i</span><span class="hljs-string">'"}'</span> &gt; Ca.<span class="hljs-variable">$i</span>.fst
<span class="hljs-keyword">done</span>

cat Ca*fst | sed <span class="hljs-string">'1i\Fst Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Combined.fst.txt

</div></code></pre>
<h3 id="fst-with-accessions-no-taxonomy-reassignment">Fst with accessions no taxonomy reassignment</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Fst/No.taxonomy.correction

<span class="hljs-comment"># Calculate Fst between subgroups</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> annuum bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum_I glabriusculum_II; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> bac.baccatum bac.pendulum chacoense chinense frutescens glabriusculum_I glabriusculum_II pubescens; <span class="hljs-keyword">do</span>
        <span class="hljs-keyword">for</span> chr <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../../vcf.clean/Chr<span class="hljs-variable">$chr</span>.clean.vcf.gz \
            --weir-fst-pop Ca.<span class="hljs-variable">$i</span>.list \
            --weir-fst-pop Ca.<span class="hljs-variable">$j</span>.list \
            --fst-window-size 100000 \
            --fst-window-step 10000 \
            --out Ca.<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$chr</span> 
        <span class="hljs-keyword">done</span> &amp;
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>


<span class="hljs-comment"># Combine the results</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *weir.fst | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.Chr/\t/g'</span> | cut -f1 | sort | uniq); <span class="hljs-keyword">do</span> 
    cat <span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | \
        awk <span class="hljs-string">'{print $7=$1"_"$2"_"$3,$1,$2,$3,$8=($2+$3)/2,$4,$5,$6}'</span> | \
        awk <span class="hljs-string">'$8 &gt; 0 {print $0}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS WEIGHTED_FST MEAN_FST'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.fst.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove chromosomal level files</span>
rm *.windowed.weir.fst *<span class="hljs-built_in">log</span> 

<span class="hljs-comment"># Calculate the mean Fst value</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *fst.txt | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.fst.txt//g'</span>); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; cat <span class="hljs-variable">$i</span>.fst.txt | grep -v MEAN | awk <span class="hljs-string">'{sum+=$8} END {print sum/NR}'</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *fst.txt-2 common.id

<span class="hljs-comment"># Prepare the file for significant test</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *fst.txt.gz | sed <span class="hljs-string">'s/.fst.txt.gz//g'</span> | sed <span class="hljs-string">'s/Ca.//g'</span>); <span class="hljs-keyword">do</span> \
    zcat Ca.<span class="hljs-variable">$i</span>.fst.txt.gz | grep -v MEAN | awk <span class="hljs-string">'{print $8,"'</span><span class="hljs-variable">$i</span><span class="hljs-string">'"}'</span> &gt; Ca.<span class="hljs-variable">$i</span>.fst
<span class="hljs-keyword">done</span>

cat Ca*fst | sed <span class="hljs-string">'1i\Fst Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Combined.fst.txt

</div></code></pre>
<h3 id="fst-with-accessions-of-pure-background">Fst with accessions of pure background</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Fst/Pure.background

<span class="hljs-comment"># Calculate Fst between BAB and CHA</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
vcftools --gzvcf /data/zhaojiantao/pepper/vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
    --weir-fst-pop /data/zhaojiantao/pepper/Pop.list/Ca.bac.baccatum.list \
    --weir-fst-pop /data/zhaojiantao/pepper/Pop.list/Ca.chacoense.list \
    --fst-window-size 100000 \
    --fst-window-step 10000 \
    --out Ca.bac.baccatum.chacoense.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *weir.fst | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.Chr/\t/g'</span> | cut -f1 | sort | uniq); <span class="hljs-keyword">do</span> 
    cat <span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | \
        awk <span class="hljs-string">'{print $7=$1"_"$2"_"$3,$1,$2,$3,$8=($2+$3)/2,$4,$5,$6}'</span> | \
        awk <span class="hljs-string">'$8 &gt; 0 {print $0}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS WEIGHTED_FST MEAN_FST'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.fst.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove chromosomal level files</span>
rm *.windowed.weir.fst *<span class="hljs-built_in">log</span> 

<span class="hljs-comment"># Calculate the mean Fst value</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *fst.txt | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.fst.txt//g'</span>); <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; cat <span class="hljs-variable">$i</span>.fst.txt | grep -v MEAN | awk <span class="hljs-string">'{sum+=$8} END {print sum/NR}'</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *fst.txt-2 common.id

<span class="hljs-comment"># Prepare the file for significant test</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ls *fst.txt.gz | sed <span class="hljs-string">'s/.fst.txt.gz//g'</span> | sed <span class="hljs-string">'s/Ca.//g'</span>); <span class="hljs-keyword">do</span> \
    zcat Ca.<span class="hljs-variable">$i</span>.fst.txt.gz | grep -v MEAN | awk <span class="hljs-string">'{print $8,"'</span><span class="hljs-variable">$i</span><span class="hljs-string">'"}'</span> &gt; Ca.<span class="hljs-variable">$i</span>.fst
<span class="hljs-keyword">done</span>

cat Ca*fst | sed <span class="hljs-string">'1i\Fst Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Combined.fst.txt

</div></code></pre>
<h3 id="fst-permutation-test">Fst permutation test</h3>
<h4 id="babcha">BAB_CHA</h4>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Fst/PermutationTest/BAB_CHA

<span class="hljs-comment"># Without permutation</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
vcftools --vcf /data/zhaojiantao/pepper/SNP.prune/Chr<span class="hljs-variable">$i</span>.prune.vcf \
    --weir-fst-pop /data/zhaojiantao/pepper/Pop.list/Ca.bac.baccatum.list \
    --weir-fst-pop /data/zhaojiantao/pepper/Pop.list/Ca.chacoense.list \
    --fst-window-size 100000 \
    --fst-window-step 10000 \
    --out Ca.bac.baccatum.chacoense.Chr<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the mean value</span>
cat Ca.bac.baccatum.chacoense.Chr*.windowed.weir.fst | grep -v CHROM | \
    awk <span class="hljs-string">'$6&gt;=0 {print}'</span> | awk <span class="hljs-string">'{sum+=$6} END {print sum/NR}'</span> &gt; Ca.bac.baccatum.chacoense.meanFst
<span class="hljs-comment"># 0.411388</span>

<span class="hljs-comment"># Remove tempory file</span>
rm *<span class="hljs-built_in">log</span> *weir.fst

<span class="hljs-comment"># Permutation test</span>
wc -l /data/zhaojiantao/pepper/Pop.list/Ca.bac.baccatum.list /data/zhaojiantao/pepper/Pop.list/Ca.chacoense.list

<span class="hljs-comment"># Prepare the list</span>
cat /data/zhaojiantao/pepper/Pop.list/Ca.bac.baccatum.list /data/zhaojiantao/pepper/Pop.list/Ca.chacoense.list \
    &gt; Ca.bac.baccatum.chacoense.list

<span class="hljs-comment"># Randomize the samples </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    sort --random-sort Ca.bac.baccatum.chacoense.list | head -12 &gt; Ca.bac.baccatum.Random<span class="hljs-variable">$i</span>.list
    grep -Fvf Ca.bac.baccatum.Random<span class="hljs-variable">$i</span>.list Ca.bac.baccatum.chacoense.list &gt; Ca.chacoense.Random<span class="hljs-variable">$i</span>.list
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run the permutation test with 100 replicates</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    vcftools --vcf /data/zhaojiantao/pepper/SNP.prune/Chr<span class="hljs-variable">$j</span>.prune.vcf \
        --weir-fst-pop Ca.bac.baccatum.Random<span class="hljs-variable">$i</span>.list \
        --weir-fst-pop Ca.chacoense.Random<span class="hljs-variable">$i</span>.list \
        --fst-window-size 100000 \
        --fst-window-step 10000 \
        --out Ca.bac.baccatum.chacoense.Random<span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the mean value</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 100); <span class="hljs-keyword">do</span>
    cat Ca.bac.baccatum.chacoense.Random<span class="hljs-variable">$i</span>.Chr*.windowed.weir.fst | \
        grep -v CHROM | awk <span class="hljs-string">'$6&gt;=0 {print}'</span> | awk <span class="hljs-string">'{sum+=$6} END {print sum/NR}'</span> \
        &gt; Ca.bac.baccatum.chacoense.Random<span class="hljs-variable">$i</span>.meanFst
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the permutation result</span>
cat *Random*.meanFst &gt; Ca.bac.baccatum.chacoense.Random.meanFst

<span class="hljs-comment"># Remove tempory file</span>
rm *<span class="hljs-built_in">log</span> *weir.fst Ca.bac.baccatum.Random*.list Ca.chacoense.Random*.list 
rm Ca.bac.baccatum.chacoense.Random0*.meanFst Ca.bac.baccatum.chacoense.Random100.meanFst

<span class="hljs-comment"># Generate scripts for significant t test in R</span>
<span class="hljs-built_in">echo</span> <span class="hljs-string">"
data &lt;- read.table("</span>Ca.bac.baccatum.chacoense.Random.meanFst<span class="hljs-string">", header=FALSE)
AA &lt;- t.test(x=data, mu=0.411388, conf.level=0.95)
chars &lt;- capture.output(print(AA))
writeLines(chars, con = file("</span>t.test.txt<span class="hljs-string">")) "</span> | \
sed <span class="hljs-string">'s/Ca.bac.baccatum.chacoense.Random.meanFst/"Ca.bac.baccatum.chacoense.Random.meanFst"/g'</span> |\
sed <span class="hljs-string">'s/t.test.txt/"t.test.txt"/g'</span> \
&gt; t.test.R

<span class="hljs-comment"># Significant t test in R</span>
Rscript t.test.R

</div></code></pre>
<h2 id="nucleotide-diversity">Nucleotide diversity</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Workind directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Pi

<span class="hljs-comment"># Calculate per-site nucleotide diversity</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../../vcf.clean/Chr<span class="hljs-variable">$j</span>.clean.vcf.gz --keep <span class="hljs-variable">$i</span>.list --site-pi --out <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate windowed pi </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.sites.pi
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results of each chromosome</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens; <span class="hljs-keyword">do</span>
    cat <span class="hljs-variable">$i</span>.Chr*.w100000.s10000.txt | \
        grep -v <span class="hljs-string">"#Chr"</span> | \
        awk <span class="hljs-string">'{print $1"_"$3"_"$4,$1,$3,$4,($3+$4)/2,$6,$7}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS PI'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.windowed.pi.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the common ids</span>
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.annuum"}'</span> Ca.annuum.windowed.pi.txt &gt; Ca.annuum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.bac.baccatum"}'</span> Ca.bac.baccatum.windowed.pi.txt &gt; Ca.bac.baccatum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.bac.pendulum"}'</span> Ca.bac.pendulum.windowed.pi.txt &gt; Ca.bac.pendulum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.chacoense"}'</span> Ca.chacoense.windowed.pi.txt &gt; Ca.chacoense.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.chinense"}'</span> Ca.chinense.windowed.pi.txt &gt; Ca.chinense.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.frutescens"}'</span> Ca.frutescens.windowed.pi.txt &gt; Ca.frutescens.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.glabriusculum_II"}'</span> Ca.glabriusculum_II.windowed.pi.txt &gt; Ca.glabriusculum_II.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.glabriusculum"}'</span> Ca.glabriusculum.windowed.pi.txt &gt; Ca.glabriusculum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"Ca.pubescens"}'</span> Ca.pubescens.windowed.pi.txt &gt; Ca.pubescens.windowed.pi.txt-2

cat *windowed.pi.txt-2 | grep -v ID | sed <span class="hljs-string">'1i\SNP Chromosome Position PI Taxonomy'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.pi.txt

<span class="hljs-comment"># Calculate the mean value of Pi</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *windowed.pi.txt ; <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ;
    grep -v ID <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{ total += $7 } END { print total/NR }'</span> 
<span class="hljs-keyword">done</span>

cat *Random*.meanFst &gt; Ca.bac.baccatum.chacoense.Random.meanFst

<span class="hljs-comment"># Remove unnecessary files</span>
rm *.windowed.pi.txt-2 common.snp *windowed.pi *.w100000.s10000.txt *sites.pi *<span class="hljs-built_in">log</span> *

</div></code></pre>
<h3 id="nucleotide-diversity-with-accessions-no-taxonomy-reassignment">Nucleotide diversity with accessions no taxonomy reassignment</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Pi/No.taxonomy.correction

<span class="hljs-comment"># Calculate per-site nucleotide diversity</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum_I Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../../vcf.clean/Chr<span class="hljs-variable">$j</span>.clean.vcf.gz --keep <span class="hljs-variable">$i</span>.list --site-pi --out <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span> &amp;
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate windowed pi </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum_I Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.sites.pi
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results of each chromosome</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum_I Ca.glabriusculum_II Ca.pubescens; <span class="hljs-keyword">do</span>
    cat <span class="hljs-variable">$i</span>.Chr*.w100000.s10000.txt | \
        grep -v <span class="hljs-string">"#Chr"</span> | \
        awk <span class="hljs-string">'{print $1"_"$3"_"$4,$1,$3,$4,($3+$4)/2,$6,$7}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS PI'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.windowed.pi.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the common ids</span>
awk <span class="hljs-string">'{print $1,$2,$3,$7,"A"}'</span> Ca.pubescens.windowed.pi.txt &gt; Ca.pubescens.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"B"}'</span> Ca.chacoense.windowed.pi.txt &gt; Ca.chacoense.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"C"}'</span> Ca.bac.baccatum.windowed.pi.txt &gt; Ca.bac.baccatum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"D"}'</span> Ca.bac.pendulum.windowed.pi.txt &gt; Ca.bac.pendulum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"E"}'</span> Ca.glabriusculum_I.windowed.pi.txt &gt; Ca.glabriusculum_I.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"F"}'</span> Ca.glabriusculum_II.windowed.pi.txt &gt; Ca.glabriusculum_II.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"G"}'</span> Ca.annuum.windowed.pi.txt &gt; Ca.annuum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"H"}'</span> Ca.frutescens.windowed.pi.txt &gt; Ca.frutescens.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"I"}'</span> Ca.chinense.windowed.pi.txt &gt; Ca.chinense.windowed.pi.txt-2

cat *windowed.pi.txt-2 | grep -v ID | sed <span class="hljs-string">'1i\SNP Chromosome Position PI Taxonomy'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.pi.nocorrect.txt

<span class="hljs-comment"># Calculate the mean value of Pi</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *windowed.pi.txt ; <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ;
    grep -v ID <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{ total += $7 } END { print total/NR }'</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the mean value of pi for all </span>
grep -v SNP Group.pi.nocorrect.txt | awk <span class="hljs-string">'{ total += $4 } END { print total/NR }'</span> 

<span class="hljs-comment"># Clean the files</span>
rm *w100000.s10000.txt *<span class="hljs-built_in">log</span> *-2
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum_I Ca.glabriusculum_II Ca.pubescens; <span class="hljs-keyword">do</span>
    cat <span class="hljs-variable">$i</span>.Chr*.sites.pi | gzip &gt; <span class="hljs-variable">$i</span>.sites.pi.gz
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="nucleotide-diversity-with-accessions-of-pure-background">Nucleotide diversity with accessions of pure background</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Pi/Pure.background

<span class="hljs-comment"># Calculate per-site nucleotide diversity</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        vcftools --gzvcf ../../vcf.clean/Chr<span class="hljs-variable">$j</span>.clean.vcf.gz --keep <span class="hljs-variable">$i</span>.list --site-pi --out <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate windowed pi </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum_I Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
        perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.sites.pi
    <span class="hljs-keyword">done</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the results of each chromosome</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum_I Ca.glabriusculum_II Ca.pubescens; <span class="hljs-keyword">do</span>
    cat <span class="hljs-variable">$i</span>.Chr*.w100000.s10000.txt | \
        grep -v <span class="hljs-string">"#Chr"</span> | \
        awk <span class="hljs-string">'{print $1"_"$3"_"$4,$1,$3,$4,($3+$4)/2,$6,$7}'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr0/, "", $2)} 1'</span> | \
        awk <span class="hljs-string">'{gsub(/Chr/, "", $2)} 1'</span> | \
        sed <span class="hljs-string">'1i\ID CHROM BIN_START BIN_END BIN_Middle N_VARIANTS PI'</span> | \
        sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.windowed.pi.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the common ids</span>
awk <span class="hljs-string">'{print $1,$2,$3,$7,"A"}'</span> Ca.pubescens.windowed.pi.txt &gt; Ca.pubescens.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"B"}'</span> Ca.chacoense.windowed.pi.txt &gt; Ca.chacoense.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"C"}'</span> Ca.bac.baccatum.windowed.pi.txt &gt; Ca.bac.baccatum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"D"}'</span> Ca.bac.pendulum.windowed.pi.txt &gt; Ca.bac.pendulum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"E"}'</span> Ca.glabriusculum_I.windowed.pi.txt &gt; Ca.glabriusculum_I.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"F"}'</span> Ca.glabriusculum_II.windowed.pi.txt &gt; Ca.glabriusculum_II.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"G"}'</span> Ca.annuum.windowed.pi.txt &gt; Ca.annuum.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"H"}'</span> Ca.frutescens.windowed.pi.txt &gt; Ca.frutescens.windowed.pi.txt-2
awk <span class="hljs-string">'{print $1,$2,$3,$7,"I"}'</span> Ca.chinense.windowed.pi.txt &gt; Ca.chinense.windowed.pi.txt-2

cat *windowed.pi.txt-2 | grep -v ID | sed <span class="hljs-string">'1i\SNP Chromosome Position PI Taxonomy'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Group.pi.pure.txt

<span class="hljs-comment"># Calculate the mean value of Pi</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> *windowed.pi.txt ; <span class="hljs-keyword">do</span>
    <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span> ;
    grep -v ID <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{ total += $7 } END { print total/NR }'</span> 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the mean value of pi for all </span>
grep -v SNP Group.pi.pure.txt | awk <span class="hljs-string">'{ total += $4 } END { print total/NR }'</span> 

</div></code></pre>
<h2 id="treemix-analysis">Treemix analysis</h2>
<p>Reference: <a href="https://speciationgenomics.github.io/Treemix/">treemix analysis</a></p>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/treemix

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Step 16.1 Create chromosome map</span>
    <span class="hljs-built_in">echo</span> Chr<span class="hljs-variable">$i</span> Chr<span class="hljs-variable">$i</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr<span class="hljs-variable">$i</span>.clean-map.txt ; 
    <span class="hljs-comment"># Step 16.2 Make ped file using this chromosome map</span>
    vcftools --gzvcf ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
        --keep ../Pop.list/Pop8File.txt \
        --max-missing 1 \
        --plink --chrom-map Chr<span class="hljs-variable">$i</span>.clean-map.txt \
        --out Chr<span class="hljs-variable">$i</span>.noN
    <span class="hljs-comment"># Step 16.3 It assumes unlinked SNPs and we are thus first going to prune the file for SNPs in high LD.</span>
    plink --file Chr<span class="hljs-variable">$i</span>.noN --indep-pairwise 50 10 0.2 --allow-extra-chr --out Chr<span class="hljs-variable">$i</span>.noN
    plink --file Chr<span class="hljs-variable">$i</span>.noN --extract Chr<span class="hljs-variable">$i</span>.noN.prune.in --make-bed --allow-extra-chr --out Chr<span class="hljs-variable">$i</span>.noN.pruned
    <span class="hljs-comment"># Step 16.4 Prepare the pop clust file</span>
    cat ../Pop.list/Ca.annuum.list | awk <span class="hljs-string">'{print $1,$1,"Ca.annuum"}'</span> &gt; Ca.annuum.clust
    cat ../Pop.list/Ca.bac.baccatum.list | awk <span class="hljs-string">'{print $1,$1,"Ca.bac.baccatum"}'</span> &gt; Ca.bac.baccatum.clust
    cat ../Pop.list/Ca.bac.pendulum.list | awk <span class="hljs-string">'{print $1,$1,"Ca.bac.pendulum"}'</span> &gt; Ca.bac.pendulum.clust
    cat ../Pop.list/Ca.chacoense.list | awk <span class="hljs-string">'{print $1,$1,"Ca.chacoense"}'</span> &gt; Ca.chacoense.clust
    cat ../Pop.list/Ca.chinense.list | awk <span class="hljs-string">'{print $1,$1,"Ca.chinense"}'</span> &gt; Ca.chinense.clust
    cat ../Pop.list/Ca.frutescens.list | awk <span class="hljs-string">'{print $1,$1,"Ca.frutescens"}'</span> &gt; Ca.frutescens.clust
    cat ../Pop.list/Ca.glabriusculum.list | awk <span class="hljs-string">'{print $1,$1,"Ca.glabriusculum"}'</span> &gt; Ca.glabriusculum.clust
    cat ../Pop.list/Ca.pubescens.list | awk <span class="hljs-string">'{print $1,$1,"Ca.pubescens"}'</span> &gt; Ca.pubescens.clust
    cat *clust &gt; Pop8.clust.txt
    rm *clust
    <span class="hljs-comment"># Step 16.5 Calculate the frequency</span>
    plink -bfile Chr<span class="hljs-variable">$i</span>.noN.pruned --freq --missing --allow-extra-chr --within Pop8.clust.txt --out Chr<span class="hljs-variable">$i</span>.noN.pruned
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Step 16.6 Prepare the input file for treemix</span>
cat Chr01.noN.pruned.frq.strat | sed -n <span class="hljs-string">'1p'</span> &gt; header
cat Chr*.noN.pruned.frq.strat | grep -v CHR &gt; noN.pruned.frq-1
cat header noN.pruned.frq-1 | gzip &gt; noN.pruned.frq.gz

<span class="hljs-comment"># Step 16.7 Convert frequency file to treemix file format</span>
plink2treemix.py noN.pruned.frq.gz noN.pruned.treemix.frq.gz

<span class="hljs-comment"># Step 16.8 Remove unnecessary files</span>
rm *clean-map.txt *bed *bim *fam *<span class="hljs-built_in">log</span> *map *nosex *ped *frq.strat *imiss *lmiss *prune.in *prune.out header noN.pruned.frq-1 

<span class="hljs-comment"># Step 16.9 run treemix</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/treemix/migration

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 10); <span class="hljs-keyword">do</span>  
    treemix \
        -i ../Ca.S8.noN.pruned.treemix.frq.gz \
        -m <span class="hljs-variable">$i</span> \
        -o Ca.S8.noN.pruned.treemix.<span class="hljs-variable">$i</span> \
        -root Ca.pubescens \
        -bootstrap \
        -k 1400 &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the optimal edges using [OptM R package](https://cran.r-project.org/web/packages/OptM/readme/README.html)</span>
<span class="hljs-comment"># The PL and BC model are parametric models that </span>
<span class="hljs-keyword">for</span> m <span class="hljs-keyword">in</span> {1..10}; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> {1..10}; <span class="hljs-keyword">do</span>
        <span class="hljs-comment"># Generate random seed</span>
        s=<span class="hljs-variable">$RANDOM</span>
        treemix -i Ca.S8.noN.pruned.treemix.frq.gz -o OptM/S8.<span class="hljs-variable">${i}</span>.<span class="hljs-variable">${m}</span> -m <span class="hljs-variable">${m}</span> -k 1400 -seed <span class="hljs-variable">$s</span> -bootstrap 100 -root Ca.pubescens
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

test.linear = optM(<span class="hljs-string">"OptM"</span>, method = <span class="hljs-string">"linear"</span>)
jpeg(<span class="hljs-string">"treemix.migration.linear.jpeg"</span>, height = 15, width = 20, units=<span class="hljs-string">"cm"</span>, res = 300)
plot_optM(test.linear, method = <span class="hljs-string">"linear"</span>)
dev.off()

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> {1..1000}; <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Generate random seed</span>
    s=<span class="hljs-variable">$RANDOM</span>
    treemix -i Ca.S8.noN.pruned.treemix.frq.gz -o OptM2/S8.<span class="hljs-variable">${i}</span>.1 -m 1 -k 1400 -seed <span class="hljs-variable">$s</span> -bootstrap 100 -root Ca.pubescens
<span class="hljs-keyword">done</span>

</div></code></pre>
<h2 id="abba-baba-tests">ABBA-BABA tests</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/ABBA-BABA

<span class="hljs-comment"># Convert vcf to ABAB-ABBA geno file format for BAC2FRU+CHN</span>
<span class="hljs-comment"># Extract and convert the analyzed samples</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    python /data/zhaojiantao/tools/ABBA-BABA/genomics_general-master/VCF_processing/parseVCF.py \
        -i ../Chr<span class="hljs-variable">$i</span>.clean.vcf.gz --skipIndels --minQual 30 --gtf flag=DP |\ 
        bgzip -c &gt; Chr<span class="hljs-variable">$i</span>.clean.geno.gz 
<span class="hljs-keyword">done</span>


<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    python /data/zhaojiantao/tools/ABBA-BABA/genomics_general-master/VCF_processing/parseVCF.py \
        -i ../Chr<span class="hljs-variable">$i</span>.clean.vcf.gz --skipIndels --minQual 30 --gtf flag=DP | \
        bgzip -c &gt; Chr<span class="hljs-variable">$i</span>.clean.geno.gz 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Combine the files</span>
zcat Chr01.clean.geno.gz | head -1 &gt; header
zcat Chr*.clean.geno.gz | grep -v <span class="hljs-string">"#CHROM"</span> &gt; clean.geno
cat header clean.geno | gzip &gt; clean.geno.gz

<span class="hljs-comment"># Remove unnecessary files</span>
rm header clean.geno Chr*.clean.geno.gz 
rm *clean-map.txt 

<span class="hljs-comment"># Prepare the popfile txt</span>
cat ../treemix/Pop8.clust.txt | cut -f 1,3 -d <span class="hljs-string">' '</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Pop8File.txt
<span class="hljs-comment"># Introgression between BAB to FRU/CHN</span>
cat Pop8File.txt | sed <span class="hljs-string">'s/Ca.bac.baccatum/Ca.baccatum/g'</span> | sed <span class="hljs-string">'s/Ca.bac.pendulum/Ca.baccatum/g'</span> | \
    sed <span class="hljs-string">'s/Ca.chinense/Ca.FRU.CHN/g'</span> | sed <span class="hljs-string">'s/Ca.frutescens/Ca.FRU.CHN/g'</span> &gt; Pop.BAB2ANN.txt

<span class="hljs-comment"># Calculate the allele frequency</span>
python2 /data/zhaojiantao/tools/ABBA-BABA/genomics_general-master/freq.py -t 60 \
    -g Ca.S8.clean.geno.gz \
    -p Ca.annuum -p Ca.baccatum \
    -p Ca.FRU.CHN -p Ca.pubescens \
    --popsFile Pop.BAB2ANN.txt --target derived \
    -o BAC2FRUCHN.derFreq.tsv.gz

</div></code></pre>
<h2 id="introgression">Introgression</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/ABBA-BABA

<span class="hljs-comment"># Extract and filter the SNPs</span>
plink --vcf ../vcf.clean/Chr06.clean.vcf.gz  \
    --keep &lt;(awk <span class="hljs-string">'{print $1,$1}'</span> Pop.PUB.BAB2ANN.txt) \
    --allow-extra-chr --chr Chr06 --geno 0.1 --maf 0.05 \
    --from-bp 241500000 --to-bp 247500000 \
    --recode -output-missing-genotype 9 --double-id --threads 40 \
    --out Chr06.region
awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> Chr06.region.map | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr06.region.map2
mv Chr06.region.map2 Chr06.region.map

<span class="hljs-comment"># Calculate freq for BAC</span>
plink --file Chr06.region \
    --keep &lt;(cat ../Pop.list/Ca.bac*.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr06.region.BAC

<span class="hljs-comment"># Calculate frequency for PUB</span>
plink --file Chr06.region \
    --keep &lt;(cat ../Pop.list/Ca.pubescens.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr06.region.PUB

<span class="hljs-comment"># Find the SNPs where PUB.dominant not equal to BAC.dominant allele</span>
paste Chr06.region.PUB.frq Chr06.region.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr06.region.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f2 &gt; Chr06.region.remain.snps

<span class="hljs-comment"># Transpose to tped</span>
plink --file Chr06.region --extract Chr06.region.remain.snps \
    --recode transpose --out Chr06.region

<span class="hljs-comment"># Combine every two lines</span>
cut -f5- -d <span class="hljs-string">' '</span> Chr06.region.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Chr06.region.geno-1

<span class="hljs-comment"># Chr06.region.BAC.dominant.allele</span>
grep -v CHR Chr06.region.BAC.frq | awk <span class="hljs-string">'{print $4$4}'</span> &gt; Chr06.region.BAC.dominant.allele.out
paste Chr06.region.PUB.frq Chr06.region.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr06.region.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f7 \
    &gt; Chr06.region.BAC.dominant.allele.out1

<span class="hljs-comment"># Check the SNP IDs</span>
paste Chr06.region.BAC.dominant.allele.out Chr06.region.geno-1 | \
    awk <span class="hljs-string">'{for(i=2;i&lt;=449;i=i+1) if ($i==$1){printf "1 "} else {printf "0 "}; printf "\n"}'</span> \
    &gt; Chr06.region.alleles

<span class="hljs-comment"># Change heterozygous to 0.5</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); <span class="hljs-keyword">do</span>
    cut -f <span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr06.region.geno-1 | sed <span class="hljs-string">'s/./&amp; /g'</span> | \
    paste Chr06.region.BAC.dominant.allele.out1 - | \
    awk <span class="hljs-string">'{
        if ($2==$1 &amp;&amp; $3==$1) 
        print 1 ;
        else if ($2==$1 &amp;&amp; $3!=$1)
        print 0.5
        else 
        print 0
    }'</span> &gt; Accession<span class="hljs-variable">$i</span>.geno
<span class="hljs-keyword">done</span>
paste Accession*.geno | sed <span class="hljs-string">'s/\t/ /g'</span> &gt; Chr06.region.alleles
rm Accession*.geno

<span class="hljs-comment"># Transpose the genotypes</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr06.region.alleles &gt; Chr06.region.alleles.trans

<span class="hljs-comment"># Sort the genos</span>
paste Chr06.sample.id Chr06.region.alleles.trans | \
    sed <span class="hljs-string">'s/\t/ /g'</span> | \
    sort -k3,3 -t <span class="hljs-string">' '</span> | \
    cut -f1,2,4- -d <span class="hljs-string">' '</span> \
    &gt; Chr06.region.introgression 

<span class="hljs-comment"># Transpose the snp infos</span>
cut -f4 -d <span class="hljs-string">' '</span> Chr06.region.tped | sed <span class="hljs-string">'1i\ID\nTaxonomy'</span> &gt; Chr06.region.pos
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr06.region.pos &gt; Chr06.region.pos.trans

<span class="hljs-comment"># Combine the matrix</span>
cat Chr06.region.pos.trans Chr06.region.introgression &gt; Chr06.region.introgression.txt
<span class="hljs-comment"># Transpose the data</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr06.region.introgression.txt &gt; Chr06.region.introgression.trans.txt

grep -v ID Chr06.region.introgression.trans.txt | \
    grep -v Taxonomy | \
    cut -f2- -d <span class="hljs-string">' '</span> \
    &gt; Chr06.region.introgression.trans2

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 0 35472); <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; <span class="hljs-keyword">done</span> | \
    paste - &lt;(grep -v ID Chr06.region.introgression.trans.txt | grep -v Taxonomy) | awk <span class="hljs-string">'{print 6,$1,1,$2,0}'</span> &gt; Chr06.region.introgression.trans3

<span class="hljs-comment"># Cluster the result in sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); <span class="hljs-keyword">do</span>
    perl cluster_introgression.pl \
        &lt;(cut -f<span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr06.region.introgression.trans2 | \
        paste - Chr06.region.introgression.trans3 | awk <span class="hljs-string">'{print $2,$3,$4,$5,$6,$1}'</span>) \
        -wind_size 1000000 -wind_step 100000 \
        &gt; Chr06.region.w1000000.s100000.Accession<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the result</span>
paste Chr06.region.w1000000.s100000.Accession* &gt; Chr06.region.w1000000.s100000.Accessions.region
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr06.region.w1000000.s100000.Accessions.region | grep Avg | grep -v AdjAvg &gt; Chr06.region.w1000000.s100000.Accessions.region.trans

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 2 448); <span class="hljs-keyword">do</span>
    rm Chr06.region.w1000000.s100000.Accession<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Original values</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt; Chr06.region.w1000000.s100000.Accessions.region.trans | \
    grep -v Avg &gt;Chr06.region.w1000000.s100000.Accessions.region.clean-1

awk <span class="hljs-string">'{print $1,$2,$3,($2+$3)/2}'</span> Chr06.region.w1000000.s100000.Accession001 | \
    grep -v chrID | paste - Chr06.region.w1000000.s100000.Accessions.region.clean-1 | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr06.region.w1000000.s100000.Accessions.region.clean.txt

python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt;Chr06.region.w1000000.s100000.Accessions.region.clean.txt | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr06.region.w1000000.s100000.Accessions.region.clean.trans.txt

</div></code></pre>
<h3 id="introgression-chr074900001-10000000">Introgression Chr07:4,900,001-10,000,000</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/ABBA-BABA

<span class="hljs-comment"># Extract and filter the SNPs</span>
plink --vcf ../vcf.clean/Chr07.clean.vcf.gz  \
    --keep &lt;(awk <span class="hljs-string">'{print $1,$1}'</span> Pop.PUB.BAB2ANN.txt) \
    --allow-extra-chr --chr Chr07 --geno 0.1 --maf 0.05 \
    --from-bp 3900001 --to-bp 13000000 \
    --recode -output-missing-genotype 9 --double-id \
    --out Chr07.region1
awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> Chr07.region1.map | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr07.region1.map2
mv Chr07.region1.map2 Chr07.region1.map

<span class="hljs-comment"># Calculate freq for BAC</span>
plink --file Chr07.region1 \
    --keep &lt;(cat ../Pop.list/Ca.bac*.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr07.region1.BAC

<span class="hljs-comment"># Calculate frequency for PUB</span>
plink --file Chr07.region1 \
    --keep &lt;(cat ../Pop.list/Ca.pubescens.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr07.region1.PUB

<span class="hljs-comment"># Find the SNPs where PUB.dominant not equal to BAC.dominant allele</span>
paste Chr07.region1.PUB.frq Chr07.region1.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr07.region1.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f2 &gt; Chr07.region1.remain.snps

<span class="hljs-comment"># Transpose to tped</span>
plink --file Chr07.region1 --extract Chr07.region1.remain.snps \
    --recode transpose --out Chr07.region1

<span class="hljs-comment"># Combine every two lines</span>
cut -f5- -d <span class="hljs-string">' '</span> Chr07.region1.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Chr07.region1.geno-1

<span class="hljs-comment"># Chr07.region1.BAC.dominant.allele</span>
grep -v CHR Chr07.region1.BAC.frq | awk <span class="hljs-string">'{print $4$4}'</span> &gt; Chr07.region1.BAC.dominant.allele.out
paste Chr07.region1.PUB.frq Chr07.region1.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr07.region1.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f7 \
    &gt; Chr07.region1.BAC.dominant.allele.out1

<span class="hljs-comment"># Check the SNP IDs</span>
<span class="hljs-comment">#paste Chr07.region1.BAC.dominant.allele.out Chr07.region1.geno-1 | \</span>
<span class="hljs-comment">#    awk '{for(i=2;i&lt;=449;i=i+1) if ($i==$1){printf "1 "} else {printf "0 "}; printf "\n"}' \</span>
<span class="hljs-comment">#    &gt; Chr07.region1.alleles</span>

<span class="hljs-comment"># Change heterozygous to 0.5</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); 
    <span class="hljs-keyword">do</span> cut -f <span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr07.region1.geno-1 | sed <span class="hljs-string">'s/./&amp; /g'</span> | \
    paste Chr07.region1.BAC.dominant.allele.out1 - | \
    awk <span class="hljs-string">'{
        if ($2==$1 &amp;&amp; $3==$1) 
        print 1 ;
        else if ($2==$1 &amp;&amp; $3!=$1)
        print 0.5
        else 
        print 0
    }'</span> &gt; Accession<span class="hljs-variable">$i</span>.geno &amp;
<span class="hljs-keyword">done</span>
paste Accession*.geno | sed <span class="hljs-string">'s/\t/ /g'</span> &gt; Chr07.region1.alleles
rm Accession*.geno

<span class="hljs-comment"># Transpose the genotypes</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region1.alleles &gt; Chr07.region1.alleles.trans

<span class="hljs-comment"># Sort the genos</span>
paste Chr07.sample.id Chr07.region1.alleles.trans | sed <span class="hljs-string">'s/\t/ /g'</span> | sort -k3 | cut -f1,2,4- -d <span class="hljs-string">' '</span> &gt; Chr07.region1.introgression 

<span class="hljs-comment"># Transpose the snp infos</span>
cut -f4 -d <span class="hljs-string">' '</span> Chr07.region1.tped | sed <span class="hljs-string">'1i\ID\nTaxonomy'</span> &gt; Chr07.region1.pos
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region1.pos &gt; Chr07.region1.pos.trans

<span class="hljs-comment"># Combine the matrix</span>
cat Chr07.region1.pos.trans Chr07.region1.introgression &gt; Chr07.region1.introgression.txt
<span class="hljs-comment"># Transpose the data</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region1.introgression.txt &gt; Chr07.region1.introgression.trans.txt

grep -v ID Chr07.region1.introgression.trans.txt | grep -v Taxonomy | cut -f2- -d <span class="hljs-string">' '</span> &gt; Chr07.region1.introgression.trans2

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 0 38089); <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; <span class="hljs-keyword">done</span> | \
    paste - &lt;(grep -v ID Chr07.region1.introgression.trans.txt | grep -v Taxonomy) | awk <span class="hljs-string">'{print 6,$1,1,$2,0}'</span> &gt; Chr07.region1.introgression.trans3

<span class="hljs-comment"># Cluster the result in sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); <span class="hljs-keyword">do</span>
    perl cluster_introgression.pl \
        &lt;(cut -f<span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr07.region1.introgression.trans2 | \
        paste - Chr07.region1.introgression.trans3 | awk <span class="hljs-string">'{print $2,$3,$4,$5,$6,$1}'</span>) \
        -wind_size 1000000 -wind_step 100000 \
        &gt; Chr07.region1.w1000000.s100000.Accession<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the result</span>
paste Chr07.region1.w1000000.s100000.Accession* &gt; Chr07.region1.w1000000.s100000.Accessions.region1
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region1.w1000000.s100000.Accessions.region1 | grep Avg | grep -v AdjAvg &gt; Chr07.region1.w1000000.s100000.Accessions.region1.trans

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 2 448); <span class="hljs-keyword">do</span>
    rm Chr07.region1.w1000000.s100000.Accession<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Original values</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt; Chr07.region1.w1000000.s100000.Accessions.region1.trans | \
    grep -v Avg &gt;Chr07.region1.w1000000.s100000.Accessions.region1.clean-1

awk <span class="hljs-string">'{print $1,$2,$3,($2+$3)/2}'</span> Chr07.region1.w1000000.s100000.Accession001 | \
    grep -v chrID | paste - Chr07.region1.w1000000.s100000.Accessions.region1.clean-1 | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr07.region1.w1000000.s100000.Accessions.region1.clean.txt

python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt;Chr07.region1.w1000000.s100000.Accessions.region1.clean.txt | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr07.region1.w1000000.s100000.Accessions.region1.clean.trans.txt

<span class="hljs-comment"># Prepare the plot of fd</span>
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAC.FRUCHN.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr07" &amp;&amp; $2&gt;=3900001 &amp;&amp; $3&lt;=13000000 {print $0}'</span> | grep -v <span class="hljs-string">"nan"</span> | \
    cut -f1-4,10 &gt; PUB.BAC.FRUCHN.ANN.Chr07.region1.out.txt

</div></code></pre>
<h3 id="introgression-chr07258150001-259300000">Introgression Chr07:258,150,001-259,300,000</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/ABBA-BABA

<span class="hljs-comment"># Extract and filter the SNPs</span>
plink --vcf ../vcf.clean/Chr07.clean.vcf.gz  \
    --keep &lt;(awk <span class="hljs-string">'{print $1,$1}'</span> Pop.PUB.BAB2ANN.txt) \
    --allow-extra-chr --chr Chr07 --geno 0.1 --maf 0.05 \
    --from-bp 256150001 --to-bp 261300000 \
    --recode -output-missing-genotype 9 --double-id \
    --out Chr07.region2
awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> Chr07.region2.map | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr07.region2.map2
mv Chr07.region2.map2 Chr07.region2.map

<span class="hljs-comment"># Calculate freq for BAC</span>
plink --file Chr07.region2 \
    --keep &lt;(cat ../Pop.list/Ca.bac*.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr07.region2.BAC

<span class="hljs-comment"># Calculate frequency for PUB</span>
plink --file Chr07.region2 \
    --keep &lt;(cat ../Pop.list/Ca.pubescens.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr07.region2.PUB

<span class="hljs-comment"># Find the SNPs where PUB.dominant not equal to BAC.dominant allele</span>
paste Chr07.region2.PUB.frq Chr07.region2.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr07.region2.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f2 &gt; Chr07.region2.remain.snps

<span class="hljs-comment"># Transpose to tped</span>
plink --file Chr07.region2 --extract Chr07.region2.remain.snps \
    --recode transpose --out Chr07.region2

<span class="hljs-comment"># Combine every two lines</span>
cut -f5- -d <span class="hljs-string">' '</span> Chr07.region2.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Chr07.region2.geno-1

<span class="hljs-comment"># Chr07.region2.BAC.dominant.allele</span>
grep -v CHR Chr07.region2.BAC.frq | awk <span class="hljs-string">'{print $4$4}'</span> &gt; Chr07.region2.BAC.dominant.allele.out
paste Chr07.region2.PUB.frq Chr07.region2.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr07.region2.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f7 \
    &gt; Chr07.region2.BAC.dominant.allele.out1

<span class="hljs-comment"># Check the SNP IDs</span>
paste Chr07.region2.BAC.dominant.allele.out Chr07.region2.geno-1 | \
    awk <span class="hljs-string">'{for(i=2;i&lt;=449;i=i+1) if ($i==$1){printf "1 "} else {printf "0 "}; printf "\n"}'</span> \
    &gt; Chr07.region2.alleles

<span class="hljs-comment"># Change heterozygous to 0.5</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); <span class="hljs-keyword">do</span>
    cut -f <span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr07.region2.geno-1 | sed <span class="hljs-string">'s/./&amp; /g'</span> | \
    paste Chr07.region2.BAC.dominant.allele.out1 - | \
    awk <span class="hljs-string">'{
        if ($2==$1 &amp;&amp; $3==$1) 
        print 1 ;
        else if ($2==$1 &amp;&amp; $3!=$1)
        print 0.5
        else 
        print 0
    }'</span> &gt; Accession<span class="hljs-variable">$i</span>.geno &amp;
<span class="hljs-keyword">done</span>

paste Accession*.geno | sed <span class="hljs-string">'s/\t/ /g'</span> &gt; Chr07.region2.alleles
rm Accession*.geno

<span class="hljs-comment"># Transpose the genotypes</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region2.alleles &gt; Chr07.region2.alleles.trans

<span class="hljs-comment"># Sort the genos</span>
paste Chr07.sample.id Chr07.region2.alleles.trans | sed <span class="hljs-string">'s/\t/ /g'</span> | sort -k3 | cut -f1,2,4- -d <span class="hljs-string">' '</span> &gt; Chr07.region2.introgression 

<span class="hljs-comment"># Transpose the snp infos</span>
cut -f4 -d <span class="hljs-string">' '</span> Chr07.region2.tped | sed <span class="hljs-string">'1i\ID\nTaxonomy'</span> &gt; Chr07.region2.pos
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region2.pos &gt; Chr07.region2.pos.trans

<span class="hljs-comment"># Combine the matrix</span>
cat Chr07.region2.pos.trans Chr07.region2.introgression &gt; Chr07.region2.introgression.txt
<span class="hljs-comment"># Transpose the data</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region2.introgression.txt &gt; Chr07.region2.introgression.trans.txt

grep -v ID Chr07.region2.introgression.trans.txt | grep -v Taxonomy | cut -f2- -d <span class="hljs-string">' '</span> &gt; Chr07.region2.introgression.trans2

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 0 26308); <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; <span class="hljs-keyword">done</span> | \
    paste - &lt;(grep -v ID Chr07.region2.introgression.trans.txt | grep -v Taxonomy) | awk <span class="hljs-string">'{print 6,$1,1,$2,0}'</span> &gt; Chr07.region2.introgression.trans3

<span class="hljs-comment"># Cluster the result in sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); <span class="hljs-keyword">do</span>
    perl cluster_introgression.pl \
        &lt;(cut -f<span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr07.region2.introgression.trans2 | \
        paste - Chr07.region2.introgression.trans3 | awk <span class="hljs-string">'{print $2,$3,$4,$5,$6,$1}'</span>) \
        -wind_size 1000000 -wind_step 100000 \
        &gt; Chr07.region2.w1000000.s100000.Accession<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the result</span>
paste Chr07.region2.w1000000.s100000.Accession* &gt; Chr07.region2.w1000000.s100000.Accessions.region2
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr07.region2.w1000000.s100000.Accessions.region2 | grep Avg | grep -v AdjAvg &gt; Chr07.region2.w1000000.s100000.Accessions.region2.trans

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 2 448); <span class="hljs-keyword">do</span>
    rm Chr07.region2.w1000000.s100000.Accession<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Original values</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt; Chr07.region2.w1000000.s100000.Accessions.region2.trans | \
    grep -v Avg &gt;Chr07.region2.w1000000.s100000.Accessions.region2.clean-1

awk <span class="hljs-string">'{print $1,$2,$3,($2+$3)/2}'</span> Chr07.region2.w1000000.s100000.Accession001 | \
    grep -v chrID | paste - Chr07.region2.w1000000.s100000.Accessions.region2.clean-1 | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr07.region2.w1000000.s100000.Accessions.region2.clean.txt

python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt;Chr07.region2.w1000000.s100000.Accessions.region2.clean.txt | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr07.region2.w1000000.s100000.Accessions.region2.clean.trans.txt

<span class="hljs-comment"># Prepare the plot of fd</span>
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAC.FRUCHN.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr07" &amp;&amp; $2&gt;=256150001 &amp;&amp; $3&lt;=261300000 {print $0}'</span> | grep -v <span class="hljs-string">"nan"</span> | \
    cut -f1-4,10 &gt; PUB.BAC.FRUCHN.ANN.Chr07.region2.out.txt

</div></code></pre>
<h3 id="introgression-chr12250550001-255900000">Introgression Chr12:250550001-255900000</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/ABBA-BABA
<span class="hljs-comment"># Extract and filter the SNPs</span>
plink --vcf ../vcf.clean/Chr12.clean.vcf.gz  \
    --keep &lt;(awk <span class="hljs-string">'{print $1,$1}'</span> Pop.PUB.BAB2ANN.txt) \
    --allow-extra-chr --chr Chr12 --geno 0.1 --maf 0.05 \
    --from-bp 250550001 --to-bp 255900000 \
    --recode -output-missing-genotype 9 --double-id \
    --out Chr12.region1
awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> Chr12.region1.map | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Chr12.region1.map2
mv Chr12.region1.map2 Chr12.region1.map

<span class="hljs-comment"># Calculate freq for BAC</span>
plink --file Chr12.region1 \
    --keep &lt;(cat ../Pop.list/Ca.bac*.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr12.region1.BAC

<span class="hljs-comment"># Calculate frequency for PUB</span>
plink --file Chr12.region1 \
    --keep &lt;(cat ../Pop.list/Ca.pubescens.list | awk <span class="hljs-string">'{print $1,$1}'</span>) --double-id \
    --freq --out Chr12.region1.PUB

<span class="hljs-comment"># Find the SNPs where PUB.dominant not equal to BAC.dominant allele</span>
paste Chr12.region1.PUB.frq Chr12.region1.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr12.region1.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f2 &gt; Chr12.region1.remain.snps

<span class="hljs-comment"># Transpose to tped</span>
plink --file Chr12.region1 --extract Chr12.region1.remain.snps \
    --recode transpose --out Chr12.region1

<span class="hljs-comment"># Combine every two lines</span>
cut -f5- -d <span class="hljs-string">' '</span> Chr12.region1.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Chr12.region1.geno-1

<span class="hljs-comment"># Chr12.region1.BAC.dominant.allele</span>
grep -v CHR Chr12.region1.BAC.frq | awk <span class="hljs-string">'{print $4$4}'</span> &gt; Chr12.region1.BAC.dominant.allele.out
paste Chr12.region1.PUB.frq Chr12.region1.BAC.frq | awk <span class="hljs-string">'{print $4,1-$5,$10,1-$11}'</span> | \
    grep -v A2 | paste Chr12.region1.map - | sed <span class="hljs-string">'s/ /\t/g'</span> | \
    awk <span class="hljs-string">'{if ($5==$7)print $0,"TRUE "; else print $0,"FALSE "}'</span> | grep FALSE | cut -f7 \
    &gt; Chr12.region1.BAC.dominant.allele.out1

<span class="hljs-comment"># Check the SNP IDs</span>
paste Chr12.region1.BAC.dominant.allele.out Chr12.region1.geno-1 | \
    awk <span class="hljs-string">'{for(i=2;i&lt;=449;i=i+1) if ($i==$1){printf "1 "} else {printf "0 "}; printf "\n"}'</span> \
    &gt; Chr12.region1.alleles

<span class="hljs-comment"># Change heterozygous to 0.5</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); 
    <span class="hljs-keyword">do</span> cut -f <span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr12.region1.geno-1 | sed <span class="hljs-string">'s/./&amp; /g'</span> | \
    paste Chr12.region1.BAC.dominant.allele.out1 - | \
    awk <span class="hljs-string">'{
        if ($2==$1 &amp;&amp; $3==$1) 
        print 1 ;
        else if ($2==$1 &amp;&amp; $3!=$1)
        print 0.5
        else 
        print 0
    }'</span> &gt; Accession<span class="hljs-variable">$i</span>.geno &amp;
<span class="hljs-keyword">done</span>

paste Accession*.geno | sed <span class="hljs-string">'s/\t/ /g'</span> &gt; Chr12.region1.alleles
rm Accession*.geno

<span class="hljs-comment"># Transpose the genotypes</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr12.region1.alleles &gt; Chr12.region1.alleles.trans

<span class="hljs-comment"># Sort the genos</span>
paste Chr12.sample.id Chr12.region1.alleles.trans | sed <span class="hljs-string">'s/\t/ /g'</span> | sort -k3 | cut -f1,2,4- -d <span class="hljs-string">' '</span> &gt; Chr12.region1.introgression 

<span class="hljs-comment"># Transpose the snp infos</span>
cut -f4 -d <span class="hljs-string">' '</span> Chr12.region1.tped | sed <span class="hljs-string">'1i\ID\nTaxonomy'</span> &gt; Chr12.region1.pos
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr12.region1.pos &gt; Chr12.region1.pos.trans

<span class="hljs-comment"># Combine the matrix</span>
cat Chr12.region1.pos.trans Chr12.region1.introgression &gt; Chr12.region1.introgression.txt
<span class="hljs-comment"># Transpose the data</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr12.region1.introgression.txt &gt; Chr12.region1.introgression.trans.txt

grep -v ID Chr12.region1.introgression.trans.txt | grep -v Taxonomy | cut -f2- -d <span class="hljs-string">' '</span> &gt; Chr12.region1.introgression.trans2

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq 0 26308); <span class="hljs-keyword">do</span> <span class="hljs-built_in">echo</span> <span class="hljs-variable">$i</span>; <span class="hljs-keyword">done</span> | \
    paste - &lt;(grep -v ID Chr12.region1.introgression.trans.txt | grep -v Taxonomy) | awk <span class="hljs-string">'{print 6,$1,1,$2,0}'</span> &gt; Chr12.region1.introgression.trans3

<span class="hljs-comment"># Cluster the result in sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 448); <span class="hljs-keyword">do</span>
    perl cluster_introgression.pl \
        &lt;(cut -f<span class="hljs-variable">$i</span> -d <span class="hljs-string">' '</span> Chr12.region1.introgression.trans2 | \
        paste - Chr12.region1.introgression.trans3 | awk <span class="hljs-string">'{print $2,$3,$4,$5,$6,$1}'</span>) \
        -wind_size 1000000 -wind_step 100000 \
        &gt; Chr12.region1.w1000000.s100000.Accession<span class="hljs-variable">$i</span> &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Combine the result</span>
paste Chr12.region1.w1000000.s100000.Accession* &gt; Chr12.region1.w1000000.s100000.Accessions.region1
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Chr12.region1.w1000000.s100000.Accessions.region1 | grep Avg | grep -v AdjAvg &gt; Chr12.region1.w1000000.s100000.Accessions.region1.trans

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 2 448); <span class="hljs-keyword">do</span>
    rm Chr12.region1.w1000000.s100000.Accession<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Original values</span>
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt; Chr12.region1.w1000000.s100000.Accessions.region1.trans | \
    grep -v Avg &gt;Chr12.region1.w1000000.s100000.Accessions.region1.clean-1

awk <span class="hljs-string">'{print $1,$2,$3,($2+$3)/2}'</span> Chr12.region1.w1000000.s100000.Accession001 | \
    grep -v chrID | paste - Chr12.region1.w1000000.s100000.Accessions.region1.clean-1 | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr12.region1.w1000000.s100000.Accessions.region1.clean.txt

python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> \
    &lt;Chr12.region1.w1000000.s100000.Accessions.region1.clean.txt | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; Chr12.region1.w1000000.s100000.Accessions.region1.clean.trans.txt

<span class="hljs-comment"># Prepare the plot of fd</span>
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAC.FRUCHN.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr12" &amp;&amp; $2&gt;=250550001 &amp;&amp; $3&lt;=255900000 {print $0}'</span> | grep -v <span class="hljs-string">"nan"</span> | \
    cut -f1-4,10 &gt; PUB.BAC.FRUCHN.ANN.Chr12.region1.out.txt

</div></code></pre>
<h3 id="pubbacfruchngla">PUB.BAC.FRUCHN.GLA</h3>
<pre class="hljs"><code><div>ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAC.FRUCHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 30 \
    -P1 Ca.annuum -P2 Ca.FRU.CHN -P3 Ca.baccatum -O Ca.pubescens --popsFile Pop.BAB2ANN.txt --writeFailedWindows
<span class="hljs-comment"># PUB.BAB.CHN.ANN</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.CHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.chinense -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
<span class="hljs-comment"># PUB.BAB.FRU.ANN</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.FRU.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.frutescens -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
<span class="hljs-comment"># PUB.BAP.CHN.ANN</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAP.CHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.annuum -P2 Ca.chinense -P3 Ca.bac.pendulum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows 
<span class="hljs-comment"># PUB.BAP.FRU.ANN</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAP.FRU.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.annuum -P2 Ca.frutescens -P3 Ca.bac.pendulum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows 

<span class="hljs-comment"># Select the top 5% highest fd values</span>
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAC.FRUCHN.ANN.out | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;=1 {print $1,$2,$10}'</span> \
    &gt; PUB.BAC.FRUCHN.ANN.boxplot.out

<span class="hljs-comment"># Prepare the fd boxplot</span>
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAC.FRUCHN.ANN.out | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;=1 {print $10,"PUB.BAC.FRUCHN.ANN"}'</span> \
    &gt; PUB.BAC.FRUCHN.ANN.boxplot.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAB.FRU.ANN.out | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;=1 {print $10,"PUB.BAB.FRU.ANN"}'</span> \
    &gt; PUB.BAB.FRU.ANN.boxplot.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAB.CHN.ANN.out | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;=1 {print $10,"PUB.BAB.CHN.ANN"}'</span> \
    &gt; PUB.BAB.CHN.ANN.boxplot.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAP.FRU.ANN.out | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;=1 {print $10,"PUB.BAP.FRU.ANN"}'</span> \
    &gt; PUB.BAP.FRU.ANN.boxplot.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAP.CHN.ANN.out | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;=1 {print $10,"PUB.BAP.CHN.ANN"}'</span> \
    &gt; PUB.BAP.CHN.ANN.boxplot.out
cat *boxplot.out | sed <span class="hljs-string">'1i\fd Group'</span> | awk <span class="hljs-string">'$1&gt;0 {print $0}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; BAC2FRUCHN.fd.boxplot.txt

<span class="hljs-comment"># Prepare the fd line plot on Chr06</span>
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAC.FRUCHN.GLA.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr06" &amp;&amp; $9&gt;0 &amp;&amp; $10&lt;=1 &amp;&amp; $4&gt;240000000 &amp;&amp; $4&lt;248000000 {print $1,$4,$10,"PUB.BAC.FRUCHN.GLA"}'</span> \
    &gt; PUB.BAC.FRUCHN.GLA.Chr06.region.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAB.FRU.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr06" &amp;&amp; $9&gt;0 &amp;&amp; $10&lt;=1 &amp;&amp; $4&gt;240000000 &amp;&amp; $4&lt;248000000 {print $1,$4,$10,"PUB.BAB.FRU.ANN"}'</span> \
    &gt; PUB.BAB.FRU.ANN.Chr06.region.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAB.CHN.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr06" &amp;&amp; $9&gt;0 &amp;&amp; $10&lt;=1 &amp;&amp; $4&gt;240000000 &amp;&amp; $4&lt;248000000 {print $1,$4,$10,"PUB.BAB.CHN.ANN"}'</span> \
    &gt; PUB.BAB.CHN.ANN.Chr06.region.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAP.FRU.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr06" &amp;&amp; $9&gt;0 &amp;&amp; $10&lt;=1 &amp;&amp; $4&gt;240000000 &amp;&amp; $4&lt;248000000 {print $1,$4,$10,"PUB.BAP.FRU.ANN"}'</span> \
    &gt; PUB.BAP.FRU.ANN.Chr06.region.out
sed <span class="hljs-string">'s/,/\t/g'</span> PUB.BAP.CHN.ANN.out | \
    awk <span class="hljs-string">'$1=="Ca.S8.Chr06" &amp;&amp; $9&gt;0 &amp;&amp; $10&lt;=1 &amp;&amp; $4&gt;240000000 &amp;&amp; $4&lt;248000000 {print $1,$4,$10,"PUB.BAP.CHN.ANN"}'</span> \
    &gt; PUB.BAP.CHN.ANN.Chr06.region.out
cat *region.out | sed <span class="hljs-string">'1i\Chr Pos fd Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; fd.BAC2FRUCHN.Chr06.region.txt

<span class="hljs-comment"># Calculate the average value of fd values.</span>
cat PUB.BAC.FRUCHN.GLA.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;1{print $0}'</span> | grep -v nan  | awk <span class="hljs-string">'{s+=$10} END {print s/NR}'</span>
<span class="hljs-comment"># Calculate the total value of fd</span>
cat PUB.BAC.FRUCHN.GLA.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;1{print $0}'</span> | grep -v nan  | awk <span class="hljs-string">'{s+=$6*$10} END {print s}'</span> 
<span class="hljs-comment"># Calculate the total G</span>
cat PUB.BAC.FRUCHN.GLA.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 &amp;&amp; $10&lt;1{print $0}'</span> | grep -v nan  | awk <span class="hljs-string">'{s+=$6} END {print s}'</span> 

<span class="hljs-comment"># For all the other comparisons</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.CHA.BAB.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.bac.baccatum -P3 Ca.chacoense -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows 
<span class="hljs-comment"># C.bac.pendulum as P2</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.CHA.BAP.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.bac.pendulum -P3 Ca.chacoense -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows 
<span class="hljs-comment"># Ca.chinense as P2    </span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.CHA.CHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.chinense -P3 Ca.chacoense -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
<span class="hljs-comment"># Ca.frutescens as P2</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.CHA.FRU.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.frutescens -P3 Ca.chacoense -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
<span class="hljs-comment"># Ca.glabriusculum as P2</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.CHA.GLA.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.glabriusculum -P3 Ca.chacoense -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows

<span class="hljs-comment"># Step 17.5.2 Prepare the file format for fd boxplot using CHA as P3</span>
cat PUB.CHA.BAB.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 {print $0}'</span> | grep -v nan &gt; PUB.CHA.BAB.ANN.clean.txt
cat PUB.CHA.BAP.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 {print $0}'</span> | grep -v nan &gt; PUB.CHA.BAP.ANN.clean.txt
cat PUB.CHA.CHN.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 {print $0}'</span> | grep -v nan &gt; PUB.CHA.CHN.ANN.clean.txt
cat PUB.CHA.FRU.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 {print $0}'</span> | grep -v nan &gt; PUB.CHA.FRU.ANN.clean.txt
cat PUB.CHA.GLA.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'$9&gt;0 {print $0}'</span> | grep -v nan &gt; PUB.CHA.GLA.ANN.clean.txt
cat PUB.CHA.BAB.ANN.clean.txt | awk <span class="hljs-string">'{print $1,$4,$10,"BAB"}'</span> &gt; PUB.CHA.BAB.ANN.clean-2
cat PUB.CHA.BAP.ANN.clean.txt | awk <span class="hljs-string">'{print $1,$4,$10,"BAP"}'</span> &gt; PUB.CHA.BAP.ANN.clean-2
cat PUB.CHA.CHN.ANN.clean.txt | awk <span class="hljs-string">'{print $1,$4,$10,"CHN"}'</span> &gt; PUB.CHA.CHN.ANN.clean-2
cat PUB.CHA.FRU.ANN.clean.txt | awk <span class="hljs-string">'{print $1,$4,$10,"FRU"}'</span> &gt; PUB.CHA.FRU.ANN.clean-2
cat PUB.CHA.GLA.ANN.clean.txt | awk <span class="hljs-string">'{print $1,$4,$10,"GLA"}'</span> &gt; PUB.CHA.GLA.ANN.clean-2
<span class="hljs-comment"># Prepare the bxoplot</span>
cat PUB.CHA.*.ANN.clean-2 | grep -v fd | cut -f3,4 -d <span class="hljs-string">' '</span> | sed <span class="hljs-string">'1i\fd Taxonomy'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; PUB.CHA.all.ANN.fd.txt

<span class="hljs-comment"># Step 17.5.3 Prepare the genome-wide distribution plot</span>
<span class="hljs-comment"># Genome-wide distribution of fd values for Annuum clade</span>
cat PUB.CHA.BAB.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'{print $1,$4,$9,$10,"BAB"}'</span> &gt; PUB.CHA.BAB.ANN.clean-1
cat PUB.CHA.BAP.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'{print $1,$4,$9,$10,"BAP"}'</span> &gt; PUB.CHA.BAP.ANN.clean-1
cat PUB.CHA.CHN.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'{print $1,$4,$9,$10,"CHN"}'</span> &gt; PUB.CHA.CHN.ANN.clean-1
cat PUB.CHA.FRU.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'{print $1,$4,$9,$10,"FRU"}'</span> &gt; PUB.CHA.FRU.ANN.clean-1
cat PUB.CHA.GLA.ANN.out | sed <span class="hljs-string">'s/,/ /g'</span> | awk <span class="hljs-string">'{print $1,$4,$9,$10,"GLA"}'</span> &gt; PUB.CHA.GLA.ANN.clean-1

<span class="hljs-comment"># Only chromosome 11</span>
cat PUB.CHA.*.ANN.clean-1 | grep Chr11 | \
    sed <span class="hljs-string">'s/nan/0/g'</span> | sed <span class="hljs-string">'s///g'</span> | \
    sed <span class="hljs-string">'1i\scaffold mid D fd Taxonomy'</span> &gt; PUB.CHA.all.ANN.fd.Chr11.txt

<span class="hljs-comment"># All chromosomes</span>
cat PUB.CHA.*.ANN.clean-1 | \
    sed <span class="hljs-string">'s/nan/0/g'</span> | sed <span class="hljs-string">'s///g'</span> | \
    sed <span class="hljs-string">'1i\scaffold mid D fd Taxonomy'</span> &gt; PUB.CHA.all.ANN.fd.txt

rm PUB.CHA.*.ANN.clean-1

<span class="hljs-comment"># Step 17.5.4 Calculate mean fd and PGI</span>
<span class="hljs-comment"># Calculate the average value of fd values.</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB.CHA.*.ANN.clean.txt; <span class="hljs-keyword">do</span> cat <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{s+=$10} END {print s/NR}'</span> ; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Calculate the total value of fd</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB.CHA.*.ANN.clean.txt; <span class="hljs-keyword">do</span> cat <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{s+=$6*$10} END {print s}'</span> ; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Calculate the total G</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> PUB.CHA.*.ANN.clean.txt; <span class="hljs-keyword">do</span> cat <span class="hljs-variable">$i</span> | awk <span class="hljs-string">'{s+=$6} END {print s}'</span> ; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Step 17.6 Run ABBA-BABA using C.bac.baccatum as P3</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.BAP.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.bac.pendulum -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.FRU.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.frutescens -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.CHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.chinense -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.GLA.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.glabriusculum -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows

<span class="hljs-comment"># Run ABBA-BABA using C.bac.pendulum as P3</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAP.FRU.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.frutescens -P3 Ca.bac.pendulum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAP.CHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.chinense -P3 Ca.bac.pendulum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAP.GLA.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.glabriusculum -P3 Ca.bac.pendulum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows

<span class="hljs-comment"># Step 17.8 Run ABBA-BABA using Ca.frutescens as P3</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o BAP.FRU.CHN.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.annuum -P2 Ca.chinense -P3 Ca.frutescens -O Ca.bac.pendulum --popsFile Pop8File.txt --writeFailedWindows ;
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o BAP.FRU.GLA.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 60 \
    -P1 Ca.annuum -P2 Ca.glabriusculum -P3 Ca.frutescens -O Ca.bac.pendulum --popsFile Pop8File.txt --writeFailedWindows

<span class="hljs-comment"># Run ABBA-BABA using C. chinense as P3</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o FRU.CHN.GLA.ANN.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 20 \
    -P1 Ca.annuum -P2 Ca.glabriusculum -P3 Ca.chinense -O Ca.frutescens --popsFile Pop8File.txt --writeFailedWindows 

<span class="hljs-comment"># Step 17.9 Run ABBA-BABA using ANN as P2</span>
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.CHA.ANN.GLA.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.glabriusculum -P2 Ca.annuum  -P3 Ca.chacoense -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAB.ANN.GLA.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.glabriusculum -P2 Ca.annuum  -P3 Ca.bac.baccatum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o PUB.BAP.ANN.GLA.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.glabriusculum -P2 Ca.annuum  -P3 Ca.bac.pendulum -O Ca.pubescens --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o BAP.FRU.ANN.GLA.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.glabriusculum -P2 Ca.annuum  -P3 Ca.frutescens -O Ca.bac.pendulum --popsFile Pop8File.txt --writeFailedWindows
ABBABABAwindows.py -g Ca.S8.clean.geno.gz -o BAP.CHN.ANN.GLA.out -w 500000 -s 50000 -m 100 -f phased --minData 0.5 -T 10 \
    -P1 Ca.glabriusculum -P2 Ca.annuum  -P3 Ca.chinense -O Ca.bac.pendulum --popsFile Pop8File.txt --writeFailedWindows

<span class="hljs-comment"># Combine every two columns</span>
head -1  Chr06.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span>

</div></code></pre>
<h2 id="sweed">SweeD</h2>
<h3 id="installation">Installation</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Installation</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/tools
git <span class="hljs-built_in">clone</span> https://github.com/alachins/sweed.git
<span class="hljs-built_in">cd</span> sweed
<span class="hljs-comment"># Install</span>
make -f Makefile.gcc
make -f Makefile.PTHREADS.gcc

<span class="hljs-comment"># Install DMTCP library</span>
DMTCPAWARELIB=/data/zhaojiantao/tools/sweed/dmtcp/lib/dmtcp

</div></code></pre>
<h3 id="bac-with-10k-grid">BAC with 10k grid</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># SweeD does not accept zipped vcf</span>
<span class="hljs-comment"># I will first extract the BAC accessions and filter the SNPs for a faster running</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SweeD/BAC
<span class="hljs-comment"># Convert vcf to transposed ped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
        --keep BAC.list \
        --double-id \
        --chr Chr<span class="hljs-variable">$i</span> \
        --maf 0.05 \
        --geno 0.1 \
        --recode vcf \
        --out Chr<span class="hljs-variable">$i</span> &amp; 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep -v Chr00 ../../References/Zhangshugang/Zhangshugang.fa.fai | \
    awk <span class="hljs-string">'{print $1,int($2/10000+0.5)}'</span> \
    &gt; Chr.grids

<span class="hljs-comment"># Run SweeD</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    SweeD-P -name <span class="hljs-variable">$Chr</span> -input <span class="hljs-variable">$Chr</span>.vcf -grid <span class="hljs-variable">$grid</span> -threads 10 &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "1", $0}'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "2", $0}'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "3", $0}'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "4", $0}'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "5", $0}'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "6", $0}'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "7", $0}'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "8", $0}'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "9", $0}'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "10", $0}'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "11", $0}'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "12", $0}'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | awk <span class="hljs-string">'{print $1,"Chr"$1"_"$2,$2,$3,$4,$5,$6}'</span> | \
    sed <span class="hljs-string">'1i\Chr SNP Position Likelihoood Alpha StartPos EndPos'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.BAC.10k.txt

</div></code></pre>
<h3 id="bac-with-pruned-snps">BAC with pruned SNPs</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SweeD/BAC.prune

<span class="hljs-comment"># Simplify the sample id and add SNP info</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Simplify the sample id</span>
    grep <span class="hljs-string">"#"</span> ../BAC.10k/Chr<span class="hljs-variable">$i</span>.vcf | \
        sed <span class="hljs-string">'s/Grif_9199_Grif_9199/Grif_9199/g'</span> | sed <span class="hljs-string">'s/Grif_9201_Grif_9201/Grif_9201/g'</span> | \
        sed <span class="hljs-string">'s/Grif_9352_Grif_9352/Grif_9352/g'</span> | sed <span class="hljs-string">'s/Grif_9354_Grif_9354/Grif_9354/g'</span> | \
        sed <span class="hljs-string">'s/Grif_9355_Grif_9355/Grif_9355/g'</span> | sed <span class="hljs-string">'s/PI_159235_PI_159235/PI_159235/g'</span> | \
        sed <span class="hljs-string">'s/PI_159242_PI_159242/PI_159242/g'</span> | sed <span class="hljs-string">'s/PI_159249_PI_159249/PI_159249/g'</span> | \
        sed <span class="hljs-string">'s/PI_159260_PI_159260/PI_159260/g'</span> | sed <span class="hljs-string">'s/PI_188481_PI_188481/PI_188481/g'</span> | \
        sed <span class="hljs-string">'s/PI_199506_PI_199506/PI_199506/g'</span> | sed <span class="hljs-string">'s/PI_215699_PI_215699/PI_215699/g'</span> | \
        sed <span class="hljs-string">'s/PI_215741_PI_215741/PI_215741/g'</span> | sed <span class="hljs-string">'s/PI_241679_PI_241679/PI_241679/g'</span> | \
        sed <span class="hljs-string">'s/PI_257110_PI_257110/PI_257110/g'</span> | sed <span class="hljs-string">'s/PI_257141_PI_257141/PI_257141/g'</span> | \
        sed <span class="hljs-string">'s/PI_257157_PI_257157/PI_257157/g'</span> | sed <span class="hljs-string">'s/PI_257161_PI_257161/PI_257161/g'</span> | \
        sed <span class="hljs-string">'s/PI_257173_PI_257173/PI_257173/g'</span> | sed <span class="hljs-string">'s/PI_257174_PI_257174/PI_257174/g'</span> | \
        sed <span class="hljs-string">'s/PI_260488_PI_260488/PI_260488/g'</span> | sed <span class="hljs-string">'s/PI_260535_PI_260535/PI_260535/g'</span> | \
        sed <span class="hljs-string">'s/PI_260538_PI_260538/PI_260538/g'</span> | sed <span class="hljs-string">'s/PI_260545_PI_260545/PI_260545/g'</span> | \
        sed <span class="hljs-string">'s/PI_260549_PI_260549/PI_260549/g'</span> | sed <span class="hljs-string">'s/PI_260550_PI_260550/PI_260550/g'</span> | \
        sed <span class="hljs-string">'s/PI_260551_PI_260551/PI_260551/g'</span> | sed <span class="hljs-string">'s/PI_260564_PI_260564/PI_260564/g'</span> | \
        sed <span class="hljs-string">'s/PI_260567_PI_260567/PI_260567/g'</span> | sed <span class="hljs-string">'s/PI_260575_PI_260575/PI_260575/g'</span> | \
        sed <span class="hljs-string">'s/PI_260582_PI_260582/PI_260582/g'</span> | sed <span class="hljs-string">'s/PI_260590_PI_260590/PI_260590/g'</span> | \
        sed <span class="hljs-string">'s/PI_260595_PI_260595/PI_260595/g'</span> | sed <span class="hljs-string">'s/PI_266041_PI_266041/PI_266041/g'</span> | \
        sed <span class="hljs-string">'s/PI_267729_PI_267729/PI_267729/g'</span> | sed <span class="hljs-string">'s/PI_281308_PI_281308/PI_281308/g'</span> | \
        sed <span class="hljs-string">'s/PI_281309_PI_281309/PI_281309/g'</span> | sed <span class="hljs-string">'s/PI_281310_PI_281310/PI_281310/g'</span> | \
        sed <span class="hljs-string">'s/PI_281398_PI_281398/PI_281398/g'</span> | sed <span class="hljs-string">'s/PI_281407_PI_281407/PI_281407/g'</span> | \
        sed <span class="hljs-string">'s/PI_281436_PI_281436/PI_281436/g'</span> | sed <span class="hljs-string">'s/PI_290982_PI_290982/PI_290982/g'</span> | \
        sed <span class="hljs-string">'s/PI_290983_PI_290983/PI_290983/g'</span> | sed <span class="hljs-string">'s/PI_293349_PI_293349/PI_293349/g'</span> | \
        sed <span class="hljs-string">'s/PI_315020_PI_315020/PI_315020/g'</span> | sed <span class="hljs-string">'s/PI_337524_PI_337524/PI_337524/g'</span> | \
        sed <span class="hljs-string">'s/PI_355813_PI_355813/PI_355813/g'</span> | sed <span class="hljs-string">'s/PI_370010_PI_370010/PI_370010/g'</span> | \
        sed <span class="hljs-string">'s/PI_424732_PI_424732/PI_424732/g'</span> | sed <span class="hljs-string">'s/PI_439359_PI_439359/PI_439359/g'</span> | \
        sed <span class="hljs-string">'s/PI_439360_PI_439360/PI_439360/g'</span> | sed <span class="hljs-string">'s/PI_439363_PI_439363/PI_439363/g'</span> | \
        sed <span class="hljs-string">'s/PI_439366_PI_439366/PI_439366/g'</span> | sed <span class="hljs-string">'s/PI_439367_PI_439367/PI_439367/g'</span> | \
        sed <span class="hljs-string">'s/PI_439369_PI_439369/PI_439369/g'</span> | sed <span class="hljs-string">'s/PI_439371_PI_439371/PI_439371/g'</span> | \
        sed <span class="hljs-string">'s/PI_439380_PI_439380/PI_439380/g'</span> | sed <span class="hljs-string">'s/PI_439384_PI_439384/PI_439384/g'</span> | \
        sed <span class="hljs-string">'s/PI_439386_PI_439386/PI_439386/g'</span> | sed <span class="hljs-string">'s/PI_439390_PI_439390/PI_439390/g'</span> | \
        sed <span class="hljs-string">'s/PI_439395_PI_439395/PI_439395/g'</span> | sed <span class="hljs-string">'s/PI_439398_PI_439398/PI_439398/g'</span> | \
        sed <span class="hljs-string">'s/PI_439401_PI_439401/PI_439401/g'</span> | sed <span class="hljs-string">'s/PI_439402_PI_439402/PI_439402/g'</span> | \
        sed <span class="hljs-string">'s/PI_439403_PI_439403/PI_439403/g'</span> | sed <span class="hljs-string">'s/PI_439405_PI_439405/PI_439405/g'</span> | \
        sed <span class="hljs-string">'s/PI_439407_PI_439407/PI_439407/g'</span> | sed <span class="hljs-string">'s/PI_439410_PI_439410/PI_439410/g'</span> | \
        sed <span class="hljs-string">'s/PI_439411_PI_439411/PI_439411/g'</span> | sed <span class="hljs-string">'s/PI_439472_PI_439472/PI_439472/g'</span> | \
        sed <span class="hljs-string">'s/PI_439528_PI_439528/PI_439528/g'</span> | sed <span class="hljs-string">'s/PI_441516_PI_441516/PI_441516/g'</span> | \
        sed <span class="hljs-string">'s/PI_441523_PI_441523/PI_441523/g'</span> | sed <span class="hljs-string">'s/PI_441531_PI_441531/PI_441531/g'</span> | \
        sed <span class="hljs-string">'s/PI_441532_PI_441532/PI_441532/g'</span> | sed <span class="hljs-string">'s/PI_441536_PI_441536/PI_441536/g'</span> | \
        sed <span class="hljs-string">'s/PI_441537_PI_441537/PI_441537/g'</span> | sed <span class="hljs-string">'s/PI_441541_PI_441541/PI_441541/g'</span> | \
        sed <span class="hljs-string">'s/PI_441547_PI_441547/PI_441547/g'</span> | sed <span class="hljs-string">'s/PI_441548_PI_441548/PI_441548/g'</span> | \
        sed <span class="hljs-string">'s/PI_441558_PI_441558/PI_441558/g'</span> | sed <span class="hljs-string">'s/PI_441578_PI_441578/PI_441578/g'</span> | \
        sed <span class="hljs-string">'s/PI_441590_PI_441590/PI_441590/g'</span> | sed <span class="hljs-string">'s/PI_441592_PI_441592/PI_441592/g'</span> | \
        sed <span class="hljs-string">'s/PI_441594_PI_441594/PI_441594/g'</span> | sed <span class="hljs-string">'s/PI_441597_PI_441597/PI_441597/g'</span> | \
        sed <span class="hljs-string">'s/PI_441656_PI_441656/PI_441656/g'</span> | sed <span class="hljs-string">'s/PI_441674_PI_441674/PI_441674/g'</span> | \
        sed <span class="hljs-string">'s/PI_441685_PI_441685/PI_441685/g'</span> | sed <span class="hljs-string">'s/PI_441699_PI_441699/PI_441699/g'</span> | \
        sed <span class="hljs-string">'s/PI_446900_PI_446900/PI_446900/g'</span> | sed <span class="hljs-string">'s/PI_446909_PI_446909/PI_446909/g'</span> | \
        sed <span class="hljs-string">'s/PI_497972_PI_497972/PI_497972/g'</span> | sed <span class="hljs-string">'s/PI_555611_PI_555611/PI_555611/g'</span> | \
        sed <span class="hljs-string">'s/PI_585247_PI_585247/PI_585247/g'</span> | sed <span class="hljs-string">'s/PI_585250_PI_585250/PI_585250/g'</span> | \
        sed <span class="hljs-string">'s/PI_593606_PI_593606/PI_593606/g'</span> | sed <span class="hljs-string">'s/PI_594137_PI_594137/PI_594137/g'</span> | \
        sed <span class="hljs-string">'s/PI_594138_PI_594138/PI_594138/g'</span> | sed <span class="hljs-string">'s/PI_595905_PI_595905/PI_595905/g'</span> | \
        sed <span class="hljs-string">'s/PI_596059_PI_596059/PI_596059/g'</span> | sed <span class="hljs-string">'s/PI_631146_PI_631146/PI_631146/g'</span> | \
        sed <span class="hljs-string">'s/PI_632923_PI_632923/PI_632923/g'</span> | sed <span class="hljs-string">'s/PI_632926_PI_632926/PI_632926/g'</span> | \
        sed <span class="hljs-string">'s/PI_632928_PI_632928/PI_632928/g'</span> | sed <span class="hljs-string">'s/PI_633751_PI_633751/PI_633751/g'</span> | \
        sed <span class="hljs-string">'s/PI_633755_PI_633755/PI_633755/g'</span> | sed <span class="hljs-string">'s/PI_633758_PI_633758/PI_633758/g'</span> | \
        sed <span class="hljs-string">'s/PI_639129_PI_639129/PI_639129/g'</span> | sed <span class="hljs-string">'s/PI_639648_PI_639648/PI_639648/g'</span> | \
        sed <span class="hljs-string">'s/PI_640882_PI_640882/PI_640882/g'</span> | sed <span class="hljs-string">'s/PI_640884_PI_640884/PI_640884/g'</span> | \
        sed <span class="hljs-string">'s/PI_640889_PI_640889/PI_640889/g'</span> | sed <span class="hljs-string">'s/PI_643124_PI_643124/PI_643124/g'</span> | \
        sed <span class="hljs-string">'s/PI_659105_PI_659105/PI_659105/g'</span> | sed <span class="hljs-string">'s/PI_660972_PI_660972/PI_660972/g'</span> | \
        sed <span class="hljs-string">'s/PI_688675_PI_688675/PI_688675/g'</span> \
        &gt; Chr<span class="hljs-variable">$i</span>.header
    <span class="hljs-comment"># Add SNP info</span>
    grep -v <span class="hljs-string">"#"</span> ../BAC.10k/Chr<span class="hljs-variable">$i</span>.vcf | \
        awk <span class="hljs-string">'{print $1,$2,$1"_"$2}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> | \
        paste - &lt;(grep -v <span class="hljs-string">"#"</span> ../BAC.10k/Chr<span class="hljs-variable">$i</span>.vcf) | \
        cut -f1-3,7- | \
        cat Chr<span class="hljs-variable">$i</span>.header - \
        &gt; Chr<span class="hljs-variable">$i</span>.vcf
    <span class="hljs-comment"># Remove temporary file</span>
    rm Chr<span class="hljs-variable">$i</span>.header
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># SNP pruning</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf Chr<span class="hljs-variable">$i</span>.vcf \
    --indep-pairwise 50 10 0.2 \
    --allow-extra-chr --double-id \
    --out Chr<span class="hljs-variable">$i</span>
    <span class="hljs-comment"># Extract the prunned SNPs</span>
    plink --vcf Chr<span class="hljs-variable">$i</span>.vcf \
        --extract Chr<span class="hljs-variable">$i</span>.prune.in \
        --recode vcf --double-id --allow-extra-chr \
        --out Chr<span class="hljs-variable">$i</span>.prune
    <span class="hljs-comment"># Remove temporary files</span>
    rm *<span class="hljs-built_in">log</span> *nosex *out *<span class="hljs-keyword">in</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep -v Chr00 ../../References/Zhangshugang/Zhangshugang.fa.fai | \
    awk <span class="hljs-string">'{print $1,int($2/10000+0.5)}'</span> \
    &gt; Chr.grids

<span class="hljs-comment"># Run SweeD</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    SweeD-P -name <span class="hljs-variable">$Chr</span> -input <span class="hljs-variable">$Chr</span>.prune.vcf -grid <span class="hljs-variable">$grid</span> -threads 10 &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "1", $0}'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "2", $0}'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "3", $0}'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "4", $0}'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "5", $0}'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "6", $0}'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "7", $0}'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "8", $0}'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "9", $0}'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "10", $0}'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "11", $0}'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "12", $0}'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | awk <span class="hljs-string">'{print $1,"Chr"$1"_"$2,$2,$3,$4,$5,$6}'</span> | \
    sed <span class="hljs-string">'1i\Chr SNP Position Likelihoood Alpha StartPos EndPos'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.BAC.prune.10k.txt

</div></code></pre>
<h3 id="gla-with-10k-grid">GLA with 10k grid</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># For GLA</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SweeD/GLA.10k

<span class="hljs-comment"># Prepare the sample list</span>
cat /data/zhaojiantao/pepper/Pop.list.pure/Ca.glabriusculum_I.list \
    /data/zhaojiantao/pepper/Pop.list.pure/Ca.annuum.list | \
    awk <span class="hljs-string">'{print $1,$1}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; GLA.list

<span class="hljs-comment"># Convert vcf to transposed ped</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --vcf ../../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
        --keep GLA.list \
        --double-id \
        --chr Chr<span class="hljs-variable">$i</span> \
        --maf 0.05 \
        --geno 0.1 \
        --recode vcf \
        --out Chr<span class="hljs-variable">$i</span> &amp; 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the bed file containing the chr and grid numbers</span>
grep -v Chr00 ../../References/Zhangshugang/Zhangshugang.fa.fai | \
    awk <span class="hljs-string">'{print $1,int($2/10000+0.5)}'</span> \
    &gt; Chr.grids

<span class="hljs-comment"># Run SweeD</span>
IFS=$<span class="hljs-string">'\n'</span>
<span class="hljs-keyword">for</span> LINE <span class="hljs-keyword">in</span> $(cat Chr.grids); <span class="hljs-keyword">do</span>
    Chr=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $1}'</span>)
    grid=$(<span class="hljs-built_in">echo</span> <span class="hljs-variable">${LINE}</span> | awk <span class="hljs-string">'{print $2}'</span>)
    SweeD-P -name <span class="hljs-variable">$Chr</span> -input <span class="hljs-variable">$Chr</span>.vcf -grid <span class="hljs-variable">$grid</span> -threads 10 &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Add the chromosome ids to the output</span>
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr01 | awk <span class="hljs-string">'{print "1", $0}'</span> &gt; SweeD_Report.Chr01-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr02 | awk <span class="hljs-string">'{print "2", $0}'</span> &gt; SweeD_Report.Chr02-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr03 | awk <span class="hljs-string">'{print "3", $0}'</span> &gt; SweeD_Report.Chr03-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr04 | awk <span class="hljs-string">'{print "4", $0}'</span> &gt; SweeD_Report.Chr04-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr05 | awk <span class="hljs-string">'{print "5", $0}'</span> &gt; SweeD_Report.Chr05-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr06 | awk <span class="hljs-string">'{print "6", $0}'</span> &gt; SweeD_Report.Chr06-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr07 | awk <span class="hljs-string">'{print "7", $0}'</span> &gt; SweeD_Report.Chr07-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr08 | awk <span class="hljs-string">'{print "8", $0}'</span> &gt; SweeD_Report.Chr08-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr09 | awk <span class="hljs-string">'{print "9", $0}'</span> &gt; SweeD_Report.Chr09-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr10 | awk <span class="hljs-string">'{print "10", $0}'</span> &gt; SweeD_Report.Chr10-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr11 | awk <span class="hljs-string">'{print "11", $0}'</span> &gt; SweeD_Report.Chr11-1
sed -n <span class="hljs-string">'4,$p'</span> SweeD_Report.Chr12 | awk <span class="hljs-string">'{print "12", $0}'</span> &gt; SweeD_Report.Chr12-1

<span class="hljs-comment"># Combine the results</span>
cat SweeD_Report.Chr*-1 | awk <span class="hljs-string">'{print $1,"Chr"$1"_"$2,$2,$3,$4,$5,$6}'</span> | \
    sed <span class="hljs-string">'1i\Chr SNP Position Likelihoood Alpha StartPos EndPos'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; SweeD_Report.GLA.10k.txt

</div></code></pre>
<h2 id="xp-clr-for-c-annuum">XP-CLR for C. annuum</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/XPCLR

<span class="hljs-comment"># Blast the markers to S8 to update the physical position</span>
blastall -p blastn -i cM.BP.fa -d ../pepper_genome/fa -m 8 -e 1e-40 -o cM.BP.S8.blast

<span class="hljs-comment"># Prepare the SNP format for inferring the pseudo genetic position</span>
zcat ../vcf.clean/Chr01.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr01_"$2,$2}'</span> &gt; Chr01.BP.txt
zcat ../vcf.clean/Chr02.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr02_"$2,$2}'</span> &gt; Chr02.BP.txt
zcat ../vcf.clean/Chr03.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr03_"$2,$2}'</span> &gt; Chr03.BP.txt
zcat ../vcf.clean/Chr04.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr04_"$2,$2}'</span> &gt; Chr04.BP.txt
zcat ../vcf.clean/Chr05.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr05_"$2,$2}'</span> &gt; Chr05.BP.txt
zcat ../vcf.clean/Chr06.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr06_"$2,$2}'</span> &gt; Chr06.BP.txt
zcat ../vcf.clean/Chr07.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr07_"$2,$2}'</span> &gt; Chr07.BP.txt
zcat ../vcf.clean/Chr08.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr08_"$2,$2}'</span> &gt; Chr08.BP.txt
zcat ../vcf.clean/Chr09.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr09_"$2,$2}'</span> &gt; Chr09.BP.txt
zcat ../vcf.clean/Chr10.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr10_"$2,$2}'</span> &gt; Chr10.BP.txt
zcat ../vcf.clean/Chr11.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr11_"$2,$2}'</span> &gt; Chr11.BP.txt
zcat ../vcf.clean/Chr12.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr12_"$2,$2}'</span> &gt; Chr12.BP.txt

<span class="hljs-comment"># Prepare the real genetic maps</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    cat Linkage.clean.txt | grep Chr<span class="hljs-variable">$i</span> |  awk <span class="hljs-string">'{print $2,$5,$4}'</span> &gt; Chr<span class="hljs-variable">$i</span>.map.txt ; 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Infer the genetic position </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    perl assign_genetic.pl Chr<span class="hljs-variable">$i</span>.map.txt Chr<span class="hljs-variable">$i</span>.BP.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Compare the relationship between the inferred and real genetic maps</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    cat Chr<span class="hljs-variable">$i</span>.map.txt | awk <span class="hljs-string">'{print $0,"Real"}'</span> &gt; Chr<span class="hljs-variable">$i</span>.map-1.txt
    cat Chr<span class="hljs-variable">$i</span>.snp.genetic.txt | awk <span class="hljs-string">'{print $1,$2,$3,"Inferred"}'</span> | \
        cat - Chr<span class="hljs-variable">$i</span>.map-1.txt | sed <span class="hljs-string">'1i\SNP Pos cM Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Chr<span class="hljs-variable">$i</span>.compare.txt
    rm Chr<span class="hljs-variable">$i</span>.map-1.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># SNP position files in LinkageMap folder</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    zcat ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print $1,$2,$4,$5}'</span> &gt; Chr<span class="hljs-variable">$i</span>.BP-2.txt
    paste Chr<span class="hljs-variable">$i</span>.snp.genetic.txt Chr<span class="hljs-variable">$i</span>.BP-2.txt | \
        awk <span class="hljs-string">'{print $1,$4,$3,$2,$6,$7}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; ../XPCLR/Chr<span class="hljs-variable">$i</span>.snp
    rm Chr<span class="hljs-variable">$i</span>.BP-2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the genotypes for xpclr in XPCLR folder</span>
<span class="hljs-comment"># Make ped file using this chromosome map</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum ; <span class="hljs-keyword">do</span> 
    cat ../Pop.list/<span class="hljs-variable">$j</span>.list | awk <span class="hljs-string">'{print $1 "\t" $1}'</span> &gt; <span class="hljs-variable">$j</span>.list.txt
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum ; <span class="hljs-keyword">do</span> 
        <span class="hljs-comment"># Convert vcf to transposed ped</span>
        plink --vcf ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
            --keep <span class="hljs-variable">$j</span>.list.txt  \
            --allow-extra-chr --chr Chr<span class="hljs-variable">$i</span> \
            --recode 01 transpose \
            -output-missing-genotype 9 --double-id \
            --<span class="hljs-built_in">set</span>-missing-var-ids @:<span class="hljs-comment"># --keep-allele-order \</span>
            --out <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>
        <span class="hljs-comment"># Prepare the transposed ped file to xpclr geno file format</span>
        cut -d <span class="hljs-string">" "</span> -f 5- <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>.tped  | awk <span class="hljs-string">'{print $0" "}'</span> &gt; <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>.geno
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex *tfam *tped 

<span class="hljs-comment"># XPCLR of Ca.glabriusculum vs Ca.annuum</span>
<span class="hljs-comment"># Chr01</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 9086854); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr01.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr01.0000000.geno ; 
    cat Ca.glabriusculum.Chr01.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr01.0000000.geno ; 
    cat Chr01.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr01.0000000.snp
    cat Ca.annuum.Chr01.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr01.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr01.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr01.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr01.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr01.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 9086854); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr01.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr01.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr01.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr01.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr01 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr01.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr02</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 4776664); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr02.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr02.0000000.geno ; 
    cat Ca.glabriusculum.Chr02.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr02.0000000.geno ; 
    cat Chr02.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr02.0000000.snp
    cat Ca.annuum.Chr02.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr02.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr02.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr02.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr02.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr02.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 4776664); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr02.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr02.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr02.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr02.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr02 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr02.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr03</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 8369414); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr03.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr03.0000000.geno ; 
    cat Ca.glabriusculum.Chr03.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr03.0000000.geno ; 
    cat Chr03.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr03.0000000.snp
    cat Ca.annuum.Chr03.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr03.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr03.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr03.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr03.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr03.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 8369414); <span class="hljs-keyword">do</span>
XPCLR -xpclr GLA.ANN/Ca.annuum.Chr03.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Ca.glabriusculum.Chr03.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Chr03.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr03.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr03 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr03.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr04</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7616000); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr04.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr04.0000000.geno ; 
    cat Ca.glabriusculum.Chr04.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr04.0000000.geno ; 
    cat Chr04.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr04.0000000.snp
    cat Ca.annuum.Chr04.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr04.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr04.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr04.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr04.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr04.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7616000); <span class="hljs-keyword">do</span>
XPCLR -xpclr GLA.ANN/Ca.annuum.Chr04.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Ca.glabriusculum.Chr04.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Chr04.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr04.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr04 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr04.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span> 

<span class="hljs-comment"># Chr05</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7570910); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr05.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr05.0000000.geno ; 
    cat Ca.glabriusculum.Chr05.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr05.0000000.geno ; 
    cat Chr05.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr05.0000000.snp
    cat Ca.annuum.Chr05.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr05.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr05.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr05.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr05.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr05.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7570910); <span class="hljs-keyword">do</span>
XPCLR -xpclr GLA.ANN/Ca.annuum.Chr05.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Ca.glabriusculum.Chr05.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Chr05.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr05.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr05 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr05.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr06</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7522020); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr06.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr06.0000000.geno ; 
    cat Ca.glabriusculum.Chr06.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr06.0000000.geno ; 
    cat Chr06.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr06.0000000.snp
    cat Ca.annuum.Chr06.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr06.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr06.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr06.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr06.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr06.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7522020); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr06.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr06.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr06.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr06.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr06 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr06.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr07</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7824408); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr07.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr07.0000000.geno ; 
    cat Ca.glabriusculum.Chr07.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr07.0000000.geno ; 
    cat Chr07.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr07.0000000.snp
    cat Ca.annuum.Chr07.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr07.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr07.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr07.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr07.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr07.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7824408); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr07.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr07.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr07.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr07.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr07 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr07.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr08</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 5099543); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr08.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr08.0000000.geno ; 
    cat Ca.glabriusculum.Chr08.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr08.0000000.geno ; 
    cat Chr08.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr08.0000000.snp
    cat Ca.annuum.Chr08.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr08.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr08.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr08.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr08.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr08.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 5099543); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr08.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr08.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr08.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr08.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 1000 Chr08 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr08.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr09</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 8390222); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr09.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr09.0000000.geno ; 
    cat Ca.glabriusculum.Chr09.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr09.0000000.geno ; 
    cat Chr09.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr09.0000000.snp
    cat Ca.annuum.Chr09.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr09.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr09.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr09.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr09.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr09.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 8390222); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr09.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr09.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr09.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr09.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr09 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr09.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr10</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 6143906); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr10.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr10.0000000.geno ; 
    cat Ca.glabriusculum.Chr10.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr10.0000000.geno ; 
    cat Chr10.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr10.0000000.snp
    cat Ca.annuum.Chr10.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr10.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr10.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr10.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr10.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr10.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 6143906); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr10.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr10.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr10.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr10.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr10 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr10.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr11</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 8763242); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr11.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr11.0000000.geno ; 
    cat Ca.glabriusculum.Chr11.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr11.0000000.geno ; 
    cat Chr11.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr11.0000000.snp
    cat Ca.annuum.Chr11.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr11.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr11.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr11.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr11.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr11.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 8763242); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr11.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr11.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr11.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr11.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr11 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr11.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr12</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7930008); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr12.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr12.0000000.geno ; 
    cat Ca.glabriusculum.Chr12.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr12.0000000.geno ; 
    cat Chr12.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr12.0000000.snp
    cat Ca.annuum.Chr12.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr12.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr12.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr12.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr12.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr12.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7930008); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr12.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr12.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr12.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr12.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr12 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr12.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *geno *snp *<span class="hljs-built_in">log</span>

<span class="hljs-comment"># Prepare the output format for distribution plots</span>
<span class="hljs-comment"># Average XPCLR score using 100 kb window at 10 kb step size sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    cat GLA.ANN.Chr<span class="hljs-variable">$i</span>.*.out.xpclr.txt &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.combined.out.xpclr.txt
    perl ../cluster_xpclrscore.pl GLA.ANN.Chr<span class="hljs-variable">$i</span>.combined.out.xpclr.txt \
        -wind_size 100000 -wind_step 10000 \
        &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.averaged.out.xpclr.txt
<span class="hljs-keyword">done</span>

awk <span class="hljs-string">'{print "Chr01_"$2,1,$2,$3,$7}'</span> GLA.ANN.Chr01.averaged.out.xpclr.txt &gt; GLA.ANN.Chr01.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr02_"$2,2,$2,$3,$7}'</span> GLA.ANN.Chr02.averaged.out.xpclr.txt &gt; GLA.ANN.Chr02.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr03_"$2,3,$2,$3,$7}'</span> GLA.ANN.Chr03.averaged.out.xpclr.txt &gt; GLA.ANN.Chr03.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr04_"$2,4,$2,$3,$7}'</span> GLA.ANN.Chr04.averaged.out.xpclr.txt &gt; GLA.ANN.Chr04.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr05_"$2,5,$2,$3,$7}'</span> GLA.ANN.Chr05.averaged.out.xpclr.txt &gt; GLA.ANN.Chr05.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr06_"$2,6,$2,$3,$7}'</span> GLA.ANN.Chr06.averaged.out.xpclr.txt &gt; GLA.ANN.Chr06.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr07_"$2,7,$2,$3,$7}'</span> GLA.ANN.Chr07.averaged.out.xpclr.txt &gt; GLA.ANN.Chr07.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr08_"$2,8,$2,$3,$7}'</span> GLA.ANN.Chr08.averaged.out.xpclr.txt &gt; GLA.ANN.Chr08.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr09_"$2,9,$2,$3,$7}'</span> GLA.ANN.Chr09.averaged.out.xpclr.txt &gt; GLA.ANN.Chr09.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr10_"$2,10,$2,$3,$7}'</span> GLA.ANN.Chr10.averaged.out.xpclr.txt &gt; GLA.ANN.Chr10.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr11_"$2,11,$2,$3,$7}'</span> GLA.ANN.Chr11.averaged.out.xpclr.txt &gt; GLA.ANN.Chr11.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr12_"$2,12,$2,$3,$7}'</span> GLA.ANN.Chr12.averaged.out.xpclr.txt &gt; GLA.ANN.Chr12.out.xpclr.txt-1

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
cat GLA.ANN.Chr*.out.xpclr.txt-1 | grep -v Wind | sed <span class="hljs-string">'1i\SNP Chr Pos PosEnd XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; GLA.ANN.out.xpclr.txt

<span class="hljs-comment"># Remove unnecessary files</span>
rm GLA.ANN.Chr*.out.xpclr.txt-1 GLA.ANN.Chr*.combined.out.xpclr.txt GLA.ANN.Chr*.averaged.out.xpclr.txt

<span class="hljs-comment"># Filter the selective sweep windows</span>
bedtools merge -d 100000 \
    -i &lt;(cat BAB.BAP.out.xpclr.top5.txt  | sed <span class="hljs-string">'s/_/\t/g'</span> | awk <span class="hljs-string">'{print $1,$4,$5}'</span> | grep -v XPCLR | sed <span class="hljs-string">'s/ /\t/g'</span> ) \
    &gt; BAB.BAP.out.xpclr.top5.merged.txt

bedtools merge -d 10000 \
    -i &lt;(cat BAB.BAP.out.xpclr.top5.txt  | sed <span class="hljs-string">'s/_/\t/g'</span> | awk <span class="hljs-string">'{print $1,$4,$5}'</span> | grep -v XPCLR | sed <span class="hljs-string">'s/ /\t/g'</span> ) \
    &gt; BAB.BAP.out.xpclr.top5.merged.10k.txt

bedtools merge -d 10000 \
    -i &lt;(cat GLA.ANN.out.xpclr.top5.txt  | sed <span class="hljs-string">'s/_/\t/g'</span> | awk <span class="hljs-string">'{print $1,$4,$5}'</span> | grep -v XPCLR | sed <span class="hljs-string">'s/ /\t/g'</span> ) \
    &gt; GLA.ANN.out.xpclr.top5.merged.10k.txt

<span class="hljs-comment"># Alternatively, use python script</span>
python region.py BAB.BAP.out.xpclr.top5.txt &gt; BAB.BAP.out.xpclr.top5.merged.txt

<span class="hljs-comment"># Find the overlapped region betwen BAB.BAP and GLA.ANN</span>
bedtools intersect \
    -a &lt;(cut -f1-3 BAB.BAP.out.xpclr.top5.merged.txt | grep -v Start) \
    -b &lt;(cut -f1-3 GLA.ANN.out.xpclr.top5.merged.txt | grep -v Start) \
    &gt; BAB.BAP.GLA.ANN.overlapped.txt

<span class="hljs-comment"># Find the number of genes within the overlapped regions</span>
bedtools intersect \
    -a ../References/S8/Ca.S8.gff \
    -b BAB.BAP.GLA.ANN.overlapped.txt \
    &gt; BAB.BAP.GLA.ANN.overlapped.genes.gff3

<span class="hljs-comment"># Find the number of genes within the specific regions</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *merged.txt | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.txt//g'</span>) ; <span class="hljs-keyword">do</span>
    bedtools intersect \
        -a ../References/S8/Ca.S8.gff\
        -b &lt;(cut -f1-3 <span class="hljs-variable">$i</span>.txt | grep -v Start) \
        &gt; <span class="hljs-variable">$i</span>.genes.gff3
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="xp-clr-for-c-annuum-with-all-c-annuum-var-grabriusculum">XP-CLR for C. annuum with all C. annuum var. grabriusculum</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/XPCLR/GLA.ANN.All

<span class="hljs-comment"># Blast the markers to S8 to update the physical position</span>
blastall -p blastn -i cM.BP.fa -d ../pepper_genome/fa -m 8 -e 1e-40 -o cM.BP.S8.blast

<span class="hljs-comment"># Prepare the SNP format for inferring the pseudo genetic position</span>
zcat ../vcf.clean/Chr01.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr01_"$2,$2}'</span> &gt; Chr01.BP.txt
zcat ../vcf.clean/Chr02.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr02_"$2,$2}'</span> &gt; Chr02.BP.txt
zcat ../vcf.clean/Chr03.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr03_"$2,$2}'</span> &gt; Chr03.BP.txt
zcat ../vcf.clean/Chr04.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr04_"$2,$2}'</span> &gt; Chr04.BP.txt
zcat ../vcf.clean/Chr05.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr05_"$2,$2}'</span> &gt; Chr05.BP.txt
zcat ../vcf.clean/Chr06.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr06_"$2,$2}'</span> &gt; Chr06.BP.txt
zcat ../vcf.clean/Chr07.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr07_"$2,$2}'</span> &gt; Chr07.BP.txt
zcat ../vcf.clean/Chr08.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr08_"$2,$2}'</span> &gt; Chr08.BP.txt
zcat ../vcf.clean/Chr09.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr09_"$2,$2}'</span> &gt; Chr09.BP.txt
zcat ../vcf.clean/Chr10.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr10_"$2,$2}'</span> &gt; Chr10.BP.txt
zcat ../vcf.clean/Chr11.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr11_"$2,$2}'</span> &gt; Chr11.BP.txt
zcat ../vcf.clean/Chr12.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print "Chr12_"$2,$2}'</span> &gt; Chr12.BP.txt

<span class="hljs-comment"># Prepare the real genetic maps</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    cat Linkage.clean.txt | grep Chr<span class="hljs-variable">$i</span> |  awk <span class="hljs-string">'{print $2,$5,$4}'</span> &gt; Chr<span class="hljs-variable">$i</span>.map.txt ; 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Infer the genetic position </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    perl assign_genetic.pl Chr<span class="hljs-variable">$i</span>.map.txt Chr<span class="hljs-variable">$i</span>.BP.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Compare the relationship between the inferred and real genetic maps</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    cat Chr<span class="hljs-variable">$i</span>.map.txt | awk <span class="hljs-string">'{print $0,"Real"}'</span> &gt; Chr<span class="hljs-variable">$i</span>.map-1.txt
    cat Chr<span class="hljs-variable">$i</span>.snp.genetic.txt | awk <span class="hljs-string">'{print $1,$2,$3,"Inferred"}'</span> | \
        cat - Chr<span class="hljs-variable">$i</span>.map-1.txt | sed <span class="hljs-string">'1i\SNP Pos cM Group'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; Chr<span class="hljs-variable">$i</span>.compare.txt
    rm Chr<span class="hljs-variable">$i</span>.map-1.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># SNP position files in LinkageMap folder</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span> 
    zcat ../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz | grep -v <span class="hljs-string">"#"</span> | awk <span class="hljs-string">'{print $1,$2,$4,$5}'</span> &gt; Chr<span class="hljs-variable">$i</span>.BP-2.txt
    paste Chr<span class="hljs-variable">$i</span>.snp.genetic.txt Chr<span class="hljs-variable">$i</span>.BP-2.txt | \
        awk <span class="hljs-string">'{print $1,$4,$3,$2,$6,$7}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; ../XPCLR/Chr<span class="hljs-variable">$i</span>.snp
    rm Chr<span class="hljs-variable">$i</span>.BP-2.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the genotypes for xpclr in XPCLR folder</span>
<span class="hljs-comment"># Make ped file using this chromosome map</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> Ca.annuum Ca.glabriusculum ; <span class="hljs-keyword">do</span> 
    cat ../../Pop.list/<span class="hljs-variable">$j</span>.list | awk <span class="hljs-string">'{print $1 "\t" $1}'</span> &gt; <span class="hljs-variable">$j</span>.list.txt
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> Ca.glabriusculum ; <span class="hljs-keyword">do</span> 
        <span class="hljs-comment"># Convert vcf to transposed ped</span>
        plink --vcf ../../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
            --keep <span class="hljs-variable">$j</span>.list.txt  \
            --allow-extra-chr --chr Chr<span class="hljs-variable">$i</span> \
            --recode 01 transpose \
            -output-missing-genotype 9 --double-id \
            --<span class="hljs-built_in">set</span>-missing-var-ids @:<span class="hljs-comment"># --keep-allele-order \</span>
            --out <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>
        <span class="hljs-comment"># Prepare the transposed ped file to xpclr geno file format</span>
        cut -d <span class="hljs-string">" "</span> -f 5- <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>.tped  | awk <span class="hljs-string">'{print $0" "}'</span> &gt; <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>.geno
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex *tfam *tped 

<span class="hljs-comment"># XPCLR of Ca.glabriusculum vs Ca.annuum</span>
<span class="hljs-comment"># Chr01</span>
<span class="hljs-comment"># Split the genotypes</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 9086854); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr01.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr01.0000000.geno ; 
    cat Ca.glabriusculum.Chr01.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr01.0000000.geno ; 
    cat Chr01.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr01.0000000.snp
    cat Ca.annuum.Chr01.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr01.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr01.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr01.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr01.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr01.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 9086854); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr01.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr01.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr01.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr01.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr01 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr01.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr02</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 4776664); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr02.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr02.0000000.geno ; 
    cat Ca.glabriusculum.Chr02.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr02.0000000.geno ; 
    cat Chr02.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr02.0000000.snp
    cat Ca.annuum.Chr02.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr02.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr02.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr02.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr02.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr02.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run XPCLR</span>
    <span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 4776664); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr02.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr02.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr02.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr02.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr02 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr02.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr03</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 8369414); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr03.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr03.0000000.geno ; 
    cat Ca.glabriusculum.Chr03.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr03.0000000.geno ; 
    cat Chr03.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr03.0000000.snp
    cat Ca.annuum.Chr03.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr03.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr03.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr03.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr03.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr03.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 8369414); <span class="hljs-keyword">do</span>
XPCLR -xpclr GLA.ANN/Ca.annuum.Chr03.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Ca.glabriusculum.Chr03.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Chr03.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr03.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr03 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr03.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr04</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7616000); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr04.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr04.0000000.geno ; 
    cat Ca.glabriusculum.Chr04.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr04.0000000.geno ; 
    cat Chr04.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr04.0000000.snp
    cat Ca.annuum.Chr04.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr04.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr04.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr04.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr04.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr04.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7616000); <span class="hljs-keyword">do</span>
XPCLR -xpclr GLA.ANN/Ca.annuum.Chr04.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Ca.glabriusculum.Chr04.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Chr04.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr04.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr04 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr04.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span> 

<span class="hljs-comment"># Chr05</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7570910); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr05.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr05.0000000.geno ; 
    cat Ca.glabriusculum.Chr05.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr05.0000000.geno ; 
    cat Chr05.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr05.0000000.snp
    cat Ca.annuum.Chr05.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr05.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr05.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr05.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr05.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr05.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7570910); <span class="hljs-keyword">do</span>
XPCLR -xpclr GLA.ANN/Ca.annuum.Chr05.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Ca.glabriusculum.Chr05.<span class="hljs-variable">$i</span>.geno \
    GLA.ANN/Chr05.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr05.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr05 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr05.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr06</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7522020); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr06.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr06.0000000.geno ; 
    cat Ca.glabriusculum.Chr06.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr06.0000000.geno ; 
    cat Chr06.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr06.0000000.snp
    cat Ca.annuum.Chr06.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr06.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr06.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr06.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr06.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr06.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7522020); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr06.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr06.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr06.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr06.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr06 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr06.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr07</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7824408); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr07.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr07.0000000.geno ; 
    cat Ca.glabriusculum.Chr07.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr07.0000000.geno ; 
    cat Chr07.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr07.0000000.snp
    cat Ca.annuum.Chr07.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr07.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr07.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr07.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr07.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr07.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7824408); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr07.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr07.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr07.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr07.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr07 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr07.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr08</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 5099543); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr08.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr08.0000000.geno ; 
    cat Ca.glabriusculum.Chr08.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr08.0000000.geno ; 
    cat Chr08.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr08.0000000.snp
    cat Ca.annuum.Chr08.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr08.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr08.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr08.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr08.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr08.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 5099543); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr08.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr08.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr08.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr08.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 1000 Chr08 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr08.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr09</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 8390222); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr09.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr09.0000000.geno ; 
    cat Ca.glabriusculum.Chr09.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr09.0000000.geno ; 
    cat Chr09.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr09.0000000.snp
    cat Ca.annuum.Chr09.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr09.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr09.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr09.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr09.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr09.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 8390222); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr09.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr09.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr09.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr09.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr09 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr09.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr10</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 6143906); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr10.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr10.0000000.geno ; 
    cat Ca.glabriusculum.Chr10.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr10.0000000.geno ; 
    cat Chr10.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr10.0000000.snp
    cat Ca.annuum.Chr10.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr10.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr10.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr10.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr10.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr10.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 6143906); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr10.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr10.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr10.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr10.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr10 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr10.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr11</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 8763242); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr11.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr11.0000000.geno ; 
    cat Ca.glabriusculum.Chr11.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr11.0000000.geno ; 
    cat Chr11.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr11.0000000.snp
    cat Ca.annuum.Chr11.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr11.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr11.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr11.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr11.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr11.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 8763242); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr11.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr11.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr11.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr11.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr11 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr11.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr12</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 500000 500000 7930008); <span class="hljs-keyword">do</span>
    cat Ca.annuum.Chr12.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr12.0000000.geno ; 
    cat Ca.glabriusculum.Chr12.geno | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr12.0000000.geno ; 
    cat Chr12.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; GLA.ANN/Chr12.0000000.snp
    cat Ca.annuum.Chr12.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.annuum.Chr12.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.glabriusculum.Chr12.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Ca.glabriusculum.Chr12.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr12.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+499999p"</span> &gt; GLA.ANN/Chr12.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 000000 500000 7930008); <span class="hljs-keyword">do</span>
    XPCLR -xpclr GLA.ANN/Ca.annuum.Chr12.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Ca.glabriusculum.Chr12.<span class="hljs-variable">$i</span>.geno \
        GLA.ANN/Chr12.<span class="hljs-variable">$i</span>.snp GLA.ANN/GLA.ANN.Chr12.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr12 -p0 0.7 &gt; GLA.ANN/GLA.ANN.Chr12.<span class="hljs-variable">$i</span>.out.xpclr.log &amp;
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *geno *snp *<span class="hljs-built_in">log</span>

<span class="hljs-comment"># Prepare the output format for distribution plots</span>
<span class="hljs-comment"># Average XPCLR score using 100 kb window at 10 kb step size sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    cat GLA.ANN/GLA.ANN.Chr<span class="hljs-variable">$i</span>.*.out.xpclr.txt &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.combined.out.xpclr.txt
    perl ../cluster_xpclrscore.pl GLA.ANN.Chr<span class="hljs-variable">$i</span>.combined.out.xpclr.txt \
        -wind_size 100000 -wind_step 10000 \
        &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.averaged.out.xpclr.txt
<span class="hljs-keyword">done</span>

awk <span class="hljs-string">'{print "Chr01_"$2,1,$2,$3,$7}'</span> GLA.ANN.Chr01.averaged.out.xpclr.txt &gt; GLA.ANN.Chr01.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr02_"$2,2,$2,$3,$7}'</span> GLA.ANN.Chr02.averaged.out.xpclr.txt &gt; GLA.ANN.Chr02.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr03_"$2,3,$2,$3,$7}'</span> GLA.ANN.Chr03.averaged.out.xpclr.txt &gt; GLA.ANN.Chr03.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr04_"$2,4,$2,$3,$7}'</span> GLA.ANN.Chr04.averaged.out.xpclr.txt &gt; GLA.ANN.Chr04.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr05_"$2,5,$2,$3,$7}'</span> GLA.ANN.Chr05.averaged.out.xpclr.txt &gt; GLA.ANN.Chr05.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr06_"$2,6,$2,$3,$7}'</span> GLA.ANN.Chr06.averaged.out.xpclr.txt &gt; GLA.ANN.Chr06.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr07_"$2,7,$2,$3,$7}'</span> GLA.ANN.Chr07.averaged.out.xpclr.txt &gt; GLA.ANN.Chr07.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr08_"$2,8,$2,$3,$7}'</span> GLA.ANN.Chr08.averaged.out.xpclr.txt &gt; GLA.ANN.Chr08.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr09_"$2,9,$2,$3,$7}'</span> GLA.ANN.Chr09.averaged.out.xpclr.txt &gt; GLA.ANN.Chr09.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr10_"$2,10,$2,$3,$7}'</span> GLA.ANN.Chr10.averaged.out.xpclr.txt &gt; GLA.ANN.Chr10.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr11_"$2,11,$2,$3,$7}'</span> GLA.ANN.Chr11.averaged.out.xpclr.txt &gt; GLA.ANN.Chr11.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr12_"$2,12,$2,$3,$7}'</span> GLA.ANN.Chr12.averaged.out.xpclr.txt &gt; GLA.ANN.Chr12.out.xpclr.txt-1

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
cat GLA.ANN.Chr*.out.xpclr.txt-1 | grep -v Wind | sed <span class="hljs-string">'1i\SNP Chr Pos PosEnd XPCLR'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; GLA.all.ANN.out.xpclr.txt

<span class="hljs-comment"># Remove unnecessary files</span>
rm GLA.ANN.Chr*.out.xpclr.txt-1 GLA.ANN.Chr*.combined.out.xpclr.txt GLA.ANN.Chr*.averaged.out.xpclr.txt

<span class="hljs-comment"># Extract the top 5% regions</span>
grep <span class="hljs-string">"SNP"</span> GLA.all.ANN.out.xpclr.txt | \
    cat - &lt;(grep -v <span class="hljs-string">"SNP"</span> GLA.all.ANN.out.xpclr.txt | sort -nrk5 | head -14870 | sort -nk2 -nk3) \
    &gt; GLA.all.ANN.out.xpclr.top5.txt

<span class="hljs-comment"># Merge the top 5% regions</span>
python ../region.py GLA.all.ANN.out.xpclr.top5.txt GLA.all.ANN.out.xpclr.top5.merged.txt
<span class="hljs-comment"># Alternatively use perl script</span>
perl ../merge.pl GLA.all.ANN.out.xpclr.top5.txt  &gt; GLA.all.ANN.out.xpclr.top5.merged2.txt


<span class="hljs-comment"># Filter the selective sweep windows</span>
bedtools merge -d 100000 \
    -i &lt;(cat BAB.BAP.out.xpclr.top5.txt  | sed <span class="hljs-string">'s/_/\t/g'</span> | awk <span class="hljs-string">'{print $1,$4,$5}'</span> | grep -v XPCLR | sed <span class="hljs-string">'s/ /\t/g'</span> ) \
    &gt; BAB.BAP.out.xpclr.top5.merged.txt

<span class="hljs-comment"># Find the overlapped region betwen BAB.BAP and GLA.ANN</span>
bedtools intersect \
    -a &lt;(cut -f1-3 ../BAB.BAP.out.xpclr.top5.merged.txt | grep -v Start) \
    -b &lt;(cut -f1-3 GLA.all.ANN.out.xpclr.top5.merged.txt | grep -v Start) \
    &gt; BAB.BAP.GLA.ANN.overlapped.txt

<span class="hljs-comment"># Find the number of genes with C. annuum</span>
bedtools intersect \
    -a ../../References/S8/Ca.S8.gff \
    -b &lt;(cut -f1-3 GLA.all.ANN.out.xpclr.top5.merged.txt | grep -v Chromsome) \
    &gt; GLA.all.ANN.genes.gff3

<span class="hljs-comment"># Number of genes</span>
grep ID=gene  GLA.all.ANN.genes.gff3 | wc

<span class="hljs-comment"># Find the number of genes within the overlapped regions</span>
bedtools intersect \
    -a ../../References/S8/Ca.S8.gff \
    -b BAB.BAP.GLA.ANN.overlapped.txt \
    &gt; BAB.BAP.GLA.ANN.overlapped.genes.gff3

<span class="hljs-comment"># Find the SNPs within the overlapped regions</span>
<span class="hljs-comment"># GLA.ANN</span>
bedtools intersect -a &lt;(cut -f1 GLA.all.ANN.out.xpclr.txt | \
    sed <span class="hljs-string">'s/_/\t/g'</span> | paste - GLA.all.ANN.out.xpclr.txt | \
    awk <span class="hljs-string">'{print $1,$2,$6,$3}'</span> | grep -v SNP | sed <span class="hljs-string">'s/ /\t/g'</span>) -b BAB.BAP.GLA.ANN.overlapped.txt | cut -f4 \
    &gt; GLA.all.ANN.overlapped.snps.txt

<span class="hljs-comment"># BAB.BAP</span>
bedtools intersect -a &lt;(cut -f1 ../BAB.BAP.out.xpclr.txt | \
    sed <span class="hljs-string">'s/_/\t/g'</span> | paste - ../BAB.BAP.out.xpclr.txt | \
    awk <span class="hljs-string">'{print $1,$2,$6,$3}'</span> | grep -v SNP | sed <span class="hljs-string">'s/ /\t/g'</span>) -b BAB.BAP.GLA.ANN.overlapped.txt | cut -f4 \
    &gt; BAB.BAP.overlapped.snps.txt

<span class="hljs-comment"># Find the number of genes within the specific regions</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(ll *merged.txt | awk <span class="hljs-string">'{print $9}'</span> | sed <span class="hljs-string">'s/.txt//g'</span>) ; <span class="hljs-keyword">do</span>
    bedtools intersect \
        -a ../References/S8/Ca.S8.gff\
        -b &lt;(cut -f1-3 <span class="hljs-variable">$i</span>.txt | grep -v Start) \
        &gt; <span class="hljs-variable">$i</span>.genes.gff3
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="xp-clr-for-c-bac-baccatum-versus-c-bac-pendulum">XP-CLR for C. bac. baccatum versus C. bac. pendulum</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/XPCLR/BAB.BAP

<span class="hljs-comment"># Prepare the genotypes</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> Ca.bac.baccatum Ca.bac.pendulum; <span class="hljs-keyword">do</span> 
        cat ../../Pop.list/<span class="hljs-variable">$j</span>.list | awk <span class="hljs-string">'{print $1 "\t" $1}'</span> &gt; <span class="hljs-variable">$j</span>.list.txt
        <span class="hljs-comment"># Convert vcf to transposed ped</span>
        plink --vcf ../../vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
            --keep <span class="hljs-variable">$j</span>.list.txt  \
            --allow-extra-chr --chr Chr<span class="hljs-variable">$i</span> \
            --recode 01 transpose \
            --geno 0.5 \
            --maf 0.05 \
            -output-missing-genotype 9 --double-id \
            --<span class="hljs-built_in">set</span>-missing-var-ids @:<span class="hljs-comment"># --keep-allele-order \</span>
            --out <span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Find the common SNPs between C. bac. baccatum and C. bac. pendulum</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    cut -f2 -d <span class="hljs-string">' '</span> Ca.bac.pendulum.Chr<span class="hljs-variable">$i</span>.tped &gt; BAP.Chr<span class="hljs-variable">$i</span>.snp
    cut -f2 -d <span class="hljs-string">' '</span> Ca.bac.baccatum.Chr<span class="hljs-variable">$i</span>.tped &gt; BAB.Chr<span class="hljs-variable">$i</span>.snp
    comm -12 &lt;(sort BAP.Chr<span class="hljs-variable">$i</span>.snp) &lt;(sort BAB.Chr<span class="hljs-variable">$i</span>.snp) &gt; BAB.BAP.Chr<span class="hljs-variable">$i</span>.snp
    grep -Fwf BAB.BAP.Chr<span class="hljs-variable">$i</span>.snp Ca.bac.pendulum.Chr<span class="hljs-variable">$i</span>.tped | \
        cut -d <span class="hljs-string">" "</span> -f 5- | awk <span class="hljs-string">'{print $0" "}'</span> &gt; Ca.bac.pendulum.Chr<span class="hljs-variable">$i</span>.geno
    grep -Fwf BAB.BAP.Chr<span class="hljs-variable">$i</span>.snp Ca.bac.baccatum.Chr<span class="hljs-variable">$i</span>.tped | \
        cut -d <span class="hljs-string">" "</span> -f 5- | awk <span class="hljs-string">'{print $0" "}'</span> &gt; Ca.bac.baccatum.Chr<span class="hljs-variable">$i</span>.geno
    grep -Fwf BAB.BAP.Chr<span class="hljs-variable">$i</span>.snp ../Chr<span class="hljs-variable">$i</span>.snp &gt; Chr<span class="hljs-variable">$i</span>.snp
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex *tfam *tped 

<span class="hljs-comment"># XPCLR of Ca.bac.baccatum vs Ca.bac.pendulum</span>
<span class="hljs-comment"># Chr01</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 740681); <span class="hljs-keyword">do</span>
    cat Ca.bac.pendulum.Chr01.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr01.000000.geno ; 
    cat Ca.bac.baccatum.Chr01.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr01.000000.geno ; 
    cat Chr01.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr01.000000.snp
    cat Ca.bac.pendulum.Chr01.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr01.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.bac.baccatum.Chr01.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr01.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr01.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr01.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 740681); <span class="hljs-keyword">do</span>
    XPCLR -xpclr Ca.bac.pendulum.Chr01.<span class="hljs-variable">$i</span>.geno \
        Ca.bac.baccatum.Chr01.<span class="hljs-variable">$i</span>.geno \
        Chr01.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr01.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr01 -p0 0.7 &gt; BAB.BAP.Chr01.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr02</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 291463); <span class="hljs-keyword">do</span>
    cat Ca.bac.pendulum.Chr02.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr02.000000.geno ; 
    cat Ca.bac.baccatum.Chr02.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr02.000000.geno ; 
    cat Chr02.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr02.000000.snp
    cat Ca.bac.pendulum.Chr02.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr02.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.bac.baccatum.Chr02.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr02.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr02.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr02.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 291463); <span class="hljs-keyword">do</span>
    XPCLR -xpclr Ca.bac.pendulum.Chr02.<span class="hljs-variable">$i</span>.geno \
        Ca.bac.baccatum.Chr02.<span class="hljs-variable">$i</span>.geno \
        Chr02.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr02.<span class="hljs-variable">$i</span>.out \
        -w1 0.0005 100 100 Chr02 -p0 0.7 &gt; BAB.BAP.Chr02.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr03</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 521851); <span class="hljs-keyword">do</span>
    cat Ca.bac.pendulum.Chr03.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr03.000000.geno ; 
    cat Ca.bac.baccatum.Chr03.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr03.000000.geno ; 
    cat Chr03.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr03.000000.snp
    cat Ca.bac.pendulum.Chr03.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr03.<span class="hljs-variable">$i</span>.geno ; 
    cat Ca.bac.baccatum.Chr03.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr03.<span class="hljs-variable">$i</span>.geno ; 
    cat Chr03.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr03.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 521851); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr03.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr03.<span class="hljs-variable">$i</span>.geno \
    Chr03.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr03.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr03 -p0 0.7 &gt; BAB.BAP.Chr03.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr04</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 563213); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr04.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr04.000000.geno ; 
cat Ca.bac.baccatum.Chr04.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr04.000000.geno ; 
cat Chr04.snp | sed -n <span class="hljs-string">"1,499999p"</span> &gt; Chr04.000000.snp
cat Ca.bac.pendulum.Chr04.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr04.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr04.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr04.<span class="hljs-variable">$i</span>.geno ; 
cat Chr04.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr04.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 563213); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr04.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr04.<span class="hljs-variable">$i</span>.geno \
    Chr04.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr04.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr04 -p0 0.7 &gt; BAB.BAP.Chr04.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span> 

<span class="hljs-comment"># Chr05</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 538616); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr05.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr05.000000.geno ; 
cat Ca.bac.baccatum.Chr05.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr05.000000.geno ; 
cat Chr05.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr05.000000.snp
cat Ca.bac.pendulum.Chr05.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr05.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr05.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr05.<span class="hljs-variable">$i</span>.geno ; 
cat Chr05.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr05.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 538616); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr05.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr05.<span class="hljs-variable">$i</span>.geno \
    Chr05.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr05.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr05 -p0 0.7 &gt; BAB.BAP.Chr05.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr06</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 548970); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr06.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr06.000000.geno ; 
cat Ca.bac.baccatum.Chr06.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr06.000000.geno ; 
cat Chr06.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr06.000000.snp
cat Ca.bac.pendulum.Chr06.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr06.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr06.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr06.<span class="hljs-variable">$i</span>.geno ; 
cat Chr06.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr06.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 548970); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr06.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr06.<span class="hljs-variable">$i</span>.geno \
    Chr06.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr06.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr06 -p0 0.7 &gt; BAB.BAP.Chr06.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr07</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 487786); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr07.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr07.000000.geno ; 
cat Ca.bac.baccatum.Chr07.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr07.000000.geno ; 
cat Chr07.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr07.000000.snp
cat Ca.bac.pendulum.Chr07.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr07.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr07.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr07.<span class="hljs-variable">$i</span>.geno ; 
cat Chr07.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr07.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 487786); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr07.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr07.<span class="hljs-variable">$i</span>.geno \
    Chr07.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr07.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr07 -p0 0.7 &gt; BAB.BAP.Chr07.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr08</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 238001); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr08.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr08.000000.geno ; 
cat Ca.bac.baccatum.Chr08.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr08.000000.geno ; 
cat Chr08.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr08.000000.snp
cat Ca.bac.pendulum.Chr08.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr08.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr08.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr08.<span class="hljs-variable">$i</span>.geno ; 
cat Chr08.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr08.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 238001); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr08.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr08.<span class="hljs-variable">$i</span>.geno \
    Chr08.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr08.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr08 -p0 0.7 &gt; BAB.BAP.Chr08.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr09</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 482094); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr09.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr09.000000.geno ; 
cat Ca.bac.baccatum.Chr09.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr09.000000.geno ; 
cat Chr09.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr09.000000.snp
cat Ca.bac.pendulum.Chr09.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr09.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr09.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr09.<span class="hljs-variable">$i</span>.geno ; 
cat Chr09.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr09.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 482094); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr09.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr09.<span class="hljs-variable">$i</span>.geno \
    Chr09.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr09.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr09 -p0 0.7 &gt; BAB.BAP.Chr09.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr10</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 583266); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr10.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr10.000000.geno ; 
cat Ca.bac.baccatum.Chr10.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr10.000000.geno ; 
cat Chr10.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr10.000000.snp
cat Ca.bac.pendulum.Chr10.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr10.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr10.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr10.<span class="hljs-variable">$i</span>.geno ; 
cat Chr10.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr10.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 583266); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr10.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr10.<span class="hljs-variable">$i</span>.geno \
    Chr10.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr10.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr10 -p0 0.7 &gt; BAB.BAP.Chr10.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr11</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 488040); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr11.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr11.000000.geno ; 
cat Ca.bac.baccatum.Chr11.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr11.000000.geno ; 
cat Chr11.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr11.000000.snp
cat Ca.bac.pendulum.Chr11.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr11.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr11.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr11.<span class="hljs-variable">$i</span>.geno ; 
cat Chr11.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr11.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 488040); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr11.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr11.<span class="hljs-variable">$i</span>.geno \
    Chr11.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr11.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr11 -p0 0.7 &gt; BAB.BAP.Chr11.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Chr12</span>
<span class="hljs-comment"># Prepare the input files</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 50000 50000 507900); <span class="hljs-keyword">do</span>
cat Ca.bac.pendulum.Chr12.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.pendulum.Chr12.000000.geno ; 
cat Ca.bac.baccatum.Chr12.geno | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Ca.bac.baccatum.Chr12.000000.geno ; 
cat Chr12.snp | sed -n <span class="hljs-string">"1,49999p"</span> &gt; Chr12.000000.snp
cat Ca.bac.pendulum.Chr12.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.pendulum.Chr12.<span class="hljs-variable">$i</span>.geno ; 
cat Ca.bac.baccatum.Chr12.geno | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Ca.bac.baccatum.Chr12.<span class="hljs-variable">$i</span>.geno ; 
cat Chr12.snp | sed -n <span class="hljs-string">"1p;<span class="hljs-variable">${i}</span>,+49999p"</span> &gt; Chr12.<span class="hljs-variable">$i</span>.snp ; 
<span class="hljs-keyword">done</span>
<span class="hljs-comment"># Run XPCLR</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 00000 50000 507900); <span class="hljs-keyword">do</span>
XPCLR -xpclr Ca.bac.pendulum.Chr12.<span class="hljs-variable">$i</span>.geno \
    Ca.bac.baccatum.Chr12.<span class="hljs-variable">$i</span>.geno \
    Chr12.<span class="hljs-variable">$i</span>.snp BAB.BAP.Chr12.<span class="hljs-variable">$i</span>.out \
    -w1 0.0005 100 100 Chr12 -p0 0.7 &gt; BAB.BAP.Chr12.<span class="hljs-variable">$i</span>.out.xpclr.log 
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Remove unnecessary files</span>
rm *geno *snp *<span class="hljs-built_in">log</span>

<span class="hljs-comment"># Prepare the output format for distribution plots</span>
<span class="hljs-comment"># Average XPCLR score using 100 kb window at 10 kb step size sliding window</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    cat BAB.BAP.Chr<span class="hljs-variable">$i</span>.*out.xpclr.txt &gt; BAB.BAP.Chr<span class="hljs-variable">$i</span>.combined.out.xpclr.txt
    perl ../cluster_xpclrscore.pl BAB.BAP.Chr<span class="hljs-variable">$i</span>.combined.out.xpclr.txt \
        -wind_size 100000 -wind_step 10000 \
        &gt; BAB.BAP.Chr<span class="hljs-variable">$i</span>.averaged.out.xpclr.txt
<span class="hljs-keyword">done</span>

awk <span class="hljs-string">'{print "Chr01_"$2,1,$2,$3,$7}'</span> BAB.BAP.Chr01.averaged.out.xpclr.txt &gt; BAB.BAP.Chr01.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr02_"$2,2,$2,$3,$7}'</span> BAB.BAP.Chr02.averaged.out.xpclr.txt &gt; BAB.BAP.Chr02.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr03_"$2,3,$2,$3,$7}'</span> BAB.BAP.Chr03.averaged.out.xpclr.txt &gt; BAB.BAP.Chr03.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr04_"$2,4,$2,$3,$7}'</span> BAB.BAP.Chr04.averaged.out.xpclr.txt &gt; BAB.BAP.Chr04.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr05_"$2,5,$2,$3,$7}'</span> BAB.BAP.Chr05.averaged.out.xpclr.txt &gt; BAB.BAP.Chr05.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr06_"$2,6,$2,$3,$7}'</span> BAB.BAP.Chr06.averaged.out.xpclr.txt &gt; BAB.BAP.Chr06.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr07_"$2,7,$2,$3,$7}'</span> BAB.BAP.Chr07.averaged.out.xpclr.txt &gt; BAB.BAP.Chr07.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr08_"$2,8,$2,$3,$7}'</span> BAB.BAP.Chr08.averaged.out.xpclr.txt &gt; BAB.BAP.Chr08.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr09_"$2,9,$2,$3,$7}'</span> BAB.BAP.Chr09.averaged.out.xpclr.txt &gt; BAB.BAP.Chr09.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr10_"$2,10,$2,$3,$7}'</span> BAB.BAP.Chr10.averaged.out.xpclr.txt &gt; BAB.BAP.Chr10.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr11_"$2,11,$2,$3,$7}'</span> BAB.BAP.Chr11.averaged.out.xpclr.txt &gt; BAB.BAP.Chr11.out.xpclr.txt-1
awk <span class="hljs-string">'{print "Chr12_"$2,12,$2,$3,$7}'</span> BAB.BAP.Chr12.averaged.out.xpclr.txt &gt; BAB.BAP.Chr12.out.xpclr.txt-1

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
cat BAB.BAP.Chr*.out.xpclr.txt-1 | \
    grep -v Wind | \
    sed <span class="hljs-string">'1i\SNP Chr Pos PosEnd XPCLR'</span> | \
    sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; BAB.BAP.out.xpclr.txt

<span class="hljs-comment"># Remove unnecessary files</span>
rm BAB.BAP.Chr*.out.xpclr.txt-1 BAB.BAP.Chr*.combined.out.xpclr.txt BAB.BAP.Chr*.averaged.out.xpclr.txt

</div></code></pre>
<h3 id="enrichment-analysis">Enrichment analysis</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/References/S8

<span class="hljs-comment"># Genome gene list</span>
less pepper_S8.gff3.gz | grep gene | cut -f9 | sed <span class="hljs-string">'s/;/\t/g'</span> | \
    cut -f1 | grep rna | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/Parent=//g'</span> \
    &gt; S8.genome.gene.list.txt

<span class="hljs-comment"># Prepare the gene lists</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAB.BAP.0.05.divergent.genes GLA.ANN.0.05.divergent.genes \
    GLA.ANN.0.05.narrow.convergent.genes GLA.ANN.0.05.broad.convergent.genes ; <span class="hljs-keyword">do</span>
    grep gene <span class="hljs-variable">$i</span> | cut -f9 | sed <span class="hljs-string">'s/;/.1;/g'</span> | sed <span class="hljs-string">'s/;/\t/g'</span> | \
        cut -f1 | sed <span class="hljs-string">'s/ID=//g'</span> | sed <span class="hljs-string">'s/Parent=//g'</span> | sed <span class="hljs-string">'s/gene/rna/g'</span> \
        &gt; <span class="hljs-variable">$i</span>.list.txt
<span class="hljs-keyword">done</span>

</div></code></pre>
<h2 id="effective-population-size">Effective population size</h2>
<p>Software link: <a href="https://github.com/popgenmethods/smcpp#tips-for-using-smc">SMC++</a></p>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SMC++

<span class="hljs-comment"># Install </span>
pip install git+https://github.com/popgenmethods/smcpp

<span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SMC++/GLA

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat GLA.list) ; <span class="hljs-keyword">do</span>
        <span class="hljs-comment"># Extract the tested samples from all vcf</span>
        vcftools --gzvcf /data/zhaojiantao/pepper/vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
            --keep GLA.list \
            --maf 0.05 \
            --max-missing 0.8 \
            --recode --out Chr<span class="hljs-variable">$i</span>.clean
        <span class="hljs-comment"># bgzip the vcf, not gzip</span>
        bgzip Chr<span class="hljs-variable">$i</span>.clean.recode.vcf
        <span class="hljs-comment"># Index the file</span>
        tabix Chr<span class="hljs-variable">$i</span>.clean.recode.vcf.gz
        <span class="hljs-comment"># Convert vcf to smc </span>
        smc++ vcf2smc --cores 60 --missing-cutoff 5000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            Chr<span class="hljs-variable">$i</span>.clean.recode.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            GLA:Grif_9241,PI_555616,PI_632930,Grif_9142,PI_311126,PI_511885,PI_593557,Grif_9249,PI_593526,PI_439311
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 20 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot GLA.smc.jpeg -g 1 smc.out/model.final.json

</div></code></pre>
<h3 id="smc-for-gla-with-different-geographical-background">SMC for GLA with different geographical background</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/SMC++/Accessions.diverse.background/GLA1

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(cat GLA.list) ; <span class="hljs-keyword">do</span>
        <span class="hljs-comment"># Extract the tested samples from all vcf</span>
        vcftools --gzvcf /data/zhaojiantao/pepper/vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
            --keep GLA.list \
            --maf 0.05 \
            --max-missing 0.8 \
            --recode --out Chr<span class="hljs-variable">$i</span>.clean
        <span class="hljs-comment"># bgzip the vcf, not gzip</span>
        bgzip Chr<span class="hljs-variable">$i</span>.clean.recode.vcf
        <span class="hljs-comment"># Index the file</span>
        tabix Chr<span class="hljs-variable">$i</span>.clean.recode.vcf.gz
        <span class="hljs-comment"># Convert vcf to smc </span>
        smc++ vcf2smc --cores 60 --missing-cutoff 5000 -d <span class="hljs-variable">$j</span> <span class="hljs-variable">$j</span> \
            Chr<span class="hljs-variable">$i</span>.clean.recode.vcf.gz Chr<span class="hljs-variable">$i</span>.<span class="hljs-variable">$j</span>.smc.gz Chr<span class="hljs-variable">$i</span> \
            GLA:Grif_9138,Grif_9142,PI_406948,PI_438567,PI_593526,PI_593574,PI_631136,PI_555616,PI_661082,PI_674459
    <span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Run smc++</span>
smc++ estimate -o smc.out 6.96e-9 --cores 20 *smc.gz

<span class="hljs-comment"># Generate the plots</span>
smc++ plot GLA.smc.jpeg -g 1 model.final.json

</div></code></pre>
<h2 id="gwas-analysis-of-fruit-shape">GWAS analysis of fruit shape</h2>
<h3 id="gwas-for-c-annuum">GWAS for <em>C. annuum</em></h3>
<pre class="hljs"><code><div><span class="hljs-comment"># GLA.ANN working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/GWAS/GLA.ANN

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    <span class="hljs-comment"># Convert vcf to tped</span>
    plink --vcf /data/zhaojiantao/pepper/vcf.clean/Chr<span class="hljs-variable">$i</span>.clean.vcf.gz \
        --keep GLA.ANN.list \
        --allow-extra-chr --chr Chr<span class="hljs-variable">$i</span> \
        --recode \
        --double-id \
        --geno 0.5 \
        --maf 0.05 \
        --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    sed <span class="hljs-string">'s/Chr0//g'</span> GLA.ANN.Chr<span class="hljs-variable">$i</span>.map | sed <span class="hljs-string">'s/Chr//g'</span> | \
    awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; GLA.ANN.Chr<span class="hljs-variable">$i</span>.map-1
<span class="hljs-keyword">done</span>

rm *map *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    mv GLA.ANN.Chr<span class="hljs-variable">$i</span>.map-1 GLA.ANN.Chr<span class="hljs-variable">$i</span>.map
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Make-bed</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> \
        --make-bed \
        --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    java -jar /data/zhaojiantao/tools/GeneticTypeIError/gec.jar \
        --effect-number \
        --plink-binary GLA.ANN.Chr<span class="hljs-variable">$i</span>
    mv gec.sum GLA.ANN.Chr<span class="hljs-variable">$i</span>.gec.sum
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the effective number of SNPs of all chromosomes</span>
cat *gec.sum | grep -v Effective_Number | awk <span class="hljs-string">'{ sum += $2 } END { print sum }'</span> 

<span class="hljs-comment"># Population structure</span>
plink --vcf &lt;(zcat ../../IQ-Tree504/Ca.S8.ffds.vcf.gz | grep -v Chr00) \
    --keep GLA.ANN.list \
    --allow-extra-chr \
    --recode \
    --double-id \
    --geno 0.2 \
    --maf 0.05 \
    --out GLA.ANN.ffds

sed <span class="hljs-string">'s/Chr0//g'</span> GLA.ANN.ffds.map | sed <span class="hljs-string">'s/Chr//g'</span> | awk <span class="hljs-string">'{print $1,$1"_"$4,$3,$4}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; GLA.ANN.ffds.map-1
rm GLA.ANN.ffds.map
mv GLA.ANN.ffds.map-1 GLA.ANN.ffds.map

<span class="hljs-comment"># Calculate PCA</span>
plink --file GLA.ANN.ffds --pca --out GLA.ANN.ffds

<span class="hljs-comment"># Remove unnecessary files</span>
rm *<span class="hljs-built_in">log</span> *nosex

<span class="hljs-comment"># Prepare the PCA file as cofactor</span>
awk <span class="hljs-string">'{print $1,$2,1,$3,$4,$5,$6,$7}'</span> GLA.ANN.ffds.eigenvec &gt; GLA.ANN.ffds.PCA.txt

<span class="hljs-comment"># Transpose the ped file</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    plink --file GLA.ANN.Chr<span class="hljs-variable">$i</span> --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate kinship</span>
plink --file GLA.ANN.ffds --recode12 --output-missing-genotype 0 --transpose --out GLA.ANN.ffds
emmax-kin -v -d 10 GLA.ANN.ffds

<span class="hljs-comment"># EMMAX</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
<span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> FruitLength FruitWidth FruitRatio ; <span class="hljs-keyword">do</span>
    emmax -v -d 10 -t GLA.ANN.Chr<span class="hljs-variable">$i</span> -p <span class="hljs-variable">$j</span>.txt -k GLA.ANN.ffds.BN.kinf -c GLA.ANN.ffds.PCA.txt -o GLA.ANN.<span class="hljs-variable">$j</span>.Chr<span class="hljs-variable">$i</span>
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the output for manhattan plot</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> FruitLength FruitWidth FruitRatio ; <span class="hljs-keyword">do</span>
    cat GLA.ANN.<span class="hljs-variable">$i</span>.Chr*.ps | cut -f1 | sed <span class="hljs-string">'s/_/\t/g'</span> | \
        paste - &lt;(cat GLA.ANN.<span class="hljs-variable">$i</span>.Chr*.ps) | \
        awk <span class="hljs-string">'{print $1,$2,$3,$5}'</span> | sed <span class="hljs-string">'1i\CHR BP SNP P'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
        &gt; GLA.ANN.<span class="hljs-variable">$i</span>.GWAS.txt
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the LD for the peak SNP</span>
plink --file GLA.ANN.Chr02 \
    --r2 \
    --ld-snp 2_160951499 \
    --ld-window-kb 1000 \
    --ld-window 1050600 \
    --ld-window-r2 0.351 \
    --out 2_160951499

plink --file BAB.BAP.Chr10 \
    --r2 \
    --ld-snp 10_121586106 \
    --ld-window-kb 1000 \
    --ld-window 126500 \
    --ld-window-r2 0.366 \
    --out 10_121586106

plink --file BAB.BAP.Chr02 \
    --r2 \
    --ld-snp 2_172705768 \
    --ld-window-kb 1000 \
    --ld-window 126500 \
    --ld-window-r2 0.366 \
    --out 2_172705768

<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> BAP.long BAP.bell ; <span class="hljs-keyword">do</span>
    vcftools --gzvcf ../vcf.clean/Chr10.clean.vcf.gz --keep <span class="hljs-variable">$i</span>.list --site-pi --out <span class="hljs-variable">$i</span>.Chr10
    perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr10.sites.pi
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Step 15.2 Calculate windowed pi </span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> Ca.annuum Ca.bac.baccatum Ca.bac.pendulum Ca.chacoense Ca.chinense Ca.frutescens Ca.glabriusculum Ca.glabriusculum_II Ca.pubescens ; <span class="hljs-keyword">do</span>
    <span class="hljs-keyword">for</span> j <span class="hljs-keyword">in</span> $(seq -w 1 12); <span class="hljs-keyword">do</span>
    perl PI_slide_win.pl -w 100000 -s 10000 <span class="hljs-variable">$i</span>.Chr<span class="hljs-variable">$j</span>.sites.pi
<span class="hljs-keyword">done</span>
<span class="hljs-keyword">done</span>

</div></code></pre>
<h3 id="extract-the-genotypes-for-caovate">Extract the genotypes for CaOvate</h3>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/Pi

<span class="hljs-comment"># Extract the genotype and convert to plink ped</span>
plink --vcf /data/zhaojiantao/pepper/vcf.clean/Chr10.clean.vcf.gz \
    --keep &lt;(awk <span class="hljs-string">'{print $1,$1}'</span> ../Pop.list/Ca.bac.baccatum.list | sed <span class="hljs-string">'s/ /\t/g'</span> | cat - BAP.bell.list BAP.long.list) \
    --chr Chr10 --from-mb 120 --to-mb 122.6 \
    --recode --make-bed \
    --double-id \
    --maf 0.05 \
    --out Ca.baccatum.Chr10.region

<span class="hljs-comment"># Convert to 012</span>
plink --file Ca.baccatum.Chr10.region --recode12 transpose --out Ca.baccatum.Chr10.region

<span class="hljs-comment"># Combine the genotypes</span>
cut -f5- -d <span class="hljs-string">' '</span> Ca.baccatum.Chr10.region.tped | sed <span class="hljs-string">'s! \([^ ]\+\)\( \|$\)!\1 !g'</span> &gt; Ca.baccatum.Chr10.region.tped-1

<span class="hljs-comment"># Prepare the genotype</span>
cut -f1 -d <span class="hljs-string">' '</span> Ca.baccatum.Chr10.region.tfam &gt; Ca.baccatum.Chr10.region.tfam-1

python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Ca.baccatum.Chr10.region.tfam-1 &gt; Ca.baccatum.Chr10.region.tfam.trans

cat Ca.baccatum.Chr10.region.tfam.trans Ca.baccatum.Chr10.region.tped-1 | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Ca.baccatum.Chr10.region.tped-2
python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Ca.baccatum.Chr10.region.tped-2 &gt; Ca.baccatum.Chr10.region.tped-3


<span class="hljs-comment"># BAB geno</span>
grep -Fwf ../Pop.list/Ca.bac.baccatum.list Ca.baccatum.Chr10.region.tped-3 &gt; Ca.baccatum.Chr10.region.BAB.geno
grep -Fwf &lt;(cut -f1 BAP.bell.list) Ca.baccatum.Chr10.region.tped-3 &gt; Ca.baccatum.Chr10.region.bell.geno
grep -Fwf &lt;(cut -f1 BAP.long.list) Ca.baccatum.Chr10.region.tped-3 &gt; Ca.baccatum.Chr10.region.long.geno

cat Ca.baccatum.Chr10.region.*.geno | sed <span class="hljs-string">'s/ /\t/g'</span> &gt; Ca.baccatum.Chr10.region.txt

python -c <span class="hljs-string">"import sys; print('\n'.join(' '.join(c) for c in zip(*(l.split() for l in sys.stdin.readlines() if l.strip()))))"</span> &lt; Ca.baccatum.Chr10.region.txt &gt; Ca.baccatum.Chr10.region.trans.txt


</div></code></pre>
<h3 id="manhattan-plot-for-glaann">Manhattan plot for GLA.ANN</h3>
<pre class="hljs"><code><div><span class="hljs-keyword">library</span>(qqman)
<span class="hljs-keyword">library</span>(data.table)
<span class="hljs-keyword">source</span>(<span class="hljs-string">"manhattan.R"</span>)

GLA.ANN.GWAS.FruitLength &lt;- fread(<span class="hljs-string">"GLA.ANN.FruitLength.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.FruitLength.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">20</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.GWAS.FruitLength, chr=<span class="hljs-string">"CHR"</span>, bp=<span class="hljs-string">"BP"</span>, snp=<span class="hljs-string">"SNP"</span>, p=<span class="hljs-string">"P"</span>, 
          col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>), cex=<span class="hljs-number">0.1</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">8</span>),
          genomewideline = <span class="hljs-number">6.29</span>, suggestiveline = <span class="hljs-literal">FALSE</span>)
dev.off()

GLA.ANN.GWAS.FruitWidth &lt;- fread(<span class="hljs-string">"GLA.ANN.FruitWidth.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.FruitWidth.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">20</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.GWAS.FruitWidth, chr=<span class="hljs-string">"CHR"</span>, bp=<span class="hljs-string">"BP"</span>, snp=<span class="hljs-string">"SNP"</span>, p=<span class="hljs-string">"P"</span>, 
          col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>), cex=<span class="hljs-number">0.1</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">8</span>),
          genomewideline = <span class="hljs-number">6.29</span>, suggestiveline = <span class="hljs-literal">FALSE</span>)
dev.off()

GLA.ANN.GWAS.FruitRatio &lt;- fread(<span class="hljs-string">"GLA.ANN.FruitRatio.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"GLA.ANN.FruitRatio.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">15</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(GLA.ANN.GWAS.FruitRatio, chr=<span class="hljs-string">"CHR"</span>, bp=<span class="hljs-string">"BP"</span>, snp=<span class="hljs-string">"SNP"</span>, p=<span class="hljs-string">"P"</span>, 
          col = c(<span class="hljs-string">"orange"</span>, <span class="hljs-string">"darkgreen"</span>), cex=<span class="hljs-number">0.1</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">8</span>),
          genomewideline = <span class="hljs-number">6.29</span>, suggestiveline = <span class="hljs-literal">FALSE</span>)
dev.off()

</div></code></pre>
<h3 id="manhattan-plot-for-babbap">Manhattan plot for BAB.BAP</h3>
<pre class="hljs"><code><div>BAB.BAP.GWAS.FruitLength &lt;- fread(<span class="hljs-string">"BAB.BAP.FruitLength.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.FruitLength.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">20</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.GWAS.FruitLength, chr=<span class="hljs-string">"CHR"</span>, bp=<span class="hljs-string">"BP"</span>, snp=<span class="hljs-string">"SNP"</span>, p=<span class="hljs-string">"P"</span>, 
          col = c(<span class="hljs-string">"purple"</span>, <span class="hljs-string">"darkorange"</span>), cex=<span class="hljs-number">0.1</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">8</span>),
          genomewideline = <span class="hljs-number">6.39</span>, suggestiveline = <span class="hljs-literal">FALSE</span>)
dev.off()

BAB.BAP.GWAS.FruitWidth &lt;- fread(<span class="hljs-string">"BAB.BAP.FruitWidth.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.FruitWidth.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">20</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.GWAS.FruitWidth, chr=<span class="hljs-string">"CHR"</span>, bp=<span class="hljs-string">"BP"</span>, snp=<span class="hljs-string">"SNP"</span>, p=<span class="hljs-string">"P"</span>, 
          col = c(<span class="hljs-string">"purple"</span>, <span class="hljs-string">"darkorange"</span>), cex=<span class="hljs-number">0.1</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">8</span>),
          genomewideline = <span class="hljs-number">6.39</span>, suggestiveline = <span class="hljs-literal">FALSE</span>)
dev.off()

BAB.BAP.GWAS.FruitRatio &lt;- fread(<span class="hljs-string">"BAB.BAP.FruitRatio.GWAS.txt"</span>, header = <span class="hljs-literal">TRUE</span>)
jpeg(<span class="hljs-string">"BAB.BAP.FruitRatio.jpeg"</span>, height = <span class="hljs-number">8</span>, width = <span class="hljs-number">15</span>, units=<span class="hljs-string">"cm"</span>, res = <span class="hljs-number">300</span>)
manhattan(BAB.BAP.GWAS.FruitRatio, chr=<span class="hljs-string">"CHR"</span>, bp=<span class="hljs-string">"BP"</span>, snp=<span class="hljs-string">"SNP"</span>, p=<span class="hljs-string">"P"</span>, 
          col = c(<span class="hljs-string">"purple"</span>, <span class="hljs-string">"darkorange"</span>), cex=<span class="hljs-number">0.1</span>, ylim=c(<span class="hljs-number">0</span>,<span class="hljs-number">12</span>),
          genomewideline = <span class="hljs-number">6.39</span>, suggestiveline = <span class="hljs-literal">FALSE</span>)
dev.off()

</div></code></pre>
<h2 id="transcriptome-data">Transcriptome data</h2>
<pre class="hljs"><code><div><span class="hljs-comment"># Working directory</span>
<span class="hljs-built_in">cd</span> /data/zhaojiantao/pepper/transcriptome

<span class="hljs-comment"># Process the RNAseq data for fruit shape at different developing stages</span>
<span class="hljs-comment"># Save the above command line to Qualityclean.sh and run: nohup sh Qualityclean.sh &gt; Qualityclean.log &amp;</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat fruitpeel.id); <span class="hljs-keyword">do</span>
        <span class="hljs-comment"># Trim adaptor and low quality</span>
        trimmomatic PE <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1.fq.gz <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2.fq.gz \
                <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1U.fq \
                <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2U.fq \
                -threads 60 \
                ILLUMINACLIP:/data/zhaojiantao/tools/anaconda3/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa:2:30:10:1:TRUE \
                SLIDINGWINDOW:4:20 LEADING:3 TRAILING:3 MINLEN:40
        <span class="hljs-comment"># Remove polyA</span>
        prinseq++ -threads 60 -VERBOSE 0 \
                -fastq <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.fq \
                -fastq2 <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.fq \
                -min_len 40 \
                -trim_tail_left 10 \
                -trim_tail_right 10 \
                -out_name <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>
        <span class="hljs-comment"># Remove rRNA</span>
        bowtie -v 3 -k 1 -p 60 --al <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.rRNA.fq \
                /data/zhaojiantao/database/rRNA_region/rRNA_combined \
                -q <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_good_out_R1.fastq &gt;/dev/null
        bowtie -v 3 -k 1 -p 60 --al <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.rRNA.fq \
                /data/zhaojiantao/database/rRNA_combined/rRNA_combined \
                -q <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_good_out_R2.fastq &gt;/dev/null
        grep @ <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.rRNA.fq | cut -f1 -d <span class="hljs-string">' '</span> | sed <span class="hljs-string">'s/@//g'</span> &gt; <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.rRNA.readsID 
        grep @ <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.rRNA.fq | cut -f1 -d <span class="hljs-string">' '</span> | sed <span class="hljs-string">'s/@//g'</span> &gt; <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.rRNA.readsID
        cat <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.rRNA.readsID <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.rRNA.readsID | sort | uniq &gt; <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_rRNA.sorted.readsID
        seqkit grep -vf <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_rRNA.sorted.readsID <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_good_out_R1.fastq -o <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.clean.fq.gz
        seqkit grep -vf <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_rRNA.sorted.readsID <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_good_out_R2.fastq -o <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.clean.fq.gz
        <span class="hljs-comment"># Remove unnecessary files</span>
        rm <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R*U.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R*P.fq <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R*P.rRNA.fq <span class="hljs-variable">$i</span>/*.readsID <span class="hljs-variable">$i</span>/*bad* <span class="hljs-variable">$i</span>/*good* <span class="hljs-variable">$i</span>/*single*
        <span class="hljs-comment"># Index the reference genome. Only run this for one time.</span>
        <span class="hljs-comment"># hisat2-build -f /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \</span>
        <span class="hljs-comment">#        /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta</span>
        <span class="hljs-comment">#Map the reads</span>
        hisat2 --rna-strandness RF -p 32 \
            --summary-file <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.summary \
            -x /data/zhaojiantao/pepper/References/S8/Zhangshugang.fasta \
            -1 <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R1P.clean.fq.gz \
            -2 <span class="hljs-variable">$i</span>/<span class="hljs-variable">${i}</span>_R2P.clean.fq.gz \
            -S <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.sam
        <span class="hljs-comment"># Convert sam to bam</span>
        samtools view -S -b <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.sam &gt; <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.bam
        <span class="hljs-comment"># Remove sam file</span>
        rm <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.sam
<span class="hljs-keyword">done</span>

<span class="hljs-comment"># Prepare the gene bed file </span>
cat /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff | \
    awk <span class="hljs-string">'$3=="CDS" &amp;&amp; $9~/^ID/{sub(/;.*/,"",$9);L[substr($9,4)]+=$5-$4+1}END{for(i in L){print i,L[i]}}'</span> | \
    sed <span class="hljs-string">'s/cds-//g'</span> | sort -k1 &gt; /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.trans.length

sed <span class="hljs-string">'s/;/\t/g'</span> /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff | \
    awk <span class="hljs-string">'$3=="mRNA" {print $1,$4,$5,$7,$9}'</span> | sed <span class="hljs-string">'s/ID=//g'</span> | sort -k5 | \
    paste - /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.trans.length | \
    sort -t <span class="hljs-string">' '</span> -k1,1 -k5,5 |  awk <span class="hljs-string">'{print $1,$2,$3,$5,$7,$4}'</span> | sed <span class="hljs-string">'s/ /\t/g'</span> \
    &gt; /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.bed

rm /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.trans.length

<span class="hljs-comment"># Get the longest transcripts for each gene using [AGAT](https://agat.readthedocs.io/en/latest/index.html)</span>
<span class="hljs-comment"># agat_sp_keep_longest_isoform.pl: The script aims to filter isoforms when present. For a locus: - when all isoforms have CDS we keep the one with the longest CDS. - when some isoforms have CDS some others not, we keep the one with the longest CDS. - when none of the isoforms have CDS, we keep the one with the longest concatenated exons.</span>
<span class="hljs-comment"># Source: https://agat.readthedocs.io/en/latest/tools/agat_sp_keep_longest_isoform.html</span>
agat_sp_keep_longest_isoform.pl -gff Ca.S8.gff -o Ca.S8.longestisoform.gff

<span class="hljs-comment"># Count the reads</span>
featureCounts -T 20 -s 1 -p \
    -a /data/zhaojiantao/pepper/References/S8/Ca.S8.longestisoform.gff \
    -t gene -g ID */*.bam -O \
    -o raw.count 

<span class="hljs-comment"># Generate raw count</span>
mRNAtool.pl -t count -s PS -l fr-firststrand \
    -f /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.bed */*.bam 

<span class="hljs-comment"># Prepare the mapped reads summary file</span>
<span class="hljs-comment"># Total reads</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat fruitpeel.id); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'1p'</span> <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.summary | awk <span class="hljs-string">'{print $1}'</span> ; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Pair-end concordantly exactly 1 time</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat fruitpeel.id); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'4p'</span> <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.summary | awk <span class="hljs-string">'{print $1}'</span> ; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Pair-end concordantly &gt;1 times</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat fruitpeel.id); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'5p'</span> <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.summary | awk <span class="hljs-string">'{print $1}'</span> ; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Single concordantly exactly 1 time</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat fruitpeel.id); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'13p'</span> <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.summary | awk <span class="hljs-string">'{print $1}'</span> ; <span class="hljs-keyword">done</span>
<span class="hljs-comment"># Single concordantly exactly &gt;1 time</span>
<span class="hljs-keyword">for</span> i <span class="hljs-keyword">in</span> $(cat fruitpeel.id); <span class="hljs-keyword">do</span> sed -n <span class="hljs-string">'14p'</span> <span class="hljs-variable">$i</span>/<span class="hljs-variable">$i</span>.cleaned.summary | awk <span class="hljs-string">'{print $1}'</span> ; <span class="hljs-keyword">done</span>

<span class="hljs-comment"># Calculate the reads count</span>
mRNAtool.pl -t count -s PS -l fr-firststrand \
    -f /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.bed */*.bam
    
<span class="hljs-comment"># Normalize the raw count</span>
mRNAtool.pl -t norm -f /data/zhaojiantao/pepper/References/S8/Zhangshugang.gff.bed -m 1 -u mapped_reads exp_raw_count.txt &gt; rpkm.txt

<span class="hljs-comment"># Step 6.3 Calculate correlation</span>
mRNAtool.pl -t corre rpkm.txt &gt; rpkm_corre.txt

</div></code></pre>
<h2 id="personal-scripts">Personal scripts</h2>
<h3 id="4dtvscanpl">4DTV_scan.pl</h3>
<pre class="hljs"><code><div><span class="hljs-comment">#! /usr/bin/env perl</span>
<span class="hljs-keyword">use</span> strict;
<span class="hljs-keyword">use</span> warnings;
<span class="hljs-keyword">use</span> Bio::SeqIO;
<span class="hljs-comment">#use Getopt::Std;</span>
<span class="hljs-keyword">use</span> Getopt::Long;
<span class="hljs-keyword">use</span> File::Basename;
<span class="hljs-comment"># Author: Xin Wang   20131020</span>
<span class="hljs-keyword">my</span> ($genome_file,$gff_file,$fold);
GetOptions(
				<span class="hljs-string">"g|genome=s"</span> =&gt; \$genome_file,
				<span class="hljs-string">"f|gff=s"</span> =&gt; \$gff_file,
                <span class="hljs-string">"o|output=s"</span> =&gt; \$fold,
        );
        
<span class="hljs-keyword">die</span>(qq/Usage: Get the <span class="hljs-number">4</span>DTV site from genome \n
Options: -g|genome      The genome fasta\n
         -f|gff         The GFF file\n
         -o|output      The output file\n
/) <span class="hljs-keyword">unless</span> ( $genome_file &amp;&amp; $gff_file &amp;&amp; $fold);

<span class="hljs-comment">#=@ARGV;</span>

<span class="hljs-comment">#die "Usage: $0 &lt;genome file&gt; &lt;gff file&gt; &lt;4 fold degeneration sites output&gt;\n" if(@ARGV &lt; 3);</span>

<span class="hljs-keyword">my</span> %gff;
<span class="hljs-keyword">open</span>(I,<span class="hljs-string">"&lt; $gff_file"</span>);
<span class="hljs-keyword">my</span> $no=<span class="hljs-number">0</span>;
<span class="hljs-keyword">while</span>(&lt;I&gt;){
    <span class="hljs-keyword">chomp</span>;
    <span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span>(<span class="hljs-regexp">/^#/</span>);
    <span class="hljs-keyword">my</span> @a=<span class="hljs-keyword">split</span>(<span class="hljs-regexp">/\s+/</span>);
    <span class="hljs-keyword">next</span> <span class="hljs-keyword">unless</span>($a[<span class="hljs-number">2</span>] eq <span class="hljs-string">"CDS"</span>);
    $no++;
    <span class="hljs-keyword">my</span> ($chr,$start,$end,$strand,$phase,$name)=($a[<span class="hljs-number">0</span>],$a[<span class="hljs-number">3</span>],$a[<span class="hljs-number">4</span>],$a[<span class="hljs-number">6</span>],$a[<span class="hljs-number">7</span>],$a[<span class="hljs-number">8</span>]);
    <span class="hljs-comment"># $chr=~s/chr//g;</span>
    $name=~<span class="hljs-regexp">/Parent=([^;]+)/</span>;
    $name=$1;
    $gff{$chr}{$name}{$no}<span class="hljs-string">{start}</span>=$start;
    $gff{$chr}{$name}{$no}<span class="hljs-string">{end}</span>=$end;
    $gff{$chr}{$name}{$no}<span class="hljs-string">{strand}</span>=$strand;
    $gff{$chr}{$name}{$no}<span class="hljs-string">{phase}</span>=$phase;
}
<span class="hljs-keyword">close</span> I;

<span class="hljs-keyword">print</span> STDERR <span class="hljs-string">"GFF reading complete!\n"</span>;

<span class="hljs-keyword">my</span> $fa=Bio::SeqIO-&gt;new(<span class="hljs-string">-file=&gt;</span>$genome_file,<span class="hljs-string">-format=&gt;</span><span class="hljs-string">'fasta'</span>);

<span class="hljs-keyword">my</span> $control=<span class="hljs-number">0</span>;
<span class="hljs-keyword">open</span>(O,<span class="hljs-string">"&gt; $fold"</span>);
<span class="hljs-keyword">while</span>(<span class="hljs-keyword">my</span> $seq=$fa-&gt;next_seq){
    <span class="hljs-keyword">my</span> $chr=$seq-&gt;id;
    <span class="hljs-keyword">my</span> $seq=$seq-&gt;seq;
    <span class="hljs-keyword">next</span> <span class="hljs-keyword">unless</span>(<span class="hljs-keyword">exists</span> $gff{$chr});

    <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $name(<span class="hljs-keyword">keys</span> %{$gff{$chr}}){
        <span class="hljs-keyword">my</span> $strand=<span class="hljs-string">"NA"</span>;
        <span class="hljs-keyword">my</span> $line=<span class="hljs-string">""</span>;
        <span class="hljs-comment"># die "debug" if($control++&gt;100);</span>
        <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $no(<span class="hljs-keyword">sort</span> { $gff{$chr}{$name}{$a}<span class="hljs-string">{start}</span> &lt;=&gt; $gff{$chr}{$name}{$b}<span class="hljs-string">{start}</span> } <span class="hljs-keyword">keys</span> %{$gff{$chr}{$name}}){
            <span class="hljs-keyword">if</span>($strand eq <span class="hljs-string">"NA"</span>){
	$strand=$gff{$chr}{$name}{$no}<span class="hljs-string">{strand}</span>;
            }
            <span class="hljs-keyword">my</span> $start=$gff{$chr}{$name}{$no}<span class="hljs-string">{start}</span>;
            <span class="hljs-keyword">my</span> $end=$gff{$chr}{$name}{$no}<span class="hljs-string">{end}</span>;
            <span class="hljs-keyword">my</span> $len=$end-$start+<span class="hljs-number">1</span>;
            <span class="hljs-keyword">my</span> $subline=<span class="hljs-keyword">substr</span>($seq,$start-<span class="hljs-number">1</span>,$len);
            $line.=$subline;
        }

        <span class="hljs-keyword">my</span> @fold_sites_location_in_genome;

        <span class="hljs-keyword">if</span>($strand eq <span class="hljs-string">"+"</span>){
            <span class="hljs-keyword">my</span> $pep = &amp;translate_nucl($line);

            <span class="hljs-keyword">my</span> @fold_sites_location_in_cds = &amp;get_location_in_cds($line);

            <span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span>(<span class="hljs-keyword">scalar</span>(@fold_sites_location_in_cds) == <span class="hljs-number">0</span>);

            <span class="hljs-keyword">my</span> %corresponding_postion;
            <span class="hljs-keyword">my</span> $accumulating_length=<span class="hljs-number">0</span>;
            <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $no(<span class="hljs-keyword">sort</span> { $gff{$chr}{$name}{$a}<span class="hljs-string">{start}</span> &lt;=&gt; $gff{$chr}{$name}{$b}<span class="hljs-string">{start}</span> } <span class="hljs-keyword">keys</span> %{$gff{$chr}{$name}}){
	<span class="hljs-keyword">my</span> $start=$gff{$chr}{$name}{$no}<span class="hljs-string">{start}</span>;
	<span class="hljs-keyword">my</span> $end=$gff{$chr}{$name}{$no}<span class="hljs-string">{end}</span>;
	<span class="hljs-keyword">my</span> $len=$end-$start+<span class="hljs-number">1</span>;

	<span class="hljs-keyword">my</span> $real_start=$start;
	<span class="hljs-keyword">my</span> $real_end  =$end;
	<span class="hljs-keyword">my</span> $cds_start =$accumulating_length+<span class="hljs-number">1</span>;
	<span class="hljs-keyword">my</span> $cds_end   =$accumulating_length+$len;

	<span class="hljs-keyword">my</span> $real_pos=$real_start;
	<span class="hljs-keyword">for</span>(<span class="hljs-keyword">my</span> $i=$cds_start;$i&lt;=$cds_end;$i++){
	    <span class="hljs-keyword">my</span> $cds_pos=$i;
	    $corresponding_postion{$cds_pos}=$real_pos;
	    $real_pos++;
	}
	$accumulating_length+=$len;
            }
            <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $cds_pos(@fold_sites_location_in_cds){
	<span class="hljs-keyword">my</span> $real_pos=$corresponding_postion{$cds_pos};
	<span class="hljs-keyword">push</span> @fold_sites_location_in_genome,$real_pos;
            }
        }
        <span class="hljs-keyword">elsif</span>($strand eq <span class="hljs-string">"-"</span>){
            $line=<span class="hljs-keyword">reverse</span>($line);
            $line=~<span class="hljs-regexp">tr/ATCGatcg/TAGCtagc/</span>;
            <span class="hljs-keyword">my</span> $pep = &amp;translate_nucl($line);

            <span class="hljs-keyword">my</span> @fold_sites_location_in_cds = &amp;get_location_in_cds($line);

            <span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span>(<span class="hljs-keyword">scalar</span>(@fold_sites_location_in_cds) == <span class="hljs-number">0</span>);

            <span class="hljs-keyword">my</span> %corresponding_postion;
            <span class="hljs-keyword">my</span> $accumulating_length=<span class="hljs-number">0</span>;
            <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $no(<span class="hljs-keyword">sort</span> { $gff{$chr}{$name}{$b}<span class="hljs-string">{start}</span> &lt;=&gt; $gff{$chr}{$name}{$a}<span class="hljs-string">{start}</span> } <span class="hljs-keyword">keys</span> %{$gff{$chr}{$name}}){
	<span class="hljs-keyword">my</span> $start=$gff{$chr}{$name}{$no}<span class="hljs-string">{start}</span>;
	<span class="hljs-keyword">my</span> $end=$gff{$chr}{$name}{$no}<span class="hljs-string">{end}</span>;
	<span class="hljs-keyword">my</span> $len=$end-$start+<span class="hljs-number">1</span>;

	<span class="hljs-keyword">my</span> $real_start=$start;
	<span class="hljs-keyword">my</span> $real_end  =$end;
	<span class="hljs-keyword">my</span> $cds_start =$accumulating_length+<span class="hljs-number">1</span>;
	<span class="hljs-keyword">my</span> $cds_end   =$accumulating_length+$len;

	<span class="hljs-keyword">my</span> $real_pos=$real_end;
	<span class="hljs-keyword">for</span>(<span class="hljs-keyword">my</span> $i=$cds_start;$i&lt;=$cds_end;$i++){
	    <span class="hljs-keyword">my</span> $cds_pos=$i;
	    $corresponding_postion{$cds_pos}=$real_pos;
	    $real_pos--;
	}
	$accumulating_length+=$len;
            }
            <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $cds_pos(@fold_sites_location_in_cds){
	<span class="hljs-keyword">my</span> $real_pos=$corresponding_postion{$cds_pos};
	<span class="hljs-keyword">push</span> @fold_sites_location_in_genome,$real_pos;
            }
        }

        <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $pos(@fold_sites_location_in_genome){
            <span class="hljs-keyword">print</span> O <span class="hljs-string">"$chr\t$pos\t$name\n"</span>;
        }

        <span class="hljs-keyword">print</span> STDERR <span class="hljs-string">"$chr\t$name\n"</span>;
    }
}
<span class="hljs-keyword">close</span> O;

<span class="hljs-function"><span class="hljs-keyword">sub</span> <span class="hljs-title">get_location_in_cds</span></span>{
    <span class="hljs-keyword">my</span> $line=<span class="hljs-keyword">shift</span>;
    <span class="hljs-keyword">my</span> @location;
    <span class="hljs-keyword">my</span> $len=<span class="hljs-keyword">length</span>($line);

    <span class="hljs-keyword">if</span>($len == <span class="hljs-number">0</span>){
        <span class="hljs-keyword">return</span>(@location);
    }

    <span class="hljs-keyword">my</span> @bases=<span class="hljs-keyword">split</span>(<span class="hljs-string">""</span>,$line);
    <span class="hljs-keyword">my</span> @ATCG=(<span class="hljs-string">"A"</span>,<span class="hljs-string">"T"</span>,<span class="hljs-string">"C"</span>,<span class="hljs-string">"G"</span>);

    <span class="hljs-keyword">for</span>(<span class="hljs-keyword">my</span> $i=<span class="hljs-number">0</span>;$i&lt;$len;$i+=<span class="hljs-number">3</span>){
        <span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span>(!$bases[$i] <span class="hljs-keyword">or</span> !$bases[$i+<span class="hljs-number">1</span>] <span class="hljs-keyword">or</span> !$bases[$i+<span class="hljs-number">2</span>]);
        <span class="hljs-keyword">my</span> @codon_bases=($bases[$i],$bases[$i+<span class="hljs-number">1</span>],$bases[$i+<span class="hljs-number">2</span>]);
        <span class="hljs-keyword">my</span> $codon_bases=<span class="hljs-keyword">join</span> <span class="hljs-string">""</span>,@codon_bases;
        $codon_bases=~<span class="hljs-regexp">tr/atcg/ATCG/</span>;
        <span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span>($codon_bases=~<span class="hljs-regexp">/[^ATCG]/</span>);
        <span class="hljs-keyword">my</span> $codon_pep  = &amp;translate_nucl($codon_bases);

        <span class="hljs-keyword">for</span>(<span class="hljs-keyword">my</span> $j=<span class="hljs-number">0</span>;$j&lt;<span class="hljs-number">3</span>;$j++){
            <span class="hljs-keyword">my</span> $light = <span class="hljs-number">1</span>;
            <span class="hljs-keyword">my</span> @new_codon_bases = @codon_bases;
            <span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $bases(@ATCG){
	$new_codon_bases[$j]=$bases;
	<span class="hljs-keyword">my</span> $new_codon_bases = <span class="hljs-keyword">join</span> <span class="hljs-string">""</span>,@new_codon_bases;
	<span class="hljs-keyword">my</span> $new_codon_pep   = &amp;translate_nucl($new_codon_bases);
	<span class="hljs-keyword">if</span>($new_codon_pep <span class="hljs-keyword">ne</span> $codon_pep){
	    $light=<span class="hljs-number">0</span>;
	    <span class="hljs-keyword">last</span>;
	}
            }
            <span class="hljs-keyword">if</span>($light == <span class="hljs-number">1</span>){
	<span class="hljs-keyword">my</span> $location_in_cds=$i+$j+<span class="hljs-number">1</span>;
	<span class="hljs-keyword">push</span> @location,$location_in_cds;
            }
        }
    }
    <span class="hljs-keyword">my</span> $number=@location;
    <span class="hljs-keyword">print</span> STDERR <span class="hljs-string">" $number four-fold-degenerate-sites in this gene\n"</span>;
    <span class="hljs-keyword">return</span>(@location);
}

<span class="hljs-function"><span class="hljs-keyword">sub</span> <span class="hljs-title">translate_nucl</span></span>{
    <span class="hljs-keyword">my</span> $seq=<span class="hljs-keyword">shift</span>;
    <span class="hljs-keyword">my</span> $seq_obj=Bio::Seq-&gt;new(<span class="hljs-string">-seq=&gt;</span>$seq,<span class="hljs-string">-alphabet=&gt;</span><span class="hljs-string">'dna'</span>);
    <span class="hljs-keyword">my</span> $pro=$seq_obj-&gt;translate;
    $pro=$pro-&gt;seq;
    <span class="hljs-keyword">return</span>($pro);
}
</div></code></pre>
<h3 id="assigngeneticpl">assign_genetic.pl</h3>
<pre class="hljs"><code><div><span class="hljs-comment">#!/usr/bin/perl -w</span>
<span class="hljs-keyword">use</span> strict;

<span class="hljs-keyword">my</span> ($genetic_map,$in) = @ARGV;

<span class="hljs-keyword">my</span> %map;

<span class="hljs-keyword">open</span> IN,<span class="hljs-string">"&lt;"</span>,$genetic_map;
<span class="hljs-keyword">while</span>(&lt;IN&gt;){
	<span class="hljs-keyword">chomp</span>;
	<span class="hljs-keyword">my</span> @line = <span class="hljs-keyword">split</span>;
	<span class="hljs-keyword">my</span> @infor = <span class="hljs-keyword">split</span><span class="hljs-regexp">/\_/</span>,$line[<span class="hljs-number">0</span>];
	<span class="hljs-keyword">my</span> $chr = $infor[<span class="hljs-number">0</span>];
	<span class="hljs-keyword">my</span> $pos = $infor[<span class="hljs-number">1</span>];
	<span class="hljs-keyword">my</span> $genetic_distance = $line[<span class="hljs-number">2</span>];
	$map{$chr}{$pos} = $genetic_distance;
}
<span class="hljs-keyword">close</span> IN;

<span class="hljs-keyword">my</span> %snp;
<span class="hljs-keyword">open</span> IN,<span class="hljs-string">"&lt;"</span>,$in;
<span class="hljs-keyword">while</span>(&lt;IN&gt;){
	<span class="hljs-keyword">chomp</span>;
	<span class="hljs-keyword">my</span> @line = <span class="hljs-keyword">split</span>;
	<span class="hljs-keyword">my</span> @infor = <span class="hljs-keyword">split</span><span class="hljs-regexp">/\_/</span>,$line[<span class="hljs-number">0</span>];
	<span class="hljs-keyword">my</span> $chr = $infor[<span class="hljs-number">0</span>];
	<span class="hljs-keyword">my</span> $pos = $infor[<span class="hljs-number">1</span>];
	$snp{$chr}{$pos}{$line[<span class="hljs-number">0</span>]} =<span class="hljs-number">0</span>;
}

<span class="hljs-comment">#my @test = ("SL2.50ch01");</span>
<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $chr(<span class="hljs-keyword">keys</span> %snp){
	<span class="hljs-keyword">my</span> @pos = <span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$snp{$chr}};
	<span class="hljs-keyword">my</span> $pos_length = @pos;
	<span class="hljs-keyword">my</span> @map = <span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$map{$chr}};
	<span class="hljs-keyword">my</span> $map_length = @map;
	<span class="hljs-keyword">open</span> OUT,<span class="hljs-string">"&gt;"</span>,<span class="hljs-string">"$chr.snp.genetic.txt"</span>;
	<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $pos(@pos){
		<span class="hljs-keyword">my</span> $distance = <span class="hljs-string">"NA"</span>;
		<span class="hljs-keyword">if</span>($pos &lt;=$map[<span class="hljs-number">0</span>]){
			$distance = $pos*$map{$chr}{$map[<span class="hljs-number">0</span>]}/($map[<span class="hljs-number">0</span>]);
		}
		<span class="hljs-keyword">elsif</span>($pos &gt;=$map[-<span class="hljs-number">1</span>]){
			<span class="hljs-keyword">my</span> $bin =(<span class="hljs-keyword">abs</span>($map{$chr}{$map[-<span class="hljs-number">1</span>]} - $map{$chr}{$map[-<span class="hljs-number">2</span>]}))/($map[-<span class="hljs-number">1</span>] - $map[-<span class="hljs-number">2</span>]+<span class="hljs-number">1</span>);
			$distance = $map{$chr}{$map[-<span class="hljs-number">1</span>]} + ($pos -$map[-<span class="hljs-number">1</span>])*$bin;
		}
		<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $i(<span class="hljs-number">0</span>..($map_length-<span class="hljs-number">2</span>)){
			<span class="hljs-keyword">if</span>(($pos&gt;$map[$i]) &amp;&amp; ($pos&lt;=$map[$i+<span class="hljs-number">1</span>])){
				<span class="hljs-keyword">my</span> $bin =(<span class="hljs-keyword">abs</span>($map{$chr}{$map[$i+<span class="hljs-number">1</span>]} - $map{$chr}{$map[$i]}))/($map[$i+<span class="hljs-number">1</span>] - $map[$i]+<span class="hljs-number">1</span>);
				$distance = $map{$chr}{$map[$i]} + ($pos -$map[$i])*$bin;
			}
		}
		<span class="hljs-keyword">my</span> @id = <span class="hljs-keyword">keys</span> %{$snp{$chr}{$pos}};
		$distance = <span class="hljs-keyword">sprintf</span>(<span class="hljs-string">"%.3f"</span>,$distance);
		<span class="hljs-keyword">print</span> OUT<span class="hljs-string">"$id[0]\t$pos\t$distance\n"</span>;
	}
	<span class="hljs-keyword">close</span> OUT;
}

</div></code></pre>
<h3 id="pislidewinpl">PI_slide_win.pl</h3>
<pre class="hljs"><code><div><span class="hljs-comment">#!/usr/bin/perl</span>
<span class="hljs-keyword">use</span> warnings;
<span class="hljs-keyword">use</span> strict;
<span class="hljs-keyword">use</span> Getopt::Std;
<span class="hljs-keyword">use</span> Set::IntervalTree;
$|=<span class="hljs-number">1</span>;

<span class="hljs-keyword">our</span> ($opt_i, $opt_p, $opt_s, $opt_d, $opt_w, $opt_o);
getopt(<span class="hljs-string">"s:i:p:d:w:o:"</span>);

&amp;usage <span class="hljs-keyword">if</span> (@ARGV &lt; <span class="hljs-number">1</span>);


<span class="hljs-keyword">print</span> <span class="hljs-string">"Starting $0...\n"</span>;

<span class="hljs-comment"># default parameters</span>
$opt_w = $opt_w || <span class="hljs-number">100000</span>;
$opt_s = $opt_s || <span class="hljs-number">10000</span>;


<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $infile(@ARGV){
	<span class="hljs-keyword">my</span> $time1 = <span class="hljs-keyword">time</span>();
	<span class="hljs-keyword">open</span> IN,<span class="hljs-string">"&lt;"</span>,$infile;
	<span class="hljs-keyword">my</span> %pi_per_site;
	<span class="hljs-keyword">my</span> %genome_length;
	<span class="hljs-keyword">while</span>(&lt;IN&gt;){
		<span class="hljs-keyword">chomp</span>;
		<span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span>(<span class="hljs-regexp">/^CHROM/</span>);
		<span class="hljs-keyword">my</span> @line = <span class="hljs-keyword">split</span><span class="hljs-regexp">/\t/</span>,$_;
		<span class="hljs-keyword">my</span> $chr = $line[<span class="hljs-number">0</span>];
		<span class="hljs-keyword">my</span> $pos = $line[<span class="hljs-number">1</span>];
		<span class="hljs-keyword">my</span> $pi = $line[<span class="hljs-number">2</span>];
		<span class="hljs-keyword">if</span>($pi eq <span class="hljs-string">"-nan"</span>){
			$pi =<span class="hljs-number">0</span>;
		}
		$pi_per_site{$chr}{$pos} =$pi;
		<span class="hljs-comment">#$genome_length{$chr}{$pos} =0;</span>
	}
	<span class="hljs-keyword">close</span> IN;
	<span class="hljs-keyword">our</span> %window_hash = ();	<span class="hljs-comment">#####change the my to our to avoid the Variable "%window_hash" will not stay shared </span>
	<span class="hljs-keyword">foreach</span> <span class="hljs-keyword">my</span> $chr (<span class="hljs-keyword">sort</span> <span class="hljs-keyword">keys</span> %pi_per_site){
		<span class="hljs-keyword">my</span> @pos = <span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$pi_per_site{$chr}};
		<span class="hljs-keyword">my</span> $chr_len = $pos[-<span class="hljs-number">1</span>];
	<span class="hljs-comment">#	print "\tprocessing $opt_p$chr, length: $chr_len, ";</span>
		<span class="hljs-comment"># get the window number, window start, window end</span>
		<span class="hljs-keyword">my</span> $window_num = <span class="hljs-number">0</span>;
		<span class="hljs-keyword">if</span> ($chr_len &lt;= $opt_w) {               <span class="hljs-comment">#windowsize greater than chro_length</span>
			$window_num = <span class="hljs-number">1</span>;
			($window_hash{$chr}{$window_num}<span class="hljs-string">{start}</span>, $window_hash{$chr}{$window_num}<span class="hljs-string">{end}</span>) = (<span class="hljs-number">1</span>, $chr_len);
		}<span class="hljs-keyword">else</span>{                                  <span class="hljs-comment">#windowsize less than chro_length</span>
			$window_num = &amp;split_genome($chr, $chr_len);
		}
	<span class="hljs-comment">#	print "window_number: $window_num.\n";</span>
	}
	<span class="hljs-comment">##generate interval</span>
	<span class="hljs-keyword">print</span> <span class="hljs-string">"Step1: calculating PI per window for $infile\n"</span>;
	<span class="hljs-keyword">my</span> %average_pi;	
	<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $chr(<span class="hljs-keyword">sort</span> {$a cmp $b} <span class="hljs-keyword">keys</span> %window_hash){
		<span class="hljs-keyword">my</span> $chr_tree = $chr;
		$chr_tree = Set::IntervalTree-&gt;new;
		<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $w_num(<span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$window_hash{$chr}}){
			<span class="hljs-keyword">my</span> $start = $window_hash{$chr}{$w_num}{<span class="hljs-string">'start'</span>};
			<span class="hljs-keyword">my</span> $end = $window_hash{$chr}{$w_num}{<span class="hljs-string">'end'</span>};
			<span class="hljs-comment">#print "$chr\t$w_num\t$window_hash{$chr}{$w_num}{'start'}\t$window_hash{$chr}{$w_num}{'end'}\t$N_num\n";</span>
			$chr_tree-&gt;insert(<span class="hljs-string">"$chr"</span>.<span class="hljs-string">"_"</span>.<span class="hljs-string">"$w_num"</span>.<span class="hljs-string">"_"</span>.<span class="hljs-string">"$start"</span>.<span class="hljs-string">"_"</span>.<span class="hljs-string">"$end"</span>, $start, $end);
			$average_pi{$chr}{$w_num}{$start}{$end}[<span class="hljs-number">0</span>]=<span class="hljs-number">0</span>;   <span class="hljs-comment"># variation number</span>
			$average_pi{$chr}{$w_num}{$start}{$end}[<span class="hljs-number">1</span>] =<span class="hljs-number">0</span>; <span class="hljs-comment">#PI sum</span>
		}
		<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $pos(<span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$pi_per_site{$chr}}){
			<span class="hljs-keyword">my</span> $interval = $chr_tree-&gt;fetch($pos-<span class="hljs-number">0</span>.<span class="hljs-number">5</span>,$pos+<span class="hljs-number">0</span>.<span class="hljs-number">5</span>);
			<span class="hljs-keyword">my</span> @interval_array = @$interval;
			<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $interval_infor(@interval_array){
				<span class="hljs-keyword">my</span> @infor = <span class="hljs-keyword">split</span><span class="hljs-regexp">/\_/</span>,$interval_infor;
				$average_pi{$infor[<span class="hljs-number">0</span>]}{$infor[<span class="hljs-number">1</span>]}{$infor[<span class="hljs-number">2</span>]}{$infor[<span class="hljs-number">3</span>]}[<span class="hljs-number">0</span>]++; <span class="hljs-comment"># variation number</span>
				$average_pi{$infor[<span class="hljs-number">0</span>]}{$infor[<span class="hljs-number">1</span>]}{$infor[<span class="hljs-number">2</span>]}{$infor[<span class="hljs-number">3</span>]}[<span class="hljs-number">1</span>]=$average_pi{$infor[<span class="hljs-number">0</span>]}{$infor[<span class="hljs-number">1</span>]}{$infor[<span class="hljs-number">2</span>]}{$infor[<span class="hljs-number">3</span>]}[<span class="hljs-number">1</span>] + $pi_per_site{$chr}{$pos}; <span class="hljs-comment">#Sum PI			</span>
			}
		}	
	}
	
	<span class="hljs-comment">####output average PI for one window</span>
	<span class="hljs-keyword">print</span> <span class="hljs-string">"Step2: outputing PI per window for $infile\n"</span>;
	<span class="hljs-keyword">my</span> $whole_pi =<span class="hljs-number">0</span>;
	<span class="hljs-keyword">my</span> $whole_win_num = <span class="hljs-number">0</span>;
	<span class="hljs-comment">#print "\#Chr\tWindow number\tStart\tEnd\tBin_Size\tNon_N_Bin_Size\tPI_Bin\tPI_Non_N_Bin\n";</span>
	<span class="hljs-keyword">open</span> OUT,<span class="hljs-string">"&gt;"</span>,<span class="hljs-string">"$infile.w$opt_w.s$opt_s.txt"</span>;
	<span class="hljs-keyword">print</span> OUT<span class="hljs-string">"\#Chr\tWindow number\tStart\tEnd\tBin_Size\tN_var\tPI\n"</span>;
	<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $chr(<span class="hljs-keyword">sort</span> {$a cmp $b} <span class="hljs-keyword">keys</span> %average_pi){
		<span class="hljs-comment">#print OUT"\#Chr\tWindow number\tStart\tEnd\tBin_Size\tNon_N_Bin_Size\tPI_Bin\tPI_Non_N_Bin\n";</span>
		<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $win_num(<span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$average_pi{$chr}}){
			<span class="hljs-keyword">print</span> OUT<span class="hljs-string">"$chr\t$win_num\t"</span>;
			$whole_win_num++;
			<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $start(<span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$average_pi{$chr}{$win_num}}){
				<span class="hljs-keyword">for</span> <span class="hljs-keyword">my</span> $end(<span class="hljs-keyword">sort</span> {$a &lt;=&gt; $b} <span class="hljs-keyword">keys</span> %{$average_pi{$chr}{$win_num}{$start}}){
					<span class="hljs-keyword">my</span> $bin_size = $end -$start +<span class="hljs-number">1</span>;
					<span class="hljs-keyword">my</span> $average_pi = $average_pi{$chr}{$win_num}{$start}{$end}[<span class="hljs-number">1</span>]/$bin_size;
					$whole_pi = $whole_pi + $average_pi;
					<span class="hljs-comment">#print OUT"$start\t$end\t$bin_size\t$non_n_bin_size\t$average_pi{$chr}{$win_num}{$start}{$end}[0]\t$average_pi{$chr}{$win_num}{$start}{$end}[1]\n";;</span>
					<span class="hljs-keyword">print</span> OUT<span class="hljs-string">"$start\t$end\t$bin_size\t$average_pi{$chr}{$win_num}{$start}{$end}[0]\t$average_pi\n"</span>;;		
				}
			}
		}
	}
	<span class="hljs-keyword">close</span> OUT;
	<span class="hljs-keyword">my</span> $average_whole_pi = $whole_pi/$whole_win_num;
	<span class="hljs-keyword">print</span> <span class="hljs-string">"The whole genome PI for $infile is $average_whole_pi\n\n"</span>;	
	<span class="hljs-keyword">my</span> $time2 = <span class="hljs-keyword">time</span>();
	<span class="hljs-keyword">my</span> $time  = <span class="hljs-keyword">sprintf</span> <span class="hljs-string">"%.4f"</span>, ( $time2 - $time1 ) / <span class="hljs-number">60</span>;
	<span class="hljs-keyword">print</span> <span class="hljs-string">"$0 finished! $time mins elapsed for $infile.\n"</span>;
	
	<span class="hljs-function"><span class="hljs-keyword">sub</span> <span class="hljs-title">split_genome</span> </span>{
		<span class="hljs-keyword">my</span> ($chro_id, $chro_length) = @_;
		<span class="hljs-keyword">my</span> $win_num =<span class="hljs-number">0</span>;
		<span class="hljs-keyword">for</span> (<span class="hljs-keyword">my</span> $start=<span class="hljs-number">1</span>;$start &lt; ($chro_length - $opt_w + $opt_s);$start=$start + $opt_s){
			<span class="hljs-keyword">my</span> $end = $start + $opt_w - <span class="hljs-number">1</span>;
			$win_num++;
			<span class="hljs-keyword">if</span>($end &lt; $chro_length){
				$window_hash{$chro_id}{$win_num}{<span class="hljs-string">'start'</span>}=$start;
				$window_hash{$chro_id}{$win_num}{<span class="hljs-string">'end'</span>}=$end;
			}
			<span class="hljs-keyword">else</span>{
				$end = $chro_length;
				$window_hash{$chro_id}{$win_num}{<span class="hljs-string">'start'</span>}=$start;
				$window_hash{$chro_id}{$win_num}{<span class="hljs-string">'end'</span>}=$end;
				<span class="hljs-comment">#print "$size\n";</span>
			}
		}
	}
}	

<span class="hljs-function"><span class="hljs-keyword">sub</span> <span class="hljs-title">usage</span> </span>{
<span class="hljs-keyword">print</span><span class="hljs-string">"Usage: $0 -OPTIONS VALUES infile1 infile2 infile3 ......
This script is used to caculate the average PI using sliding window.
Options:
     -w  NO   windowsize to caculate the average PI, default = 100000
     -s  NO   step size, should be greater than 0kb and less than windowsize, default = 10000 
     \n"</span>
}
</div></code></pre>
<h3 id="mergepl">merge.pl</h3>
<p>This script is used to merge the top 4% XP-CLR scores with distances within 100000</p>
<pre class="hljs"><code><div><span class="hljs-comment">#!/usr/bin/perl</span>
<span class="hljs-keyword">while</span>(&lt;&gt;) {

        <span class="hljs-keyword">chomp</span>;

        <span class="hljs-keyword">next</span> <span class="hljs-keyword">if</span> (<span class="hljs-regexp">/SNP/</span>);

        @a = <span class="hljs-keyword">split</span> <span class="hljs-string">"\t"</span>;

        <span class="hljs-keyword">if</span> ($a[<span class="hljs-number">2</span>] - $end &gt; <span class="hljs-number">100000</span> || $a[<span class="hljs-number">1</span>] <span class="hljs-keyword">ne</span> $chr || <span class="hljs-keyword">eof</span>() ) { <span class="hljs-keyword">print</span> <span class="hljs-string">"$chr\t$start\t$end\t$max\n"</span>; $start = $a[<span class="hljs-number">2</span>]; $end = $a[<span class="hljs-number">3</span>]; $max = $a[<span class="hljs-number">4</span>]; }

        <span class="hljs-keyword">else</span> { $end = $a[<span class="hljs-number">3</span>]; <span class="hljs-keyword">if</span> ($a[<span class="hljs-number">4</span>] &gt; $max) { $max = $a[<span class="hljs-number">4</span>]; } }

        $chr = $a[<span class="hljs-number">1</span>];

}

</div></code></pre>

</body>
</html>
